// lowcoder-sdk/dist/07b971d2.js
import { mJ as h, i6 as D } from "./c988e26e.js";
import M from "react";
function j(f, l) {
  for (var c = 0; c < l.length; c++) {
    const i = l[c];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const o in i)
        if (o !== "default" && !(o in f)) {
          const a = Object.getOwnPropertyDescriptor(i, o);
          a && Object.defineProperty(f, o, a.get ? a : {
            enumerable: !0,
            get: () => i[o]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, w = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "0 0 1024 1024", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M676.4 528.95L293.2 829.97c-14.25 11.2-35.2 1.1-35.2-16.95V210.97c0-18.05 20.95-28.14 35.2-16.94l383.2 301.02a21.53 21.53 0 010 33.9M694 864h64a8 8 0 008-8V168a8 8 0 00-8-8h-64a8 8 0 00-8 8v688a8 8 0 008 8" } }] }, name: "step-forward", theme: "filled" };
_.default = E;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return S;
    }
  });
  var l = /* @__PURE__ */ g(M), c = /* @__PURE__ */ a(_), i = /* @__PURE__ */ a(h);
  function o(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? t : r;
    })(e);
  }
  function g(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = O(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, u = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = u ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function P(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(u) {
        return Object.getOwnPropertyDescriptor(t, u).enumerable;
      }))), n.forEach(function(u) {
        o(e, u, t[u]);
      });
    }
    return e;
  }
  function F(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(u) {
        return Object.getOwnPropertyDescriptor(e, u).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function m(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : F(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var b = function(e, r) {
    return /* @__PURE__ */ l.createElement(i.default, m(P({}, e), {
      ref: r,
      icon: c.default
    }));
  }, v = /* @__PURE__ */ l.forwardRef(b);
  process.env.NODE_ENV !== "production" && (v.displayName = "StepForwardFilled");
  var S = v;
})(w);
(function(f, l) {
  Object.defineProperty(l, "__esModule", {
    value: !0
  }), l.default = void 0;
  const c = i(w);
  function i(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const o = c;
  l.default = o, f.exports = o;
})(s, s.exports);
var y = s.exports;
const x = /* @__PURE__ */ D(y), $ = /* @__PURE__ */ j({
  __proto__: null,
  default: x
}, [y]);
export {
  $ as S
};
