// lowcoder-sdk/dist/08ac68d2.js
import { mJ as C, i6 as M } from "./c988e26e.js";
import D from "react";
function j(u, o) {
  for (var c = 0; c < o.length; c++) {
    const i = o[c];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const f in i)
        if (f !== "default" && !(f in u)) {
          const a = Object.getOwnPropertyDescriptor(i, f);
          a && Object.defineProperty(u, f, a.get ? a : {
            enumerable: !0,
            get: () => i[f]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(u, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, v = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var z = { icon: function(o, c) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zM327.6 701.7c-2 .9-4.4 0-5.3-2.1-.4-1-.4-2.2 0-3.2L421 470.9 553.1 603l-225.5 98.7zm375.1-375.1L604 552.1 471.9 420l225.5-98.7c2-.9 4.4 0 5.3 2.1.4 1 .4 2.1 0 3.2z", fill: c } }, { tag: "path", attrs: { d: "M322.3 696.4c-.4 1-.4 2.2 0 3.2.9 2.1 3.3 3 5.3 2.1L553.1 603 421 470.9l-98.7 225.5zm375.1-375.1L471.9 420 604 552.1l98.7-225.5c.4-1.1.4-2.2 0-3.2-.9-2.1-3.3-3-5.3-2.1z", fill: o } }, { tag: "path", attrs: { d: "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z", fill: o } }] };
}, name: "compass", theme: "twotone" };
_.default = z;
(function(u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), Object.defineProperty(u, "default", {
    enumerable: !0,
    get: function() {
      return h;
    }
  });
  var o = /* @__PURE__ */ w(D), c = /* @__PURE__ */ a(_), i = /* @__PURE__ */ a(C);
  function f(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function w(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, p = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var l in e)
      if (l !== "default" && Object.prototype.hasOwnProperty.call(e, l)) {
        var s = p ? Object.getOwnPropertyDescriptor(e, l) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, l, s) : n[l] = e[l];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function y(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(p) {
        return Object.getOwnPropertyDescriptor(r, p).enumerable;
      }))), n.forEach(function(p) {
        f(e, p, r[p]);
      });
    }
    return e;
  }
  function P(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(p) {
        return Object.getOwnPropertyDescriptor(e, p).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function T(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : P(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var b = function(e, t) {
    return /* @__PURE__ */ o.createElement(i.default, T(y({}, e), {
      ref: t,
      icon: c.default
    }));
  }, m = /* @__PURE__ */ o.forwardRef(b);
  process.env.NODE_ENV !== "production" && (m.displayName = "CompassTwoTone");
  var h = m;
})(v);
(function(u, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const c = i(v);
  function i(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const f = c;
  o.default = f, u.exports = f;
})(d, d.exports);
var g = d.exports;
const E = /* @__PURE__ */ M(g), N = /* @__PURE__ */ j({
  __proto__: null,
  default: E
}, [g]);
export {
  N as C
};
