// lowcoder-sdk/dist/145a13e9.js
import { Y as o, aC as c, aD as e, aE as i, aF as r, aG as u, aH as w, aI as l, aJ as f, aK as y, aL as g, aM as N, aN as h, aO as b, aP as D, aQ as R, aR as F, aS as k, aT as I, aU as K, aV as $, aW as S, aX as L, aY as x, aZ as T, a_ as v, a$ as E, b0 as j, b1 as V, b2 as A, b3 as B, b4 as M, b5 as P, b6 as H, b7 as J, b8 as q, b9 as G, ba as O, bb as U, bc as Q, bd as W, be as Y, bf as a, bg as z, bh as X, bi as Z, bj as _, bk as oo, bl as eo, bm as mo, bn as io, bo as ro, bp as ao, bq as po, br as to, bs as no, bt as so, bu as Co, bv as co, bw as uo, bx as wo, by as lo, bz as fo, bA as yo, bB as go, bC as No, bD as p, bE as ho, bF as bo, bG as Do } from "./c988e26e.js";
import { ButtonComp as Ro } from "./7c950db2.js";
import { DropdownComp as Fo } from "./44a7bc45.js";
import { LinkComp as ko } from "./4c3ee702.js";
import { ContainerComp as t, defaultContainerData as Io } from "./5ed415ad.js";
import { CustomComp as Ko } from "./f4dd517e.js";
import { DatePickerComp as $o, DateRangeComp as So } from "./62fdf1a5.js";
import { DividerComp as Lo } from "./ede9fb49.js";
import { FileComp as xo } from "./e1e86b1c.js";
import { FileViewerComp as To } from "./2211d28b.js";
import { ImageComp as vo } from "./82dff944.js";
import { JsonSchemaFormComp as Eo } from "./ab9066e0.js";
import { NumberInputComp as jo } from "./75cd6c85.js";
import { RangeSliderComp as Vo } from "./4c8fb1cc.js";
import { SliderComp as Ao } from "./fe51b167.js";
import { ProgressCircleComp as Bo } from "./3ad031d5.js";
import { ProgressComp as Mo } from "./0c3664e6.js";
import { RatingComp as Po } from "./637b37d8.js";
import { RichTextEditorComp as Ho } from "./91f867d0.js";
import { CascaderWithDefault as Jo } from "./48a13bee.js";
import { CheckboxComp as qo } from "./2088ebd5.js";
import { MultiSelectComp as Go } from "./3e042f08.js";
import { RadioComp as Oo } from "./2c7dce0f.js";
import { SegmentedControlComp as Uo } from "./317cb09e.js";
import { SelectComp as Qo } from "./719447be.js";
import { SwitchComp as Wo } from "./f588d8b4.js";
import { defaultTableData as Yo } from "./b416b998.js";
import { TabbedContainerComp as zo } from "./85c8d626.js";
import { TextComp as Xo } from "./f9495ea0.js";
import { InputComp as Zo } from "./2d3be21e.js";
import { PasswordComp as _o } from "./0fe59fe4.js";
import { TextAreaComp as oe } from "./48824b90.js";
import { TimePickerComp as ee, TimeRangeComp as me } from "./c3947343.js";
import { FormComp as ie, defaultFormData as re } from "./44c19c99.js";
import { IFrameComp as ae } from "./ac887e72.js";
import { GridComp as pe, defaultGridData as te } from "./57a2ad53.js";
import { ListViewComp as ne, defaultListViewData as se } from "./62fe8227.js";
import { NavComp as Ce } from "./5e5bb16d.js";
import { TableComp as de } from "./e830a3ab.js";
import { QRCodeComp as ce } from "./92be2a34.js";
import { JsonExplorerComp as ue } from "./27e4a09e.js";
import { JsonEditorComp as we } from "./45952886.js";
import { TreeComp as le } from "./22ff0462.js";
import { TreeSelectComp as fe } from "./98e6b95a.js";
import { AudioComp as ye } from "./8dc71748.js";
import { VideoComp as ge } from "./3e5c99b7.js";
import { CarouselComp as Ne } from "./24046e7b.js";
import { ToggleButtonComp as he } from "./0b1ff108.js";
import { defaultCollapsibleContainerData as be } from "./6559ac3d.js";
import { ScannerComp as De } from "./8a01f194.js";
import { SignatureComp as Re } from "./c2f0a2ba.js";
import { TimeLineComp as Fe } from "./be8baf2f.js";
import { CommentComp as ke } from "./5830a3e5.js";
import { MentionComp as Ie } from "./19b5cdb2.js";
import { AutoCompleteComp as Ke } from "./83d1a1b5.js";
import { JsonLottieComp as $e } from "./252f669a.js";
import { ResponsiveLayoutComp as Se } from "./e7645a61.js";
import { ControlButton as Le } from "./55bdce46.js";
import "react";
import "react-dom";
import "./975a080d.js";
import "./44a79fed.js";
import "./63085759.js";
import "./ef17c1a9.js";
import "./06b2ca8b.js";
import "./30970d5d.js";
import "./85e61f03.js";
import "./7032c5eb.js";
import "./034fe3cf.js";
import "./89fd4803.js";
import "./8c5a44af.js";
import "./d2612a35.js";
import "./14897c7e.js";
import "./182e4b4f.js";
import "./8f0aa994.js";
import "./70c18940.js";
import "./9caec5b5.js";
import "./99e2f395.js";
import "./af4bbc3e.js";
import "./d118cfee.js";
import "./45befefc.js";
import "./442c49ea.js";
import "./e27658d4.js";
import "./1daf7233.js";
import "./2ae2291e.js";
import "./70f9cad3.js";
import "./a789fa2f.js";
import "./a9feb7db.js";
import "./33706f61.js";
import "./325a473d.js";
import "./d3b0afba.js";
import "./36c2248f.js";
import "./fcc9e192.js";
import "./f915fcd2.js";
import "./22fa10de.js";
import "./8c51b61b.js";
import "./5230605f.js";
import "./be799735.js";
import "./0417d842.js";
import "./82a4bca8.js";
import "./71bdd49a.js";
import "./80c67fbe.js";
import "./094988fd.js";
import "./734fbd0f.js";
import "./80229ccd.js";
import "./16b77c22.js";
import "./b25be8d9.js";
import "./c504e788.js";
import "./ad2208e2.js";
import "./e5bcb9b0.js";
import "./51dea655.js";
import "./765e17fb.js";
import "./98885dce.js";
import "./9ee13d0e.js";
import "./9f467dd0.js";
import "./765a8ca5.js";
import "./3e9c3444.js";
import "./464239f7.js";
import "./608865ad.js";
import "./f0d61f39.js";
import "./5ce38ef7.js";
import "./6c369ea4.js";
import "./f52ee168.js";
import "./de2e4e92.js";
import "./bf2ffa97.js";
import "./90e6d3f7.js";
import "./3eacf14a.js";
import "./a353f3e4.js";
import "./a6ed71f1.js";
import "./dca7e75f.js";
import "./911053df.js";
import "./f26c56d5.js";
import "./842a0c82.js";
import "./9a75bf68.js";
import "./2d92f8b6.js";
import "./6a2c46a1.js";
import "./b0ddbdcf.js";
import "./ada1f35c.js";
import "./68ba86a3.js";
import "./0a417d48.js";
import "./66703dd9.js";
import "./16599e37.js";
import "./58025b46.js";
import "./8cb32968.js";
import "./2856edc9.js";
import "./7d1c8402.js";
import "./20782254.js";
import "./15a00e24.js";
import "./d3b6c877.js";
import "./29d70112.js";
const m = {
  source: "npm",
  isRemote: !0,
  packageName: "lowcoder-comps"
};
var n = {
  // Dashboards
  chart: {
    name: o("uiComp.chartCompName"),
    enName: "Chart",
    description: o("uiComp.chartCompDesc"),
    categories: ["dashboards"],
    icon: c,
    comp: e({
      ...m,
      compName: "chart"
    }),
    keywords: o("uiComp.chartCompKeywords"),
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  sharingcomponent: {
    name: o("meeting.sharingCompName"),
    enName: "Sharing",
    description: o("meeting.sharingCompName"),
    categories: ["collaboration"],
    icon: i,
    keywords: o("meeting.meetingCompKeywords"),
    // comp: VideoSharingStreamComp,
    comp: e({
      ...m,
      compName: "meetingSharing"
    }),
    withoutLoading: !0,
    layoutInfo: {
      w: 12,
      h: 50
    }
  },
  videocomponent: {
    name: o("meeting.videoCompName"),
    enName: "Video",
    description: o("meeting.videoCompName"),
    categories: ["collaboration"],
    icon: i,
    keywords: o("meeting.meetingCompKeywords"),
    // comp: VideoMeetingStreamComp,
    comp: e({
      ...m,
      compName: "meetingStream"
    }),
    withoutLoading: !0,
    layoutInfo: {
      w: 6,
      h: 32
    }
  },
  meeting: {
    name: o("meeting.meetingCompName"),
    enName: "Drawer",
    comp: e({
      ...m,
      compName: "meetingController"
    }),
    description: o("meeting.meetingCompDesc"),
    categories: ["collaboration"],
    icon: r,
    keywords: o("meeting.meetingCompKeywords"),
    // comp: VideoMeetingControllerComp,
    withoutLoading: !0
  },
  mermaid: {
    name: o("uiComp.mermaidCompName"),
    enName: "Mermaid Charts",
    comp: e({
      ...m,
      compName: "mermaid"
    }),
    description: o("uiComp.mermaidCompDesc"),
    categories: ["dashboards"],
    icon: u,
    keywords: o("uiComp.mermaidCompKeywords"),
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  timeline: {
    name: o("uiComp.timelineCompName"),
    enName: "timeline",
    description: o("uiComp.timelineCompDesc"),
    categories: ["dashboards"],
    icon: w,
    keywords: o("uiComp.timelineCompKeywords"),
    comp: Fe,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  table: {
    name: o("uiComp.tableCompName"),
    enName: "Table",
    description: o("uiComp.tableCompDesc"),
    categories: ["dashboards", "projectmanagement"],
    icon: l,
    keywords: o("uiComp.tableCompKeywords"),
    comp: de,
    layoutInfo: {
      w: 12,
      h: 40
    },
    withoutLoading: !0,
    defaultDataFn: Yo
  },
  slider: {
    name: o("uiComp.sliderCompName"),
    enName: "Slider",
    description: o("uiComp.sliderCompDesc"),
    categories: ["dashboards"],
    icon: f,
    keywords: o("uiComp.sliderCompKeywords"),
    comp: Ao,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  rangeSlider: {
    name: o("uiComp.rangeSliderCompName"),
    enName: "Range Slider",
    description: o("uiComp.rangeSliderCompDesc"),
    categories: ["dashboards"],
    icon: y,
    keywords: o("uiComp.rangeSliderCompKeywords"),
    comp: Vo,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  // Layout
  responsiveLayout: {
    name: o("uiComp.responsiveLayoutCompName"),
    enName: "Responsive Layout",
    description: o("uiComp.responsiveLayoutCompDesc"),
    categories: ["layout"],
    icon: g,
    keywords: o("uiComp.responsiveLayoutCompKeywords"),
    comp: Se,
    withoutLoading: !0,
    layoutInfo: {
      w: 24,
      h: 25,
      delayCollision: !0
    }
  },
  container: {
    name: o("uiComp.containerCompName"),
    enName: "Container",
    description: o("uiComp.containerCompDesc"),
    categories: ["layout"],
    icon: N,
    keywords: o("uiComp.containerCompKeywords"),
    comp: t,
    withoutLoading: !0,
    layoutInfo: {
      w: 12,
      h: 25,
      // static: true,
      delayCollision: !0
    },
    defaultDataFn: Io
  },
  tabbedContainer: {
    name: o("uiComp.tabbedContainerCompName"),
    enName: "Tabbed Container",
    description: o("uiComp.tabbedContainerCompDesc"),
    categories: ["layout"],
    icon: h,
    keywords: o("uiComp.tabbedContainerCompKeywords"),
    comp: zo,
    withoutLoading: !0,
    layoutInfo: {
      w: 12,
      h: 25,
      // static: true,
      delayCollision: !0
    }
  },
  collapsibleContainer: {
    name: o("uiComp.collapsibleContainerCompName"),
    enName: "Collapsible Container",
    description: o("uiComp.collapsibleContainerCompDesc"),
    categories: ["layout"],
    icon: b,
    keywords: o("uiComp.collapsibleContainerCompKeywords"),
    comp: t,
    withoutLoading: !0,
    layoutInfo: {
      w: 12,
      h: 25,
      // static: true,
      delayCollision: !0
    },
    defaultDataFn: be
  },
  listView: {
    name: o("uiComp.listViewCompName"),
    enName: "List View",
    icon: D,
    description: o("uiComp.listViewCompDesc"),
    categories: ["layout"],
    keywords: o("uiComp.listViewCompKeywords"),
    comp: ne,
    layoutInfo: {
      w: 12,
      h: 40,
      delayCollision: !0
    },
    defaultDataFn: se
  },
  grid: {
    name: o("uiComp.gridCompName"),
    enName: "Grid",
    icon: R,
    description: o("uiComp.gridCompDesc"),
    categories: ["layout"],
    keywords: o("uiComp.gridCompKeywords"),
    comp: pe,
    layoutInfo: {
      w: 12,
      h: 40,
      delayCollision: !0
    },
    defaultDataFn: te
  },
  modal: {
    name: o("uiComp.modalCompName"),
    enName: "Modal",
    icon: F,
    description: o("uiComp.modalCompDesc"),
    categories: ["layout"],
    keywords: o("uiComp.modalCompKeywords"),
    comp: k,
    withoutLoading: !0
  },
  drawer: {
    name: o("uiComp.drawerCompName"),
    enName: "Drawer",
    description: o("uiComp.drawerCompDesc"),
    categories: ["layout"],
    icon: r,
    keywords: o("uiComp.drawerCompKeywords"),
    comp: I,
    withoutLoading: !0
  },
  navigation: {
    name: o("uiComp.navigationCompName"),
    enName: "Navigation",
    description: o("uiComp.navigationCompDesc"),
    icon: K,
    categories: ["layout"],
    keywords: o("uiComp.navigationCompKeywords"),
    comp: Ce,
    layoutInfo: {
      w: 24,
      h: 5
    }
  },
  cascader: {
    name: o("uiComp.cascaderCompName"),
    enName: "Cascader",
    description: o("uiComp.cascaderCompDesc"),
    categories: ["layout"],
    icon: $,
    keywords: o("uiComp.cascaderCompKeywords"),
    comp: Jo,
    layoutInfo: {
      w: 9,
      h: 5
    }
  },
  link: {
    name: o("uiComp.linkCompName"),
    enName: "Link",
    description: o("uiComp.linkCompDesc"),
    categories: ["layout"],
    icon: S,
    keywords: o("uiComp.linkCompKeywords"),
    comp: ko,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  divider: {
    name: o("uiComp.dividerCompName"),
    enName: "Divider",
    description: o("uiComp.dividerCompDesc"),
    categories: ["layout"],
    icon: L,
    keywords: o("uiComp.dividerCompKeywords"),
    comp: Lo,
    layoutInfo: {
      w: 12,
      h: 1
    }
  },
  // Scheduling
  calendar: {
    name: o("uiComp.calendarCompName"),
    enName: "Calendar",
    description: o("uiComp.calendarCompDesc"),
    categories: ["scheduling", "projectmanagement"],
    icon: x,
    keywords: o("uiComp.calendarCompKeywords"),
    comp: e({
      ...m,
      compName: "calendar"
    }),
    layoutInfo: {
      w: 19,
      h: 60
    }
  },
  // Collaboration
  // sharingcomponent: {
  //   name: trans("meeting.sharingCompName"),
  //   enName: "Sharing",
  //   description: trans("meeting.sharingCompName"),
  //   categories: ["collaboration"],
  //   icon: VideoCompIcon,
  //   keywords: trans("meeting.meetingCompKeywords"),
  //   comp: VideoSharingStreamComp,
  //   withoutLoading: true,
  //   layoutInfo: {
  //     w: 12,
  //     h: 50,
  //   },
  // },
  // videocomponent: {
  //   name: trans("meeting.videoCompName"),
  //   enName: "Video",
  //   description: trans("meeting.videoCompName"),
  //   categories: ["collaboration"],
  //   icon: VideoCompIcon,
  //   keywords: trans("meeting.meetingCompKeywords"),
  //   comp: VideoMeetingStreamComp,
  //   withoutLoading: true,
  //   layoutInfo: {
  //     w: 6,
  //     h: 32,
  //   },
  // },
  // meeting: {
  //   name: trans("meeting.meetingCompName"),
  //   enName: "Drawer",
  //   description: trans("meeting.meetingCompDesc"),
  //   categories: ["collaboration"],
  //   icon: DrawerCompIcon,
  //   keywords: trans("meeting.meetingCompKeywords"),
  //   comp: VideoMeetingControllerComp,
  //   withoutLoading: true,
  // },
  comment: {
    name: o("uiComp.commentCompName"),
    enName: "comment",
    description: o("uiComp.commentCompDesc"),
    categories: ["forms", "collaboration"],
    icon: T,
    keywords: o("uiComp.commentCompKeywords"),
    comp: ke,
    layoutInfo: {
      w: 13,
      h: 55
    }
  },
  mention: {
    name: o("uiComp.mentionCompName"),
    enName: "mention",
    description: o("uiComp.mentionCompDesc"),
    categories: ["forms", "collaboration"],
    icon: v,
    keywords: o("uiComp.mentionCompKeywords"),
    comp: Ie
  },
  // Forms
  form: {
    name: o("uiComp.formCompName"),
    enName: "Form",
    description: o("uiComp.formCompDesc"),
    categories: ["forms"],
    icon: E,
    keywords: o("uiComp.formCompKeywords"),
    comp: ie,
    withoutLoading: !0,
    layoutInfo: {
      w: 12,
      h: 50,
      // static: true,
      delayCollision: !0
    },
    defaultDataFn: re
  },
  jsonSchemaForm: {
    name: o("uiComp.jsonSchemaFormCompName"),
    enName: "JSON Schema Form",
    description: o("uiComp.jsonSchemaFormCompDesc"),
    categories: ["forms"],
    icon: j,
    keywords: o("uiComp.jsonSchemaFormCompKeywords"),
    comp: Eo,
    layoutInfo: {
      w: 12,
      h: 50
    }
  },
  jsonEditor: {
    name: o("uiComp.jsonEditorCompName"),
    enName: "JSON Editor",
    description: o("uiComp.jsonEditorCompDesc"),
    categories: ["forms"],
    icon: V,
    keywords: o("uiComp.jsonEditorCompKeywords"),
    comp: we,
    layoutInfo: {
      w: 12,
      h: 50
    }
  },
  jsonExplorer: {
    name: o("uiComp.jsonExplorerCompName"),
    enName: "JSON Explorer",
    description: o("uiComp.jsonExplorerCompDesc"),
    categories: ["forms"],
    icon: A,
    keywords: o("uiComp.jsonExplorerCompKeywords"),
    comp: ue,
    layoutInfo: {
      w: 12,
      h: 50
    }
  },
  richTextEditor: {
    name: o("uiComp.richTextEditorCompName"),
    enName: "Rich Text Editor",
    categories: ["forms"],
    description: o("uiComp.richTextEditorCompDesc"),
    icon: B,
    keywords: o("uiComp.richTextEditorCompKeywords"),
    comp: Ho,
    layoutInfo: {
      w: 12,
      h: 50
    }
  },
  input: {
    name: o("uiComp.inputCompName"),
    enName: "Input",
    description: o("uiComp.inputCompDesc"),
    categories: ["forms"],
    icon: M,
    keywords: o("uiComp.inputCompKeywords"),
    comp: Zo,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  password: {
    name: o("uiComp.passwordCompName"),
    enName: "Password",
    description: o("uiComp.passwordCompDesc"),
    categories: ["forms"],
    icon: P,
    keywords: o("uiComp.passwordCompKeywords"),
    comp: _o,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  numberInput: {
    name: o("uiComp.numberInputCompName"),
    enName: "Number Input",
    description: o("uiComp.numberInputCompDesc"),
    categories: ["forms"],
    icon: H,
    keywords: o("uiComp.numberInputCompKeywords"),
    comp: jo,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  textArea: {
    name: o("uiComp.textAreaCompName"),
    enName: "Text Area",
    description: o("uiComp.textAreaCompDesc"),
    categories: ["forms"],
    icon: J,
    keywords: o("uiComp.textAreaCompKeywords"),
    comp: oe,
    layoutInfo: {
      w: 6,
      h: 12
    }
  },
  switch: {
    name: o("uiComp.switchCompName"),
    enName: "Switch",
    description: o("uiComp.switchCompDesc"),
    categories: ["forms"],
    icon: q,
    keywords: o("uiComp.switchCompKeywords"),
    comp: Wo,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  checkbox: {
    name: o("uiComp.checkboxCompName"),
    enName: "Checkbox",
    description: o("uiComp.checkboxCompDesc"),
    categories: ["forms"],
    icon: G,
    keywords: o("uiComp.checkboxCompKeywords"),
    comp: qo,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  radio: {
    name: o("uiComp.radioCompName"),
    enName: "Radio",
    description: o("uiComp.radioCompDesc"),
    categories: ["forms"],
    icon: O,
    keywords: o("uiComp.radioCompKeywords"),
    comp: Oo,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  date: {
    name: o("uiComp.dateCompName"),
    enName: "Date",
    description: o("uiComp.dateCompDesc"),
    categories: ["forms", "scheduling"],
    icon: U,
    keywords: o("uiComp.dateCompKeywords"),
    comp: $o,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  dateRange: {
    name: o("uiComp.dateRangeCompName"),
    enName: "Date Range",
    description: o("uiComp.dateRangeCompDesc"),
    categories: ["forms", "scheduling"],
    icon: Q,
    keywords: o("uiComp.dateRangeCompKeywords"),
    comp: So,
    layoutInfo: {
      w: 12,
      h: 6
    }
  },
  time: {
    name: o("uiComp.timeCompName"),
    enName: "Time",
    description: o("uiComp.timeCompDesc"),
    categories: ["forms", "scheduling"],
    icon: W,
    keywords: o("uiComp.timeCompKeywords"),
    comp: ee,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  timeRange: {
    name: o("uiComp.timeRangeCompName"),
    enName: "Time Range",
    categories: ["forms", "scheduling"],
    description: o("uiComp.timeRangeCompDesc"),
    icon: Y,
    keywords: o("uiComp.timeRangeCompKeywords"),
    comp: me,
    layoutInfo: {
      w: 12,
      h: 6
    }
  },
  button: {
    name: o("uiComp.buttonCompName"),
    enName: "Button",
    description: o("uiComp.buttonCompDesc"),
    categories: ["forms"],
    icon: a,
    keywords: o("uiComp.buttonCompKeywords"),
    comp: Ro,
    layoutInfo: {
      w: 6,
      h: 6
    },
    withoutLoading: !0
  },
  controlButton: {
    name: o("meeting.meetingControlCompName"),
    enName: "Controls",
    description: o("meeting.meetingCompDesc"),
    categories: ["forms", "collaboration"],
    icon: a,
    keywords: o("meeting.meetingCompKeywords"),
    comp: Le,
    withoutLoading: !0,
    layoutInfo: {
      w: 3,
      h: 6
    }
  },
  dropdown: {
    name: o("uiComp.dropdownCompName"),
    enName: "Dropdown",
    description: o("uiComp.dropdownCompDesc"),
    categories: ["forms"],
    icon: z,
    keywords: o("uiComp.dropdownCompKeywords"),
    comp: Fo,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  toggleButton: {
    name: o("uiComp.toggleButtonCompName"),
    enName: "Toggle Button",
    description: o("uiComp.toggleButtonCompDesc"),
    categories: ["forms"],
    icon: X,
    keywords: o("uiComp.toggleButtonCompKeywords"),
    comp: he,
    layoutInfo: {
      w: 3,
      h: 6
    }
  },
  segmentedControl: {
    name: o("uiComp.segmentedControlCompName"),
    enName: "Segmented Control",
    description: o("uiComp.segmentedControlCompDesc"),
    categories: ["forms"],
    icon: Z,
    keywords: o("uiComp.segmentedControlCompKeywords"),
    comp: Uo,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  rating: {
    name: o("uiComp.ratingCompName"),
    enName: "Rating",
    description: o("uiComp.ratingCompDesc"),
    categories: ["forms"],
    icon: _,
    keywords: o("uiComp.ratingCompKeywords"),
    comp: Po,
    layoutInfo: {
      w: 6,
      h: 6
    }
  },
  autocomplete: {
    name: o("uiComp.autoCompleteCompName"),
    enName: "autoComplete",
    description: o("uiComp.autoCompleteCompDesc"),
    categories: ["forms", "collaboration"],
    icon: oo,
    keywords: eo.spell(o("uiComp.autoCompleteCompName"), "first", "low").toString(),
    comp: Ke,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  // Project Management
  progress: {
    name: o("uiComp.progressCompName"),
    enName: "Progress",
    description: o("uiComp.progressCompDesc"),
    categories: ["dashboards", "projectmanagement"],
    icon: mo,
    keywords: o("uiComp.progressCompKeywords"),
    comp: Mo,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  progressCircle: {
    name: o("uiComp.progressCircleCompName"),
    enName: "Process Circle",
    description: o("uiComp.progressCircleCompDesc"),
    categories: ["dashboards", "projectmanagement"],
    icon: io,
    keywords: o("uiComp.progressCircleCompKeywords"),
    comp: Bo,
    layoutInfo: {
      w: 6,
      h: 20
    }
  },
  // Document handling
  file: {
    name: o("uiComp.fileUploadCompName"),
    enName: "File Upload",
    description: o("uiComp.fileUploadCompDesc"),
    categories: ["documents"],
    icon: ro,
    keywords: o("uiComp.fileUploadCompKeywords"),
    comp: xo,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  fileViewer: {
    name: o("uiComp.fileViewerCompName"),
    enName: "File Viewer",
    description: o("uiComp.fileViewerCompDesc"),
    categories: ["documents"],
    icon: ao,
    keywords: o("uiComp.fileViewerCompKeywords"),
    comp: To,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  // Multimedia
  image: {
    name: o("uiComp.imageCompName"),
    enName: "Image",
    description: o("uiComp.imageCompDesc"),
    categories: ["multimedia"],
    icon: po,
    keywords: o("uiComp.imageCompKeywords"),
    comp: vo,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  carousel: {
    name: o("uiComp.carouselCompName"),
    enName: "Carousel",
    description: o("uiComp.carouselCompDesc"),
    categories: ["multimedia"],
    icon: to,
    keywords: o("uiComp.drawerCompKeywords"),
    comp: Ne,
    withoutLoading: !0,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  audio: {
    name: o("uiComp.audioCompName"),
    enName: "Audio",
    description: o("uiComp.audioCompDesc"),
    categories: ["multimedia"],
    icon: no,
    keywords: o("uiComp.audioCompKeywords"),
    comp: ye,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  video: {
    name: o("uiComp.videoCompName"),
    enName: "Video",
    description: o("uiComp.videoCompDesc"),
    categories: ["multimedia"],
    icon: i,
    keywords: o("uiComp.videoCompKeywords"),
    comp: ge,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  jsonLottie: {
    name: o("uiComp.jsonLottieCompName"),
    enName: "Lottie Animation",
    description: o("uiComp.jsonLottieCompDesc"),
    categories: ["multimedia"],
    icon: so,
    keywords: o("uiComp.jsonLottieCompKeywords"),
    comp: $e,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  imageEditor: {
    name: o("uiComp.imageEditorCompName"),
    enName: "Image Editor",
    comp: e({
      ...m,
      compName: "imageEditor"
    }),
    description: o("uiComp.imageEditorCompDesc"),
    categories: ["multimedia"],
    icon: Co,
    keywords: o("uiComp.imageEditorCompKeywords"),
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  // item Handling
  qrCode: {
    name: o("uiComp.qrCodeCompName"),
    enName: "QR Code",
    description: o("uiComp.qrCodeCompDesc"),
    categories: ["itemHandling", "documents"],
    icon: co,
    keywords: o("uiComp.qrCodeCompKeywords"),
    comp: ce,
    layoutInfo: {
      w: 6,
      h: 32
    }
  },
  scanner: {
    name: o("uiComp.scannerCompName"),
    enName: "Scanner",
    description: o("uiComp.scannerCompDesc"),
    categories: ["itemHandling"],
    icon: uo,
    keywords: o("uiComp.scannerCompKeywords"),
    comp: De,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  signature: {
    name: o("uiComp.signatureCompName"),
    enName: "Signature",
    description: o("uiComp.signatureCompDesc"),
    categories: ["itemHandling"],
    icon: wo,
    keywords: o("uiComp.signatureCompKeywords"),
    comp: Re,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  select: {
    name: o("uiComp.selectCompName"),
    enName: "Select",
    description: o("uiComp.selectCompDesc"),
    categories: ["forms", "itemHandling"],
    icon: lo,
    keywords: o("uiComp.selectCompKeywords"),
    comp: Qo,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  multiSelect: {
    name: o("uiComp.multiSelectCompName"),
    enName: "Multiselect",
    description: o("uiComp.multiSelectCompDesc"),
    categories: ["forms", "itemHandling"],
    icon: fo,
    keywords: o("uiComp.multiSelectCompKeywords"),
    comp: Go,
    layoutInfo: {
      w: 6,
      h: 5
    }
  },
  tree: {
    name: o("uiComp.treeCompName"),
    enName: "Tree",
    description: o("uiComp.treeCompDesc"),
    categories: ["layout", "itemHandling", "documents"],
    icon: yo,
    keywords: o("uiComp.treeCompKeywords"),
    comp: le,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  treeSelect: {
    name: o("uiComp.treeSelectCompName"),
    enName: "Tree Select",
    description: o("uiComp.treeSelectCompDesc"),
    categories: ["layout", "itemHandling", "documents"],
    icon: go,
    keywords: o("uiComp.treeSelectCompKeywords"),
    comp: fe,
    layoutInfo: {
      w: 12,
      h: 5
    }
  },
  // Integration
  iframe: {
    name: o("uiComp.iframeCompName"),
    enName: "IFrame",
    description: o("uiComp.iframeCompDesc"),
    icon: No,
    categories: ["integration"],
    keywords: o("uiComp.iframeCompKeywords"),
    comp: ae,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  custom: {
    name: o("uiComp.customCompName"),
    enName: "Custom Component",
    description: o("uiComp.customCompDesc"),
    icon: p,
    categories: ["integration"],
    keywords: o("uiComp.customCompKeywords"),
    comp: Ko,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  module: {
    name: o("uiComp.moduleCompName"),
    enName: "Module",
    icon: p,
    description: o("uiComp.moduleCompDesc"),
    categories: [],
    keywords: o("uiComp.moduleCompKeywords"),
    comp: ho,
    layoutInfo: {
      w: 12,
      h: 40
    }
  },
  // mixed
  text: {
    name: o("uiComp.textCompName"),
    enName: "Text",
    description: o("uiComp.textCompDesc"),
    categories: ["dashboards", "layout", "multimedia"],
    icon: bo,
    keywords: o("uiComp.textCompKeywords"),
    comp: Xo,
    layoutInfo: {
      w: 6,
      h: 24
    }
  }
};
function Nr() {
  if (!n)
    return;
  const s = Object.entries(n);
  for (const [C, d] of s)
    Do(C, d);
}
export {
  Nr as loadComps
};
