// lowcoder-sdk/dist/14897c7e.js
import { lx as o, i6 as g } from "./c988e26e.js";
var j = typeof o == "object" && o && o.Object === Object && o, v = j, T = v, y = typeof self == "object" && self && self.Object === Object && self, O = T || y || Function("return this")(), S = O, u = S, d = u.Symbol, c = d, a = c, b = Object.prototype, p = b.hasOwnProperty, m = b.toString, e = a ? a.toStringTag : void 0;
function $(t) {
  var r = p.call(t, e), s = t[e];
  try {
    t[e] = void 0;
    var f = !0;
  } catch {
  }
  var l = m.call(t);
  return f && (r ? t[e] = s : delete t[e]), l;
}
var _ = $, w = Object.prototype, G = w.toString;
function h(t) {
  return G.call(t);
}
var P = h, n = c, x = _, R = P, k = "[object Null]", F = "[object Undefined]", i = n ? n.toStringTag : void 0;
function L(t) {
  return t == null ? t === void 0 ? F : k : i && i in Object(t) ? x(t) : R(t);
}
var q = L;
function C(t) {
  return t != null && typeof t == "object";
}
var z = C;
function D(t) {
  var r = typeof t;
  return t != null && (r == "object" || r == "function");
}
var E = D;
const A = /* @__PURE__ */ g(E);
export {
  q as _,
  E as a,
  c as b,
  S as c,
  v as d,
  A as e,
  z as i
};
