// lowcoder-sdk/dist/15a00e24.js
import { a8 as m, Y as t, F as n, fd as r } from "./c988e26e.js";
import "react";
import "react-dom";
const p = /* @__PURE__ */ m("li", {
  children: [t("comment.Introduction"), ":", /* @__PURE__ */ n("br", {}), "1. user - ", t("comment.helpUser"), /* @__PURE__ */ n("br", {}), " .name - ", t("comment.helpname"), /* @__PURE__ */ n("br", {}), " .avatar - ", t("comment.helpavatar"), /* @__PURE__ */ n("br", {}), " .displayName - ", t("comment.helpdisplayName"), /* @__PURE__ */ n("br", {}), "2. value - ", t("comment.helpvalue"), /* @__PURE__ */ n("br", {}), "3. createdAt - ", t("comment.helpcreatedAt")]
}), h = /* @__PURE__ */ m("li", {
  children: [t("comment.Introduction"), ":", /* @__PURE__ */ n("br", {}), "1.name - ", t("comment.helpname"), /* @__PURE__ */ n("br", {}), "2.avatar - ", t("comment.helpavatar"), /* @__PURE__ */ n("br", {}), "3.displayName - ", t("comment.helpdisplayName")]
}), v = [{
  user: {
    name: "Li Lei",
    avatar: "https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
  },
  value: "What is the use of this component?",
  createdAt: "2023-06-15T08:40:41.658Z"
}, {
  user: {
    name: "mou"
  },
  value: "This component allows you to post or delete comments, as well as mention people who appear in the chat.",
  createdAt: "2023-06-16T08:43:42.658Z"
}, {
  user: {
    name: "Han Meimei",
    displayName: "Han"
  },
  value: "I want to give it a try",
  createdAt: "2023-06-17T08:49:01.658Z"
}, {
  user: {
    name: "mou"
  },
  value: "Enter the content in the input box below and press shift+enter to send it immediately",
  createdAt: "2023-06-18T08:50:11.658Z"
}];
function f(e) {
  return e === "" ? [] : i(e) ?? [];
}
function i(e, o) {
  return r(e, ["array", "undefined"], o, (a, c) => (r(a, ["object"], c), r(a.user, ["object"], "user"), r(a.value, ["string", "undefined"], "value"), r(a.createdAt, ["string", "undefined"], "createdAt"), a));
}
function b(e) {
  return r(e == null ? void 0 : e.name, ["string"], "name"), r(e == null ? void 0 : e.avatar, ["string", "undefined"], "avatar"), e;
}
function d(e) {
  if (e === "")
    return {};
  for (const o in e)
    r(e[o], ["array"], o, (a) => (r(a, ["string"]), a));
  return e;
}
export {
  p as CommentDataTooltip,
  h as CommentUserDataTooltip,
  d as checkMentionListData,
  b as checkUserInfoData,
  v as commentDate,
  f as convertCommentData
};
