// lowcoder-sdk/dist/15c1fb73.js
import { mJ as F, i6 as L } from "./c988e26e.js";
import M from "react";
function k(u, l) {
  for (var o = 0; o < l.length; o++) {
    const f = l[o];
    if (typeof f != "string" && !Array.isArray(f)) {
      for (const a in f)
        if (a !== "default" && !(a in u)) {
          const i = Object.getOwnPropertyDescriptor(f, a);
          i && Object.defineProperty(u, a, i.get ? i : {
            enumerable: !0,
            get: () => f[a]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(u, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, y = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var j = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1168.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z" } }] }, name: "linkedin", theme: "filled" };
_.default = j;
(function(u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), Object.defineProperty(u, "default", {
    enumerable: !0,
    get: function() {
      return D;
    }
  });
  var l = /* @__PURE__ */ P(M), o = /* @__PURE__ */ i(_), f = /* @__PURE__ */ i(F);
  function a(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function i(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function P(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, c = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var d in e)
      if (d !== "default" && Object.prototype.hasOwnProperty.call(e, d)) {
        var p = c ? Object.getOwnPropertyDescriptor(e, d) : null;
        p && (p.get || p.set) ? Object.defineProperty(n, d, p) : n[d] = e[d];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(c) {
        return Object.getOwnPropertyDescriptor(r, c).enumerable;
      }))), n.forEach(function(c) {
        a(e, c, r[c]);
      });
    }
    return e;
  }
  function b(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(c) {
        return Object.getOwnPropertyDescriptor(e, c).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function h(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : b(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ l.createElement(f.default, h(m({}, e), {
      ref: t,
      icon: o.default
    }));
  }, v = /* @__PURE__ */ l.forwardRef(w);
  process.env.NODE_ENV !== "production" && (v.displayName = "LinkedinFilled");
  var D = v;
})(y);
(function(u, l) {
  Object.defineProperty(l, "__esModule", {
    value: !0
  }), l.default = void 0;
  const o = f(y);
  function f(i) {
    return i && i.__esModule ? i : { default: i };
  }
  const a = o;
  l.default = a, u.exports = a;
})(s, s.exports);
var g = s.exports;
const E = /* @__PURE__ */ L(g), z = /* @__PURE__ */ k({
  __proto__: null,
  default: E
}, [g]);
export {
  z as L
};
