// lowcoder-sdk/dist/182e4b4f.js
import { i6 as Q } from "./c988e26e.js";
import { h as X, j as Z, e as k, k as U, l as F, m as z, n as ee, o as M, p as re, s as te, q as ae, a as ne, d as A, r as K, t as D, c as oe, i as se, b as ie, g as ce } from "./8f0aa994.js";
import { a as T, c as fe, b as le, i as N } from "./14897c7e.js";
var ye = X, ge = ye(Object.getPrototypeOf, Object), G = ge;
function ue(e, r) {
  for (var t = -1, n = e == null ? 0 : e.length; ++t < n && r(e[t], t, e) !== !1; )
    ;
  return e;
}
var be = ue, pe = Z, ve = function() {
  try {
    var e = pe(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}(), $e = ve, I = $e;
function _e(e, r, t) {
  r == "__proto__" && I ? I(e, r, {
    configurable: !0,
    enumerable: !0,
    value: t,
    writable: !0
  }) : e[r] = t;
}
var V = _e, Ae = V, Te = k, je = Object.prototype, de = je.hasOwnProperty;
function Oe(e, r, t) {
  var n = e[r];
  (!(de.call(e, r) && Te(n, t)) || t === void 0 && !(r in e)) && Ae(e, r, t);
}
var R = Oe, Se = R, Ie = V;
function he(e, r, t, n) {
  var f = !t;
  t || (t = {});
  for (var i = -1, o = r.length; ++i < o; ) {
    var s = r[i], c = n ? n(t[s], e[s], s, t, e) : void 0;
    c === void 0 && (c = e[s]), f ? Ie(t, s, c) : Se(t, s, c);
  }
  return t;
}
var _ = he, me = _, we = U;
function Pe(e, r) {
  return e && me(r, we(r), e);
}
var Ce = Pe;
function Ee(e) {
  var r = [];
  if (e != null)
    for (var t in Object(e))
      r.push(t);
  return r;
}
var xe = Ee, Be = T, Le = F, Ue = xe, Fe = Object.prototype, Me = Fe.hasOwnProperty;
function Ke(e) {
  if (!Be(e))
    return Ue(e);
  var r = Le(e), t = [];
  for (var n in e)
    n == "constructor" && (r || !Me.call(e, n)) || t.push(n);
  return t;
}
var De = Ke, Ne = z, Ge = De, Ve = ee;
function Re(e) {
  return Ve(e) ? Ne(e, !0) : Ge(e);
}
var j = Re, qe = _, Ye = j;
function We(e, r) {
  return e && qe(r, Ye(r), e);
}
var He = We, $ = { exports: {} };
$.exports;
(function(e, r) {
  var t = fe, n = r && !r.nodeType && r, f = n && !0 && e && !e.nodeType && e, i = f && f.exports === n, o = i ? t.Buffer : void 0, s = o ? o.allocUnsafe : void 0;
  function c(u, p) {
    if (p)
      return u.slice();
    var l = u.length, b = s ? s(l) : new u.constructor(l);
    return u.copy(b), b;
  }
  e.exports = c;
})($, $.exports);
var Je = $.exports;
function Qe(e, r) {
  var t = -1, n = e.length;
  for (r || (r = Array(n)); ++t < n; )
    r[t] = e[t];
  return r;
}
var Xe = Qe, Ze = _, ke = M;
function ze(e, r) {
  return Ze(e, ke(e), r);
}
var er = ze, rr = re, tr = G, ar = M, nr = te, or = Object.getOwnPropertySymbols, sr = or ? function(e) {
  for (var r = []; e; )
    rr(r, ar(e)), e = tr(e);
  return r;
} : nr, q = sr, ir = _, cr = q;
function fr(e, r) {
  return ir(e, cr(e), r);
}
var lr = fr, yr = ae, gr = q, ur = j;
function br(e) {
  return yr(e, ur, gr);
}
var pr = br, vr = Object.prototype, $r = vr.hasOwnProperty;
function _r(e) {
  var r = e.length, t = new e.constructor(r);
  return r && typeof e[0] == "string" && $r.call(e, "index") && (t.index = e.index, t.input = e.input), t;
}
var Ar = _r, h = ne;
function Tr(e) {
  var r = new e.constructor(e.byteLength);
  return new h(r).set(new h(e)), r;
}
var d = Tr, jr = d;
function dr(e, r) {
  var t = r ? jr(e.buffer) : e.buffer;
  return new e.constructor(t, e.byteOffset, e.byteLength);
}
var Or = dr, Sr = /\w*$/;
function Ir(e) {
  var r = new e.constructor(e.source, Sr.exec(e));
  return r.lastIndex = e.lastIndex, r;
}
var hr = Ir, m = le, w = m ? m.prototype : void 0, P = w ? w.valueOf : void 0;
function mr(e) {
  return P ? Object(P.call(e)) : {};
}
var wr = mr, Pr = d;
function Cr(e, r) {
  var t = r ? Pr(e.buffer) : e.buffer;
  return new e.constructor(t, e.byteOffset, e.length);
}
var Er = Cr, xr = d, Br = Or, Lr = hr, Ur = wr, Fr = Er, Mr = "[object Boolean]", Kr = "[object Date]", Dr = "[object Map]", Nr = "[object Number]", Gr = "[object RegExp]", Vr = "[object Set]", Rr = "[object String]", qr = "[object Symbol]", Yr = "[object ArrayBuffer]", Wr = "[object DataView]", Hr = "[object Float32Array]", Jr = "[object Float64Array]", Qr = "[object Int8Array]", Xr = "[object Int16Array]", Zr = "[object Int32Array]", kr = "[object Uint8Array]", zr = "[object Uint8ClampedArray]", et = "[object Uint16Array]", rt = "[object Uint32Array]";
function tt(e, r, t) {
  var n = e.constructor;
  switch (r) {
    case Yr:
      return xr(e);
    case Mr:
    case Kr:
      return new n(+e);
    case Wr:
      return Br(e, t);
    case Hr:
    case Jr:
    case Qr:
    case Xr:
    case Zr:
    case kr:
    case zr:
    case et:
    case rt:
      return Fr(e, t);
    case Dr:
      return new n();
    case Nr:
    case Rr:
      return new n(e);
    case Gr:
      return Lr(e);
    case Vr:
      return new n();
    case qr:
      return Ur(e);
  }
}
var at = tt, nt = T, C = Object.create, ot = function() {
  function e() {
  }
  return function(r) {
    if (!nt(r))
      return {};
    if (C)
      return C(r);
    e.prototype = r;
    var t = new e();
    return e.prototype = void 0, t;
  };
}(), st = ot, it = st, ct = G, ft = F;
function lt(e) {
  return typeof e.constructor == "function" && !ft(e) ? it(ct(e)) : {};
}
var yt = lt, gt = A, ut = N, bt = "[object Map]";
function pt(e) {
  return ut(e) && gt(e) == bt;
}
var vt = pt, $t = vt, _t = D, E = K, x = E && E.isMap, At = x ? _t(x) : $t, Tt = At, jt = A, dt = N, Ot = "[object Set]";
function St(e) {
  return dt(e) && jt(e) == Ot;
}
var It = St, ht = It, mt = D, B = K, L = B && B.isSet, wt = L ? mt(L) : ht, Pt = wt, Ct = oe, Et = be, xt = R, Bt = Ce, Lt = He, Ut = Je, Ft = Xe, Mt = er, Kt = lr, Dt = ie, Nt = pr, Gt = A, Vt = Ar, Rt = at, qt = yt, Yt = ce, Wt = se, Ht = Tt, Jt = T, Qt = Pt, Xt = U, Zt = j, kt = 1, zt = 2, ea = 4, Y = "[object Arguments]", ra = "[object Array]", ta = "[object Boolean]", aa = "[object Date]", na = "[object Error]", W = "[object Function]", oa = "[object GeneratorFunction]", sa = "[object Map]", ia = "[object Number]", H = "[object Object]", ca = "[object RegExp]", fa = "[object Set]", la = "[object String]", ya = "[object Symbol]", ga = "[object WeakMap]", ua = "[object ArrayBuffer]", ba = "[object DataView]", pa = "[object Float32Array]", va = "[object Float64Array]", $a = "[object Int8Array]", _a = "[object Int16Array]", Aa = "[object Int32Array]", Ta = "[object Uint8Array]", ja = "[object Uint8ClampedArray]", da = "[object Uint16Array]", Oa = "[object Uint32Array]", a = {};
a[Y] = a[ra] = a[ua] = a[ba] = a[ta] = a[aa] = a[pa] = a[va] = a[$a] = a[_a] = a[Aa] = a[sa] = a[ia] = a[H] = a[ca] = a[fa] = a[la] = a[ya] = a[Ta] = a[ja] = a[da] = a[Oa] = !0;
a[na] = a[W] = a[ga] = !1;
function v(e, r, t, n, f, i) {
  var o, s = r & kt, c = r & zt, u = r & ea;
  if (t && (o = f ? t(e, n, f, i) : t(e)), o !== void 0)
    return o;
  if (!Jt(e))
    return e;
  var p = Yt(e);
  if (p) {
    if (o = Vt(e), !s)
      return Ft(e, o);
  } else {
    var l = Gt(e), b = l == W || l == oa;
    if (Wt(e))
      return Ut(e, s);
    if (l == H || l == Y || b && !f) {
      if (o = c || b ? {} : qt(e), !s)
        return c ? Kt(e, Lt(o, e)) : Mt(e, Bt(o, e));
    } else {
      if (!a[l])
        return f ? e : {};
      o = Rt(e, l, s);
    }
  }
  i || (i = new Ct());
  var O = i.get(e);
  if (O)
    return O;
  i.set(e, o), Qt(e) ? e.forEach(function(y) {
    o.add(v(y, r, t, y, e, i));
  }) : Ht(e) && e.forEach(function(y, g) {
    o.set(g, v(y, r, t, g, e, i));
  });
  var J = u ? c ? Nt : Dt : c ? Zt : Xt, S = p ? void 0 : J(e);
  return Et(S || e, function(y, g) {
    S && (g = y, y = e[g]), xt(o, g, v(y, r, t, g, e, i));
  }), o;
}
var Sa = v, Ia = Sa, ha = 1, ma = 4;
function wa(e) {
  return Ia(e, ha | ma);
}
var Pa = wa;
const Ba = /* @__PURE__ */ Q(Pa);
export {
  G as _,
  $e as a,
  _ as b,
  Ba as c,
  pr as d,
  Sa as e,
  R as f,
  be as g,
  st as h,
  V as i,
  Je as j,
  j as k,
  Er as l,
  Xe as m,
  yt as n,
  Pa as o
};
