// lowcoder-sdk/dist/1c23a771.js
import { mJ as C, i6 as M } from "./c988e26e.js";
import D from "react";
function j(c, u) {
  for (var l = 0; l < u.length; l++) {
    const o = u[l];
    if (typeof o != "string" && !Array.isArray(o)) {
      for (const i in o)
        if (i !== "default" && !(i in c)) {
          const a = Object.getOwnPropertyDescriptor(o, i);
          a && Object.defineProperty(c, i, a.get ? a : {
            enumerable: !0,
            get: () => o[i]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(c, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, m = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M912 302.3L784 376V224c0-35.3-28.7-64-64-64H128c-35.3 0-64 28.7-64 64v576c0 35.3 28.7 64 64 64h592c35.3 0 64-28.7 64-64V648l128 73.7c21.3 12.3 48-3.1 48-27.6V330c0-24.6-26.7-40-48-27.7zM712 792H136V232h576v560zm176-167l-104-59.8V458.9L888 399v226zM208 360h112c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H208c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z" } }] }, name: "video-camera", theme: "outlined" };
s.default = E;
(function(c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), Object.defineProperty(c, "default", {
    enumerable: !0,
    get: function() {
      return V;
    }
  });
  var u = /* @__PURE__ */ g(D), l = /* @__PURE__ */ a(s), o = /* @__PURE__ */ a(C);
  function i(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function g(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var d in e)
      if (d !== "default" && Object.prototype.hasOwnProperty.call(e, d)) {
        var p = f ? Object.getOwnPropertyDescriptor(e, d) : null;
        p && (p.get || p.set) ? Object.defineProperty(n, d, p) : n[d] = e[d];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(f) {
        return Object.getOwnPropertyDescriptor(r, f).enumerable;
      }))), n.forEach(function(f) {
        i(e, f, r[f]);
      });
    }
    return e;
  }
  function b(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function w(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : b(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var h = function(e, t) {
    return /* @__PURE__ */ u.createElement(o.default, w(P({}, e), {
      ref: t,
      icon: l.default
    }));
  }, v = /* @__PURE__ */ u.forwardRef(h);
  process.env.NODE_ENV !== "production" && (v.displayName = "VideoCameraOutlined");
  var V = v;
})(m);
(function(c, u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), u.default = void 0;
  const l = o(m);
  function o(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const i = l;
  u.default = i, c.exports = i;
})(O, O.exports);
var y = O.exports;
const S = /* @__PURE__ */ M(y), N = /* @__PURE__ */ j({
  __proto__: null,
  default: S
}, [y]);
export {
  N as V
};
