// lowcoder-sdk/dist/1c5ee0c7.js
const e = {
  ui: {
    layout: [{
      w: 4,
      h: 5,
      x: 15,
      y: 0,
      i: "f1a550ef",
      moved: !1,
      static: !1,
      isDraggable: !0,
      resizeHandles: []
    }, {
      w: 6,
      h: 5,
      x: 8,
      y: 2,
      i: "91816a3",
      moved: !1,
      static: !1,
      isDraggable: !0,
      resizeHandles: []
    }, {
      w: 6,
      h: 5,
      x: 0,
      y: 2,
      i: "18f4f3f6",
      moved: !1,
      static: !1,
      isDraggable: !0,
      resizeHandles: []
    }, {
      w: 18,
      h: 140,
      x: 0,
      y: 11,
      i: "5ea4f2dc",
      moved: !1,
      static: !1,
      isDraggable: !0,
      resizeHandles: []
    }],
    items: {
      f1a550ef: {
        compType: "button",
        comp: {
          buttonType: "default",
          text: "button",
          onClick: {
            compType: "empty",
            comp: {}
          },
          hidden: "{{switch1.checked}}"
        },
        name: "button1"
      },
      "91816a3": {
        compType: "switch",
        comp: {
          checked: {
            value: !1,
            defaultValue: ""
          },
          label: {
            text: "Label",
            tooltip: "",
            align: "right"
          },
          disabled: !1
        },
        name: "switch1"
      },
      "18f4f3f6": {
        compType: "select",
        comp: {
          value: {
            value: "",
            defaultValue: "number1"
          },
          labels: '["option1","option2","option3"]',
          values: '["number1","number2","number3"]',
          label: {
            text: "label",
            tooltip: "",
            align: "right"
          },
          placeholder: ""
        },
        name: "select1"
      },
      "5ea4f2dc": {
        compType: "table",
        comp: {
          hideBordered: !1,
          hideHeader: !1,
          data: `[
 {
  "ID": 1,
  "name": "a",
  "date": "2021-05-17",
  "mail": "romario70@qq.com",
  "phone": "17137564247",
  "sex": "male"
 },
 {
  "ID": 2,
  "name": "b",
  "date": "2019-07-16",
  "mail": "lovely.luzixuan@163.com",
  "phone": "17110230998",
  "sex": "female"
 },
 {
  "ID": 3,
  "name": "b",
  "date": "2020-12-27",
  "mail": "gisel_bin34@qq.com",
  "phone": "18875259170",
  "sex": "male"
 },
 {
  "ID": 4,
  "name": "c",
  "date": "2019-11-28",
  "mail": "wilma68@163.com",
  "phone": "17111461612",
  "sex": "male"
 },
 {
  "ID": 5,
  "name": "d",
  "date": "2019-12-03",
  "mail": "silencer_gong@gmail.com",
  "phone": "16683012636",
  "sex": "male"
 },
 {
  "ID": 6,
  "name": "e",
  "date": "2019-06-03",
  "mail": "liu_reynolds34@126.com",
  "phone": "18888119131",
  "sex": "female"
 },
 {
  "ID": 7,
  "name": "f",
  "date": "2020-02-09",
  "mail": "realmadrID_xiao1995@gmail.com",
  "phone": "18871452348",
  "sex": "female"
 },
 {
  "name": "g",
  "date": "2019-07-14",
  "mail": "rachelle5@gmail.com",
  "phone": "13254036165",
  "sex": "male",
  "ID": 8
 },
 {
  "name": "h",
  "date": "2019-07-11",
  "mail": "mingzhe67@sina.com",
  "phone": "17155089400",
  "sex": "female",
  "ID": 9
 },
 {
  "name": "i",
  "date": "2021-05-29",
  "mail": "zelma.dai@hotmail.com",
  "phone": "18849094541",
  "sex": "female",
  "ID": 10
 },
 {
  "name": "j",
  "date": "2020-05-02",
  "mail": "joannie_beatty88@gmail.com",
  "phone": "13234021406",
  "sex": "female",
  "ID": 11
 },
 {
  "name": "k",
  "date": "2022-03-08",
  "mail": "carlos_crona66@sina.com",
  "phone": "13256894564",
  "sex": "female",
  "ID": 12
 },
 {
  "name": "l",
  "date": "2019-11-28",
  "mail": "garland_labadie@sohu.com",
  "phone": "13104286197",
  "sex": "male",
  "ID": 13
 },
 {
  "name": "m",
  "date": "2021-01-02",
  "mail": "mitchel75@sohu.com",
  "phone": "13121266906",
  "sex": "female",
  "ID": 14
 },
 {
  "name": "n",
  "date": "2020-10-03",
  "mail": "trent_runte@hotmail.com",
  "phone": "17122462652",
  "sex": "female",
  "ID": 15
 },
 {
  "name": "h",
  "date": "2021-04-02",
  "mail": "jean10@gmail.com",
  "phone": "17109314490",
  "sex": "female",
  "ID": 16
 },
 {
  "name": "e",
  "date": "2019-08-24",
  "mail": "libbie58@hotmail.com",
  "phone": "16623924476",
  "sex": "female",
  "ID": 17
 },
 {
  "name": "w",
  "date": "2020-07-23",
  "mail": "alycia_ferry74@sohu.com",
  "phone": "13997971072",
  "sex": "male",
  "ID": 18
 }
]`,
          columns: [{
            title: "ID",
            isCustom: !1,
            dataIndex: "ID",
            hide: !1,
            sortable: !1,
            width: 91.3359375,
            render: {
              compType: "text",
              comp: {
                text: "{{currentCell}}"
              }
            },
            align: "left"
          }, {
            title: "date",
            isCustom: !1,
            dataIndex: "date",
            hide: !1,
            sortable: !0,
            width: 146.640625,
            render: {
              compType: "text",
              comp: {
                text: "{{currentCell}}"
              }
            },
            align: "left"
          }, {
            title: "name",
            isCustom: !1,
            dataIndex: "name",
            hide: !1,
            sortable: !1,
            width: 42,
            render: {
              compType: "text",
              comp: {
                text: "{{currentCell}}"
              }
            },
            align: "left"
          }, {
            title: "mail",
            isCustom: !1,
            dataIndex: "mail",
            hide: !1,
            sortable: !1,
            width: 20,
            render: {
              compType: "text",
              comp: {
                text: "{{currentCell}}"
              }
            },
            align: "left"
          }, {
            title: "phone",
            isCustom: !1,
            dataIndex: "phone",
            hide: !1,
            sortable: !1,
            width: -1,
            render: {
              compType: "text",
              comp: {
                text: "{{currentRow.ID}}"
              }
            },
            align: "left"
          }, {
            title: "sex",
            isCustom: !1,
            dataIndex: "sex",
            hide: !1,
            sortable: !1,
            width: -1,
            render: {
              compType: "text",
              comp: {
                text: "{{currentOriginalIndex}}"
              }
            },
            align: "left"
          }, {
            title: "custom column",
            isCustom: !0,
            dataIndex: "69924dc2",
            hide: !1,
            sortable: !1,
            width: -1,
            render: {
              compType: "text",
              comp: {
                text: "{{currentRow.ID}}"
              }
            },
            align: "left"
          }],
          size: "large",
          selection: {
            mode: "single",
            selectedRowKey: "0",
            selectedRowKeys: []
          },
          pagination: {
            showQuickJumper: !1,
            showSizeChanger: !1,
            changeablePageSize: {
              value: 0,
              defaultValue: ""
            },
            pageSize: "10",
            total: "",
            pageNo: 1,
            pageSizeOptions: "[10, 20, 50, 100]"
          },
          hidden: ""
        },
        name: "table1"
      }
    }
  },
  queries: [{
    compType: "mysql",
    comp: {
      sql: "",
      timeout: "10000"
    },
    id: "2qu36dzm29jd2vm2ihmr6crv",
    name: "query2",
    code: 1,
    message: "",
    data: {},
    extra: {},
    isFetching: !1,
    datasourceId: "623878d5320b9d7726fdac5d",
    triggerType: "automatic",
    successTrigger: {
      compType: "empty",
      comp: {}
    }
  }]
};
export {
  e as ROOT_COMP_VALUE
};
