// lowcoder-sdk/dist/1deadd6d.js
import { mJ as C, i6 as D } from "./c988e26e.js";
import M from "react";
function j(f, i) {
  for (var l = 0; l < i.length; l++) {
    const c = i[l];
    if (typeof c != "string" && !Array.isArray(c)) {
      for (const u in c)
        if (u !== "default" && !(u in f)) {
          const a = Object.getOwnPropertyDescriptor(c, u);
          a && Object.defineProperty(f, u, a.get ? a : {
            enumerable: !0,
            get: () => c[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, y = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-616-64h536c4.4 0 8-3.6 8-8V284c0-7.2-8.7-10.7-13.7-5.7L592 488.6l-125.4-124a8.03 8.03 0 00-11.3 0l-189 189.6a7.87 7.87 0 00-2.3 5.6V720c0 4.4 3.6 8 8 8z" } }] }, name: "area-chart", theme: "outlined" };
s.default = E;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return A;
    }
  });
  var i = /* @__PURE__ */ g(M), l = /* @__PURE__ */ a(s), c = /* @__PURE__ */ a(C);
  function u(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function g(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, o = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = o ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(o) {
        return Object.getOwnPropertyDescriptor(r, o).enumerable;
      }))), n.forEach(function(o) {
        u(e, o, r[o]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(o) {
        return Object.getOwnPropertyDescriptor(e, o).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ i.createElement(c.default, b(P({}, e), {
      ref: t,
      icon: l.default
    }));
  }, v = /* @__PURE__ */ i.forwardRef(w);
  process.env.NODE_ENV !== "production" && (v.displayName = "AreaChartOutlined");
  var A = v;
})(y);
(function(f, i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), i.default = void 0;
  const l = c(y);
  function c(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const u = l;
  i.default = u, f.exports = u;
})(O, O.exports);
var h = O.exports;
const S = /* @__PURE__ */ D(h), R = /* @__PURE__ */ j({
  __proto__: null,
  default: S
}, [h]);
export {
  R as A
};
