// lowcoder-sdk/dist/1eee6c5e.js
import { mJ as S, i6 as C } from "./c988e26e.js";
import M from "react";
function D(a, o) {
  for (var i = 0; i < o.length; i++) {
    const f = o[i];
    if (typeof f != "string" && !Array.isArray(f)) {
      for (const u in f)
        if (u !== "default" && !(u in a)) {
          const c = Object.getOwnPropertyDescriptor(f, u);
          c && Object.defineProperty(a, u, c.get ? c : {
            enumerable: !0,
            get: () => f[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(a, Symbol.toStringTag, { value: "Module" }));
}
var v = { exports: {} }, m = {}, d = {};
Object.defineProperty(d, "__esModule", { value: !0 });
var j = { icon: function(o, i) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M696 632h128v192H696zm-496 0h128v192H200z", fill: i } }, { tag: "path", attrs: { d: "M512 128c-212.1 0-384 171.9-384 384v360c0 13.3 10.7 24 24 24h184c35.3 0 64-28.7 64-64V624c0-35.3-28.7-64-64-64H200v-48c0-172.3 139.7-312 312-312s312 139.7 312 312v48H688c-35.3 0-64 28.7-64 64v208c0 35.3 28.7 64 64 64h184c13.3 0 24-10.7 24-24V512c0-212.1-171.9-384-384-384zM328 632v192H200V632h128zm496 192H696V632h128v192z", fill: o } }] };
}, name: "customer-service", theme: "twotone" };
d.default = j;
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var o = /* @__PURE__ */ y(M), i = /* @__PURE__ */ c(d), f = /* @__PURE__ */ c(S);
  function u(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function c(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? t : r;
    })(e);
  }
  function y(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = _(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, l = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = l ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function g(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(l) {
        return Object.getOwnPropertyDescriptor(t, l).enumerable;
      }))), n.forEach(function(l) {
        u(e, l, t[l]);
      });
    }
    return e;
  }
  function P(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(l) {
        return Object.getOwnPropertyDescriptor(e, l).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function T(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : P(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var h = function(e, r) {
    return /* @__PURE__ */ o.createElement(f.default, T(g({}, e), {
      ref: r,
      icon: i.default
    }));
  }, O = /* @__PURE__ */ o.forwardRef(h);
  process.env.NODE_ENV !== "production" && (O.displayName = "CustomerServiceTwoTone");
  var b = O;
})(m);
(function(a, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const i = f(m);
  function f(c) {
    return c && c.__esModule ? c : { default: c };
  }
  const u = i;
  o.default = u, a.exports = u;
})(v, v.exports);
var w = v.exports;
const E = /* @__PURE__ */ C(w), x = /* @__PURE__ */ D({
  __proto__: null,
  default: E
}, [w]);
export {
  x as C
};
