// lowcoder-sdk/dist/2211d28b.js
import { F as w, a8 as Ft, X as xi, S as zc, Z as on, li as Xc, az as Jc, U as Zc, cs as Qc, Y as te, G as vn, a9 as Gr, aa as Kr, ah as cn, ab as rl, ac as el, ae as tl, ad as al, av as nl, aw as ol, ax as il } from "./c988e26e.js";
import { useRef as Ae, useState as Ni, useEffect as ln, useContext as sn } from "react";
import "react-dom";
var un = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, kr = function(r) {
  return r && r.Math == Math && r;
}, $ = (
  // eslint-disable-next-line es-x/no-global-this -- safe
  kr(typeof globalThis == "object" && globalThis) || kr(typeof window == "object" && window) || // eslint-disable-next-line no-restricted-globals -- safe
  kr(typeof self == "object" && self) || kr(typeof un == "object" && un) || // eslint-disable-next-line no-new-func -- fallback
  function() {
    return this;
  }() || Function("return this")()
), $r = {}, E = function(r) {
  try {
    return !!r();
  } catch {
    return !0;
  }
}, vl = E, U = !vl(function() {
  return Object.defineProperty({}, 1, { get: function() {
    return 7;
  } })[1] != 7;
}), cl = E, be = !cl(function() {
  var r = (function() {
  }).bind();
  return typeof r != "function" || r.hasOwnProperty("prototype");
}), ll = be, Vr = Function.prototype.call, T = ll ? Vr.bind(Vr) : function() {
  return Vr.apply(Vr, arguments);
}, Ai = {}, Di = {}.propertyIsEnumerable, Li = Object.getOwnPropertyDescriptor, sl = Li && !Di.call({ 1: 2 }, 1);
Ai.f = sl ? function(e) {
  var t = Li(this, e);
  return !!t && t.enumerable;
} : Di;
var Mi = function(r, e) {
  return {
    enumerable: !(r & 1),
    configurable: !(r & 2),
    writable: !(r & 4),
    value: e
  };
}, Fi = be, Ui = Function.prototype, ul = Ui.bind, Ut = Ui.call, fl = Fi && ul.bind(Ut, Ut), m = Fi ? function(r) {
  return r && fl(r);
} : function(r) {
  return r && function() {
    return Ut.apply(r, arguments);
  };
}, Bi = m, $l = Bi({}.toString), dl = Bi("".slice), jr = function(r) {
  return dl($l(r), 8, -1);
}, pl = $, yl = m, bl = E, gl = jr, De = pl.Object, hl = yl("".split), ml = bl(function() {
  return !De("z").propertyIsEnumerable(0);
}) ? function(r) {
  return gl(r) == "String" ? hl(r, "") : De(r);
} : De, Ol = $, Sl = Ol.TypeError, J = function(r) {
  if (r == null)
    throw Sl("Can't call method on " + r);
  return r;
}, El = ml, Tl = J, ge = function(r) {
  return El(Tl(r));
}, O = function(r) {
  return typeof r == "function";
}, Il = O, B = function(r) {
  return typeof r == "object" ? r !== null : Il(r);
}, Le = $, wl = O, Pl = function(r) {
  return wl(r) ? r : void 0;
}, G = function(r, e) {
  return arguments.length < 2 ? Pl(Le[r]) : Le[r] && Le[r][e];
}, Rl = m, ga = Rl({}.isPrototypeOf), Cl = G, he = Cl("navigator", "userAgent") || "", Gi = $, Me = he, fn = Gi.process, $n = Gi.Deno, dn = fn && fn.versions || $n && $n.version, pn = dn && dn.v8, j, ce;
pn && (j = pn.split("."), ce = j[0] > 0 && j[0] < 4 ? 1 : +(j[0] + j[1]));
!ce && Me && (j = Me.match(/Edge\/(\d+)/), (!j || j[1] >= 74) && (j = Me.match(/Chrome\/(\d+)/), j && (ce = +j[1])));
var Ki = ce, yn = Ki, jl = E, ki = !!Object.getOwnPropertySymbols && !jl(function() {
  var r = Symbol();
  return !String(r) || !(Object(r) instanceof Symbol) || // Chrome 38-40 symbols are not inherited from DOM collections prototypes to instances
  !Symbol.sham && yn && yn < 41;
}), _l = ki, Vi = _l && !Symbol.sham && typeof Symbol.iterator == "symbol", xl = $, Nl = G, Al = O, Dl = ga, Ll = Vi, Ml = xl.Object, Wi = Ll ? function(r) {
  return typeof r == "symbol";
} : function(r) {
  var e = Nl("Symbol");
  return Al(e) && Dl(e.prototype, Ml(r));
}, Fl = $, Ul = Fl.String, me = function(r) {
  try {
    return Ul(r);
  } catch {
    return "Object";
  }
}, Bl = $, Gl = O, Kl = me, kl = Bl.TypeError, Z = function(r) {
  if (Gl(r))
    return r;
  throw kl(Kl(r) + " is not a function");
}, Vl = Z, _r = function(r, e) {
  var t = r[e];
  return t == null ? void 0 : Vl(t);
}, Wl = $, Fe = T, Ue = O, Be = B, Yl = Wl.TypeError, Hl = function(r, e) {
  var t, a;
  if (e === "string" && Ue(t = r.toString) && !Be(a = Fe(t, r)) || Ue(t = r.valueOf) && !Be(a = Fe(t, r)) || e !== "string" && Ue(t = r.toString) && !Be(a = Fe(t, r)))
    return a;
  throw Yl("Can't convert object to primitive value");
}, Oe = { exports: {} }, bn = $, ql = Object.defineProperty, ha = function(r, e) {
  try {
    ql(bn, r, { value: e, configurable: !0, writable: !0 });
  } catch {
    bn[r] = e;
  }
  return e;
}, zl = $, Xl = ha, gn = "__core-js_shared__", Jl = zl[gn] || Xl(gn, {}), ma = Jl, hn = ma;
(Oe.exports = function(r, e) {
  return hn[r] || (hn[r] = e !== void 0 ? e : {});
})("versions", []).push({
  version: "3.22.2",
  mode: "global",
  copyright: "© 2014-2022 Denis Pushkarev (zloirock.ru)",
  license: "https://github.com/zloirock/core-js/blob/v3.22.2/LICENSE",
  source: "https://github.com/zloirock/core-js"
});
var Zl = $, Ql = J, rs = Zl.Object, Yi = function(r) {
  return rs(Ql(r));
}, es = m, ts = Yi, as = es({}.hasOwnProperty), D = Object.hasOwn || function(e, t) {
  return as(ts(e), t);
}, ns = m, os = 0, is = Math.random(), vs = ns(1 .toString), Hi = function(r) {
  return "Symbol(" + (r === void 0 ? "" : r) + ")_" + vs(++os + is, 36);
}, cs = $, ls = Oe.exports, mn = D, ss = Hi, On = ki, qi = Vi, er = ls("wks"), H = cs.Symbol, Sn = H && H.for, us = qi ? H : H && H.withoutSetter || ss, S = function(r) {
  if (!mn(er, r) || !(On || typeof er[r] == "string")) {
    var e = "Symbol." + r;
    On && mn(H, r) ? er[r] = H[r] : qi && Sn ? er[r] = Sn(e) : er[r] = us(e);
  }
  return er[r];
}, fs = $, $s = T, En = B, Tn = Wi, ds = _r, ps = Hl, ys = S, bs = fs.TypeError, gs = ys("toPrimitive"), hs = function(r, e) {
  if (!En(r) || Tn(r))
    return r;
  var t = ds(r, gs), a;
  if (t) {
    if (e === void 0 && (e = "default"), a = $s(t, r, e), !En(a) || Tn(a))
      return a;
    throw bs("Can't convert object to primitive value");
  }
  return e === void 0 && (e = "number"), ps(r, e);
}, ms = hs, Os = Wi, zi = function(r) {
  var e = ms(r, "string");
  return Os(e) ? e : e + "";
}, Ss = $, In = B, Bt = Ss.document, Es = In(Bt) && In(Bt.createElement), Oa = function(r) {
  return Es ? Bt.createElement(r) : {};
}, Ts = U, Is = E, ws = Oa, Xi = !Ts && !Is(function() {
  return Object.defineProperty(ws("div"), "a", {
    get: function() {
      return 7;
    }
  }).a != 7;
}), Ps = U, Rs = T, Cs = Ai, js = Mi, _s = ge, xs = zi, Ns = D, As = Xi, wn = Object.getOwnPropertyDescriptor;
$r.f = Ps ? wn : function(e, t) {
  if (e = _s(e), t = xs(t), As)
    try {
      return wn(e, t);
    } catch {
    }
  if (Ns(e, t))
    return js(!Rs(Cs.f, e, t), e[t]);
};
var Q = {}, Ds = U, Ls = E, Ji = Ds && Ls(function() {
  return Object.defineProperty(function() {
  }, "prototype", {
    value: 42,
    writable: !1
  }).prototype != 42;
}), Zi = $, Ms = B, Fs = Zi.String, Us = Zi.TypeError, I = function(r) {
  if (Ms(r))
    return r;
  throw Us(Fs(r) + " is not an object");
}, Bs = $, Gs = U, Ks = Xi, ks = Ji, Wr = I, Pn = zi, Vs = Bs.TypeError, Ge = Object.defineProperty, Ws = Object.getOwnPropertyDescriptor, Ke = "enumerable", ke = "configurable", Ve = "writable";
Q.f = Gs ? ks ? function(e, t, a) {
  if (Wr(e), t = Pn(t), Wr(a), typeof e == "function" && t === "prototype" && "value" in a && Ve in a && !a[Ve]) {
    var n = Ws(e, t);
    n && n[Ve] && (e[t] = a.value, a = {
      configurable: ke in a ? a[ke] : n[ke],
      enumerable: Ke in a ? a[Ke] : n[Ke],
      writable: !1
    });
  }
  return Ge(e, t, a);
} : Ge : function(e, t, a) {
  if (Wr(e), t = Pn(t), Wr(a), Ks)
    try {
      return Ge(e, t, a);
    } catch {
    }
  if ("get" in a || "set" in a)
    throw Vs("Accessors not supported");
  return "value" in a && (e[t] = a.value), e;
};
var Ys = U, Hs = Q, qs = Mi, Se = Ys ? function(r, e, t) {
  return Hs.f(r, e, qs(1, t));
} : function(r, e, t) {
  return r[e] = t, r;
}, dr = { exports: {} }, zs = m, Xs = O, Gt = ma, Js = zs(Function.toString);
Xs(Gt.inspectSource) || (Gt.inspectSource = function(r) {
  return Js(r);
});
var Ee = Gt.inspectSource, Zs = $, Qs = O, ru = Ee, Rn = Zs.WeakMap, eu = Qs(Rn) && /native code/.test(ru(Rn)), tu = Oe.exports, au = Hi, Cn = tu("keys"), Qi = function(r) {
  return Cn[r] || (Cn[r] = au(r));
}, Sa = {}, nu = eu, rv = $, We = m, ou = B, iu = Se, Ye = D, He = ma, vu = Qi, cu = Sa, jn = "Object already initialized", Kt = rv.TypeError, lu = rv.WeakMap, le, Pr, se, su = function(r) {
  return se(r) ? Pr(r) : le(r, {});
}, uu = function(r) {
  return function(e) {
    var t;
    if (!ou(e) || (t = Pr(e)).type !== r)
      throw Kt("Incompatible receiver, " + r + " required");
    return t;
  };
};
if (nu || He.state) {
  var K = He.state || (He.state = new lu()), fu = We(K.get), _n = We(K.has), $u = We(K.set);
  le = function(r, e) {
    if (_n(K, r))
      throw new Kt(jn);
    return e.facade = r, $u(K, r, e), e;
  }, Pr = function(r) {
    return fu(K, r) || {};
  }, se = function(r) {
    return _n(K, r);
  };
} else {
  var tr = vu("state");
  cu[tr] = !0, le = function(r, e) {
    if (Ye(r, tr))
      throw new Kt(jn);
    return e.facade = r, iu(r, tr, e), e;
  }, Pr = function(r) {
    return Ye(r, tr) ? r[tr] : {};
  }, se = function(r) {
    return Ye(r, tr);
  };
}
var Ea = {
  set: le,
  get: Pr,
  has: se,
  enforce: su,
  getterFor: uu
}, kt = U, du = D, ev = Function.prototype, pu = kt && Object.getOwnPropertyDescriptor, Ta = du(ev, "name"), yu = Ta && (function() {
}).name === "something", bu = Ta && (!kt || kt && pu(ev, "name").configurable), gu = {
  EXISTS: Ta,
  PROPER: yu,
  CONFIGURABLE: bu
}, hu = $, xn = O, mu = D, Nn = Se, Ou = ha, Su = Ee, tv = Ea, Eu = gu.CONFIGURABLE, Tu = tv.get, Iu = tv.enforce, wu = String(String).split("String");
(dr.exports = function(r, e, t, a) {
  var n = a ? !!a.unsafe : !1, o = a ? !!a.enumerable : !1, i = a ? !!a.noTargetGet : !1, v = a && a.name !== void 0 ? a.name : e, s;
  if (xn(t) && (String(v).slice(0, 7) === "Symbol(" && (v = "[" + String(v).replace(/^Symbol\(([^)]*)\)/, "$1") + "]"), (!mu(t, "name") || Eu && t.name !== v) && Nn(t, "name", v), s = Iu(t), s.source || (s.source = wu.join(typeof v == "string" ? v : ""))), r === hu) {
    o ? r[e] = t : Ou(e, t);
    return;
  } else
    n ? !i && r[e] && (o = !0) : delete r[e];
  o ? r[e] = t : Nn(r, e, t);
})(Function.prototype, "toString", function() {
  return xn(this) && Tu(this).source || Su(this);
});
var av = {}, Pu = Math.ceil, Ru = Math.floor, Te = function(r) {
  var e = +r;
  return e !== e || e === 0 ? 0 : (e > 0 ? Ru : Pu)(e);
}, Cu = Te, ju = Math.max, _u = Math.min, xu = function(r, e) {
  var t = Cu(r);
  return t < 0 ? ju(t + e, 0) : _u(t, e);
}, Nu = Te, Au = Math.min, xr = function(r) {
  return r > 0 ? Au(Nu(r), 9007199254740991) : 0;
}, Du = xr, nv = function(r) {
  return Du(r.length);
}, Lu = ge, Mu = xu, Fu = nv, An = function(r) {
  return function(e, t, a) {
    var n = Lu(e), o = Fu(n), i = Mu(a, o), v;
    if (r && t != t) {
      for (; o > i; )
        if (v = n[i++], v != v)
          return !0;
    } else
      for (; o > i; i++)
        if ((r || i in n) && n[i] === t)
          return r || i || 0;
    return !r && -1;
  };
}, ov = {
  // `Array.prototype.includes` method
  // https://tc39.es/ecma262/#sec-array.prototype.includes
  includes: An(!0),
  // `Array.prototype.indexOf` method
  // https://tc39.es/ecma262/#sec-array.prototype.indexof
  indexOf: An(!1)
}, Uu = m, qe = D, Bu = ge, Gu = ov.indexOf, Ku = Sa, Dn = Uu([].push), iv = function(r, e) {
  var t = Bu(r), a = 0, n = [], o;
  for (o in t)
    !qe(Ku, o) && qe(t, o) && Dn(n, o);
  for (; e.length > a; )
    qe(t, o = e[a++]) && (~Gu(n, o) || Dn(n, o));
  return n;
}, Ia = [
  "constructor",
  "hasOwnProperty",
  "isPrototypeOf",
  "propertyIsEnumerable",
  "toLocaleString",
  "toString",
  "valueOf"
], ku = iv, Vu = Ia, Wu = Vu.concat("length", "prototype");
av.f = Object.getOwnPropertyNames || function(e) {
  return ku(e, Wu);
};
var vv = {};
vv.f = Object.getOwnPropertySymbols;
var Yu = G, Hu = m, qu = av, zu = vv, Xu = I, Ju = Hu([].concat), Zu = Yu("Reflect", "ownKeys") || function(e) {
  var t = qu.f(Xu(e)), a = zu.f;
  return a ? Ju(t, a(e)) : t;
}, Ln = D, Qu = Zu, rf = $r, ef = Q, tf = function(r, e, t) {
  for (var a = Qu(e), n = ef.f, o = rf.f, i = 0; i < a.length; i++) {
    var v = a[i];
    !Ln(r, v) && !(t && Ln(t, v)) && n(r, v, o(e, v));
  }
}, af = E, nf = O, of = /#|\.prototype\./, Nr = function(r, e) {
  var t = cf[vf(r)];
  return t == sf ? !0 : t == lf ? !1 : nf(e) ? af(e) : !!e;
}, vf = Nr.normalize = function(r) {
  return String(r).replace(of, ".").toLowerCase();
}, cf = Nr.data = {}, lf = Nr.NATIVE = "N", sf = Nr.POLYFILL = "P", cv = Nr, ze = $, uf = $r.f, ff = Se, $f = dr.exports, df = ha, pf = tf, yf = cv, N = function(r, e) {
  var t = r.target, a = r.global, n = r.stat, o, i, v, s, l, c;
  if (a ? i = ze : n ? i = ze[t] || df(t, {}) : i = (ze[t] || {}).prototype, i)
    for (v in e) {
      if (l = e[v], r.noTargetGet ? (c = uf(i, v), s = c && c.value) : s = i[v], o = yf(a ? v : t + (n ? "." : "#") + v, r.forced), !o && s !== void 0) {
        if (typeof l == typeof s)
          continue;
        pf(l, s);
      }
      (r.sham || s && s.sham) && ff(l, "sham", !0), $f(i, v, l, r);
    }
}, bf = jr, gf = $, wa = bf(gf.process) == "process", hf = dr.exports, mf = function(r, e, t) {
  for (var a in e)
    hf(r, a, e[a], t);
  return r;
}, lv = $, Of = O, Sf = lv.String, Ef = lv.TypeError, Tf = function(r) {
  if (typeof r == "object" || Of(r))
    return r;
  throw Ef("Can't set " + Sf(r) + " as a prototype");
}, If = m, wf = I, Pf = Tf, Rf = Object.setPrototypeOf || ("__proto__" in {} ? function() {
  var r = !1, e = {}, t;
  try {
    t = If(Object.getOwnPropertyDescriptor(Object.prototype, "__proto__").set), t(e, []), r = e instanceof Array;
  } catch {
  }
  return function(n, o) {
    return wf(n), Pf(o), r ? t(n, o) : n.__proto__ = o, n;
  };
}() : void 0), Cf = Q.f, jf = D, _f = S, Mn = _f("toStringTag"), xf = function(r, e, t) {
  r && !t && (r = r.prototype), r && !jf(r, Mn) && Cf(r, Mn, { configurable: !0, value: e });
}, Nf = G, Af = Q, Df = S, Lf = U, Fn = Df("species"), Mf = function(r) {
  var e = Nf(r), t = Af.f;
  Lf && e && !e[Fn] && t(e, Fn, {
    configurable: !0,
    get: function() {
      return this;
    }
  });
}, Ff = $, Uf = ga, Bf = Ff.TypeError, Gf = function(r, e) {
  if (Uf(e, r))
    return r;
  throw Bf("Incorrect invocation");
}, Kf = S, kf = Kf("toStringTag"), sv = {};
sv[kf] = "z";
var Vf = String(sv) === "[object z]", Wf = $, Yf = Vf, Hf = O, ae = jr, qf = S, zf = qf("toStringTag"), Xf = Wf.Object, Jf = ae(function() {
  return arguments;
}()) == "Arguments", Zf = function(r, e) {
  try {
    return r[e];
  } catch {
  }
}, Pa = Yf ? ae : function(r) {
  var e, t, a;
  return r === void 0 ? "Undefined" : r === null ? "Null" : typeof (t = Zf(e = Xf(r), zf)) == "string" ? t : Jf ? ae(e) : (a = ae(e)) == "Object" && Hf(e.callee) ? "Arguments" : a;
}, Qf = m, r$ = E, uv = O, e$ = Pa, t$ = G, a$ = Ee, fv = function() {
}, n$ = [], $v = t$("Reflect", "construct"), Ra = /^\s*(?:class|function)\b/, o$ = Qf(Ra.exec), i$ = !Ra.exec(fv), hr = function(e) {
  if (!uv(e))
    return !1;
  try {
    return $v(fv, n$, e), !0;
  } catch {
    return !1;
  }
}, dv = function(e) {
  if (!uv(e))
    return !1;
  switch (e$(e)) {
    case "AsyncFunction":
    case "GeneratorFunction":
    case "AsyncGeneratorFunction":
      return !1;
  }
  try {
    return i$ || !!o$(Ra, a$(e));
  } catch {
    return !0;
  }
};
dv.sham = !0;
var v$ = !$v || r$(function() {
  var r;
  return hr(hr.call) || !hr(Object) || !hr(function() {
    r = !0;
  }) || r;
}) ? dv : hr, c$ = $, l$ = v$, s$ = me, u$ = c$.TypeError, f$ = function(r) {
  if (l$(r))
    return r;
  throw u$(s$(r) + " is not a constructor");
}, Un = I, $$ = f$, d$ = S, p$ = d$("species"), y$ = function(r, e) {
  var t = Un(r).constructor, a;
  return t === void 0 || (a = Un(t)[p$]) == null ? e : $$(a);
}, b$ = be, pv = Function.prototype, Bn = pv.apply, Gn = pv.call, yv = typeof Reflect == "object" && Reflect.apply || (b$ ? Gn.bind(Bn) : function() {
  return Gn.apply(Bn, arguments);
}), Kn = m, g$ = Z, h$ = be, m$ = Kn(Kn.bind), Ca = function(r, e) {
  return g$(r), e === void 0 ? r : h$ ? m$(r, e) : function() {
    return r.apply(e, arguments);
  };
}, O$ = G, bv = O$("document", "documentElement"), S$ = m, E$ = S$([].slice), T$ = $, I$ = T$.TypeError, w$ = function(r, e) {
  if (r < e)
    throw I$("Not enough arguments");
  return r;
}, P$ = he, gv = /(?:ipad|iphone|ipod).*applewebkit/i.test(P$), P = $, R$ = yv, C$ = Ca, kn = O, j$ = D, _$ = E, Vn = bv, x$ = E$, Wn = Oa, N$ = w$, A$ = gv, D$ = wa, Vt = P.setImmediate, Wt = P.clearImmediate, L$ = P.process, Xe = P.Dispatch, M$ = P.Function, Yn = P.MessageChannel, F$ = P.String, Je = 0, Tr = {}, Hn = "onreadystatechange", Rr, k, Ze, Qe;
try {
  Rr = P.location;
} catch {
}
var ja = function(r) {
  if (j$(Tr, r)) {
    var e = Tr[r];
    delete Tr[r], e();
  }
}, rt = function(r) {
  return function() {
    ja(r);
  };
}, qn = function(r) {
  ja(r.data);
}, zn = function(r) {
  P.postMessage(F$(r), Rr.protocol + "//" + Rr.host);
};
(!Vt || !Wt) && (Vt = function(e) {
  N$(arguments.length, 1);
  var t = kn(e) ? e : M$(e), a = x$(arguments, 1);
  return Tr[++Je] = function() {
    R$(t, void 0, a);
  }, k(Je), Je;
}, Wt = function(e) {
  delete Tr[e];
}, D$ ? k = function(r) {
  L$.nextTick(rt(r));
} : Xe && Xe.now ? k = function(r) {
  Xe.now(rt(r));
} : Yn && !A$ ? (Ze = new Yn(), Qe = Ze.port2, Ze.port1.onmessage = qn, k = C$(Qe.postMessage, Qe)) : P.addEventListener && kn(P.postMessage) && !P.importScripts && Rr && Rr.protocol !== "file:" && !_$(zn) ? (k = zn, P.addEventListener("message", qn, !1)) : Hn in Wn("script") ? k = function(r) {
  Vn.appendChild(Wn("script"))[Hn] = function() {
    Vn.removeChild(this), ja(r);
  };
} : k = function(r) {
  setTimeout(rt(r), 0);
});
var hv = {
  set: Vt,
  clear: Wt
}, U$ = he, B$ = $, G$ = /ipad|iphone|ipod/i.test(U$) && B$.Pebble !== void 0, K$ = he, k$ = /web0s(?!.*chrome)/i.test(K$), z = $, Xn = Ca, V$ = $r.f, et = hv.set, W$ = gv, Y$ = G$, H$ = k$, tt = wa, Jn = z.MutationObserver || z.WebKitMutationObserver, Zn = z.document, Qn = z.process, Yr = z.Promise, ro = V$(z, "queueMicrotask"), mv = ro && ro.value, mr, W, Ir, vr, at, nt, Hr, eo;
mv || (mr = function() {
  var r, e;
  for (tt && (r = Qn.domain) && r.exit(); W; ) {
    e = W.fn, W = W.next;
    try {
      e();
    } catch (t) {
      throw W ? vr() : Ir = void 0, t;
    }
  }
  Ir = void 0, r && r.enter();
}, !W$ && !tt && !H$ && Jn && Zn ? (at = !0, nt = Zn.createTextNode(""), new Jn(mr).observe(nt, { characterData: !0 }), vr = function() {
  nt.data = at = !at;
}) : !Y$ && Yr && Yr.resolve ? (Hr = Yr.resolve(void 0), Hr.constructor = Yr, eo = Xn(Hr.then, Hr), vr = function() {
  eo(mr);
}) : tt ? vr = function() {
  Qn.nextTick(mr);
} : (et = Xn(et, z), vr = function() {
  et(mr);
}));
var q$ = mv || function(r) {
  var e = { fn: r, next: void 0 };
  Ir && (Ir.next = e), W || (W = e, vr()), Ir = e;
}, z$ = $, X$ = function(r, e) {
  var t = z$.console;
  t && t.error && (arguments.length == 1 ? t.error(r) : t.error(r, e));
}, _a = function(r) {
  try {
    return { error: !1, value: r() };
  } catch (e) {
    return { error: !0, value: e };
  }
}, Ov = function() {
  this.head = null, this.tail = null;
};
Ov.prototype = {
  add: function(r) {
    var e = { item: r, next: null };
    this.head ? this.tail.next = e : this.head = e, this.tail = e;
  },
  get: function() {
    var r = this.head;
    if (r)
      return this.head = r.next, this.tail === r && (this.tail = null), r.item;
  }
};
var J$ = Ov, Z$ = $, Ie = Z$.Promise, Q$ = typeof window == "object" && typeof Deno != "object", rd = $, wr = Ie, ed = O, td = cv, ad = Ee, nd = S, od = Q$, to = Ki;
wr && wr.prototype;
var id = nd("species"), Yt = !1, Sv = ed(rd.PromiseRejectionEvent), vd = td("Promise", function() {
  var r = ad(wr), e = r !== String(wr);
  if (!e && to === 66)
    return !0;
  if (to >= 51 && /native code/.test(r))
    return !1;
  var t = new wr(function(o) {
    o(1);
  }), a = function(o) {
    o(function() {
    }, function() {
    });
  }, n = t.constructor = {};
  return n[id] = a, Yt = t.then(function() {
  }) instanceof a, Yt ? !e && od && !Sv : !0;
}), Ar = {
  CONSTRUCTOR: vd,
  REJECTION_EVENT: Sv,
  SUBCLASSING: Yt
}, pr = {}, ao = Z, cd = function(r) {
  var e, t;
  this.promise = new r(function(a, n) {
    if (e !== void 0 || t !== void 0)
      throw TypeError("Bad Promise constructor");
    e = a, t = n;
  }), this.resolve = ao(e), this.reject = ao(t);
};
pr.f = function(r) {
  return new cd(r);
};
var ld = N, ue = wa, F = $, fr = T, sd = dr.exports, ud = mf, no = Rf, fd = xf, $d = Mf, dd = Z, ne = O, pd = B, yd = Gf, bd = y$, Ev = hv.set, xa = q$, gd = X$, hd = _a, md = J$, Tv = Ea, fe = Ie, Na = Ar, Iv = pr, we = "Promise", wv = Na.CONSTRUCTOR, Od = Na.REJECTION_EVENT, Sd = Na.SUBCLASSING, ot = Tv.getterFor(we), Ed = Tv.set, cr = fe && fe.prototype, Y = fe, qr = cr, Pv = F.TypeError, Ht = F.document, Aa = F.process, qt = Iv.f, Td = qt, Id = !!(Ht && Ht.createEvent && F.dispatchEvent), Rv = "unhandledrejection", wd = "rejectionhandled", oo = 0, Cv = 1, Pd = 2, Da = 1, jv = 2, zr, io, Rd, vo, _v = function(r) {
  var e;
  return pd(r) && ne(e = r.then) ? e : !1;
}, xv = function(r, e) {
  var t = e.value, a = e.state == Cv, n = a ? r.ok : r.fail, o = r.resolve, i = r.reject, v = r.domain, s, l, c;
  try {
    n ? (a || (e.rejection === jv && jd(e), e.rejection = Da), n === !0 ? s = t : (v && v.enter(), s = n(t), v && (v.exit(), c = !0)), s === r.promise ? i(Pv("Promise-chain cycle")) : (l = _v(s)) ? fr(l, s, o, i) : o(s)) : i(t);
  } catch (u) {
    v && !c && v.exit(), i(u);
  }
}, Nv = function(r, e) {
  r.notified || (r.notified = !0, xa(function() {
    for (var t = r.reactions, a; a = t.get(); )
      xv(a, r);
    r.notified = !1, e && !r.rejection && Cd(r);
  }));
}, Av = function(r, e, t) {
  var a, n;
  Id ? (a = Ht.createEvent("Event"), a.promise = e, a.reason = t, a.initEvent(r, !1, !0), F.dispatchEvent(a)) : a = { promise: e, reason: t }, !Od && (n = F["on" + r]) ? n(a) : r === Rv && gd("Unhandled promise rejection", t);
}, Cd = function(r) {
  fr(Ev, F, function() {
    var e = r.facade, t = r.value, a = co(r), n;
    if (a && (n = hd(function() {
      ue ? Aa.emit("unhandledRejection", t, e) : Av(Rv, e, t);
    }), r.rejection = ue || co(r) ? jv : Da, n.error))
      throw n.value;
  });
}, co = function(r) {
  return r.rejection !== Da && !r.parent;
}, jd = function(r) {
  fr(Ev, F, function() {
    var e = r.facade;
    ue ? Aa.emit("rejectionHandled", e) : Av(wd, e, r.value);
  });
}, lr = function(r, e, t) {
  return function(a) {
    r(e, a, t);
  };
}, ur = function(r, e, t) {
  r.done || (r.done = !0, t && (r = t), r.value = e, r.state = Pd, Nv(r, !0));
}, zt = function(r, e, t) {
  if (!r.done) {
    r.done = !0, t && (r = t);
    try {
      if (r.facade === e)
        throw Pv("Promise can't be resolved itself");
      var a = _v(e);
      a ? xa(function() {
        var n = { done: !1 };
        try {
          fr(
            a,
            e,
            lr(zt, n, r),
            lr(ur, n, r)
          );
        } catch (o) {
          ur(n, o, r);
        }
      }) : (r.value = e, r.state = Cv, Nv(r, !1));
    } catch (n) {
      ur({ done: !1 }, n, r);
    }
  }
};
if (wv && (Y = function(e) {
  yd(this, qr), dd(e), fr(zr, this);
  var t = ot(this);
  try {
    e(lr(zt, t), lr(ur, t));
  } catch (a) {
    ur(t, a);
  }
}, qr = Y.prototype, zr = function(e) {
  Ed(this, {
    type: we,
    done: !1,
    notified: !1,
    parent: !1,
    reactions: new md(),
    rejection: !1,
    state: oo,
    value: void 0
  });
}, zr.prototype = ud(qr, {
  // `Promise.prototype.then` method
  // https://tc39.es/ecma262/#sec-promise.prototype.then
  // eslint-disable-next-line unicorn/no-thenable -- safe
  then: function(e, t) {
    var a = ot(this), n = qt(bd(this, Y));
    return a.parent = !0, n.ok = ne(e) ? e : !0, n.fail = ne(t) && t, n.domain = ue ? Aa.domain : void 0, a.state == oo ? a.reactions.add(n) : xa(function() {
      xv(n, a);
    }), n.promise;
  }
}), io = function() {
  var r = new zr(), e = ot(r);
  this.promise = r, this.resolve = lr(zt, e), this.reject = lr(ur, e);
}, Iv.f = qt = function(r) {
  return r === Y || r === Rd ? new io(r) : Td(r);
}, ne(fe) && cr !== Object.prototype)) {
  vo = cr.then, Sd || sd(cr, "then", function(e, t) {
    var a = this;
    return new Y(function(n, o) {
      fr(vo, a, n, o);
    }).then(e, t);
  }, { unsafe: !0 });
  try {
    delete cr.constructor;
  } catch {
  }
  no && no(cr, qr);
}
ld({ global: !0, wrap: !0, forced: wv }, {
  Promise: Y
});
fd(Y, we, !1);
$d(we);
var Dv = {}, _d = S, xd = Dv, Nd = _d("iterator"), Ad = Array.prototype, Dd = function(r) {
  return r !== void 0 && (xd.Array === r || Ad[Nd] === r);
}, Ld = Pa, lo = _r, Md = Dv, Fd = S, Ud = Fd("iterator"), Lv = function(r) {
  if (r != null)
    return lo(r, Ud) || lo(r, "@@iterator") || Md[Ld(r)];
}, Bd = $, Gd = T, Kd = Z, kd = I, Vd = me, Wd = Lv, Yd = Bd.TypeError, Hd = function(r, e) {
  var t = arguments.length < 2 ? Wd(r) : e;
  if (Kd(t))
    return kd(Gd(t, r));
  throw Yd(Vd(r) + " is not iterable");
}, qd = T, so = I, zd = _r, Xd = function(r, e, t) {
  var a, n;
  so(r);
  try {
    if (a = zd(r, "return"), !a) {
      if (e === "throw")
        throw t;
      return t;
    }
    a = qd(a, r);
  } catch (o) {
    n = !0, a = o;
  }
  if (e === "throw")
    throw t;
  if (n)
    throw a;
  return so(a), t;
}, Jd = $, Zd = Ca, Qd = T, rp = I, ep = me, tp = Dd, ap = nv, uo = ga, np = Hd, op = Lv, fo = Xd, ip = Jd.TypeError, oe = function(r, e) {
  this.stopped = r, this.result = e;
}, $o = oe.prototype, Mv = function(r, e, t) {
  var a = t && t.that, n = !!(t && t.AS_ENTRIES), o = !!(t && t.IS_ITERATOR), i = !!(t && t.INTERRUPTED), v = Zd(e, a), s, l, c, u, f, y, d, p = function(b) {
    return s && fo(s, "normal", b), new oe(!0, b);
  }, g = function(b) {
    return n ? (rp(b), i ? v(b[0], b[1], p) : v(b[0], b[1])) : i ? v(b, p) : v(b);
  };
  if (o)
    s = r;
  else {
    if (l = op(r), !l)
      throw ip(ep(r) + " is not iterable");
    if (tp(l)) {
      for (c = 0, u = ap(r); u > c; c++)
        if (f = g(r[c]), f && uo($o, f))
          return f;
      return new oe(!1);
    }
    s = np(r, l);
  }
  for (y = s.next; !(d = Qd(y, s)).done; ) {
    try {
      f = g(d.value);
    } catch (b) {
      fo(s, "throw", b);
    }
    if (typeof f == "object" && f && uo($o, f))
      return f;
  }
  return new oe(!1);
}, vp = S, Fv = vp("iterator"), Uv = !1;
try {
  var cp = 0, po = {
    next: function() {
      return { done: !!cp++ };
    },
    return: function() {
      Uv = !0;
    }
  };
  po[Fv] = function() {
    return this;
  }, Array.from(po, function() {
    throw 2;
  });
} catch {
}
var lp = function(r, e) {
  if (!e && !Uv)
    return !1;
  var t = !1;
  try {
    var a = {};
    a[Fv] = function() {
      return {
        next: function() {
          return { done: t = !0 };
        }
      };
    }, r(a);
  } catch {
  }
  return t;
}, sp = Ie, up = lp, fp = Ar.CONSTRUCTOR, Bv = fp || !up(function(r) {
  sp.all(r).then(void 0, function() {
  });
}), $p = N, dp = T, pp = Z, yp = pr, bp = _a, gp = Mv, hp = Bv;
$p({ target: "Promise", stat: !0, forced: hp }, {
  all: function(e) {
    var t = this, a = yp.f(t), n = a.resolve, o = a.reject, i = bp(function() {
      var v = pp(t.resolve), s = [], l = 0, c = 1;
      gp(e, function(u) {
        var f = l++, y = !1;
        c++, dp(v, t, u).then(function(d) {
          y || (y = !0, s[f] = d, --c || n(s));
        }, o);
      }), --c || n(s);
    });
    return i.error && o(i.value), a.promise;
  }
});
var mp = N, Op = Ar.CONSTRUCTOR, Xt = Ie, Sp = G, Ep = O, Tp = dr.exports, yo = Xt && Xt.prototype;
mp({ target: "Promise", proto: !0, forced: Op, real: !0 }, {
  catch: function(r) {
    return this.then(void 0, r);
  }
});
if (Ep(Xt)) {
  var bo = Sp("Promise").prototype.catch;
  yo.catch !== bo && Tp(yo, "catch", bo, { unsafe: !0 });
}
var Ip = N, wp = T, Pp = Z, Rp = pr, Cp = _a, jp = Mv, _p = Bv;
Ip({ target: "Promise", stat: !0, forced: _p }, {
  race: function(e) {
    var t = this, a = Rp.f(t), n = a.reject, o = Cp(function() {
      var i = Pp(t.resolve);
      jp(e, function(v) {
        wp(i, t, v).then(a.resolve, n);
      });
    });
    return o.error && n(o.value), a.promise;
  }
});
var xp = N, Np = T, Ap = pr, Dp = Ar.CONSTRUCTOR;
xp({ target: "Promise", stat: !0, forced: Dp }, {
  reject: function(e) {
    var t = Ap.f(this);
    return Np(t.reject, void 0, e), t.promise;
  }
});
var Lp = I, Mp = B, Fp = pr, Up = function(r, e) {
  if (Lp(r), Mp(e) && e.constructor === r)
    return e;
  var t = Fp.f(r), a = t.resolve;
  return a(e), t.promise;
}, Bp = N, Gp = G, Kp = Ar.CONSTRUCTOR, kp = Up;
Gp("Promise");
Bp({ target: "Promise", stat: !0, forced: Kp }, {
  resolve: function(e) {
    return kp(this, e);
  }
});
var Vp = $, Wp = Pa, Yp = Vp.String, yr = function(r) {
  if (Wp(r) === "Symbol")
    throw TypeError("Cannot convert a Symbol value to a string");
  return Yp(r);
}, Hp = I, qp = function() {
  var r = Hp(this), e = "";
  return r.hasIndices && (e += "d"), r.global && (e += "g"), r.ignoreCase && (e += "i"), r.multiline && (e += "m"), r.dotAll && (e += "s"), r.unicode && (e += "u"), r.sticky && (e += "y"), e;
}, La = E, zp = $, Ma = zp.RegExp, Fa = La(function() {
  var r = Ma("a", "y");
  return r.lastIndex = 2, r.exec("abcd") != null;
}), Xp = Fa || La(function() {
  return !Ma("a", "y").sticky;
}), Jp = Fa || La(function() {
  var r = Ma("^r", "gy");
  return r.lastIndex = 2, r.exec("str") != null;
}), Zp = {
  BROKEN_CARET: Jp,
  MISSED_STICKY: Xp,
  UNSUPPORTED_Y: Fa
}, Gv = {}, Qp = iv, ry = Ia, ey = Object.keys || function(e) {
  return Qp(e, ry);
}, ty = U, ay = Ji, ny = Q, oy = I, iy = ge, vy = ey;
Gv.f = ty && !ay ? Object.defineProperties : function(e, t) {
  oy(e);
  for (var a = iy(t), n = vy(t), o = n.length, i = 0, v; o > i; )
    ny.f(e, v = n[i++], a[v]);
  return e;
};
var cy = I, ly = Gv, go = Ia, sy = Sa, uy = bv, fy = Oa, $y = Qi, ho = ">", mo = "<", Jt = "prototype", Zt = "script", Kv = $y("IE_PROTO"), it = function() {
}, kv = function(r) {
  return mo + Zt + ho + r + mo + "/" + Zt + ho;
}, Oo = function(r) {
  r.write(kv("")), r.close();
  var e = r.parentWindow.Object;
  return r = null, e;
}, dy = function() {
  var r = fy("iframe"), e = "java" + Zt + ":", t;
  return r.style.display = "none", uy.appendChild(r), r.src = String(e), t = r.contentWindow.document, t.open(), t.write(kv("document.F=Object")), t.close(), t.F;
}, Xr, ie = function() {
  try {
    Xr = new ActiveXObject("htmlfile");
  } catch {
  }
  ie = typeof document < "u" ? document.domain && Xr ? Oo(Xr) : dy() : Oo(Xr);
  for (var r = go.length; r--; )
    delete ie[Jt][go[r]];
  return ie();
};
sy[Kv] = !0;
var Vv = Object.create || function(e, t) {
  var a;
  return e !== null ? (it[Jt] = cy(e), a = new it(), it[Jt] = null, a[Kv] = e) : a = ie(), t === void 0 ? a : ly.f(a, t);
}, py = E, yy = $, by = yy.RegExp, gy = py(function() {
  var r = by(".", "s");
  return !(r.dotAll && r.exec(`
`) && r.flags === "s");
}), hy = E, my = $, Oy = my.RegExp, Sy = hy(function() {
  var r = Oy("(?<a>b)", "g");
  return r.exec("b").groups.a !== "b" || "b".replace(r, "$<a>c") !== "bc";
}), sr = T, Pe = m, Ey = yr, Ty = qp, Iy = Zp, wy = Oe.exports, Py = Vv, Ry = Ea.get, Cy = gy, jy = Sy, _y = wy("native-string-replace", String.prototype.replace), $e = RegExp.prototype.exec, Qt = $e, xy = Pe("".charAt), Ny = Pe("".indexOf), Ay = Pe("".replace), vt = Pe("".slice), ra = function() {
  var r = /a/, e = /b*/g;
  return sr($e, r, "a"), sr($e, e, "a"), r.lastIndex !== 0 || e.lastIndex !== 0;
}(), Wv = Iy.BROKEN_CARET, ea = /()??/.exec("")[1] !== void 0, Dy = ra || ea || Wv || Cy || jy;
Dy && (Qt = function(e) {
  var t = this, a = Ry(t), n = Ey(e), o = a.raw, i, v, s, l, c, u, f;
  if (o)
    return o.lastIndex = t.lastIndex, i = sr(Qt, o, n), t.lastIndex = o.lastIndex, i;
  var y = a.groups, d = Wv && t.sticky, p = sr(Ty, t), g = t.source, b = 0, C = n;
  if (d && (p = Ay(p, "y", ""), Ny(p, "g") === -1 && (p += "g"), C = vt(n, t.lastIndex), t.lastIndex > 0 && (!t.multiline || t.multiline && xy(n, t.lastIndex - 1) !== `
`) && (g = "(?: " + g + ")", C = " " + C, b++), v = new RegExp("^(?:" + g + ")", p)), ea && (v = new RegExp("^" + g + "$(?!\\s)", p)), ra && (s = t.lastIndex), l = sr($e, d ? v : t, C), d ? l ? (l.input = vt(l.input, b), l[0] = vt(l[0], b), l.index = t.lastIndex, t.lastIndex += l[0].length) : t.lastIndex = 0 : ra && l && (t.lastIndex = t.global ? l.index + l[0].length : s), ea && l && l.length > 1 && sr(_y, l[0], v, function() {
    for (c = 1; c < arguments.length - 2; c++)
      arguments[c] === void 0 && (l[c] = void 0);
  }), l && y)
    for (l.groups = u = Py(null), c = 0; c < y.length; c++)
      f = y[c], u[f[0]] = l[f[1]];
  return l;
});
var Ua = Qt, Ly = N, So = Ua;
Ly({ target: "RegExp", proto: !0, forced: /./.exec !== So }, {
  exec: So
});
var Eo = m, To = dr.exports, My = Ua, Io = E, Yv = S, Fy = Se, Uy = Yv("species"), ct = RegExp.prototype, Hv = function(r, e, t, a) {
  var n = Yv(r), o = !Io(function() {
    var l = {};
    return l[n] = function() {
      return 7;
    }, ""[r](l) != 7;
  }), i = o && !Io(function() {
    var l = !1, c = /a/;
    return r === "split" && (c = {}, c.constructor = {}, c.constructor[Uy] = function() {
      return c;
    }, c.flags = "", c[n] = /./[n]), c.exec = function() {
      return l = !0, null;
    }, c[n](""), !l;
  });
  if (!o || !i || t) {
    var v = Eo(/./[n]), s = e(n, ""[r], function(l, c, u, f, y) {
      var d = Eo(l), p = c.exec;
      return p === My || p === ct.exec ? o && !y ? { done: !0, value: v(c, u, f) } : { done: !0, value: d(u, c, f) } : { done: !1 };
    });
    To(String.prototype, r, s[0]), To(ct, n, s[1]);
  }
  a && Fy(ct[n], "sham", !0);
}, Ba = m, By = Te, Gy = yr, Ky = J, ky = Ba("".charAt), wo = Ba("".charCodeAt), Vy = Ba("".slice), Po = function(r) {
  return function(e, t) {
    var a = Gy(Ky(e)), n = By(t), o = a.length, i, v;
    return n < 0 || n >= o ? r ? "" : void 0 : (i = wo(a, n), i < 55296 || i > 56319 || n + 1 === o || (v = wo(a, n + 1)) < 56320 || v > 57343 ? r ? ky(a, n) : i : r ? Vy(a, n, n + 2) : (i - 55296 << 10) + (v - 56320) + 65536);
  };
}, Wy = {
  // `String.prototype.codePointAt` method
  // https://tc39.es/ecma262/#sec-string.prototype.codepointat
  codeAt: Po(!1),
  // `String.prototype.at` method
  // https://github.com/mathiasbynens/String.prototype.at
  charAt: Po(!0)
}, Yy = Wy.charAt, qv = function(r, e, t) {
  return e + (t ? Yy(r, e).length : 1);
}, Hy = $, Ro = T, qy = I, zy = O, Xy = jr, Jy = Ua, Zy = Hy.TypeError, zv = function(r, e) {
  var t = r.exec;
  if (zy(t)) {
    var a = Ro(t, r, e);
    return a !== null && qy(a), a;
  }
  if (Xy(r) === "RegExp")
    return Ro(Jy, r, e);
  throw Zy("RegExp#exec called on incompatible receiver");
}, Qy = T, rb = Hv, eb = I, tb = xr, lt = yr, ab = J, nb = _r, ob = qv, Co = zv;
rb("match", function(r, e, t) {
  return [
    // `String.prototype.match` method
    // https://tc39.es/ecma262/#sec-string.prototype.match
    function(n) {
      var o = ab(this), i = n == null ? void 0 : nb(n, r);
      return i ? Qy(i, n, o) : new RegExp(n)[r](lt(o));
    },
    // `RegExp.prototype[@@match]` method
    // https://tc39.es/ecma262/#sec-regexp.prototype-@@match
    function(a) {
      var n = eb(this), o = lt(a), i = t(e, n, o);
      if (i.done)
        return i.value;
      if (!n.global)
        return Co(n, o);
      var v = n.unicode;
      n.lastIndex = 0;
      for (var s = [], l = 0, c; (c = Co(n, o)) !== null; ) {
        var u = lt(c[0]);
        s[l] = u, u === "" && (n.lastIndex = ob(o, tb(n.lastIndex), v)), l++;
      }
      return l === 0 ? null : s;
    }
  ];
});
var Ga = m, ib = Yi, vb = Math.floor, st = Ga("".charAt), cb = Ga("".replace), ut = Ga("".slice), lb = /\$([$&'`]|\d{1,2}|<[^>]*>)/g, sb = /\$([$&'`]|\d{1,2})/g, ub = function(r, e, t, a, n, o) {
  var i = t + r.length, v = a.length, s = sb;
  return n !== void 0 && (n = ib(n), s = lb), cb(o, s, function(l, c) {
    var u;
    switch (st(c, 0)) {
      case "$":
        return "$";
      case "&":
        return r;
      case "`":
        return ut(e, 0, t);
      case "'":
        return ut(e, i);
      case "<":
        u = n[ut(c, 1, -1)];
        break;
      default:
        var f = +c;
        if (f === 0)
          return l;
        if (f > v) {
          var y = vb(f / 10);
          return y === 0 ? l : y <= v ? a[y - 1] === void 0 ? st(c, 1) : a[y - 1] + st(c, 1) : l;
        }
        u = a[f - 1];
    }
    return u === void 0 ? "" : u;
  });
}, fb = yv, jo = T, Re = m, $b = Hv, db = E, pb = I, yb = O, bb = Te, gb = xr, ar = yr, hb = J, mb = qv, Ob = _r, Sb = ub, Eb = zv, Tb = S, ta = Tb("replace"), Ib = Math.max, wb = Math.min, Pb = Re([].concat), ft = Re([].push), _o = Re("".indexOf), xo = Re("".slice), Rb = function(r) {
  return r === void 0 ? r : String(r);
}, Cb = function() {
  return "a".replace(/./, "$0") === "$0";
}(), No = function() {
  return /./[ta] ? /./[ta]("a", "$0") === "" : !1;
}(), jb = !db(function() {
  var r = /./;
  return r.exec = function() {
    var e = [];
    return e.groups = { a: "7" }, e;
  }, "".replace(r, "$<a>") !== "7";
});
$b("replace", function(r, e, t) {
  var a = No ? "$" : "$0";
  return [
    // `String.prototype.replace` method
    // https://tc39.es/ecma262/#sec-string.prototype.replace
    function(o, i) {
      var v = hb(this), s = o == null ? void 0 : Ob(o, ta);
      return s ? jo(s, o, v, i) : jo(e, ar(v), o, i);
    },
    // `RegExp.prototype[@@replace]` method
    // https://tc39.es/ecma262/#sec-regexp.prototype-@@replace
    function(n, o) {
      var i = pb(this), v = ar(n);
      if (typeof o == "string" && _o(o, a) === -1 && _o(o, "$<") === -1) {
        var s = t(e, i, v, o);
        if (s.done)
          return s.value;
      }
      var l = yb(o);
      l || (o = ar(o));
      var c = i.global;
      if (c) {
        var u = i.unicode;
        i.lastIndex = 0;
      }
      for (var f = []; ; ) {
        var y = Eb(i, v);
        if (y === null || (ft(f, y), !c))
          break;
        var d = ar(y[0]);
        d === "" && (i.lastIndex = mb(v, gb(i.lastIndex), u));
      }
      for (var p = "", g = 0, b = 0; b < f.length; b++) {
        y = f[b];
        for (var C = ar(y[0]), gr = Ib(wb(bb(y.index), v.length), 0), _e = [], xe = 1; xe < y.length; xe++)
          ft(_e, Rb(y[xe]));
        var Ne = y.groups;
        if (l) {
          var an = Pb([C], _e, gr, v);
          Ne !== void 0 && ft(an, Ne);
          var nn = ar(fb(o, void 0, an));
        } else
          nn = Sb(C, v, gr, _e, Ne, o);
        gr >= g && (p += xo(v, g, gr) + nn, g = gr + C.length);
      }
      return p + xo(v, g);
    }
  ];
}, !jb || !Cb || No);
var _b = B, xb = jr, Nb = S, Ab = Nb("match"), Db = function(r) {
  var e;
  return _b(r) && ((e = r[Ab]) !== void 0 ? !!e : xb(r) == "RegExp");
}, Lb = $, Mb = Db, Fb = Lb.TypeError, Xv = function(r) {
  if (Mb(r))
    throw Fb("The method doesn't accept regular expressions");
  return r;
}, Ub = S, Bb = Ub("match"), Jv = function(r) {
  var e = /./;
  try {
    "/./"[r](e);
  } catch {
    try {
      return e[Bb] = !1, "/./"[r](e);
    } catch {
    }
  }
  return !1;
}, Gb = N, Zv = m, Kb = $r.f, kb = xr, Ao = yr, Vb = Xv, Wb = J, Yb = Jv, Do = Zv("".startsWith), Hb = Zv("".slice), qb = Math.min, Qv = Yb("startsWith"), zb = !Qv && !!function() {
  var r = Kb(String.prototype, "startsWith");
  return r && !r.writable;
}();
Gb({ target: "String", proto: !0, forced: !zb && !Qv }, {
  startsWith: function(e) {
    var t = Ao(Wb(this));
    Vb(e);
    var a = kb(qb(arguments.length > 1 ? arguments[1] : void 0, t.length)), n = Ao(e);
    return Do ? Do(t, n, a) : Hb(t, a, a + n.length) === n;
  }
});
var Xb = S, Jb = Vv, Zb = Q, aa = Xb("unscopables"), na = Array.prototype;
na[aa] == null && Zb.f(na, aa, {
  configurable: !0,
  value: Jb(null)
});
var Qb = function(r) {
  na[aa][r] = !0;
}, rg = N, eg = ov.includes, tg = Qb;
rg({ target: "Array", proto: !0 }, {
  includes: function(e) {
    return eg(this, e, arguments.length > 1 ? arguments[1] : void 0);
  }
});
tg("includes");
var ag = N, rc = m, ng = $r.f, og = xr, Lo = yr, ig = Xv, vg = J, cg = Jv, Mo = rc("".endsWith), lg = rc("".slice), sg = Math.min, ec = cg("endsWith"), ug = !ec && !!function() {
  var r = ng(String.prototype, "endsWith");
  return r && !r.writable;
}();
ag({ target: "String", proto: !0, forced: !ug && !ec }, {
  endsWith: function(e) {
    var t = Lo(vg(this));
    ig(e);
    var a = arguments.length > 1 ? arguments[1] : void 0, n = t.length, o = a === void 0 ? n : sg(og(a), n), i = Lo(e);
    return Mo ? Mo(t, i, o) : lg(t, o - i.length, o) === i;
  }
});
function fg(r, e, t, a) {
  function n(o) {
    return o instanceof t ? o : new t(function(i) {
      i(o);
    });
  }
  return new (t || (t = Promise))(function(o, i) {
    function v(c) {
      try {
        l(a.next(c));
      } catch (u) {
        i(u);
      }
    }
    function s(c) {
      try {
        l(a.throw(c));
      } catch (u) {
        i(u);
      }
    }
    function l(c) {
      c.done ? o(c.value) : n(c.value).then(v, s);
    }
    l((a = a.apply(r, e || [])).next());
  });
}
const $g = (r) => new Promise((e, t) => {
  try {
    const a = new XMLHttpRequest();
    a.open("GET", r, !0), a.responseType = "blob", a.onload = () => {
      const n = new FileReader();
      n.readAsArrayBuffer(a.response), n.onloadend = () => {
        e(n.result);
      };
    }, a.send();
  } catch {
    t(`error while retrieving file ${r}.`);
  }
}), dg = (r) => {
  if (r) {
    const e = r.src;
    r.src = "about:blank", setTimeout(() => {
      r && (r.src = e);
    }, 100);
  }
}, oa = (r) => {
  const e = r.match(/^(https?\:)\/\/(([^:\/?#]*)(?:\:([0-9]+))?)([\/]{0,1}[^?#]*)(\?[^#]*|)(#.*|)$/);
  return e && {
    href: r,
    protocol: e[1],
    host: e[2],
    hostname: e[3],
    port: e[4],
    pathname: e[5],
    search: e[6],
    hash: e[7]
  };
}, pg = (r) => fg(void 0, void 0, void 0, function* () {
  mammoth || console.error("Please install mammoth and make sure mammoth.browser.min.js is loaded.");
  const e = yield $g(r);
  return (yield mammoth.convertToHtml({
    arrayBuffer: e
  })).value;
}), yg = () => {
  let r = null, e = 0;
  return {
    subscribe: (t, a = 3e3, n = 5) => {
      if (tc(t))
        r && clearInterval(r);
      else
        return r = setInterval(() => {
          e++, e >= n && clearInterval(r), dg(t);
        }, a), r;
    },
    unsubscribe: () => {
      r && clearInterval(r);
    }
  };
}, tc = (r) => {
  var e;
  let t = !1;
  try {
    bg() ? t = !(!((e = r == null ? void 0 : r.contentWindow) === null || e === void 0) && e.document) : t = !(r != null && r.contentDocument);
  } catch {
  }
  return t;
}, bg = () => /MSIE (\d+\.\d+);/.test(navigator.userAgent) || navigator.userAgent.indexOf("Trident/") > -1, Fo = (r, e = "google", t = "", a = "") => {
  switch (e) {
    case "google":
      a = "https://docs.google.com/gview?url=%URL%&embedded=true";
      break;
    case "office": {
      a = "https://view.officeapps.live.com/op/embed.aspx?src=%URL%";
      break;
    }
    case "pdf": {
      a = "";
      break;
    }
  }
  const n = e === "google" || e === "office" || e === "url", o = r != null && r.indexOf("/") ? encodeURIComponent(r) : r;
  let i = a ? a.replace("%URL%", o) : r;
  if (t && n && e !== "url") {
    const v = t.startsWith("&") ? "" : "&";
    i = `${i}${v}${t}`;
  }
  return {
    url: i,
    externalViewer: n
  };
}, gg = (r, e) => {
  const t = oa(r), a = oa(e);
  return t && a ? r.replace(t.port ? `${t.hostname}:${t.port}` : t.hostname, a.port ? `${a.hostname}:${a.port}` : a.hostname) : r;
}, hg = (r) => {
  const e = oa(r), t = (e == null ? void 0 : e.hostname) || "";
  return ["localhost", "127.0.0.1", "", "::1"].includes(t) || t.startsWith("192.168.") || t.startsWith("10.0.") || t.endsWith(".local");
};
var Uo = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, Jr = function(r) {
  return r && r.Math == Math && r;
}, h = (
  // eslint-disable-next-line es-x/no-global-this -- safe
  Jr(typeof globalThis == "object" && globalThis) || Jr(typeof window == "object" && window) || // eslint-disable-next-line no-restricted-globals -- safe
  Jr(typeof self == "object" && self) || Jr(typeof Uo == "object" && Uo) || // eslint-disable-next-line no-new-func -- fallback
  function() {
    return this;
  }() || Function("return this")()
), A = function(r) {
  try {
    return !!r();
  } catch {
    return !0;
  }
}, mg = A, ac = !mg(function() {
  var r = (function() {
  }).bind();
  return typeof r != "function" || r.hasOwnProperty("prototype");
}), nc = ac, oc = Function.prototype, Og = oc.bind, ia = oc.call, Sg = nc && Og.bind(ia, ia), x = nc ? function(r) {
  return r && Sg(r);
} : function(r) {
  return r && function() {
    return ia.apply(r, arguments);
  };
}, ic = x, Eg = ic({}.toString), Tg = ic("".slice), Ig = function(r) {
  return Tg(Eg(r), 8, -1);
}, wg = h, Pg = x, Rg = A, Cg = Ig, $t = wg.Object, jg = Pg("".split), vc = Rg(function() {
  return !$t("z").propertyIsEnumerable(0);
}) ? function(r) {
  return Cg(r) == "String" ? jg(r, "") : $t(r);
} : $t, _g = h, xg = _g.TypeError, cc = function(r) {
  if (r == null)
    throw xg("Can't call method on " + r);
  return r;
}, Ng = vc, Ag = cc, Dr = function(r) {
  return Ng(Ag(r));
}, Ka = { exports: {} }, Bo = h, Dg = Object.defineProperty, ka = function(r, e) {
  try {
    Dg(Bo, r, { value: e, configurable: !0, writable: !0 });
  } catch {
    Bo[r] = e;
  }
  return e;
}, Lg = h, Mg = ka, Go = "__core-js_shared__", Fg = Lg[Go] || Mg(Go, {}), Va = Fg, Ko = Va;
(Ka.exports = function(r, e) {
  return Ko[r] || (Ko[r] = e !== void 0 ? e : {});
})("versions", []).push({
  version: "3.22.2",
  mode: "global",
  copyright: "© 2014-2022 Denis Pushkarev (zloirock.ru)",
  license: "https://github.com/zloirock/core-js/blob/v3.22.2/LICENSE",
  source: "https://github.com/zloirock/core-js"
});
var Ug = h, Bg = cc, Gg = Ug.Object, Wa = function(r) {
  return Gg(Bg(r));
}, Kg = x, kg = Wa, Vg = Kg({}.hasOwnProperty), L = Object.hasOwn || function(e, t) {
  return Vg(kg(e), t);
}, Wg = x, Yg = 0, Hg = Math.random(), qg = Wg(1 .toString), lc = function(r) {
  return "Symbol(" + (r === void 0 ? "" : r) + ")_" + qg(++Yg + Hg, 36);
}, R = function(r) {
  return typeof r == "function";
}, dt = h, zg = R, Xg = function(r) {
  return zg(r) ? r : void 0;
}, Ce = function(r, e) {
  return arguments.length < 2 ? Xg(dt[r]) : dt[r] && dt[r][e];
}, Jg = Ce, Zg = Jg("navigator", "userAgent") || "", sc = h, pt = Zg, ko = sc.process, Vo = sc.Deno, Wo = ko && ko.versions || Vo && Vo.version, Yo = Wo && Wo.v8, _, de;
Yo && (_ = Yo.split("."), de = _[0] > 0 && _[0] < 4 ? 1 : +(_[0] + _[1]));
!de && pt && (_ = pt.match(/Edge\/(\d+)/), (!_ || _[1] >= 74) && (_ = pt.match(/Chrome\/(\d+)/), _ && (de = +_[1])));
var Qg = de, Ho = Qg, rh = A, uc = !!Object.getOwnPropertySymbols && !rh(function() {
  var r = Symbol();
  return !String(r) || !(Object(r) instanceof Symbol) || // Chrome 38-40 symbols are not inherited from DOM collections prototypes to instances
  !Symbol.sham && Ho && Ho < 41;
}), eh = uc, fc = eh && !Symbol.sham && typeof Symbol.iterator == "symbol", th = h, ah = Ka.exports, qo = L, nh = lc, zo = uc, $c = fc, nr = ah("wks"), q = th.Symbol, Xo = q && q.for, oh = $c ? q : q && q.withoutSetter || nh, br = function(r) {
  if (!qo(nr, r) || !(zo || typeof nr[r] == "string")) {
    var e = "Symbol." + r;
    zo && qo(q, r) ? nr[r] = q[r] : $c && Xo ? nr[r] = Xo(e) : nr[r] = oh(e);
  }
  return nr[r];
}, ih = R, Lr = function(r) {
  return typeof r == "object" ? r !== null : ih(r);
}, dc = h, vh = Lr, ch = dc.String, lh = dc.TypeError, Mr = function(r) {
  if (vh(r))
    return r;
  throw lh(ch(r) + " is not an object");
}, pc = {}, sh = A, M = !sh(function() {
  return Object.defineProperty({}, 1, { get: function() {
    return 7;
  } })[1] != 7;
}), uh = M, fh = A, yc = uh && fh(function() {
  return Object.defineProperty(function() {
  }, "prototype", {
    value: 42,
    writable: !1
  }).prototype != 42;
}), rr = {}, $h = h, Jo = Lr, va = $h.document, dh = Jo(va) && Jo(va.createElement), Ya = function(r) {
  return dh ? va.createElement(r) : {};
}, ph = M, yh = A, bh = Ya, bc = !ph && !yh(function() {
  return Object.defineProperty(bh("div"), "a", {
    get: function() {
      return 7;
    }
  }).a != 7;
}), gh = ac, Zr = Function.prototype.call, Fr = gh ? Zr.bind(Zr) : function() {
  return Zr.apply(Zr, arguments);
}, hh = x, mh = hh({}.isPrototypeOf), Oh = h, Sh = Ce, Eh = R, Th = mh, Ih = fc, wh = Oh.Object, gc = Ih ? function(r) {
  return typeof r == "symbol";
} : function(r) {
  var e = Sh("Symbol");
  return Eh(e) && Th(e.prototype, wh(r));
}, Ph = h, Rh = Ph.String, Ch = function(r) {
  try {
    return Rh(r);
  } catch {
    return "Object";
  }
}, jh = h, _h = R, xh = Ch, Nh = jh.TypeError, Ah = function(r) {
  if (_h(r))
    return r;
  throw Nh(xh(r) + " is not a function");
}, Dh = Ah, Lh = function(r, e) {
  var t = r[e];
  return t == null ? void 0 : Dh(t);
}, Mh = h, yt = Fr, bt = R, gt = Lr, Fh = Mh.TypeError, Uh = function(r, e) {
  var t, a;
  if (e === "string" && bt(t = r.toString) && !gt(a = yt(t, r)) || bt(t = r.valueOf) && !gt(a = yt(t, r)) || e !== "string" && bt(t = r.toString) && !gt(a = yt(t, r)))
    return a;
  throw Fh("Can't convert object to primitive value");
}, Bh = h, Gh = Fr, Zo = Lr, Qo = gc, Kh = Lh, kh = Uh, Vh = br, Wh = Bh.TypeError, Yh = Vh("toPrimitive"), Hh = function(r, e) {
  if (!Zo(r) || Qo(r))
    return r;
  var t = Kh(r, Yh), a;
  if (t) {
    if (e === void 0 && (e = "default"), a = Gh(t, r, e), !Zo(a) || Qo(a))
      return a;
    throw Wh("Can't convert object to primitive value");
  }
  return e === void 0 && (e = "number"), kh(r, e);
}, qh = Hh, zh = gc, hc = function(r) {
  var e = qh(r, "string");
  return zh(e) ? e : e + "";
}, Xh = h, Jh = M, Zh = bc, Qh = yc, Qr = Mr, ri = hc, r1 = Xh.TypeError, ht = Object.defineProperty, e1 = Object.getOwnPropertyDescriptor, mt = "enumerable", Ot = "configurable", St = "writable";
rr.f = Jh ? Qh ? function(e, t, a) {
  if (Qr(e), t = ri(t), Qr(a), typeof e == "function" && t === "prototype" && "value" in a && St in a && !a[St]) {
    var n = e1(e, t);
    n && n[St] && (e[t] = a.value, a = {
      configurable: Ot in a ? a[Ot] : n[Ot],
      enumerable: mt in a ? a[mt] : n[mt],
      writable: !1
    });
  }
  return ht(e, t, a);
} : ht : function(e, t, a) {
  if (Qr(e), t = ri(t), Qr(a), Zh)
    try {
      return ht(e, t, a);
    } catch {
    }
  if ("get" in a || "set" in a)
    throw r1("Accessors not supported");
  return "value" in a && (e[t] = a.value), e;
};
var t1 = Math.ceil, a1 = Math.floor, mc = function(r) {
  var e = +r;
  return e !== e || e === 0 ? 0 : (e > 0 ? a1 : t1)(e);
}, n1 = mc, o1 = Math.max, i1 = Math.min, v1 = function(r, e) {
  var t = n1(r);
  return t < 0 ? o1(t + e, 0) : i1(t, e);
}, c1 = mc, l1 = Math.min, s1 = function(r) {
  return r > 0 ? l1(c1(r), 9007199254740991) : 0;
}, u1 = s1, f1 = function(r) {
  return u1(r.length);
}, $1 = Dr, d1 = v1, p1 = f1, ei = function(r) {
  return function(e, t, a) {
    var n = $1(e), o = p1(n), i = d1(a, o), v;
    if (r && t != t) {
      for (; o > i; )
        if (v = n[i++], v != v)
          return !0;
    } else
      for (; o > i; i++)
        if ((r || i in n) && n[i] === t)
          return r || i || 0;
    return !r && -1;
  };
}, y1 = {
  // `Array.prototype.includes` method
  // https://tc39.es/ecma262/#sec-array.prototype.includes
  includes: ei(!0),
  // `Array.prototype.indexOf` method
  // https://tc39.es/ecma262/#sec-array.prototype.indexof
  indexOf: ei(!1)
}, Ha = {}, b1 = x, Et = L, g1 = Dr, h1 = y1.indexOf, m1 = Ha, ti = b1([].push), Oc = function(r, e) {
  var t = g1(r), a = 0, n = [], o;
  for (o in t)
    !Et(m1, o) && Et(t, o) && ti(n, o);
  for (; e.length > a; )
    Et(t, o = e[a++]) && (~h1(n, o) || ti(n, o));
  return n;
}, qa = [
  "constructor",
  "hasOwnProperty",
  "isPrototypeOf",
  "propertyIsEnumerable",
  "toLocaleString",
  "toString",
  "valueOf"
], O1 = Oc, S1 = qa, Sc = Object.keys || function(e) {
  return O1(e, S1);
}, E1 = M, T1 = yc, I1 = rr, w1 = Mr, P1 = Dr, R1 = Sc;
pc.f = E1 && !T1 ? Object.defineProperties : function(e, t) {
  w1(e);
  for (var a = P1(t), n = R1(t), o = n.length, i = 0, v; o > i; )
    I1.f(e, v = n[i++], a[v]);
  return e;
};
var C1 = Ce, j1 = C1("document", "documentElement"), _1 = Ka.exports, x1 = lc, ai = _1("keys"), za = function(r) {
  return ai[r] || (ai[r] = x1(r));
}, N1 = Mr, A1 = pc, ni = qa, D1 = Ha, L1 = j1, M1 = Ya, F1 = za, oi = ">", ii = "<", ca = "prototype", la = "script", Ec = F1("IE_PROTO"), Tt = function() {
}, Tc = function(r) {
  return ii + la + oi + r + ii + "/" + la + oi;
}, vi = function(r) {
  r.write(Tc("")), r.close();
  var e = r.parentWindow.Object;
  return r = null, e;
}, U1 = function() {
  var r = M1("iframe"), e = "java" + la + ":", t;
  return r.style.display = "none", L1.appendChild(r), r.src = String(e), t = r.contentWindow.document, t.open(), t.write(Tc("document.F=Object")), t.close(), t.F;
}, re, ve = function() {
  try {
    re = new ActiveXObject("htmlfile");
  } catch {
  }
  ve = typeof document < "u" ? document.domain && re ? vi(re) : U1() : vi(re);
  for (var r = ni.length; r--; )
    delete ve[ca][ni[r]];
  return ve();
};
D1[Ec] = !0;
var Ic = Object.create || function(e, t) {
  var a;
  return e !== null ? (Tt[ca] = N1(e), a = new Tt(), Tt[ca] = null, a[Ec] = e) : a = ve(), t === void 0 ? a : A1.f(a, t);
}, B1 = br, G1 = Ic, K1 = rr, sa = B1("unscopables"), ua = Array.prototype;
ua[sa] == null && K1.f(ua, sa, {
  configurable: !0,
  value: G1(null)
});
var k1 = function(r) {
  ua[sa][r] = !0;
}, Xa = {}, V1 = x, W1 = R, fa = Va, Y1 = V1(Function.toString);
W1(fa.inspectSource) || (fa.inspectSource = function(r) {
  return Y1(r);
});
var wc = fa.inspectSource, H1 = h, q1 = R, z1 = wc, ci = H1.WeakMap, X1 = q1(ci) && /native code/.test(z1(ci)), Ja = function(r, e) {
  return {
    enumerable: !(r & 1),
    configurable: !(r & 2),
    writable: !(r & 4),
    value: e
  };
}, J1 = M, Z1 = rr, Q1 = Ja, Ur = J1 ? function(r, e, t) {
  return Z1.f(r, e, Q1(1, t));
} : function(r, e, t) {
  return r[e] = t, r;
}, rm = X1, Pc = h, It = x, em = Lr, tm = Ur, wt = L, Pt = Va, am = za, nm = Ha, li = "Object already initialized", $a = Pc.TypeError, om = Pc.WeakMap, pe, Cr, ye, im = function(r) {
  return ye(r) ? Cr(r) : pe(r, {});
}, vm = function(r) {
  return function(e) {
    var t;
    if (!em(e) || (t = Cr(e)).type !== r)
      throw $a("Incompatible receiver, " + r + " required");
    return t;
  };
};
if (rm || Pt.state) {
  var V = Pt.state || (Pt.state = new om()), cm = It(V.get), si = It(V.has), lm = It(V.set);
  pe = function(r, e) {
    if (si(V, r))
      throw new $a(li);
    return e.facade = r, lm(V, r, e), e;
  }, Cr = function(r) {
    return cm(V, r) || {};
  }, ye = function(r) {
    return si(V, r);
  };
} else {
  var or = am("state");
  nm[or] = !0, pe = function(r, e) {
    if (wt(r, or))
      throw new $a(li);
    return e.facade = r, tm(r, or, e), e;
  }, Cr = function(r) {
    return wt(r, or) ? r[or] : {};
  }, ye = function(r) {
    return wt(r, or);
  };
}
var Rc = {
  set: pe,
  get: Cr,
  has: ye,
  enforce: im,
  getterFor: vm
}, Za = {}, Qa = {}, Cc = {}.propertyIsEnumerable, jc = Object.getOwnPropertyDescriptor, sm = jc && !Cc.call({ 1: 2 }, 1);
Qa.f = sm ? function(e) {
  var t = jc(this, e);
  return !!t && t.enumerable;
} : Cc;
var um = M, fm = Fr, $m = Qa, dm = Ja, pm = Dr, ym = hc, bm = L, gm = bc, ui = Object.getOwnPropertyDescriptor;
Za.f = um ? ui : function(e, t) {
  if (e = pm(e), t = ym(t), gm)
    try {
      return ui(e, t);
    } catch {
    }
  if (bm(e, t))
    return dm(!fm($m.f, e, t), e[t]);
};
var je = { exports: {} }, da = M, hm = L, _c = Function.prototype, mm = da && Object.getOwnPropertyDescriptor, rn = hm(_c, "name"), Om = rn && (function() {
}).name === "something", Sm = rn && (!da || da && mm(_c, "name").configurable), xc = {
  EXISTS: rn,
  PROPER: Om,
  CONFIGURABLE: Sm
}, Em = h, fi = R, Tm = L, $i = Ur, Im = ka, wm = wc, Nc = Rc, Pm = xc.CONFIGURABLE, Rm = Nc.get, Cm = Nc.enforce, jm = String(String).split("String");
(je.exports = function(r, e, t, a) {
  var n = a ? !!a.unsafe : !1, o = a ? !!a.enumerable : !1, i = a ? !!a.noTargetGet : !1, v = a && a.name !== void 0 ? a.name : e, s;
  if (fi(t) && (String(v).slice(0, 7) === "Symbol(" && (v = "[" + String(v).replace(/^Symbol\(([^)]*)\)/, "$1") + "]"), (!Tm(t, "name") || Pm && t.name !== v) && $i(t, "name", v), s = Cm(t), s.source || (s.source = jm.join(typeof v == "string" ? v : ""))), r === Em) {
    o ? r[e] = t : Im(e, t);
    return;
  } else
    n ? !i && r[e] && (o = !0) : delete r[e];
  o ? r[e] = t : $i(r, e, t);
})(Function.prototype, "toString", function() {
  return fi(this) && Rm(this).source || wm(this);
});
var Ac = {}, _m = Oc, xm = qa, Nm = xm.concat("length", "prototype");
Ac.f = Object.getOwnPropertyNames || function(e) {
  return _m(e, Nm);
};
var en = {};
en.f = Object.getOwnPropertySymbols;
var Am = Ce, Dm = x, Lm = Ac, Mm = en, Fm = Mr, Um = Dm([].concat), Bm = Am("Reflect", "ownKeys") || function(e) {
  var t = Lm.f(Fm(e)), a = Mm.f;
  return a ? Um(t, a(e)) : t;
}, di = L, Gm = Bm, Km = Za, km = rr, Vm = function(r, e, t) {
  for (var a = Gm(e), n = km.f, o = Km.f, i = 0; i < a.length; i++) {
    var v = a[i];
    !di(r, v) && !(t && di(t, v)) && n(r, v, o(e, v));
  }
}, Wm = A, Ym = R, Hm = /#|\.prototype\./, Br = function(r, e) {
  var t = zm[qm(r)];
  return t == Jm ? !0 : t == Xm ? !1 : Ym(e) ? Wm(e) : !!e;
}, qm = Br.normalize = function(r) {
  return String(r).replace(Hm, ".").toLowerCase();
}, zm = Br.data = {}, Xm = Br.NATIVE = "N", Jm = Br.POLYFILL = "P", Zm = Br, Rt = h, Qm = Za.f, rO = Ur, eO = je.exports, tO = ka, aO = Vm, nO = Zm, Dc = function(r, e) {
  var t = r.target, a = r.global, n = r.stat, o, i, v, s, l, c;
  if (a ? i = Rt : n ? i = Rt[t] || tO(t, {}) : i = (Rt[t] || {}).prototype, i)
    for (v in e) {
      if (l = e[v], r.noTargetGet ? (c = Qm(i, v), s = c && c.value) : s = i[v], o = nO(a ? v : t + (n ? "." : "#") + v, r.forced), !o && s !== void 0) {
        if (typeof l == typeof s)
          continue;
        aO(l, s);
      }
      (r.sham || s && s.sham) && rO(l, "sham", !0), eO(i, v, l, r);
    }
}, oO = A, iO = !oO(function() {
  function r() {
  }
  return r.prototype.constructor = null, Object.getPrototypeOf(new r()) !== r.prototype;
}), vO = h, cO = L, lO = R, sO = Wa, uO = za, fO = iO, pi = uO("IE_PROTO"), pa = vO.Object, $O = pa.prototype, Lc = fO ? pa.getPrototypeOf : function(r) {
  var e = sO(r);
  if (cO(e, pi))
    return e[pi];
  var t = e.constructor;
  return lO(t) && e instanceof t ? t.prototype : e instanceof pa ? $O : null;
}, dO = A, pO = R, yi = Lc, yO = je.exports, bO = br, ya = bO("iterator"), Mc = !1, X, Ct, jt;
[].keys && (jt = [].keys(), "next" in jt ? (Ct = yi(yi(jt)), Ct !== Object.prototype && (X = Ct)) : Mc = !0);
var gO = X == null || dO(function() {
  var r = {};
  return X[ya].call(r) !== r;
});
gO && (X = {});
pO(X[ya]) || yO(X, ya, function() {
  return this;
});
var Fc = {
  IteratorPrototype: X,
  BUGGY_SAFARI_ITERATORS: Mc
}, hO = rr.f, mO = L, OO = br, bi = OO("toStringTag"), Uc = function(r, e, t) {
  r && !t && (r = r.prototype), r && !mO(r, bi) && hO(r, bi, { configurable: !0, value: e });
}, SO = Fc.IteratorPrototype, EO = Ic, TO = Ja, IO = Uc, wO = Xa, PO = function() {
  return this;
}, RO = function(r, e, t, a) {
  var n = e + " Iterator";
  return r.prototype = EO(SO, { next: TO(+!a, t) }), IO(r, n, !1), wO[n] = PO, r;
}, Bc = h, CO = R, jO = Bc.String, _O = Bc.TypeError, xO = function(r) {
  if (typeof r == "object" || CO(r))
    return r;
  throw _O("Can't set " + jO(r) + " as a prototype");
}, NO = x, AO = Mr, DO = xO, LO = Object.setPrototypeOf || ("__proto__" in {} ? function() {
  var r = !1, e = {}, t;
  try {
    t = NO(Object.getOwnPropertyDescriptor(Object.prototype, "__proto__").set), t(e, []), r = e instanceof Array;
  } catch {
  }
  return function(n, o) {
    return AO(n), DO(o), r ? t(n, o) : n.__proto__ = o, n;
  };
}() : void 0), MO = Dc, FO = Fr, Gc = xc, UO = R, BO = RO, gi = Lc, hi = LO, GO = Uc, KO = Ur, _t = je.exports, kO = br, VO = Xa, Kc = Fc, WO = Gc.PROPER, YO = Gc.CONFIGURABLE, mi = Kc.IteratorPrototype, ee = Kc.BUGGY_SAFARI_ITERATORS, Or = kO("iterator"), Oi = "keys", Sr = "values", Si = "entries", HO = function() {
  return this;
}, qO = function(r, e, t, a, n, o, i) {
  BO(t, e, a);
  var v = function(b) {
    if (b === n && f)
      return f;
    if (!ee && b in c)
      return c[b];
    switch (b) {
      case Oi:
        return function() {
          return new t(this, b);
        };
      case Sr:
        return function() {
          return new t(this, b);
        };
      case Si:
        return function() {
          return new t(this, b);
        };
    }
    return function() {
      return new t(this);
    };
  }, s = e + " Iterator", l = !1, c = r.prototype, u = c[Or] || c["@@iterator"] || n && c[n], f = !ee && u || v(n), y = e == "Array" && c.entries || u, d, p, g;
  if (y && (d = gi(y.call(new r())), d !== Object.prototype && d.next && (gi(d) !== mi && (hi ? hi(d, mi) : UO(d[Or]) || _t(d, Or, HO)), GO(d, s, !0))), WO && n == Sr && u && u.name !== Sr && (YO ? KO(c, "name", Sr) : (l = !0, f = function() {
    return FO(u, this);
  })), n)
    if (p = {
      values: v(Sr),
      keys: o ? f : v(Oi),
      entries: v(Si)
    }, i)
      for (g in p)
        (ee || l || !(g in c)) && _t(c, g, p[g]);
    else
      MO({ target: e, proto: !0, forced: ee || l }, p);
  return c[Or] !== f && _t(c, Or, f, { name: n }), VO[e] = f, p;
}, zO = Dr, tn = k1, Ei = Xa, kc = Rc, XO = rr.f, JO = qO, ZO = M, Vc = "Array Iterator", QO = kc.set, rS = kc.getterFor(Vc), eS = JO(Array, "Array", function(r, e) {
  QO(this, {
    type: Vc,
    target: zO(r),
    // target
    index: 0,
    // next index
    kind: e
    // kind
  });
}, function() {
  var r = rS(this), e = r.target, t = r.kind, a = r.index++;
  return !e || a >= e.length ? (r.target = void 0, { value: void 0, done: !0 }) : t == "keys" ? { value: a, done: !1 } : t == "values" ? { value: e[a], done: !1 } : { value: [a, e[a]], done: !1 };
}, "values"), Ti = Ei.Arguments = Ei.Array;
tn("keys");
tn("values");
tn("entries");
if (ZO && Ti.name !== "values")
  try {
    XO(Ti, "name", { value: "values" });
  } catch {
  }
var tS = {
  CSSRuleList: 0,
  CSSStyleDeclaration: 0,
  CSSValueList: 0,
  ClientRectList: 0,
  DOMRectList: 0,
  DOMStringList: 0,
  DOMTokenList: 1,
  DataTransferItemList: 0,
  FileList: 0,
  HTMLAllCollection: 0,
  HTMLCollection: 0,
  HTMLFormElement: 0,
  HTMLSelectElement: 0,
  MediaList: 0,
  MimeTypeArray: 0,
  NamedNodeMap: 0,
  NodeList: 1,
  PaintRequestList: 0,
  Plugin: 0,
  PluginArray: 0,
  SVGLengthList: 0,
  SVGNumberList: 0,
  SVGPathSegList: 0,
  SVGPointList: 0,
  SVGStringList: 0,
  SVGTransformList: 0,
  SourceBufferList: 0,
  StyleSheetList: 0,
  TextTrackCueList: 0,
  TextTrackList: 0,
  TouchList: 0
}, aS = Ya, xt = aS("span").classList, Ii = xt && xt.constructor && xt.constructor.prototype, nS = Ii === Object.prototype ? void 0 : Ii, wi = h, Wc = tS, oS = nS, Er = eS, Nt = Ur, Yc = br, At = Yc("iterator"), Pi = Yc("toStringTag"), Dt = Er.values, Hc = function(r, e) {
  if (r) {
    if (r[At] !== Dt)
      try {
        Nt(r, At, Dt);
      } catch {
        r[At] = Dt;
      }
    if (r[Pi] || Nt(r, Pi, e), Wc[e]) {
      for (var t in Er)
        if (r[t] !== Er[t])
          try {
            Nt(r, t, Er[t]);
          } catch {
            r[t] = Er[t];
          }
    }
  }
};
for (var Lt in Wc)
  Hc(wi[Lt] && wi[Lt].prototype, Lt);
Hc(oS, "DOMTokenList");
var Ri = M, iS = x, vS = Fr, cS = A, Mt = Sc, lS = en, sS = Qa, uS = Wa, fS = vc, ir = Object.assign, Ci = Object.defineProperty, $S = iS([].concat), dS = !ir || cS(function() {
  if (Ri && ir({ b: 1 }, ir(Ci({}, "a", {
    enumerable: !0,
    get: function() {
      Ci(this, "b", {
        value: 3,
        enumerable: !1
      });
    }
  }), { b: 2 })).b !== 1)
    return !0;
  var r = {}, e = {}, t = Symbol(), a = "abcdefghijklmnopqrst";
  return r[t] = 7, a.split("").forEach(function(n) {
    e[n] = n;
  }), ir({}, r)[t] != 7 || Mt(ir({}, e)).join("") != a;
}) ? function(e, t) {
  for (var a = uS(e), n = arguments.length, o = 1, i = lS.f, v = sS.f; n > o; )
    for (var s = fS(arguments[o++]), l = i ? $S(Mt(s), i(s)) : Mt(s), c = l.length, u = 0, f; c > u; )
      f = l[u++], (!Ri || vS(v, s, f)) && (a[f] = s[f]);
  return a;
} : ir, pS = Dc, ji = dS;
pS({ target: "Object", stat: !0, forced: Object.assign !== ji }, {
  assign: ji
});
function _i(r, e, t, a) {
  function n(o) {
    return o instanceof t ? o : new t(function(i) {
      i(o);
    });
  }
  return new (t || (t = Promise))(function(o, i) {
    function v(c) {
      try {
        l(a.next(c));
      } catch (u) {
        i(u);
      }
    }
    function s(c) {
      try {
        l(a.throw(c));
      } catch (u) {
        i(u);
      }
    }
    function l(c) {
      c.done ? o(c.value) : n(c.value).then(v, s);
    }
    l((a = a.apply(r, e || [])).next());
  });
}
const yS = {
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  loaded: () => {
  },
  googleCheckContentLoaded: !0,
  googleCheckInterval: 3e3,
  queryParams: "",
  url: "",
  overrideLocalhost: "",
  googleMaxChecks: 5,
  viewer: "google",
  viewerUrl: "",
  style: {
    width: "100%",
    height: "100%"
  },
  className: ""
}, bS = (r) => {
  var e, t, a, n, o;
  const i = Ae(null), [v, s] = Ni({
    url: "",
    isSwitching: !1,
    externalViewer: !0,
    docHtml: {
      __html: ""
    }
  }), l = Ae(), c = Ae(), u = (d) => _i(void 0, void 0, void 0, function* () {
    if (c.current) {
      const p = i.current;
      l && l.current && l.current.unsubscribe(), c.current.viewer === "google" && c.current.googleCheckContentLoaded === !0 && f(p, d.url, c.current.googleCheckInterval, c.current.googleMaxChecks);
    }
  });
  ln(() => {
    let d;
    return v.isSwitching && (s((p) => Object.assign(Object.assign({}, p), {
      isSwitching: !1
    })), d = setTimeout(() => {
      u({
        url: v.url,
        externalViewer: v.externalViewer
      });
    }, 500)), () => {
      d && clearTimeout(d);
    };
  }, [v.isSwitching]), ln(() => {
    c.current = Object.assign(Object.assign({}, yS), r);
    let d = Fo(c.current.url, c.current.viewer, c.current.queryParams, c.current.viewerUrl);
    if (d.externalViewer && c.current.overrideLocalhost && hg(c.current.url)) {
      const p = gg(c.current.url, c.current.overrideLocalhost);
      d = Fo(p, c.current.viewer, c.current.queryParams, c.current.viewerUrl);
    }
    if (s({
      url: d.url,
      externalViewer: d.externalViewer,
      isSwitching: !1,
      docHtml: {
        __html: ""
      }
    }), i && i.current) {
      const p = i.current;
      p.src && p.src !== `${window.location.protocol}//${window.location.host}/` && p.src !== d.url ? s((g) => Object.assign(Object.assign({}, g), {
        isSwitching: !0
      })) : u(d);
    } else
      c.current.viewer === "mammoth" && _i(void 0, void 0, void 0, function* () {
        const g = {
          __html: yield pg(d.url)
        };
        s({
          url: "",
          docHtml: g,
          isSwitching: !1,
          externalViewer: !1
        });
      });
    return () => {
      l && l.current && l.current.unsubscribe();
    };
  }, [r]);
  const f = (d, p, g, b) => {
    l.current = yg(), l.current.subscribe(d, g, b);
  }, y = () => {
    c.current && i && i.current !== null && tc(i.current) && (c.current.loaded && c.current.loaded(), l.current && l.current.unsubscribe());
  };
  return v.isSwitching ? null : v.externalViewer ? w("iframe", {
    style: (e = c.current) === null || e === void 0 ? void 0 : e.style,
    className: (t = c.current) === null || t === void 0 ? void 0 : t.className,
    ref: i,
    onLoad: () => {
      y();
    },
    id: "iframe",
    title: "iframe",
    frameBorder: "0",
    src: v.url
  }) : ((a = c.current) === null || a === void 0 ? void 0 : a.viewer) !== "pdf" ? w("div", {
    dangerouslySetInnerHTML: v.docHtml
  }) : v.url ? w("object", Object.assign({
    data: v.url,
    style: (n = c.current) === null || n === void 0 ? void 0 : n.style,
    className: (o = c.current) === null || o === void 0 ? void 0 : o.className,
    type: "application/pdf",
    width: "100%",
    height: "100%"
  }, {
    children: Ft("p", {
      children: ["Your browser does not support PDFs.", w("a", Object.assign({
        href: v.url
      }, {
        children: "Download the PDF"
      })), "."]
    })
  })) : null;
}, qc = (r) => nl`
    width: ${ol(r.margin)};	
    height: ${il(r.margin)};	
    margin: ${r.margin};	
    padding: ${r.padding};

    overflow: hidden;
    background-color: ${r.background};
    border: ${(e) => r.borderWidth ? r.borderWidth : "1px"} solid ${r.border};
    border-radius: calc(min(${r.radius}, 20px));
  `, gS = xi.div`
  display: flex;
  height: 100%;
  justify-content: center;
  align-items: center;
  ${(r) => r.$style && qc(r.$style)}
  ${(r) => r.$animationStyle}
`, hS = xi.div`
  height: 100%;
  ${(r) => r.$style && qc(r.$style)}
`, mS = (r) => {
  const [e, t] = Ni(!1);
  return /* @__PURE__ */ w(hS, {
    $style: r.style,
    onClick: (a) => t(!0),
    onMouseLeave: (a) => t(!1),
    children: /* @__PURE__ */ w(bS, {
      style: {
        pointerEvents: e ? "auto" : "none",
        width: "100%",
        height: "100%"
      },
      url: r.src,
      viewer: "url"
    })
  });
};
let ba = function() {
  const r = {
    src: zc,
    style: on(Xc, "style"),
    animationStyle: on(Jc, "animationStyle")
  };
  return new Zc(r, (e) => Qc.isEmpty(e.src) ? /* @__PURE__ */ w(gS, {
    $style: e.style,
    $animationStyle: e.animationStyle,
    children: te("fileViewer.invalidURL")
  }) : /* @__PURE__ */ w(mS, {
    src: e.src,
    style: e.style,
    animationStyle: e.animationStyle
  })).setPropertyViewFn((e) => /* @__PURE__ */ Ft(vn, {
    children: [/* @__PURE__ */ w(Gr, {
      name: Kr.basic,
      children: e.src.propertyView({
        label: te("fileViewer.src"),
        tooltip: /* @__PURE__ */ w("span", {
          children: te("fileViewer.srcTooltip")
        })
      })
    }), ["logic", "both"].includes(sn(cn).editorModeStatus) && /* @__PURE__ */ w(Gr, {
      name: Kr.interaction,
      children: rl(e)
    }), ["layout", "both"].includes(sn(cn).editorModeStatus) && /* @__PURE__ */ Ft(vn, {
      children: [/* @__PURE__ */ w(Gr, {
        name: Kr.style,
        children: e.style.getPropertyView()
      }), /* @__PURE__ */ w(Gr, {
        name: Kr.animationStyle,
        hasTooltip: !0,
        children: e.animationStyle.getPropertyView()
      })]
    })]
  })).build();
}();
ba = class extends ba {
  autoHeight() {
    return !1;
  }
};
const TS = el(ba, [new tl("src", te("fileViewer.srcDesc")), al]);
export {
  TS as FileViewerComp
};
