// lowcoder-sdk/dist/225b8bb4.js
import { X as l, ah as q, dm as D, F as d, eT as M, hC as X, hD as J, a8 as x, hE as v, G as C, dn as h, dp as k, dq as t, hF as z, dw as O, av as K, dv as f } from "./c988e26e.js";
import { useContext as Q, useRef as Y, useEffect as Z } from "react";
import "react-dom";
const {
  Header: _,
  Content: A,
  Footer: S,
  Sider: P
} = v, F = l.div`
  width: 100%;
  top: 0;
  max-width: inherit;
  overflow: auto;
  height: 100%;
`, y = (o) => K`
    border-color: ${o.border};
    border-width: ${o.borderWidth};
    border-style: ${o.borderStyle};
    border-radius: ${o.radius};
    overflow: hidden;
    padding: ${o.padding};
    ${o.background && `background-color: ${o.background};`}
    ${o.backgroundImage && `background-image: url(${o.backgroundImage});`}
    ${o.backgroundImageRepeat && `background-repeat: ${o.backgroundImageRepeat};`}
    ${o.backgroundImageSize && `background-size: ${o.backgroundImageSize};`}
    ${o.backgroundImagePosition && `background-position: ${o.backgroundImagePosition};`}
    ${o.backgroundImageOrigin && `background-origin: ${o.backgroundImageOrigin};`}
  `, rr = l.div`
  display: flex;
  flex-flow: column;
  height: 100%;
  border: 1px solid #d7d9e0;
  border-radius: 4px;
  ${(o) => o.$style && y(o.$style)}
  ${(o) => o.$animationStyle}
`, or = l(f)`
  overflow: visible;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
`, w = l(f)`
  overflow: auto;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
  ${(o) => o.$siderBackgroundImage && `background-image: url(${o.$siderBackgroundImage});`}
  ${(o) => o.$siderBackgroundImageRepeat && `background-repeat: ${o.$siderBackgroundImageRepeat};`}
  ${(o) => o.$siderBackgroundImageSize && `background-size: ${o.$siderBackgroundImageSize};`}
  ${(o) => o.$siderBackgroundImagePosition && `background-position: ${o.$siderBackgroundImagePosition};`}
  ${(o) => o.$siderBackgroundImageOrigin && `background-origin: ${o.$siderBackgroundImageOrigin};`}
`, G = l(f)`
  border-top: ${(o) => `${o.$showBorder ? o.$borderWidth : 0} solid ${o.$borderColor}`};
  flex: 1;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
`, ar = l(f)`
  border-top: ${(o) => `${o.$showBorder ? o.$borderWidth : 0} solid ${o.$borderColor}`};
  overflow: visible;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
  ${(o) => o.$footerBackgroundImage && `background-image: url(${o.$footerBackgroundImage});`}
  ${(o) => o.$footerBackgroundImageRepeat && `background-repeat: ${o.$footerBackgroundImageRepeat};`}
  ${(o) => o.$footerBackgroundImageSize && `background-size: ${o.$footerBackgroundImageSize};`}
  ${(o) => o.$footerBackgroundImagePosition && `background-position: ${o.$footerBackgroundImagePosition};`}
  ${(o) => o.$footerBackgroundImageOrigin && `background-origin: ${o.$footerBackgroundImageOrigin};`}
`;
function gr(o) {
  const {
    container: a,
    siderCollapsed: $,
    setSiderCollapsed: b,
    animationStyle: N
  } = o, {
    showHeader: u,
    showFooter: p,
    showSider: s
  } = a, {
    items: L,
    ...T
  } = a.header, {
    items: H,
    ...R
  } = a.body[0].children.view.getView(), {
    items: U,
    ...E
  } = a.footer, {
    items: B,
    ...I
  } = a.sider, {
    style: e,
    headerStyle: m,
    siderStyle: r,
    bodyStyle: i,
    footerStyle: n,
    horizontalGridCells: c
  } = a, V = Q(q).getAppSettings().maxWidth;
  D(V);
  const W = Y();
  function j(g) {
    o.setSiderCollapsed(g);
  }
  return Z(() => {
    b(a.siderCollapsed);
  }, [a.siderCollapsed]), /* @__PURE__ */ d("div", {
    style: {
      padding: e.margin,
      height: "100%"
    },
    children: /* @__PURE__ */ d(M, {
      theme: {
        components: {
          Layout: {
            triggerBg: X(a.siderStyle.siderBackground),
            triggerColor: J(a.siderStyle.siderBackground, "black", "white"),
            siderBg: a.siderStyle.siderBackground
          }
        }
      },
      children: /* @__PURE__ */ x(rr, {
        $style: e,
        $animationStyle: N,
        children: [/* @__PURE__ */ x(v, {
          style: {
            padding: "0px"
          },
          hasSider: s && !a.innerSider,
          children: [s && !a.innerSider && !a.siderRight && /* @__PURE__ */ d(C, {
            children: /* @__PURE__ */ d(h.Provider, {
              value: r == null ? void 0 : r.siderBackground,
              children: /* @__PURE__ */ d(P, {
                width: a.siderWidth,
                style: {
                  padding: "0px",
                  margin: "0px",
                  backgroundColor: (r == null ? void 0 : r.siderBackground) || "transparent"
                },
                collapsible: a.siderCollapsible,
                breakpoint: "sm",
                collapsedWidth: a.siderCollapsedWidth,
                collapsed: $,
                onCollapse: (g) => j(g),
                children: /* @__PURE__ */ d(k, {
                  style: {
                    height: a.autoHeight ? "auto" : "100%",
                    margin: "0px",
                    padding: "0px"
                  },
                  hideScrollbar: !a.siderScrollbars,
                  children: /* @__PURE__ */ d(w, {
                    ...I,
                    items: t(B),
                    horizontalGridCells: c,
                    autoHeight: !0,
                    emptyRows: 30,
                    minHeight: `calc(100vh - ${e.padding}px)`,
                    containerPadding: [0, 0],
                    showName: {
                      bottom: p ? 20 : 0
                    },
                    $backgroundColor: (r == null ? void 0 : r.siderBackground) || "transparent",
                    $siderBackgroundImage: r == null ? void 0 : r.siderBackgroundImage,
                    $siderBackgroundImageRepeat: r == null ? void 0 : r.siderBackgroundImageRepeat,
                    $siderBackgroundImageSize: r == null ? void 0 : r.siderBackgroundImageSize,
                    $siderBackgroundImagePosition: r == null ? void 0 : r.siderBackgroundImagePosition,
                    $siderBackgroundImageOrigin: r == null ? void 0 : r.siderBackgroundImageOrigin,
                    style: {
                      padding: r.containerSiderPadding
                    }
                  })
                })
              })
            })
          }), /* @__PURE__ */ x(v, {
            style: {
              padding: "0px"
            },
            children: [u && /* @__PURE__ */ d(C, {
              children: /* @__PURE__ */ d(h.Provider, {
                value: e.background,
                children: /* @__PURE__ */ d(_, {
                  style: {
                    backgroundColor: (m == null ? void 0 : m.headerBackground) || "transparent",
                    padding: "0px",
                    margin: "0px"
                  },
                  children: /* @__PURE__ */ d(or, {
                    ...T,
                    items: t(L),
                    horizontalGridCells: c,
                    autoHeight: !0,
                    emptyRows: 5,
                    minHeight: "46px",
                    containerPadding: [0, 0],
                    showName: {
                      bottom: p ? 20 : 0
                    },
                    $backgroundColor: (m == null ? void 0 : m.headerBackground) || "transparent",
                    style: {
                      padding: m.containerHeaderPadding
                    }
                  })
                })
              })
            }), s && a.innerSider ? /* @__PURE__ */ d(C, {
              children: /* @__PURE__ */ x(v, {
                style: {
                  padding: "0px"
                },
                hasSider: s && a.innerSider,
                children: [s && !a.siderRight && /* @__PURE__ */ d(h.Provider, {
                  value: r == null ? void 0 : r.siderBackground,
                  children: /* @__PURE__ */ d(P, {
                    width: a.siderWidth,
                    style: {
                      padding: "0px",
                      margin: "0px",
                      marginTop: e.borderWidth,
                      backgroundColor: (r == null ? void 0 : r.siderBackground) || "transparent"
                    },
                    collapsible: a.siderCollapsible,
                    breakpoint: "sm",
                    collapsedWidth: a.siderCollapsedWidth,
                    collapsed: $,
                    onCollapse: (g) => b(g),
                    children: /* @__PURE__ */ d(k, {
                      style: {
                        height: a.autoHeight ? "auto" : "100%",
                        margin: "0px",
                        padding: "0px"
                      },
                      hideScrollbar: !a.siderScrollbars,
                      children: /* @__PURE__ */ d(w, {
                        ...I,
                        items: t(B),
                        horizontalGridCells: c,
                        autoHeight: !0,
                        emptyRows: 30,
                        minHeight: `calc(100vh - ${e.padding}px)`,
                        containerPadding: [0, 0],
                        showName: {
                          bottom: p ? 20 : 0
                        },
                        $backgroundColor: (r == null ? void 0 : r.siderBackground) || "transparent",
                        $siderBackgroundImage: r == null ? void 0 : r.siderBackgroundImage,
                        $siderBackgroundImageRepeat: r == null ? void 0 : r.siderBackgroundImageRepeat,
                        $siderBackgroundImageSize: r == null ? void 0 : r.siderBackgroundImageSize,
                        $siderBackgroundImagePosition: r == null ? void 0 : r.siderBackgroundImagePosition,
                        $siderBackgroundImageOrigin: r == null ? void 0 : r.siderBackgroundImageOrigin,
                        style: {
                          padding: r.containerSiderPadding
                        }
                      })
                    })
                  })
                }), /* @__PURE__ */ d(A, {
                  style: {
                    padding: "0px",
                    margin: "0px",
                    backgroundColor: (i == null ? void 0 : i.background) || "transparent"
                  },
                  children: /* @__PURE__ */ d(k, {
                    style: {
                      height: a.autoHeight ? "auto" : "100%",
                      margin: "0px",
                      padding: "0px"
                    },
                    hideScrollbar: !a.contentScrollbars,
                    children: a.showApp && a.contentApp != "" ? /* @__PURE__ */ d(h.Provider, {
                      value: i == null ? void 0 : i.background,
                      children: /* @__PURE__ */ d(F, {
                        children: /* @__PURE__ */ d(z, {
                          ref: W,
                          appId: a.contentApp,
                          baseUrl: a.baseUrl
                        })
                      })
                    }) : /* @__PURE__ */ d(G, {
                      $showBorder: u,
                      ...R,
                      items: t(H),
                      horizontalGridCells: c,
                      autoHeight: a.autoHeight,
                      emptyRows: 14,
                      minHeight: u ? "143px" : "142px",
                      containerPadding: [0, 0],
                      hintPlaceholder: o.hintPlaceholder ?? O,
                      $backgroundColor: (i == null ? void 0 : i.background) || "transparent",
                      $borderColor: e == null ? void 0 : e.border,
                      $borderWidth: e == null ? void 0 : e.borderWidth,
                      style: {
                        padding: i.containerBodyPadding
                      }
                    })
                  })
                }), s && a.siderRight && /* @__PURE__ */ d(h.Provider, {
                  value: r == null ? void 0 : r.siderBackground,
                  children: /* @__PURE__ */ d(P, {
                    width: a.siderWidth,
                    style: {
                      padding: "0px",
                      margin: "0px",
                      backgroundColor: (r == null ? void 0 : r.siderBackground) || "transparent"
                    },
                    collapsible: a.siderCollapsible,
                    breakpoint: "sm",
                    collapsedWidth: a.siderCollapsedWidth,
                    reverseArrow: !0,
                    collapsed: $,
                    onCollapse: (g) => b(g),
                    children: /* @__PURE__ */ d(k, {
                      style: {
                        height: a.autoHeight ? "auto" : "100%",
                        margin: "0px",
                        padding: "0px"
                      },
                      hideScrollbar: !a.siderScrollbars,
                      children: /* @__PURE__ */ d(w, {
                        ...I,
                        items: t(B),
                        horizontalGridCells: c,
                        autoHeight: !0,
                        emptyRows: 30,
                        minHeight: `calc(100vh - ${e.padding}px)`,
                        containerPadding: [0, 0],
                        showName: {
                          bottom: p ? 20 : 0
                        },
                        $backgroundColor: (r == null ? void 0 : r.siderBackground) || "transparent",
                        $siderBackgroundImage: r == null ? void 0 : r.siderBackgroundImage,
                        $siderBackgroundImageRepeat: r == null ? void 0 : r.siderBackgroundImageRepeat,
                        $siderBackgroundImageSize: r == null ? void 0 : r.siderBackgroundImageSize,
                        $siderBackgroundImagePosition: r == null ? void 0 : r.siderBackgroundImagePosition,
                        $siderBackgroundImageOrigin: r == null ? void 0 : r.siderBackgroundImageOrigin,
                        style: {
                          padding: r.containerSiderPadding
                        }
                      })
                    })
                  })
                })]
              })
            }) : /* @__PURE__ */ d(A, {
              style: {
                padding: "0px",
                margin: "0px"
              },
              children: /* @__PURE__ */ d(k, {
                style: {
                  height: a.autoHeight ? "auto" : "100%",
                  margin: "0px",
                  padding: "0px"
                },
                hideScrollbar: !a.contentScrollbars,
                children: a.showApp && a.contentApp != "" ? /* @__PURE__ */ d(h.Provider, {
                  value: i == null ? void 0 : i.background,
                  children: /* @__PURE__ */ d(F, {
                    children: /* @__PURE__ */ d(z, {
                      ref: W,
                      appId: a.contentApp,
                      baseUrl: a.baseUrl
                    })
                  })
                }) : /* @__PURE__ */ d(G, {
                  $showBorder: u,
                  ...R,
                  items: t(H),
                  horizontalGridCells: c,
                  autoHeight: a.autoHeight,
                  emptyRows: 14,
                  minHeight: u ? "143px" : "142px",
                  containerPadding: [0, 0],
                  hintPlaceholder: o.hintPlaceholder ?? O,
                  $backgroundColor: (i == null ? void 0 : i.background) || "transparent",
                  $borderColor: e == null ? void 0 : e.border,
                  $borderWidth: e == null ? void 0 : e.borderWidth,
                  style: {
                    padding: i.containerBodyPadding
                  }
                })
              })
            }), p && /* @__PURE__ */ d(S, {
              style: {
                textAlign: "center",
                padding: "0px",
                margin: "0px",
                backgroundColor: (n == null ? void 0 : n.footerBackground) || "transparent"
              },
              children: /* @__PURE__ */ d(ar, {
                $showBorder: u,
                ...E,
                items: t(U),
                horizontalGridCells: c,
                autoHeight: !0,
                emptyRows: 5,
                minHeight: "48px",
                containerPadding: [0, 0],
                showName: {
                  top: u ? 20 : 0
                },
                $backgroundColor: (n == null ? void 0 : n.footerBackground) || "transparent",
                $footerBackgroundImage: n == null ? void 0 : n.footerBackgroundImage,
                $footerBackgroundImageRepeat: n == null ? void 0 : n.footerBackgroundImageRepeat,
                $footerBackgroundImageSize: n == null ? void 0 : n.footerBackgroundImageSize,
                $footerBackgroundImagePosition: n == null ? void 0 : n.footerBackgroundImagePosition,
                $footerBackgroundImageOrigin: n == null ? void 0 : n.footerBackgroundImageOrigin,
                $borderColor: e == null ? void 0 : e.border,
                $borderWidth: e == null ? void 0 : e.borderWidth,
                style: {
                  padding: n.containerFooterPadding
                }
              })
            })]
          }), s && !a.innerSider && a.siderRight && /* @__PURE__ */ d(C, {
            children: /* @__PURE__ */ d(h.Provider, {
              value: r == null ? void 0 : r.siderBackground,
              children: /* @__PURE__ */ d(P, {
                width: a.siderWidth,
                style: {
                  padding: "0px",
                  margin: "0px",
                  backgroundColor: (r == null ? void 0 : r.siderBackground) || "transparent"
                },
                collapsible: a.siderCollapsible,
                breakpoint: "sm",
                collapsedWidth: a.siderCollapsedWidth,
                reverseArrow: !0,
                collapsed: $,
                onCollapse: (g) => b(g),
                children: /* @__PURE__ */ d(k, {
                  style: {
                    height: a.autoHeight ? "auto" : "100%",
                    margin: "0px",
                    padding: "0px"
                  },
                  hideScrollbar: !a.siderScrollbars,
                  children: /* @__PURE__ */ d(w, {
                    ...I,
                    items: t(B),
                    horizontalGridCells: c,
                    autoHeight: !0,
                    emptyRows: 30,
                    minHeight: `calc(100vh - ${e.padding}px)`,
                    containerPadding: [0, 0],
                    showName: {
                      bottom: p ? 20 : 0
                    },
                    $backgroundColor: (r == null ? void 0 : r.siderBackground) || "transparent",
                    $siderBackgroundImage: r == null ? void 0 : r.siderBackgroundImage,
                    $siderBackgroundImageRepeat: r == null ? void 0 : r.siderBackgroundImageRepeat,
                    $siderBackgroundImageSize: r == null ? void 0 : r.siderBackgroundImageSize,
                    $siderBackgroundImagePosition: r == null ? void 0 : r.siderBackgroundImagePosition,
                    $siderBackgroundImageOrigin: r == null ? void 0 : r.siderBackgroundImageOrigin,
                    style: {
                      padding: r.containerSiderPadding
                    }
                  })
                })
              })
            })
          })]
        }), /* @__PURE__ */ d("style", {
          children: `
              .ant-layout-sider-trigger {
                bottom: 4px !important;
                ${a.siderRight ? `
                  right: 8px !important;
                ` : `
                  left: 8px !important;}
                `}
                
              }
            `
        })]
      })
    })
  });
}
export {
  gr as PageLayout
};
