// lowcoder-sdk/dist/2315b22f.js
import { mJ as D, i6 as j } from "./c988e26e.js";
import E from "react";
function S(o, a) {
  for (var l = 0; l < a.length; l++) {
    const f = a[l];
    if (typeof f != "string" && !Array.isArray(f)) {
      for (const i in f)
        if (i !== "default" && !(i in o)) {
          const u = Object.getOwnPropertyDescriptor(f, i);
          u && Object.defineProperty(o, i, u.get ? u : {
            enumerable: !0,
            get: () => f[i]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(o, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, y = {}, O = {};
Object.defineProperty(O, "__esModule", { value: !0 });
var x = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z" } }] }, name: "minus", theme: "outlined" };
O.default = x;
(function(o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), Object.defineProperty(o, "default", {
    enumerable: !0,
    get: function() {
      return h;
    }
  });
  var a = /* @__PURE__ */ P(E), l = /* @__PURE__ */ u(O), f = /* @__PURE__ */ u(D);
  function i(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function u(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function P(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, c = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = c ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(c) {
        return Object.getOwnPropertyDescriptor(r, c).enumerable;
      }))), n.forEach(function(c) {
        i(e, c, r[c]);
      });
    }
    return e;
  }
  function M(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(c) {
        return Object.getOwnPropertyDescriptor(e, c).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : M(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ a.createElement(f.default, b(m({}, e), {
      ref: t,
      icon: l.default
    }));
  }, v = /* @__PURE__ */ a.forwardRef(w);
  process.env.NODE_ENV !== "production" && (v.displayName = "MinusOutlined");
  var h = v;
})(y);
(function(o, a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), a.default = void 0;
  const l = f(y);
  function f(u) {
    return u && u.__esModule ? u : { default: u };
  }
  const i = l;
  a.default = i, o.exports = i;
})(d, d.exports);
var g = d.exports;
const N = /* @__PURE__ */ j(g), q = /* @__PURE__ */ S({
  __proto__: null,
  default: N
}, [g]);
export {
  q as M
};
