// lowcoder-sdk/dist/24046e7b.js
import { X as g, w as s, O as d, cw as P, cZ as S, dg as $, Z as m, ev as V, az as x, U as v, F as e, aB as D, ew as E, a8 as o, G as u, a9 as n, aa as i, Y as r, ah as y, ab as B, ac as H, ae as k, ad as z } from "./c988e26e.js";
import { useRef as F, useState as N, useContext as h } from "react";
import { formDataChildren as R, FormDataPropertyView as M } from "./85e61f03.js";
import "react-dom";
const j = g.div`
  background: ${(a) => a.$src && `url(${a.$src})`} no-repeat 50% 50%;
  background-size: contain;
`, A = g.div`
  &,
  .ant-carousel {
    height: 100%;
    background-color: ${(a) => a.$bg};
    ${(a) => a.$animationStyle}
  }
`;
let c = function() {
  const a = {
    autoPlay: s(d, !0),
    data: s(P, JSON.stringify(["https://temp.im/403x192", "https://temp.im/403x192"])),
    onEvent: S,
    showDots: s(d, !0),
    dotPosition: s($, "bottom"),
    style: m(V, "style"),
    animationStyle: m(x, "animationStyle"),
    ...R
  };
  return new v(a, (t) => {
    const l = F(null), [C, w] = N(0), p = () => {
      l.current && w(l.current.clientHeight);
    };
    return /* @__PURE__ */ e(A, {
      ref: l,
      $bg: t.style.background,
      $animationStyle: t.animationStyle,
      children: /* @__PURE__ */ e(D, {
        onResize: p,
        children: /* @__PURE__ */ e(E, {
          dots: t.showDots,
          dotPosition: t.dotPosition,
          autoplay: t.autoPlay,
          afterChange: () => t.onEvent("change"),
          children: t.data.map((b, f) => /* @__PURE__ */ e("div", {
            children: /* @__PURE__ */ e(j, {
              $src: b,
              style: {
                height: C
              }
            })
          }, f))
        })
      })
    });
  }).setPropertyViewFn((t) => /* @__PURE__ */ o(u, {
    children: [/* @__PURE__ */ e(n, {
      name: i.basic,
      children: t.data.propertyView({
        label: r("data")
      })
    }), ["logic", "both"].includes(h(y).editorModeStatus) && /* @__PURE__ */ o(u, {
      children: [/* @__PURE__ */ e(M, {
        ...t
      }), /* @__PURE__ */ o(n, {
        name: i.interaction,
        children: [t.onEvent.getPropertyView(), B(t), t.autoPlay.propertyView({
          label: r("carousel.autoPlay")
        })]
      })]
    }), ["layout", "both"].includes(h(y).editorModeStatus) && /* @__PURE__ */ o(u, {
      children: [/* @__PURE__ */ o(n, {
        name: i.layout,
        children: [t.showDots.propertyView({
          label: r("carousel.showDots")
        }), t.dotPosition.propertyView({
          label: r("carousel.dotPosition"),
          radioButton: !0
        })]
      }), /* @__PURE__ */ e(n, {
        name: i.style,
        children: t.style.getPropertyView()
      }), /* @__PURE__ */ e(n, {
        name: i.animationStyle,
        hasTooltip: !0,
        children: t.animationStyle.getPropertyView()
      })]
    })]
  })).build();
}();
c = class extends c {
  autoHeight() {
    return !1;
  }
};
const G = H(c, [new k("data", r("data")), z]);
export {
  G as CarouselComp
};
