// lowcoder-sdk/dist/252f669a.js
import { Y as t, w as s, ed as h, a4 as m, a2 as c, Z as b, eW as g, az as v, O as S, U as L, F as a, a8 as u, G as d, a9 as o, aa as n, ah as i, ab as p, ac as x, ae as w, ad as C } from "./c988e26e.js";
import { lazy as f, useContext as l } from "react";
import { defaultLottie as j } from "./0a417d48.js";
import "react-dom";
const F = f(() => import("./e0100d10.js").then((r) => ({
  default: r.Player
}))), V = [{
  label: t("jsonLottie.auto"),
  value: "auto"
}, {
  label: t("jsonLottie.onHover"),
  value: "on hover"
}], k = [{
  label: t("jsonLottie.singlePlay"),
  value: "single"
}, {
  label: t("jsonLottie.endlessLoop"),
  value: "endless"
}], O = [{
  label: "0.5x",
  value: "0.5"
}, {
  label: "0.75x",
  value: "0.75"
}, {
  label: "1x",
  value: "1"
}, {
  label: "1.5x",
  value: "1.5"
}, {
  label: "2x",
  value: "2"
}, {
  label: "2.5x",
  value: "2.5"
}, {
  label: "5x",
  value: "5"
}, {
  label: "10x",
  value: "10"
}];
let y = function() {
  const r = {
    value: s(h, JSON.stringify(j, null, 2)),
    speed: m(O, "1"),
    width: s(c, 100),
    height: s(c, 100),
    container: b(g, "container"),
    animationStyle: b(v, "animationStyle"),
    animationStart: m(V, "auto"),
    loop: m(k, "single"),
    keepLastFrame: S.DEFAULT_TRUE
  };
  return new L(r, (e) => /* @__PURE__ */ a("div", {
    style: {
      padding: `${e.container.margin}`,
      animation: e.animationStyle.animation,
      animationDelay: e.animationStyle.animationDelay,
      animationDuration: e.animationStyle.animationDuration,
      animationIterationCount: e.animationStyle.animationIterationCount
    },
    children: /* @__PURE__ */ a("div", {
      style: {
        height: "100%",
        display: "flex",
        justifyContent: "center",
        backgroundColor: `${e.container.background}`,
        padding: `${e.container.padding}`,
        rotate: e.container.rotation
      },
      children: /* @__PURE__ */ a(F, {
        keepLastFrame: e.keepLastFrame,
        autoplay: e.animationStart === "auto" && !0,
        hover: e.animationStart === "on hover" && !0,
        loop: e.loop !== "single",
        speed: Number(e.speed),
        src: e.value,
        style: {
          height: "100%",
          width: "100%",
          maxWidth: "100%",
          maxHeight: "100%"
        }
      }, [e.speed, e.animationStart, e.loop, e.value, e.keepLastFrame])
    })
  })).setPropertyViewFn((e) => /* @__PURE__ */ u(d, {
    children: [/* @__PURE__ */ a(o, {
      name: n.basic,
      children: e.value.propertyView({
        label: t("jsonLottie.lottieJson")
      })
    }), (l(i).editorModeStatus === "logic" || l(i).editorModeStatus === "both") && /* @__PURE__ */ a(d, {
      children: /* @__PURE__ */ u(o, {
        name: n.interaction,
        children: [e.speed.propertyView({
          label: t("jsonLottie.speed")
        }), e.loop.propertyView({
          label: t("jsonLottie.loop")
        }), e.animationStart.propertyView({
          label: t("jsonLottie.animationStart")
        }), e.keepLastFrame.propertyView({
          label: t("jsonLottie.keepLastFrame")
        }), p(e)]
      })
    }), (l(i).editorModeStatus === "layout" || l(i).editorModeStatus === "both") && /* @__PURE__ */ u(d, {
      children: [/* @__PURE__ */ a(o, {
        name: n.style,
        children: e.container.getPropertyView()
      }), /* @__PURE__ */ a(o, {
        name: n.animationStyle,
        hasTooltip: !0,
        children: e.animationStyle.getPropertyView()
      })]
    })]
  })).build();
}();
y = class extends y {
  autoHeight() {
    return !1;
  }
};
const M = x(y, [new w("value", t("jsonLottie.valueDesc")), C]);
export {
  M as JsonLottieComp
};
