// lowcoder-sdk/dist/27e4a09e.js
import { Y as a, X as f, w as s, ed as c, aA as m, O as h, a2 as g, a4 as y, Z as b, az as x, U as S, F as t, dp as w, ee as v, a8 as d, G as u, a9 as o, aa as r, ah as l, ab as E, ac as C, ae as V, ad as j } from "./c988e26e.js";
import { useContext as n } from "react";
import { defaultData as T } from "./0a417d48.js";
import "react-dom";
const D = [{
  label: a("jsonExplorer.default"),
  value: "shapeshifter:inverted"
}, {
  label: a("jsonExplorer.defaultDark"),
  value: "shapeshifter"
}, {
  label: a("jsonExplorer.neutralLight"),
  value: "grayscale:inverted"
}, {
  label: a("jsonExplorer.neutralDark"),
  value: "grayscale"
}, {
  label: a("jsonExplorer.azure"),
  value: "apathy:inverted"
}, {
  label: a("jsonExplorer.darkBlue"),
  value: "flat"
}], M = {
  "shapeshifter:inverted": "#ffffff",
  shapeshifter: "#000000",
  "grayscale:inverted": "#ffffff",
  grayscale: "#000000",
  "apathy:inverted": "#efffff",
  flat: "#2c3e50"
}, H = f.div`
  ${(i) => i.$animationStyle}
  height: 100%;
  overflow-y: scroll;
  background-color: ${(i) => M[i.$theme] || "#ffffff"};
  border: 1px solid #d7d9e0;
  border-radius: 4px;
  padding: 10px;
`;
let p = function() {
  const i = {
    value: s(c, JSON.stringify(T, null, 2)),
    autoHeight: s(m, "auto"),
    showVerticalScrollbar: h,
    indent: s(g, 4),
    expandToggle: h.DEFAULT_TRUE,
    theme: y(D, "shapeshifter:inverted"),
    animationStyle: b(x, "animationStyle")
  };
  return new S(i, (e) => /* @__PURE__ */ t(H, {
    $theme: e.theme,
    $animationStyle: e.animationStyle,
    children: /* @__PURE__ */ t(w, {
      hideScrollbar: !e.showVerticalScrollbar,
      children: /* @__PURE__ */ t(v, {
        name: !1,
        src: e.value,
        theme: e.theme,
        collapsed: !e.expandToggle,
        displayDataTypes: !1,
        indentWidth: e.indent
      })
    })
  })).setPropertyViewFn((e) => /* @__PURE__ */ d(u, {
    children: [/* @__PURE__ */ t(o, {
      name: r.basic,
      children: e.value.propertyView({
        label: a("export.jsonEditorDesc")
      })
    }), (n(l).editorModeStatus === "logic" || n(l).editorModeStatus === "both") && /* @__PURE__ */ d(o, {
      name: r.interaction,
      children: [E(e), e.expandToggle.propertyView({
        label: a("jsonExplorer.expandToggle")
      })]
    }), (n(l).editorModeStatus === "logic" || n(l).editorModeStatus === "both") && /* @__PURE__ */ t(o, {
      name: r.advanced,
      children: e.indent.propertyView({
        label: a("jsonExplorer.indent")
      })
    }), /* @__PURE__ */ t(o, {
      name: a("prop.height"),
      children: e.autoHeight.propertyView({
        label: a("prop.height")
      })
    }), !e.autoHeight.getView() && /* @__PURE__ */ t(o, {
      name: r.layout,
      children: e.showVerticalScrollbar.propertyView({
        label: a("prop.showVerticalScrollbar")
      })
    }), (n(l).editorModeStatus === "layout" || n(l).editorModeStatus === "both") && /* @__PURE__ */ d(u, {
      children: [/* @__PURE__ */ t(o, {
        name: r.style,
        children: e.theme.propertyView({
          label: a("jsonExplorer.theme")
        })
      }), /* @__PURE__ */ t(o, {
        name: r.animationStyle,
        hasTooltip: !0,
        children: e.animationStyle.getPropertyView()
      })]
    })]
  })).build();
}();
p = class extends p {
  autoHeight() {
    return this.children.autoHeight.getView();
  }
};
const A = C(p, [new V("value", a("jsonExplorer.valueDesc")), j]);
export {
  A as JsonExplorerComp
};
