// lowcoder-sdk/dist/2a50c1a7.js
import { X as t, ah as l, dm as I, F as n, dn as p, dp as x, a8 as f, aw as v, dq as w, dw as C, av as P, i2 as z, dv as S } from "./c988e26e.js";
import { useContext as R, useState as W, useEffect as B } from "react";
import "react-dom";
const $ = (r) => P`
    border-color: ${r.border};
    border-width: ${r.borderWidth};
    border-radius: ${r.radius};
    overflow: hidden;
    padding: ${r.padding};
    ${r.background && `background-color: ${r.background};`}
    ${r.backgroundImage && `background-image: ${r.backgroundImage};`}
    ${r.backgroundImageRepeat && `background-repeat: ${r.backgroundImageRepeat};`}
    ${r.backgroundImageSize && `background-size: ${r.backgroundImageSize};`}
    ${r.backgroundImagePosition && `background-position: ${r.backgroundImagePosition};`}
    ${r.backgroundImageOrigin && `background-origin: ${r.backgroundImageOrigin};`}
  `, H = t.div`
  display: flex;
  flex-flow: column;
  height: 100%;
  border: 1px solid #d7d9e0;
  border-radius: 4px;
  ${(r) => r.$style && $(r.$style)}
`, O = t(z)`;
  ${(r) => r.$style && $(r.$style)}
`, G = t(S)`
  border-top: ${(r) => `${r.$showBorder ? r.$borderWidth : 0} solid ${r.$borderColor}`};
  flex: 1;
  ${(r) => r.$backgroundColor && `background-color: ${r.$backgroundColor};`}
  border-radius: 0;
  ${(r) => r.$backgroundImage && `background-image: ${r.$backgroundImage};`}
  ${(r) => r.$backgroundImageRepeat && `background-repeat: ${r.$backgroundImageRepeat};`}
  ${(r) => r.$backgroundImageSize && `background-size: ${r.$backgroundImageSize};`}
  ${(r) => r.$backgroundImagePosition && `background-position: ${r.$backgroundImagePosition};`}
  ${(r) => r.$backgroundImageOrigin && `background-origin: ${r.$backgroundImageOrigin};`}
`;
function A(r) {
  const {
    container: i,
    icon: g
  } = r, b = i.scrollbars, {
    items: s,
    ...m
  } = i.body[0].children.view.getView(), {
    style: o,
    headerStyle: j,
    bodyStyle: a,
    footerStyle: E
  } = i, k = R(l).getAppSettings().maxWidth;
  I(k);
  let [d, h] = W({
    value: "star",
    index: 0
  });
  return B(() => {
    var c, u;
    if ((c = g.props) != null && c.value) {
      let e = (u = g.props) == null ? void 0 : u.value;
      h({
        index: parseInt(e == null ? void 0 : e.split("_")[0]),
        value: e == null ? void 0 : e.split("_")[1]
      });
    }
  }, [g.props]), /* @__PURE__ */ n("div", {
    style: {
      padding: o.margin,
      height: "100%"
    },
    children: /* @__PURE__ */ n(H, {
      $style: o,
      children: /* @__PURE__ */ n(p.Provider, {
        value: a.background,
        children: /* @__PURE__ */ n(x, {
          style: {
            height: i.autoHeight ? "auto" : "100%",
            margin: "0px",
            padding: "0px"
          },
          hideScrollbar: !b,
          children: /* @__PURE__ */ f("div", {
            style: {
              position: "relative",
              height: "100%"
            },
            children: [/* @__PURE__ */ n(O, {
              $style: o,
              type: d == null ? void 0 : d.value,
              noise: !1,
              index: d.index,
              styles: {
                position: "absolute",
                top: "0",
                left: "50%",
                transform: "translateX(-50%)",
                width: v(o.margin),
                height: "100%"
              }
            }), /* @__PURE__ */ n(G, {
              $showBorder: !1,
              ...m,
              items: w(s),
              autoHeight: i.autoHeight,
              emptyRows: 14,
              minHeight: "142px",
              hintPlaceholder: r.hintPlaceholder ?? C,
              $backgroundColor: (a == null ? void 0 : a.background) || "transparent",
              $borderColor: o == null ? void 0 : o.border,
              $borderWidth: o == null ? void 0 : o.borderWidth,
              $backgroundImage: a == null ? void 0 : a.backgroundImage,
              $backgroundImageRepeat: a == null ? void 0 : a.backgroundImageRepeat,
              $backgroundImageSize: a == null ? void 0 : a.backgroundImageSize,
              $backgroundImagePosition: a == null ? void 0 : a.backgroundImagePosition,
              $backgroundImageOrigin: a == null ? void 0 : a.backgroundImageOrigin,
              style: {
                zIndex: 999
              }
            })]
          })
        })
      })
    })
  });
}
export {
  A as ShapeTriContainer
};
