// lowcoder-sdk/dist/3ca2e906.js
import { mJ as F, i6 as M } from "./c988e26e.js";
import U from "react";
function j(u, i) {
  for (var o = 0; o < i.length; o++) {
    const l = i[o];
    if (typeof l != "string" && !Array.isArray(l)) {
      for (const c in l)
        if (c !== "default" && !(c in u)) {
          const a = Object.getOwnPropertyDescriptor(l, c);
          a && Object.defineProperty(u, c, a.get ? a : {
            enumerable: !0,
            get: () => l[c]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(u, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, b = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M408 312h48c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm352 120V144c0-17.7-14.3-32-32-32H296c-17.7 0-32 14.3-32 32v288c-66.2 0-120 52.1-120 116v356c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8V548c0-63.9-53.8-116-120-116zm-72 0H336V184h352v248zM568 312h48c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z" } }] }, name: "usb", theme: "filled" };
_.default = E;
(function(u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), Object.defineProperty(u, "default", {
    enumerable: !0,
    get: function() {
      return D;
    }
  });
  var i = /* @__PURE__ */ g(U), o = /* @__PURE__ */ a(_), l = /* @__PURE__ */ a(F);
  function c(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function g(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = f ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(f) {
        return Object.getOwnPropertyDescriptor(r, f).enumerable;
      }))), n.forEach(function(f) {
        c(e, f, r[f]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function h(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ i.createElement(l.default, h(P({}, e), {
      ref: t,
      icon: o.default
    }));
  }, v = /* @__PURE__ */ i.forwardRef(w);
  process.env.NODE_ENV !== "production" && (v.displayName = "UsbFilled");
  var D = v;
})(b);
(function(u, i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), i.default = void 0;
  const o = l(b);
  function l(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const c = o;
  i.default = c, u.exports = c;
})(d, d.exports);
var y = d.exports;
const S = /* @__PURE__ */ M(y), N = /* @__PURE__ */ j({
  __proto__: null,
  default: S
}, [y]);
export {
  N as U
};
