// lowcoder-sdk/dist/3eacf14a.js
import { generateInsertSql as o } from "./bf2ffa97.js";
import { numberCompSelection as a, timeCompSelection as n, timestampCompSelection as s, dateTimeCompSelection as r, dateCompSelection as i, booleanCompSelection as m, setCompSelection as l, enumCompSelection as u, stringCompSelection as p } from "./90e6d3f7.js";
function b(e) {
  if (e)
    switch (e.toLowerCase()) {
      case "char":
      case "varchar":
      case "binary":
      case "varbinary":
      case "tinyblob":
      case "tinytext":
      case "text":
      case "blob":
      case "mediumtext":
      case "mediumblob":
      case "longtext":
      case "longblob":
        return p();
      case "enum":
        return u();
      case "set":
        return l();
      case "bool":
      case "boolean":
        return m();
      case "bit":
      case "tinyint":
      case "smallint":
      case "mediumint":
      case "int":
      case "integer":
      case "bigint":
      case "serial":
      case "float":
      case "double":
      case "double precision":
      case "decimal":
      case "dec":
        return a(!0);
      case "date":
        return i();
      case "datetime":
        return r();
      case "timestamp":
        return s();
      case "time":
        return n();
      case "year":
        return a();
    }
}
function d(e, t, c) {
  return {
    compType: "mysql",
    comp: {
      sql: o(t, c),
      commandType: "INSERT",
      mode: "GUI",
      command: {
        table: t,
        changeSet: {
          compType: "OBJECT",
          comp: `{{ ${e}.data }}`
        }
      }
    }
  };
}
const S = {
  type: "mysql",
  getCompSelection: b,
  getQueryInitData: d
};
export {
  S as mysqlConfig
};
