// lowcoder-sdk/dist/4015b4a8.js
import { mJ as S, i6 as M } from "./c988e26e.js";
import D from "react";
function j(f, o) {
  for (var u = 0; u < o.length; u++) {
    const i = o[u];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const a in i)
        if (a !== "default" && !(a in f)) {
          const c = Object.getOwnPropertyDescriptor(i, a);
          c && Object.defineProperty(f, a, c.get ? c : {
            enumerable: !0,
            get: () => i[a]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, h = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var z = { icon: function(o, u) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M839.5 344h-655c-.3 0-.5.2-.5.5v91.2c0 59.8 49 108.3 109.3 108.3 40.7 0 76.2-22 95.1-54.7 2.9-5.1 8.4-8.3 14.3-8.3s11.3 3.2 14.3 8.3c18.8 32.7 54.3 54.7 95 54.7 40.8 0 76.4-22.1 95.1-54.9 2.9-5 8.2-8.1 13.9-8.1h.6c5.8 0 11 3.1 13.9 8.1 18.8 32.8 54.4 54.9 95.2 54.9C791 544 840 495.5 840 435.7v-91.2c0-.3-.2-.5-.5-.5z", fill: u } }, { tag: "path", attrs: { d: "M882 272.1V144c0-17.7-14.3-32-32-32H174c-17.7 0-32 14.3-32 32v128.1c-16.7 1-30 14.9-30 31.9v131.7a177 177 0 0014.4 70.4c4.3 10.2 9.6 19.8 15.6 28.9v345c0 17.6 14.3 32 32 32h676c17.7 0 32-14.3 32-32V535a175 175 0 0015.6-28.9c9.5-22.3 14.4-46 14.4-70.4V304c0-17-13.3-30.9-30-31.9zM214 184h596v88H214v-88zm362 656.1H448V736h128v104.1zm234.4 0H640V704c0-17.7-14.3-32-32-32H416c-17.7 0-32 14.3-32 32v136.1H214V597.9c2.9 1.4 5.9 2.8 9 4 22.3 9.4 46 14.1 70.4 14.1 24.4 0 48-4.7 70.4-14.1 13.8-5.8 26.8-13.2 38.7-22.1.2-.1.4-.1.6 0a180.4 180.4 0 0038.7 22.1c22.3 9.4 46 14.1 70.4 14.1s48-4.7 70.4-14.1c13.8-5.8 26.8-13.2 38.7-22.1.2-.1.4-.1.6 0a180.4 180.4 0 0038.7 22.1c22.3 9.4 46 14.1 70.4 14.1s48-4.7 70.4-14.1c3-1.3 6-2.6 9-4v242.2zM840 435.7c0 59.8-49 108.3-109.3 108.3-40.8 0-76.4-22.1-95.2-54.9-2.9-5-8.1-8.1-13.9-8.1h-.6c-5.7 0-11 3.1-13.9 8.1A109.24 109.24 0 01512 544c-40.7 0-76.2-22-95-54.7-3-5.1-8.4-8.3-14.3-8.3s-11.4 3.2-14.3 8.3a109.63 109.63 0 01-95.1 54.7C233 544 184 495.5 184 435.7v-91.2c0-.3.2-.5.5-.5h655c.3 0 .5.2.5.5v91.2z", fill: o } }] };
}, name: "shop", theme: "twotone" };
_.default = z;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var o = /* @__PURE__ */ y(D), u = /* @__PURE__ */ c(_), i = /* @__PURE__ */ c(S);
  function a(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function c(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function v(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (v = function(n) {
      return n ? r : t;
    })(e);
  }
  function y(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = v(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, p = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var l in e)
      if (l !== "default" && Object.prototype.hasOwnProperty.call(e, l)) {
        var s = p ? Object.getOwnPropertyDescriptor(e, l) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, l, s) : n[l] = e[l];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function g(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(p) {
        return Object.getOwnPropertyDescriptor(r, p).enumerable;
      }))), n.forEach(function(p) {
        a(e, p, r[p]);
      });
    }
    return e;
  }
  function P(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(p) {
        return Object.getOwnPropertyDescriptor(e, p).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function m(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : P(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var T = function(e, t) {
    return /* @__PURE__ */ o.createElement(i.default, m(g({}, e), {
      ref: t,
      icon: u.default
    }));
  }, O = /* @__PURE__ */ o.forwardRef(T);
  process.env.NODE_ENV !== "production" && (O.displayName = "ShopTwoTone");
  var b = O;
})(h);
(function(f, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const u = i(h);
  function i(c) {
    return c && c.__esModule ? c : { default: c };
  }
  const a = u;
  o.default = a, f.exports = a;
})(d, d.exports);
var w = d.exports;
const E = /* @__PURE__ */ M(w), x = /* @__PURE__ */ j({
  __proto__: null,
  default: E
}, [w]);
export {
  x as S
};
