// lowcoder-sdk/dist/44a79fed.js
import { X as t, ah as z, dm as R, F as i, a8 as v, dn as m, dq as b, dp as H, dw as O, av as p, dv as I } from "./c988e26e.js";
import { useContext as W } from "react";
import "react-dom";
const G = (o) => p`
    border-color: ${o.border};
    border-width: ${o.borderWidth};
    border-radius: ${o.radius};
    border-style: ${o.borderStyle};
    overflow: hidden;
    padding: ${o.padding};
    ${o.background && `background-color: ${o.background};`}
    ${o.backgroundImage && `background-image: url(${o.backgroundImage});`}
    ${o.backgroundImageRepeat && `background-repeat: ${o.backgroundImageRepeat};`}
    ${o.backgroundImageSize && `background-size: ${o.backgroundImageSize};`}
    ${o.backgroundImagePosition && `background-position: ${o.backgroundImagePosition};`}
    ${o.backgroundImageOrigin && `background-origin: ${o.backgroundImageOrigin};`}
  `, F = t.div`
  display: flex;
  flex-flow: column;
  height: 100%;
  border: 1px solid #d7d9e0;
  border-radius: 4px;
  ${(o) => o.$style && G(o.$style)}
  ${(o) => o.$animationStyle && o.$animationStyle}
`, M = t(I)`
  overflow: visible;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
  ${(o) => o.$headerBackgroundImage && `background-image: url(${o.$headerBackgroundImage});`}
  ${(o) => o.$headerBackgroundImageRepeat && `background-repeat: ${o.$headerBackgroundImageRepeat};`}
  ${(o) => o.$headerBackgroundImageSize && `background-size: ${o.$headerBackgroundImageSize};`}
  ${(o) => o.$headerBackgroundImagePosition && `background-position: ${o.$headerBackgroundImagePosition};`}
  ${(o) => o.$headerBackgroundImageOrigin && `background-origin: ${o.$headerBackgroundImageOrigin};`}
`, j = t(I)`
  border-top: ${(o) => `${o.$showBorder ? o.$borderWidth : 0} solid ${o.$borderColor}`};
  flex: 1;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
  ${(o) => o.$backgroundImage && `background-image: url(${o.$backgroundImage});`}
  ${(o) => o.$backgroundImageRepeat && `background-repeat: ${o.$backgroundImageRepeat};`}
  ${(o) => o.$backgroundImageSize && `background-size: ${o.$backgroundImageSize};`}
  ${(o) => o.$backgroundImagePosition && `background-position: ${o.$backgroundImagePosition};`}
  ${(o) => o.$backgroundImageOrigin && `background-origin: ${o.$backgroundImageOrigin};`}
`, N = t(I)`
  border-top: ${(o) => `${o.$showBorder ? o.$borderWidth : 0} solid ${o.$borderColor}`};
  overflow: visible;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
  ${(o) => o.$footerBackgroundImage && `background-image: url(${o.$footerBackgroundImage});`}
  ${(o) => o.$footerBackgroundImageRepeat && `background-repeat: ${o.$footerBackgroundImageRepeat};`}
  ${(o) => o.$footerBackgroundImageSize && `background-size: ${o.$footerBackgroundImageSize};`}
  ${(o) => o.$footerBackgroundImagePosition && `background-position: ${o.$footerBackgroundImagePosition};`}
  ${(o) => o.$footerBackgroundImageOrigin && `background-origin: ${o.$footerBackgroundImageOrigin};`}
`;
function X(o) {
  const {
    container: d,
    animationStyle: h
  } = o, {
    showHeader: g,
    showFooter: $,
    horizontalGridCells: k
  } = d, c = d.showBody || !g && !$, s = d.scrollbars, {
    items: B,
    ...P
  } = d.header, {
    items: l,
    ...f
  } = d.body[0].children.view.getView(), {
    items: w,
    ...x
  } = d.footer, {
    style: n,
    headerStyle: r,
    bodyStyle: a,
    footerStyle: e
  } = d, C = W(z).getAppSettings().maxWidth, u = R(C) ? 8 : 0;
  return /* @__PURE__ */ i("div", {
    style: {
      padding: n.margin,
      height: "100%"
    },
    children: /* @__PURE__ */ v(F, {
      $style: n,
      $animationStyle: h,
      children: [g && /* @__PURE__ */ i(m.Provider, {
        value: r.headerBackground,
        children: /* @__PURE__ */ i(M, {
          ...P,
          horizontalGridCells: k,
          items: b(B),
          autoHeight: !0,
          emptyRows: 5,
          minHeight: "46px",
          containerPadding: [u, 3],
          showName: {
            bottom: c || $ ? 20 : 0
          },
          $backgroundColor: (r == null ? void 0 : r.headerBackground) || "transparent",
          $headerBackgroundImage: r == null ? void 0 : r.headerBackgroundImage,
          $headerBackgroundImageRepeat: r == null ? void 0 : r.headerBackgroundImageRepeat,
          $headerBackgroundImageSize: r == null ? void 0 : r.headerBackgroundImageSize,
          $headerBackgroundImagePosition: r == null ? void 0 : r.headerBackgroundImagePosition,
          $headerBackgroundImageOrigin: r == null ? void 0 : r.headerBackgroundImageOrigin,
          style: {
            padding: r.containerHeaderPadding
          }
        })
      }), c && /* @__PURE__ */ i(m.Provider, {
        value: a.background,
        children: /* @__PURE__ */ i(H, {
          style: {
            height: d.autoHeight ? "auto" : "100%",
            margin: "0px",
            padding: "0px"
          },
          hideScrollbar: !s,
          children: /* @__PURE__ */ i(j, {
            $showBorder: g,
            ...f,
            horizontalGridCells: k,
            items: b(l),
            autoHeight: d.autoHeight,
            emptyRows: 14,
            minHeight: g ? "143px" : "142px",
            containerPadding: g && $ || g ? [u, 11.5] : [u, 11],
            hintPlaceholder: o.hintPlaceholder ?? O,
            $backgroundColor: (a == null ? void 0 : a.background) || "transparent",
            $borderColor: n == null ? void 0 : n.border,
            $borderWidth: n == null ? void 0 : n.borderWidth,
            $backgroundImage: a == null ? void 0 : a.backgroundImage,
            $backgroundImageRepeat: a == null ? void 0 : a.backgroundImageRepeat,
            $backgroundImageSize: a == null ? void 0 : a.backgroundImageSize,
            $backgroundImagePosition: a == null ? void 0 : a.backgroundImagePosition,
            $backgroundImageOrigin: a == null ? void 0 : a.backgroundImageOrigin,
            style: {
              padding: a.containerBodyPadding
            }
          })
        })
      }), $ && /* @__PURE__ */ i(m.Provider, {
        value: e.footerBackground,
        children: /* @__PURE__ */ i(N, {
          $showBorder: g || c,
          ...x,
          horizontalGridCells: k,
          items: b(w),
          autoHeight: !0,
          emptyRows: 5,
          minHeight: c ? "47px" : "46px",
          containerPadding: c || g ? [u, 3.5] : [u, 3],
          showName: {
            top: g || c ? 20 : 0
          },
          $backgroundColor: (e == null ? void 0 : e.footerBackground) || "transparent",
          $footerBackgroundImage: e == null ? void 0 : e.footerBackgroundImage,
          $footerBackgroundImageRepeat: e == null ? void 0 : e.footerBackgroundImageRepeat,
          $footerBackgroundImageSize: e == null ? void 0 : e.footerBackgroundImageSize,
          $footerBackgroundImagePosition: e == null ? void 0 : e.footerBackgroundImagePosition,
          $footerBackgroundImageOrigin: e == null ? void 0 : e.footerBackgroundImageOrigin,
          $borderColor: n == null ? void 0 : n.border,
          $borderWidth: n == null ? void 0 : n.borderWidth,
          style: {
            padding: e.containerFooterPadding
          }
        })
      })]
    })
  });
}
export {
  X as TriContainer
};
