// lowcoder-sdk/dist/4dcb2b3a.js
import { mJ as D, i6 as M } from "./c988e26e.js";
import j from "react";
function E(c, o) {
  for (var l = 0; l < o.length; l++) {
    const a = o[l];
    if (typeof a != "string" && !Array.isArray(a)) {
      for (const i in a)
        if (i !== "default" && !(i in c)) {
          const u = Object.getOwnPropertyDescriptor(a, i);
          u && Object.defineProperty(c, i, u.get ? u : {
            enumerable: !0,
            get: () => a[i]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(c, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, y = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var S = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M842 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1zM512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm-94-392c0-50.6 41.9-92 94-92s94 41.4 94 92v224c0 50.6-41.9 92-94 92s-94-41.4-94-92V232z" } }] }, name: "audio", theme: "outlined" };
s.default = S;
(function(c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), Object.defineProperty(c, "default", {
    enumerable: !0,
    get: function() {
      return A;
    }
  });
  var o = /* @__PURE__ */ P(j), l = /* @__PURE__ */ u(s), a = /* @__PURE__ */ u(D);
  function i(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function u(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function P(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var d in e)
      if (d !== "default" && Object.prototype.hasOwnProperty.call(e, d)) {
        var p = f ? Object.getOwnPropertyDescriptor(e, d) : null;
        p && (p.get || p.set) ? Object.defineProperty(n, d, p) : n[d] = e[d];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(f) {
        return Object.getOwnPropertyDescriptor(r, f).enumerable;
      }))), n.forEach(function(f) {
        i(e, f, r[f]);
      });
    }
    return e;
  }
  function b(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function w(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : b(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var h = function(e, t) {
    return /* @__PURE__ */ o.createElement(a.default, w(m({}, e), {
      ref: t,
      icon: l.default
    }));
  }, v = /* @__PURE__ */ o.forwardRef(h);
  process.env.NODE_ENV !== "production" && (v.displayName = "AudioOutlined");
  var A = v;
})(y);
(function(c, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const l = a(y);
  function a(u) {
    return u && u.__esModule ? u : { default: u };
  }
  const i = l;
  o.default = i, c.exports = i;
})(O, O.exports);
var g = O.exports;
const x = /* @__PURE__ */ M(g), V = /* @__PURE__ */ E({
  __proto__: null,
  default: x
}, [g]);
export {
  V as A
};
