// lowcoder-sdk/dist/4fe456d5.js
import { X as C, a8 as g, F as a, dN as u, Y as s, gn as M, c1 as f, dM as b, hI as w, cs as d } from "./c988e26e.js";
import { getMobileStyle as T } from "./7032c5eb.js";
import { a as I } from "./fc1052e0.js";
import "react";
import "react-dom";
const k = (e = 1) => [...d.range(0, 24, e).map((t) => {
  const n = (t < 10 ? "0" : "") + t;
  return {
    label: n,
    value: n
  };
})], $ = (e = 1) => [...d.range(0, 12, e).map((t) => {
  const n = (t < 10 ? "0" : "") + t;
  return {
    label: n,
    value: n
  };
})], h = (e = 1) => d.range(0, 60, e).map((t) => {
  const n = (t < 10 ? "0" : "") + t;
  return {
    label: n,
    value: n
  };
}), j = [{
  label: "AM",
  value: "am"
}, {
  label: "PM",
  value: "pm"
}], c = async (e) => {
  const t = (await import("./2242e3df.js").then((l) => l.i)).default, {
    disabledHours: n,
    disabledMinutes: y,
    disabledSeconds: x
  } = e.disabledTime();
  t.prompt({
    getContainer: () => document.querySelector(`#${M}`) || document.body,
    mouseWheel: !0,
    destroyOnClose: !0,
    closeOnMaskClick: !0,
    columns: (l) => {
      const i = f(l.join(":"), b);
      return [(e.use12Hours ? $ : k)(e.hourStep).filter(({
        label: m,
        value: o
      }) => !n().includes(Number(o) + (l[3] === "pm" ? 12 : 0))), h(e.minuteStep).filter(({
        label: m,
        value: o
      }) => !y(i.hour()).includes(Number(o))), h(e.secondStep).filter(({
        label: m,
        value: o
      }) => !x(i.hour(), i.minute()).includes(Number(o))), ...e.use12Hours ? [j] : []];
    },
    defaultValue: e.value ? e.value.format(e.use12Hours ? w : u).split(":") : void 0,
    onConfirm: (l) => {
      const i = f(l.join(":"), b);
      e.onChange(i);
    },
    onClose: e.onBlur
  }), e.onFocus();
}, v = C.div`
  height: 32px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 4px 8px;
  background-color: #ffffff;
  cursor: pointer;
  border-radius: 4px;
  border: 1px solid #d7d9e0;

  ${(e) => e.$style && T(e.$style)}
`, r = C.div`
  overflow: hidden;
  white-space: nowrap;
  flex-grow: 1;
  display: flex;
  justify-content: center;
`, F = (e) => /* @__PURE__ */ g(v, {
  ref: e.viewRef,
  $style: e.$style,
  onClick: () => {
    c(e);
  },
  children: [/* @__PURE__ */ a(r, {
    children: e.value ? e.value.format(e.format || u) : s("time.placeholder")
  }), e.suffixIcon]
}), N = (e) => /* @__PURE__ */ g(v, {
  ref: e.viewRef,
  $style: e.$style,
  children: [/* @__PURE__ */ a(r, {
    onClick: () => c({
      ...e,
      value: e.start,
      onChange: (t) => e.onChange(t, e.end)
    }),
    children: e.start ? e.start.format(e.format || u) : s("time.startTime")
  }), /* @__PURE__ */ a(I, {}), /* @__PURE__ */ a(r, {
    onClick: () => c({
      ...e,
      value: e.end,
      onChange: (t) => e.onChange(e.start, t)
    }),
    children: e.end ? e.end.format(e.format || u) : s("time.endTime")
  }), e.suffixIcon]
});
export {
  F as TimeMobileUIView,
  N as TimeRangeMobileUIView
};
