// lowcoder-sdk/dist/525aed20.js
import { mJ as D, i6 as F } from "./c988e26e.js";
import M from "react";
function j(p, a) {
  for (var u = 0; u < a.length; u++) {
    const l = a[u];
    if (typeof l != "string" && !Array.isArray(l)) {
      for (const o in l)
        if (o !== "default" && !(o in p)) {
          const i = Object.getOwnPropertyDescriptor(l, o);
          i && Object.defineProperty(p, o, i.get ? i : {
            enumerable: !0,
            get: () => l[o]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(p, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, v = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M832 312H696v-16c0-101.6-82.4-184-184-184s-184 82.4-184 184v16H192c-17.7 0-32 14.3-32 32v536c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V344c0-17.7-14.3-32-32-32zm-208 0H400v-16c0-61.9 50.1-112 112-112s112 50.1 112 112v16z" } }] }, name: "shopping", theme: "filled" };
_.default = E;
(function(p) {
  Object.defineProperty(p, "__esModule", {
    value: !0
  }), Object.defineProperty(p, "default", {
    enumerable: !0,
    get: function() {
      return S;
    }
  });
  var a = /* @__PURE__ */ h(M), u = /* @__PURE__ */ i(_), l = /* @__PURE__ */ i(D);
  function o(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function i(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function h(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var c in e)
      if (c !== "default" && Object.prototype.hasOwnProperty.call(e, c)) {
        var d = f ? Object.getOwnPropertyDescriptor(e, c) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, c, d) : n[c] = e[c];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(f) {
        return Object.getOwnPropertyDescriptor(r, f).enumerable;
      }))), n.forEach(function(f) {
        o(e, f, r[f]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ a.createElement(l.default, b(P({}, e), {
      ref: t,
      icon: u.default
    }));
  }, g = /* @__PURE__ */ a.forwardRef(w);
  process.env.NODE_ENV !== "production" && (g.displayName = "ShoppingFilled");
  var S = g;
})(v);
(function(p, a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), a.default = void 0;
  const u = l(v);
  function l(i) {
    return i && i.__esModule ? i : { default: i };
  }
  const o = u;
  a.default = o, p.exports = o;
})(s, s.exports);
var y = s.exports;
const x = /* @__PURE__ */ F(y), $ = /* @__PURE__ */ j({
  __proto__: null,
  default: x
}, [y]);
export {
  $ as S
};
