// lowcoder-sdk/dist/52d7c09b.js
import { mJ as S, i6 as q } from "./c988e26e.js";
import D from "react";
function M(f, i) {
  for (var l = 0; l < i.length; l++) {
    const c = i[l];
    if (typeof c != "string" && !Array.isArray(c)) {
      for (const a in c)
        if (a !== "default" && !(a in f)) {
          const u = Object.getOwnPropertyDescriptor(c, a);
          u && Object.defineProperty(f, a, u.get ? u : {
            enumerable: !0,
            get: () => c[a]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, v = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var j = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M412.7 696.5l246-178c4.4-3.2 4.4-9.7 0-12.9l-246-178c-5.3-3.8-12.7 0-12.7 6.5V381c0 10.2 4.9 19.9 13.2 25.9L558.6 512 413.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5z" } }, { tag: "path", attrs: { d: "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z" } }] }, name: "right-square", theme: "outlined" };
s.default = j;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return R;
    }
  });
  var i = /* @__PURE__ */ h(D), l = /* @__PURE__ */ u(s), c = /* @__PURE__ */ u(S);
  function a(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function u(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function h(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, o = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = o ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(o) {
        return Object.getOwnPropertyDescriptor(r, o).enumerable;
      }))), n.forEach(function(o) {
        a(e, o, r[o]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(o) {
        return Object.getOwnPropertyDescriptor(e, o).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ i.createElement(c.default, b(P({}, e), {
      ref: t,
      icon: l.default
    }));
  }, g = /* @__PURE__ */ i.forwardRef(w);
  process.env.NODE_ENV !== "production" && (g.displayName = "RightSquareOutlined");
  var R = g;
})(v);
(function(f, i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), i.default = void 0;
  const l = c(v);
  function c(u) {
    return u && u.__esModule ? u : { default: u };
  }
  const a = l;
  i.default = a, f.exports = a;
})(O, O.exports);
var y = O.exports;
const E = /* @__PURE__ */ q(y), V = /* @__PURE__ */ M({
  __proto__: null,
  default: E
}, [y]);
export {
  V as R
};
