// lowcoder-sdk/dist/5e4371ee.js
import { mJ as M, i6 as D } from "./c988e26e.js";
import z from "react";
function j(u, o) {
  for (var f = 0; f < o.length; f++) {
    const a = o[f];
    if (typeof a != "string" && !Array.isArray(a)) {
      for (const c in a)
        if (c !== "default" && !(c in u)) {
          const i = Object.getOwnPropertyDescriptor(a, c);
          i && Object.defineProperty(u, c, i.get ? i : {
            enumerable: !0,
            get: () => a[c]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(u, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, h = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M676 565c-50.8 0-92 41.2-92 92s41.2 92 92 92 92-41.2 92-92-41.2-92-92-92zm0 126c-18.8 0-34-15.2-34-34s15.2-34 34-34 34 15.2 34 34-15.2 34-34 34zm204-523H668c0-30.9-25.1-56-56-56h-80c-30.9 0-56 25.1-56 56H264c-17.7 0-32 14.3-32 32v200h-88c-17.7 0-32 14.3-32 32v448c0 17.7 14.3 32 32 32h336c17.7 0 32-14.3 32-32v-16h368c17.7 0 32-14.3 32-32V200c0-17.7-14.3-32-32-32zm-412 64h72v-56h64v56h72v48H468v-48zm-20 616H176V616h272v232zm0-296H176v-88h272v88zm392 240H512V432c0-17.7-14.3-32-32-32H304V240h100v104h336V240h100v552zM704 408v96c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-96c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8zM592 512h48c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8z" } }] }, name: "reconciliation", theme: "outlined" };
s.default = E;
(function(u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), Object.defineProperty(u, "default", {
    enumerable: !0,
    get: function() {
      return R;
    }
  });
  var o = /* @__PURE__ */ g(z), f = /* @__PURE__ */ i(s), a = /* @__PURE__ */ i(M);
  function c(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function i(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function v(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (v = function(n) {
      return n ? r : t;
    })(e);
  }
  function g(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = v(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, l = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = l ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(l) {
        return Object.getOwnPropertyDescriptor(r, l).enumerable;
      }))), n.forEach(function(l) {
        c(e, l, r[l]);
      });
    }
    return e;
  }
  function P(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(l) {
        return Object.getOwnPropertyDescriptor(e, l).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : P(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ o.createElement(a.default, b(m({}, e), {
      ref: t,
      icon: f.default
    }));
  }, _ = /* @__PURE__ */ o.forwardRef(w);
  process.env.NODE_ENV !== "production" && (_.displayName = "ReconciliationOutlined");
  var R = _;
})(h);
(function(u, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const f = a(h);
  function a(i) {
    return i && i.__esModule ? i : { default: i };
  }
  const c = f;
  o.default = c, u.exports = c;
})(O, O.exports);
var y = O.exports;
const H = /* @__PURE__ */ D(y), x = /* @__PURE__ */ j({
  __proto__: null,
  default: H
}, [y]);
export {
  x as R
};
