// lowcoder-sdk/dist/5ed415ad.js
import { bI as u, Z as y, az as p, F as t, bW as w, a8 as s, G as d, ah as n, a9 as o, aa as l, bJ as h, ab as C, ac as b, ad as f, bX as P, Y as S, B as g } from "./c988e26e.js";
import { useContext as i } from "react";
import { TriContainer as x } from "./44a79fed.js";
import { ContainerCompBuilder as V } from "./63085759.js";
import "react-dom";
import "./ef17c1a9.js";
import "./06b2ca8b.js";
const v = function() {
  const a = {
    disabled: u,
    animationStyle: y(p)
  };
  return new V(a, (e, r) => /* @__PURE__ */ t(w.Provider, {
    value: e.disabled,
    children: /* @__PURE__ */ t(x, {
      ...e
    })
  })).setPropertyViewFn((e) => /* @__PURE__ */ s(d, {
    children: [(i(n).editorModeStatus === "logic" || i(n).editorModeStatus === "both") && /* @__PURE__ */ s(o, {
      name: l.interaction,
      children: [h(e), C(e)]
    }), (i(n).editorModeStatus === "layout" || i(n).editorModeStatus === "both") && /* @__PURE__ */ s(d, {
      children: [/* @__PURE__ */ t(o, {
        name: l.layout,
        children: e.container.getPropertyView()
      }), /* @__PURE__ */ t(o, {
        name: l.style,
        children: e.container.stylePropertyView()
      }), e.container.children.showHeader.getView() && /* @__PURE__ */ t(o, {
        name: "Header Style",
        children: e.container.headerStylePropertyView()
      }), e.container.children.showBody.getView() && /* @__PURE__ */ t(o, {
        name: "Body Style",
        children: e.container.bodyStylePropertyView()
      }), e.container.children.showFooter.getView() && /* @__PURE__ */ t(o, {
        name: "Footer Style",
        children: e.container.footerStylePropertyView()
      })]
    })]
  })).build();
}();
function B(a) {
  let e = g(a);
  if (e.value) {
    const r = e.value.container;
    if (r && (r.hasOwnProperty("layout") || r.hasOwnProperty("items"))) {
      const c = e.value.autoHeight, m = e.value.scrollbars;
      return {
        ...e,
        value: {
          container: {
            showHeader: !0,
            body: {
              0: {
                view: r
              }
            },
            showBody: !0,
            showFooter: !1,
            autoHeight: c,
            scrollbars: m
          }
        }
      };
    }
  }
  return e;
}
class F extends v {
  constructor(e) {
    super(B(e));
  }
}
const j = b(F, [f]);
function E(a, e) {
  return {
    container: {
      header: P([{
        item: {
          compType: "text",
          name: e.genItemName("containerTitle"),
          comp: {
            text: "### " + S("container.title")
          }
        },
        layoutItem: {
          i: "",
          h: 5,
          w: 24,
          x: 0,
          y: 0
        }
      }])
    }
  };
}
export {
  v as ContainerBaseComp,
  j as ContainerComp,
  E as defaultContainerData
};
