// lowcoder-sdk/dist/5f5b5fb1.js
import { mJ as D, i6 as F } from "./c988e26e.js";
import M from "react";
function k(f, l) {
  for (var o = 0; o < l.length; o++) {
    const i = l[o];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const c in i)
        if (c !== "default" && !(c in f)) {
          const a = Object.getOwnPropertyDescriptor(i, c);
          a && Object.defineProperty(f, c, a.get ? a : {
            enumerable: !0,
            get: () => i[c]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, w = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var j = { icon: { tag: "svg", attrs: { viewBox: "0 0 1024 1024", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M485.6 249.9L198.2 498c-8.3 7.1-8.3 20.8 0 27.9l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9zm320 0L518.2 498a18.6 18.6 0 00-6.2 14c0 5.2 2.1 10.4 6.2 14l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9z" } }] }, name: "backward", theme: "filled" };
_.default = j;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return B;
    }
  });
  var l = /* @__PURE__ */ g(M), o = /* @__PURE__ */ a(_), i = /* @__PURE__ */ a(D);
  function c(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? t : r;
    })(e);
  }
  function g(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = O(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, u = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var d in e)
      if (d !== "default" && Object.prototype.hasOwnProperty.call(e, d)) {
        var p = u ? Object.getOwnPropertyDescriptor(e, d) : null;
        p && (p.get || p.set) ? Object.defineProperty(n, d, p) : n[d] = e[d];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function P(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(u) {
        return Object.getOwnPropertyDescriptor(t, u).enumerable;
      }))), n.forEach(function(u) {
        c(e, u, t[u]);
      });
    }
    return e;
  }
  function m(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(u) {
        return Object.getOwnPropertyDescriptor(e, u).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function b(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : m(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var h = function(e, r) {
    return /* @__PURE__ */ l.createElement(i.default, b(P({}, e), {
      ref: r,
      icon: o.default
    }));
  }, v = /* @__PURE__ */ l.forwardRef(h);
  process.env.NODE_ENV !== "production" && (v.displayName = "BackwardFilled");
  var B = v;
})(w);
(function(f, l) {
  Object.defineProperty(l, "__esModule", {
    value: !0
  }), l.default = void 0;
  const o = i(w);
  function i(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const c = o;
  l.default = c, f.exports = c;
})(s, s.exports);
var y = s.exports;
const E = /* @__PURE__ */ F(y), N = /* @__PURE__ */ k({
  __proto__: null,
  default: E
}, [y]);
export {
  N as B
};
