// lowcoder-sdk/dist/6559ac3d.js
import { bX as t, Y as l } from "./c988e26e.js";
import "react";
import "react-dom";
function p(o, e) {
  return {
    container: {
      header: t([{
        item: {
          compType: "text",
          name: e.genItemName("collapsibleTitle"),
          comp: {
            text: "### " + l("container.title")
          }
        },
        layoutItem: {
          i: "",
          h: 8,
          w: 13,
          x: 0,
          y: 0
        }
      }, {
        item: {
          compType: "toggleButton",
          name: e.genItemName("collapsibleToggle"),
          comp: {
            value: "true",
            showText: !1,
            alignment: "right",
            showBorder: !1
          }
        },
        layoutItem: {
          i: "",
          h: 5,
          w: 10,
          x: 14,
          y: 0
        }
      }]),
      showBody: `{{collapsibleToggle${Number(e.genItemName("collapsibleToggle").slice(-1)) - 1}.value}}`
    }
  };
}
export {
  p as defaultCollapsibleContainerData
};
