// lowcoder-sdk/dist/67976ea6.js
import { mJ as B, i6 as M } from "./c988e26e.js";
import D from "react";
function k(u, a) {
  for (var c = 0; c < a.length; c++) {
    const i = a[c];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const f in i)
        if (f !== "default" && !(f in u)) {
          const o = Object.getOwnPropertyDescriptor(i, f);
          o && Object.defineProperty(u, f, o.get ? o : {
            enumerable: !0,
            get: () => i[f]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(u, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, w = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var j = { icon: function(a, c) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M240.9 393.9h542.2L512 196.7z", fill: c } }, { tag: "path", attrs: { d: "M894 462c30.9 0 43.8-39.7 18.7-58L530.8 126.2a31.81 31.81 0 00-37.6 0L111.3 404c-25.1 18.2-12.2 58 18.8 58H192v374h-72c-4.4 0-8 3.6-8 8v52c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-52c0-4.4-3.6-8-8-8h-72V462h62zM381 836H264V462h117v374zm189 0H453V462h117v374zm190 0H642V462h118v374zM240.9 393.9L512 196.7l271.1 197.2H240.9z", fill: a } }] };
}, name: "bank", theme: "twotone" };
_.default = j;
(function(u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), Object.defineProperty(u, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var a = /* @__PURE__ */ g(D), c = /* @__PURE__ */ o(_), i = /* @__PURE__ */ o(B);
  function f(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function o(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function g(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, l = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = l ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(l) {
        return Object.getOwnPropertyDescriptor(r, l).enumerable;
      }))), n.forEach(function(l) {
        f(e, l, r[l]);
      });
    }
    return e;
  }
  function h(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(l) {
        return Object.getOwnPropertyDescriptor(e, l).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function m(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : h(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var T = function(e, t) {
    return /* @__PURE__ */ a.createElement(i.default, m(P({}, e), {
      ref: t,
      icon: c.default
    }));
  }, v = /* @__PURE__ */ a.forwardRef(T);
  process.env.NODE_ENV !== "production" && (v.displayName = "BankTwoTone");
  var b = v;
})(w);
(function(u, a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), a.default = void 0;
  const c = i(w);
  function i(o) {
    return o && o.__esModule ? o : { default: o };
  }
  const f = c;
  a.default = f, u.exports = f;
})(s, s.exports);
var y = s.exports;
const z = /* @__PURE__ */ M(y), x = /* @__PURE__ */ k({
  __proto__: null,
  default: z
}, [y]);
export {
  x as B
};
