// lowcoder-sdk/dist/68d21f9b.js
import { mJ as D, i6 as M } from "./c988e26e.js";
import j from "react";
function E(f, o) {
  for (var l = 0; l < o.length; l++) {
    const i = o[l];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const u in i)
        if (u !== "default" && !(u in f)) {
          const a = Object.getOwnPropertyDescriptor(i, u);
          a && Object.defineProperty(f, u, a.get ? a : {
            enumerable: !0,
            get: () => i[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, w = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var S = { icon: { tag: "svg", attrs: { viewBox: "0 0 1024 1024", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M825.8 498L538.4 249.9c-10.7-9.2-26.4-.9-26.4 14v496.3c0 14.9 15.7 23.2 26.4 14L825.8 526c8.3-7.2 8.3-20.8 0-28zm-320 0L218.4 249.9c-10.7-9.2-26.4-.9-26.4 14v496.3c0 14.9 15.7 23.2 26.4 14L505.8 526c4.1-3.6 6.2-8.8 6.2-14 0-5.2-2.1-10.4-6.2-14z" } }] }, name: "forward", theme: "outlined" };
s.default = S;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return F;
    }
  });
  var o = /* @__PURE__ */ g(j), l = /* @__PURE__ */ a(s), i = /* @__PURE__ */ a(D);
  function u(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? t : r;
    })(e);
  }
  function g(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = _(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, c = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var d in e)
      if (d !== "default" && Object.prototype.hasOwnProperty.call(e, d)) {
        var p = c ? Object.getOwnPropertyDescriptor(e, d) : null;
        p && (p.get || p.set) ? Object.defineProperty(n, d, p) : n[d] = e[d];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function P(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(c) {
        return Object.getOwnPropertyDescriptor(t, c).enumerable;
      }))), n.forEach(function(c) {
        u(e, c, t[c]);
      });
    }
    return e;
  }
  function m(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(c) {
        return Object.getOwnPropertyDescriptor(e, c).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function b(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : m(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var h = function(e, r) {
    return /* @__PURE__ */ o.createElement(i.default, b(P({}, e), {
      ref: r,
      icon: l.default
    }));
  }, v = /* @__PURE__ */ o.forwardRef(h);
  process.env.NODE_ENV !== "production" && (v.displayName = "ForwardOutlined");
  var F = v;
})(w);
(function(f, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const l = i(w);
  function i(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const u = l;
  o.default = u, f.exports = u;
})(O, O.exports);
var y = O.exports;
const x = /* @__PURE__ */ M(y), $ = /* @__PURE__ */ E({
  __proto__: null,
  default: x
}, [y]);
export {
  $ as F
};
