// lowcoder-sdk/dist/6a8a3621.js
import { mJ as D, i6 as F } from "./c988e26e.js";
import M from "react";
function j(l, a) {
  for (var f = 0; f < a.length; f++) {
    const o = a[f];
    if (typeof o != "string" && !Array.isArray(o)) {
      for (const u in o)
        if (u !== "default" && !(u in l)) {
          const i = Object.getOwnPropertyDescriptor(o, u);
          i && Object.defineProperty(l, u, i.get ? i : {
            enumerable: !0,
            get: () => o[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(l, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, y = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z" } }] }, name: "audio", theme: "filled" };
_.default = E;
(function(l) {
  Object.defineProperty(l, "__esModule", {
    value: !0
  }), Object.defineProperty(l, "default", {
    enumerable: !0,
    get: function() {
      return A;
    }
  });
  var a = /* @__PURE__ */ P(M), f = /* @__PURE__ */ i(_), o = /* @__PURE__ */ i(D);
  function u(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function i(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function P(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, c = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var d in e)
      if (d !== "default" && Object.prototype.hasOwnProperty.call(e, d)) {
        var p = c ? Object.getOwnPropertyDescriptor(e, d) : null;
        p && (p.get || p.set) ? Object.defineProperty(n, d, p) : n[d] = e[d];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(c) {
        return Object.getOwnPropertyDescriptor(r, c).enumerable;
      }))), n.forEach(function(c) {
        u(e, c, r[c]);
      });
    }
    return e;
  }
  function b(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(c) {
        return Object.getOwnPropertyDescriptor(e, c).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function w(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : b(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var h = function(e, t) {
    return /* @__PURE__ */ a.createElement(o.default, w(m({}, e), {
      ref: t,
      icon: f.default
    }));
  }, v = /* @__PURE__ */ a.forwardRef(h);
  process.env.NODE_ENV !== "production" && (v.displayName = "AudioFilled");
  var A = v;
})(y);
(function(l, a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), a.default = void 0;
  const f = o(y);
  function o(i) {
    return i && i.__esModule ? i : { default: i };
  }
  const u = f;
  a.default = u, l.exports = u;
})(s, s.exports);
var g = s.exports;
const S = /* @__PURE__ */ F(g), R = /* @__PURE__ */ j({
  __proto__: null,
  default: S
}, [g]);
export {
  R as A
};
