// lowcoder-sdk/dist/6b965096.js
import { i6 as be, lV as Fe, F as u, lA as T, lW as Ue, X as R, go as V, it as $e, iY as Ve, ah as Be, id as Ye, iS as X, i_ as Ge, lX as qe, lY as We, lZ as ze, l9 as Xe, l_ as ae, l$ as Ke, m0 as se, m1 as Qe, m2 as Je, m3 as Ze, a8 as b, m4 as et, G as ce, m5 as tt, m6 as nt, m7 as rt, al as ot, iu as le, m8 as ue, dp as it, cv as at, Y as de, m9 as st, ma as ct, mb as we, mc as Ae, md as lt } from "./c988e26e.js";
import B, { lazy as w, useContext as pe, useState as M, useCallback as N, useMemo as ut, useLayoutEffect as dt, Suspense as pt } from "react";
import { i as ft } from "./951b5247.js";
import "react-dom";
function ht(t) {
  return t && typeof t == "object" && "default" in t ? t.default : t;
}
var Se = B, mt = ht(Se);
function fe(t, e, r) {
  return e in t ? Object.defineProperty(t, e, {
    value: r,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : t[e] = r, t;
}
function gt(t, e) {
  t.prototype = Object.create(e.prototype), t.prototype.constructor = t, t.__proto__ = e;
}
var Tt = !!(typeof window < "u" && window.document && window.document.createElement);
function yt(t, e, r) {
  if (typeof t != "function")
    throw new Error("Expected reducePropsToState to be a function.");
  if (typeof e != "function")
    throw new Error("Expected handleStateChangeOnClient to be a function.");
  if (typeof r < "u" && typeof r != "function")
    throw new Error("Expected mapStateOnServer to either be undefined or a function.");
  function n(o) {
    return o.displayName || o.name || "Component";
  }
  return function(l) {
    if (typeof l != "function")
      throw new Error("Expected WrappedComponent to be a React component.");
    var a = [], s;
    function i() {
      s = t(a.map(function(d) {
        return d.props;
      })), c.canUseDOM ? e(s) : r && (s = r(s));
    }
    var c = /* @__PURE__ */ function(d) {
      gt(p, d);
      function p() {
        return d.apply(this, arguments) || this;
      }
      p.peek = function() {
        return s;
      }, p.rewind = function() {
        if (p.canUseDOM)
          throw new Error("You may only call rewind() on the server. Call peek() to read the current state.");
        var v = s;
        return s = void 0, a = [], v;
      };
      var h = p.prototype;
      return h.UNSAFE_componentWillMount = function() {
        a.push(this), i();
      }, h.componentDidUpdate = function() {
        i();
      }, h.componentWillUnmount = function() {
        var v = a.indexOf(this);
        a.splice(v, 1), i();
      }, h.render = function() {
        return mt.createElement(l, this.props);
      }, p;
    }(Se.PureComponent);
    return fe(c, "displayName", "SideEffect(" + n(l) + ")"), fe(c, "canUseDOM", Tt), c;
  };
}
var vt = yt;
const Et = /* @__PURE__ */ be(vt);
var bt = Fe();
const wt = /* @__PURE__ */ be(bt);
var I = {
  BODY: "bodyAttributes",
  HTML: "htmlAttributes",
  TITLE: "titleAttributes"
}, f = {
  BASE: "base",
  BODY: "body",
  HEAD: "head",
  HTML: "html",
  LINK: "link",
  META: "meta",
  NOSCRIPT: "noscript",
  SCRIPT: "script",
  STYLE: "style",
  TITLE: "title"
}, he = Object.keys(f).map(function(t) {
  return f[t];
}), y = {
  CHARSET: "charset",
  CSS_TEXT: "cssText",
  HREF: "href",
  HTTPEQUIV: "http-equiv",
  INNER_HTML: "innerHTML",
  ITEM_PROP: "itemprop",
  NAME: "name",
  PROPERTY: "property",
  REL: "rel",
  SRC: "src",
  TARGET: "target"
}, Y = {
  accesskey: "accessKey",
  charset: "charSet",
  class: "className",
  contenteditable: "contentEditable",
  contextmenu: "contextMenu",
  "http-equiv": "httpEquiv",
  itemprop: "itemProp",
  tabindex: "tabIndex"
}, F = {
  DEFAULT_TITLE: "defaultTitle",
  DEFER: "defer",
  ENCODE_SPECIAL_CHARACTERS: "encodeSpecialCharacters",
  ON_CHANGE_CLIENT_STATE: "onChangeClientState",
  TITLE_TEMPLATE: "titleTemplate"
}, At = Object.keys(Y).reduce(function(t, e) {
  return t[Y[e]] = e, t;
}, {}), St = [f.NOSCRIPT, f.SCRIPT, f.STYLE], L = "data-react-helmet", Ct = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(t) {
  return typeof t;
} : function(t) {
  return t && typeof Symbol == "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
}, Pt = function(t, e) {
  if (!(t instanceof e))
    throw new TypeError("Cannot call a class as a function");
}, Ot = function() {
  function t(e, r) {
    for (var n = 0; n < r.length; n++) {
      var o = r[n];
      o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, o.key, o);
    }
  }
  return function(e, r, n) {
    return r && t(e.prototype, r), n && t(e, n), e;
  };
}(), A = Object.assign || function(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e];
    for (var n in r)
      Object.prototype.hasOwnProperty.call(r, n) && (t[n] = r[n]);
  }
  return t;
}, Lt = function(t, e) {
  if (typeof e != "function" && e !== null)
    throw new TypeError("Super expression must either be null or a function, not " + typeof e);
  t.prototype = Object.create(e && e.prototype, {
    constructor: {
      value: t,
      enumerable: !1,
      writable: !0,
      configurable: !0
    }
  }), e && (Object.setPrototypeOf ? Object.setPrototypeOf(t, e) : t.__proto__ = e);
}, me = function(t, e) {
  var r = {};
  for (var n in t)
    e.indexOf(n) >= 0 || Object.prototype.hasOwnProperty.call(t, n) && (r[n] = t[n]);
  return r;
}, xt = function(t, e) {
  if (!t)
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  return e && (typeof e == "object" || typeof e == "function") ? e : t;
}, Q = function(e) {
  var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : !0;
  return r === !1 ? String(e) : String(e).replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#x27;");
}, Rt = function(e) {
  var r = k(e, f.TITLE), n = k(e, F.TITLE_TEMPLATE);
  if (n && r)
    return n.replace(/%s/g, function() {
      return Array.isArray(r) ? r.join("") : r;
    });
  var o = k(e, F.DEFAULT_TITLE);
  return r || o || void 0;
}, Ht = function(e) {
  return k(e, F.ON_CHANGE_CLIENT_STATE) || function() {
  };
}, K = function(e, r) {
  return r.filter(function(n) {
    return typeof n[e] < "u";
  }).map(function(n) {
    return n[e];
  }).reduce(function(n, o) {
    return A({}, n, o);
  }, {});
}, It = function(e, r) {
  return r.filter(function(n) {
    return typeof n[f.BASE] < "u";
  }).map(function(n) {
    return n[f.BASE];
  }).reverse().reduce(function(n, o) {
    if (!n.length)
      for (var l = Object.keys(o), a = 0; a < l.length; a++) {
        var s = l[a], i = s.toLowerCase();
        if (e.indexOf(i) !== -1 && o[i])
          return n.concat(o);
      }
    return n;
  }, []);
}, D = function(e, r, n) {
  var o = {};
  return n.filter(function(l) {
    return Array.isArray(l[e]) ? !0 : (typeof l[e] < "u" && Z("Helmet: " + e + ' should be of type "Array". Instead found type "' + Ct(l[e]) + '"'), !1);
  }).map(function(l) {
    return l[e];
  }).reverse().reduce(function(l, a) {
    var s = {};
    a.filter(function(h) {
      for (var m = void 0, v = Object.keys(h), P = 0; P < v.length; P++) {
        var C = v[P], O = C.toLowerCase();
        r.indexOf(O) !== -1 && !(m === y.REL && h[m].toLowerCase() === "canonical") && !(O === y.REL && h[O].toLowerCase() === "stylesheet") && (m = O), r.indexOf(C) !== -1 && (C === y.INNER_HTML || C === y.CSS_TEXT || C === y.ITEM_PROP) && (m = C);
      }
      if (!m || !h[m])
        return !1;
      var H = h[m].toLowerCase();
      return o[m] || (o[m] = {}), s[m] || (s[m] = {}), o[m][H] ? !1 : (s[m][H] = !0, !0);
    }).reverse().forEach(function(h) {
      return l.push(h);
    });
    for (var i = Object.keys(s), c = 0; c < i.length; c++) {
      var d = i[c], p = wt({}, o[d], s[d]);
      o[d] = p;
    }
    return l;
  }, []).reverse();
}, k = function(e, r) {
  for (var n = e.length - 1; n >= 0; n--) {
    var o = e[n];
    if (o.hasOwnProperty(r))
      return o[r];
  }
  return null;
}, Mt = function(e) {
  return {
    baseTag: It([y.HREF, y.TARGET], e),
    bodyAttributes: K(I.BODY, e),
    defer: k(e, F.DEFER),
    encode: k(e, F.ENCODE_SPECIAL_CHARACTERS),
    htmlAttributes: K(I.HTML, e),
    linkTags: D(f.LINK, [y.REL, y.HREF], e),
    metaTags: D(f.META, [y.NAME, y.CHARSET, y.HTTPEQUIV, y.PROPERTY, y.ITEM_PROP], e),
    noscriptTags: D(f.NOSCRIPT, [y.INNER_HTML], e),
    onChangeClientState: Ht(e),
    scriptTags: D(f.SCRIPT, [y.SRC, y.INNER_HTML], e),
    styleTags: D(f.STYLE, [y.CSS_TEXT], e),
    title: Rt(e),
    titleAttributes: K(I.TITLE, e)
  };
}, J = function() {
  var t = Date.now();
  return function(e) {
    var r = Date.now();
    r - t > 16 ? (t = r, e(r)) : setTimeout(function() {
      J(e);
    }, 0);
  };
}(), ge = function(e) {
  return clearTimeout(e);
}, _t = typeof window < "u" ? window.requestAnimationFrame && window.requestAnimationFrame.bind(window) || window.webkitRequestAnimationFrame || window.mozRequestAnimationFrame || J : global.requestAnimationFrame || J, kt = typeof window < "u" ? window.cancelAnimationFrame || window.webkitCancelAnimationFrame || window.mozCancelAnimationFrame || ge : global.cancelAnimationFrame || ge, Z = function(e) {
  return console && typeof console.warn == "function" && console.warn(e);
}, j = null, Nt = function(e) {
  j && kt(j), e.defer ? j = _t(function() {
    Te(e, function() {
      j = null;
    });
  }) : (Te(e), j = null);
}, Te = function(e, r) {
  var n = e.baseTag, o = e.bodyAttributes, l = e.htmlAttributes, a = e.linkTags, s = e.metaTags, i = e.noscriptTags, c = e.onChangeClientState, d = e.scriptTags, p = e.styleTags, h = e.title, m = e.titleAttributes;
  ee(f.BODY, o), ee(f.HTML, l), Dt(h, m);
  var v = {
    baseTag: _(f.BASE, n),
    linkTags: _(f.LINK, a),
    metaTags: _(f.META, s),
    noscriptTags: _(f.NOSCRIPT, i),
    scriptTags: _(f.SCRIPT, d),
    styleTags: _(f.STYLE, p)
  }, P = {}, C = {};
  Object.keys(v).forEach(function(O) {
    var H = v[O], E = H.newTags, U = H.oldTags;
    E.length && (P[O] = E), U.length && (C[O] = v[O].oldTags);
  }), r && r(), c(e, P, C);
}, Ce = function(e) {
  return Array.isArray(e) ? e.join("") : e;
}, Dt = function(e, r) {
  typeof e < "u" && document.title !== e && (document.title = Ce(e)), ee(f.TITLE, r);
}, ee = function(e, r) {
  var n = document.getElementsByTagName(e)[0];
  if (n) {
    for (var o = n.getAttribute(L), l = o ? o.split(",") : [], a = [].concat(l), s = Object.keys(r), i = 0; i < s.length; i++) {
      var c = s[i], d = r[c] || "";
      n.getAttribute(c) !== d && n.setAttribute(c, d), l.indexOf(c) === -1 && l.push(c);
      var p = a.indexOf(c);
      p !== -1 && a.splice(p, 1);
    }
    for (var h = a.length - 1; h >= 0; h--)
      n.removeAttribute(a[h]);
    l.length === a.length ? n.removeAttribute(L) : n.getAttribute(L) !== s.join(",") && n.setAttribute(L, s.join(","));
  }
}, _ = function(e, r) {
  var n = document.head || document.querySelector(f.HEAD), o = n.querySelectorAll(e + "[" + L + "]"), l = Array.prototype.slice.call(o), a = [], s = void 0;
  return r && r.length && r.forEach(function(i) {
    var c = document.createElement(e);
    for (var d in i)
      if (i.hasOwnProperty(d))
        if (d === y.INNER_HTML)
          c.innerHTML = i.innerHTML;
        else if (d === y.CSS_TEXT)
          c.styleSheet ? c.styleSheet.cssText = i.cssText : c.appendChild(document.createTextNode(i.cssText));
        else {
          var p = typeof i[d] > "u" ? "" : i[d];
          c.setAttribute(d, p);
        }
    c.setAttribute(L, "true"), l.some(function(h, m) {
      return s = m, c.isEqualNode(h);
    }) ? l.splice(s, 1) : a.push(c);
  }), l.forEach(function(i) {
    return i.parentNode.removeChild(i);
  }), a.forEach(function(i) {
    return n.appendChild(i);
  }), {
    oldTags: l,
    newTags: a
  };
}, Pe = function(e) {
  return Object.keys(e).reduce(function(r, n) {
    var o = typeof e[n] < "u" ? n + '="' + e[n] + '"' : "" + n;
    return r ? r + " " + o : o;
  }, "");
}, jt = function(e, r, n, o) {
  var l = Pe(n), a = Ce(r);
  return l ? "<" + e + " " + L + '="true" ' + l + ">" + Q(a, o) + "</" + e + ">" : "<" + e + " " + L + '="true">' + Q(a, o) + "</" + e + ">";
}, Ft = function(e, r, n) {
  return r.reduce(function(o, l) {
    var a = Object.keys(l).filter(function(c) {
      return !(c === y.INNER_HTML || c === y.CSS_TEXT);
    }).reduce(function(c, d) {
      var p = typeof l[d] > "u" ? d : d + '="' + Q(l[d], n) + '"';
      return c ? c + " " + p : p;
    }, ""), s = l.innerHTML || l.cssText || "", i = St.indexOf(e) === -1;
    return o + "<" + e + " " + L + '="true" ' + a + (i ? "/>" : ">" + s + "</" + e + ">");
  }, "");
}, Oe = function(e) {
  var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {};
  return Object.keys(e).reduce(function(n, o) {
    return n[Y[o] || o] = e[o], n;
  }, r);
}, Ut = function(e) {
  var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {};
  return Object.keys(e).reduce(function(n, o) {
    return n[At[o] || o] = e[o], n;
  }, r);
}, $t = function(e, r, n) {
  var o, l = (o = {
    key: r
  }, o[L] = !0, o), a = Oe(n, l);
  return [/* @__PURE__ */ u(f.TITLE, {
    ...a,
    children: r
  })];
}, Vt = function(e, r) {
  return r.map(function(n, o) {
    var l, a = (l = {
      key: o
    }, l[L] = !0, l);
    return Object.keys(n).forEach(function(s) {
      var i = Y[s] || s;
      if (i === y.INNER_HTML || i === y.CSS_TEXT) {
        var c = n.innerHTML || n.cssText;
        a.dangerouslySetInnerHTML = {
          __html: c
        };
      } else
        a[i] = n[s];
    }), B.createElement(e, a);
  });
}, x = function(e, r, n) {
  switch (e) {
    case f.TITLE:
      return {
        toComponent: function() {
          return $t(e, r.title, r.titleAttributes);
        },
        toString: function() {
          return jt(e, r.title, r.titleAttributes, n);
        }
      };
    case I.BODY:
    case I.HTML:
      return {
        toComponent: function() {
          return Oe(r);
        },
        toString: function() {
          return Pe(r);
        }
      };
    default:
      return {
        toComponent: function() {
          return Vt(e, r);
        },
        toString: function() {
          return Ft(e, r, n);
        }
      };
  }
}, Le = function(e) {
  var r = e.baseTag, n = e.bodyAttributes, o = e.encode, l = e.htmlAttributes, a = e.linkTags, s = e.metaTags, i = e.noscriptTags, c = e.scriptTags, d = e.styleTags, p = e.title, h = p === void 0 ? "" : p, m = e.titleAttributes;
  return {
    base: x(f.BASE, r, o),
    bodyAttributes: x(I.BODY, n, o),
    htmlAttributes: x(I.HTML, l, o),
    link: x(f.LINK, a, o),
    meta: x(f.META, s, o),
    noscript: x(f.NOSCRIPT, i, o),
    script: x(f.SCRIPT, c, o),
    style: x(f.STYLE, d, o),
    title: x(f.TITLE, {
      title: h,
      titleAttributes: m
    }, o)
  };
}, Bt = function(e) {
  var r, n;
  return n = r = function(o) {
    Lt(l, o);
    function l() {
      return Pt(this, l), xt(this, o.apply(this, arguments));
    }
    return l.prototype.shouldComponentUpdate = function(s) {
      return !ft(this.props, s);
    }, l.prototype.mapNestedChildrenToProps = function(s, i) {
      if (!i)
        return null;
      switch (s.type) {
        case f.SCRIPT:
        case f.NOSCRIPT:
          return {
            innerHTML: i
          };
        case f.STYLE:
          return {
            cssText: i
          };
      }
      throw new Error("<" + s.type + " /> elements are self-closing and can not contain children. Refer to our API for more information.");
    }, l.prototype.flattenArrayTypeChildren = function(s) {
      var i, c = s.child, d = s.arrayTypeChildren, p = s.newChildProps, h = s.nestedChildren;
      return A({}, d, (i = {}, i[c.type] = [].concat(d[c.type] || [], [A({}, p, this.mapNestedChildrenToProps(c, h))]), i));
    }, l.prototype.mapObjectTypeChildren = function(s) {
      var i, c, d = s.child, p = s.newProps, h = s.newChildProps, m = s.nestedChildren;
      switch (d.type) {
        case f.TITLE:
          return A({}, p, (i = {}, i[d.type] = m, i.titleAttributes = A({}, h), i));
        case f.BODY:
          return A({}, p, {
            bodyAttributes: A({}, h)
          });
        case f.HTML:
          return A({}, p, {
            htmlAttributes: A({}, h)
          });
      }
      return A({}, p, (c = {}, c[d.type] = A({}, h), c));
    }, l.prototype.mapArrayTypeChildrenToProps = function(s, i) {
      var c = A({}, i);
      return Object.keys(s).forEach(function(d) {
        var p;
        c = A({}, c, (p = {}, p[d] = s[d], p));
      }), c;
    }, l.prototype.warnOnInvalidChildren = function(s, i) {
      if (process.env.NODE_ENV !== "production") {
        if (!he.some(function(c) {
          return s.type === c;
        }))
          return typeof s.type == "function" ? Z("You may be attempting to nest <Helmet> components within each other, which is not allowed. Refer to our API for more information.") : Z("Only elements types " + he.join(", ") + " are allowed. Helmet does not support rendering <" + s.type + "> elements. Refer to our API for more information.");
        if (i && typeof i != "string" && (!Array.isArray(i) || i.some(function(c) {
          return typeof c != "string";
        })))
          throw new Error("Helmet expects a string as a child of <" + s.type + ">. Did you forget to wrap your children in braces? ( <" + s.type + ">{``}</" + s.type + "> ) Refer to our API for more information.");
      }
      return !0;
    }, l.prototype.mapChildrenToProps = function(s, i) {
      var c = this, d = {};
      return B.Children.forEach(s, function(p) {
        if (!(!p || !p.props)) {
          var h = p.props, m = h.children, v = me(h, ["children"]), P = Ut(v);
          switch (c.warnOnInvalidChildren(p, m), p.type) {
            case f.LINK:
            case f.META:
            case f.NOSCRIPT:
            case f.SCRIPT:
            case f.STYLE:
              d = c.flattenArrayTypeChildren({
                child: p,
                arrayTypeChildren: d,
                newChildProps: P,
                nestedChildren: m
              });
              break;
            default:
              i = c.mapObjectTypeChildren({
                child: p,
                newProps: i,
                newChildProps: P,
                nestedChildren: m
              });
              break;
          }
        }
      }), i = this.mapArrayTypeChildrenToProps(d, i), i;
    }, l.prototype.render = function() {
      var s = this.props, i = s.children, c = me(s, ["children"]), d = A({}, c);
      return i && (d = this.mapChildrenToProps(i, d)), /* @__PURE__ */ u(e, {
        ...d
      });
    }, Ot(l, null, [{
      key: "canUseDOM",
      // Component.peek comes from react-side-effect:
      // For testing, you may use a static peek() method available on the returned component.
      // It lets you get the current state without resetting the mounted instance stack.
      // Don’t use it for anything other than testing.
      /**
       * @param {Object} base: {"target": "_blank", "href": "http://mysite.com/"}
       * @param {Object} bodyAttributes: {"className": "root"}
       * @param {String} defaultTitle: "Default Title"
       * @param {Boolean} defer: true
       * @param {Boolean} encodeSpecialCharacters: true
       * @param {Object} htmlAttributes: {"lang": "en", "amp": undefined}
       * @param {Array} link: [{"rel": "canonical", "href": "http://mysite.com/example"}]
       * @param {Array} meta: [{"name": "description", "content": "Test description"}]
       * @param {Array} noscript: [{"innerHTML": "<img src='http://mysite.com/js/test.js'"}]
       * @param {Function} onChangeClientState: "(newState) => console.log(newState)"
       * @param {Array} script: [{"type": "text/javascript", "src": "http://mysite.com/js/test.js"}]
       * @param {Array} style: [{"type": "text/css", "cssText": "div { display: block; color: blue; }"}]
       * @param {String} title: "Title"
       * @param {Object} titleAttributes: {"itemprop": "name"}
       * @param {String} titleTemplate: "MySite.com - %s"
       */
      set: function(s) {
        e.canUseDOM = s;
      }
    }]), l;
  }(B.Component), r.propTypes = {
    base: T.object,
    bodyAttributes: T.object,
    children: T.oneOfType([T.arrayOf(T.node), T.node]),
    defaultTitle: T.string,
    defer: T.bool,
    encodeSpecialCharacters: T.bool,
    htmlAttributes: T.object,
    link: T.arrayOf(T.object),
    meta: T.arrayOf(T.object),
    noscript: T.arrayOf(T.object),
    onChangeClientState: T.func,
    script: T.arrayOf(T.object),
    style: T.arrayOf(T.object),
    title: T.string,
    titleAttributes: T.object,
    titleTemplate: T.string
  }, r.defaultProps = {
    defer: !0,
    encodeSpecialCharacters: !0
  }, r.peek = e.peek, r.rewind = function() {
    var o = e.rewind();
    return o || (o = Le({
      baseTag: [],
      bodyAttributes: {},
      encodeSpecialCharacters: !0,
      htmlAttributes: {},
      linkTags: [],
      metaTags: [],
      noscriptTags: [],
      scriptTags: [],
      styleTags: [],
      title: "",
      titleAttributes: {}
    })), o;
  }, n;
}, Yt = function() {
  return null;
}, Gt = Et(Mt, Nt, Le)(Yt), G = Bt(Gt);
G.renderStatic = G.rewind;
const qt = (t) => ({
  type: Ue.SET_EDITOR_EXTERNAL_STATE,
  payload: t
}), Wt = w(() => import("./32f97763.js").then((t) => ({
  default: t.LeftContent
}))), zt = w(() => import("./6d3d41c3.js").then((t) => ({
  default: t.LeftLayersContent
}))), Xt = w(() => import("./c988e26e.js").then((t) => t.EQ)), Kt = w(() => import("./cb3b2fca.js")), Qt = w(() => import("./a79eca63.js")), ye = w(() => import("./8fbb4207.js").then((t) => ({
  default: t.CustomShortcutWrapper
}))), Jt = w(() => import("./8fbb4207.js").then((t) => ({
  default: t.EditorGlobalHotKeys
}))), Zt = w(() => import("./8fbb4207.js").then((t) => ({
  default: t.EditorHotKeys
}))), en = w(() => import("./c988e26e.js").then((t) => t.Dl).then((t) => ({
  default: t.Body
}))), tn = w(() => import("./c988e26e.js").then((t) => t.Dl).then((t) => ({
  default: t.EditorContainer
}))), ve = w(() => import("./c988e26e.js").then((t) => t.Dl).then((t) => ({
  default: t.EditorContainerWithViewMode
}))), nn = w(() => import("./c988e26e.js").then((t) => t.Dl).then((t) => ({
  default: t.Height100Div
}))), rn = w(() => import("./c988e26e.js").then((t) => t.Dl).then((t) => ({
  default: t.LeftPanel
}))), on = w(() => import("./c988e26e.js").then((t) => t.Dl).then((t) => ({
  default: t.MiddlePanel
}))), an = w(() => import("./91a8b2fa.js").then((t) => ({
  default: t.HelpDropdown
}))), sn = w(() => import("./a759a2f6.js").then((t) => ({
  default: t.PreviewHeader
}))), cn = R.div`
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  contain: paint;
  z-index: ${V.hooksCompContainer};
`, Ee = R.div`
  height: ${(t) => `calc(${t.$height ? t.$height + "px" : "100vh"} - env(safe-area-inset-bottom) -
      ${t.$hideBodyHeader ? "0px" : $e}
  )`};
`, ln = R.div`
  .ant-menu {
    background-color: #393b47;
    height: calc(100vh - 48px);

    .ant-menu-item {
      padding: 0 7px !important;
      width: 40px;
      height: 26px;
      margin: 12px 0 0 0;

      svg {
        height: 26px;
        width: 26px;
        padding: 5px;
      }

      &.ant-menu-item-selected,
      &:hover,
      &:active {
        background-color: #393b47;

        svg {
          background: #8b8fa37f;
          border-radius: 4px;
        }
      }
    }
  }

  z-index: ${V.leftToolbar};
`, un = R.div`
  > div {
    left: 6px;
    right: auto;
    height: 28px;
    bottom: 36px;

    > div.shortcutList {
      left: 42px;
      bottom: 2px;
    }
  }
`;
R.div`
  > div {
    left: 6px;
    right: auto;
    height: 28px;
    top: 15px;
  }
`;
const dn = R.div`
  display: flex;
  flex-direction: column;
  height: 100%;

  .ant-divider {
    margin: 16px 0;
    border-color: #e1e3eb;
  }
`, pn = R.div`
  font-weight: 500;
  font-size: 14px;
  color: #8b8fa3;
  line-height: 14px;
  margin: 16px;
`, fn = R.div`
  margin: 16px 8px;
  height: 33px;
  display: flex;
  align-items: center;
  padding: 0 8px 0 7px;
  color: #333;
  cursor: pointer;

  svg {
    margin-right: 8px;

    path {
      fill: #222222;
    }
  }

  &:hover,
  &:active {
    background: #f2f7fc;
    border-radius: 4px;

    svg path {
      fill: #315efb;
    }
  }
`, hn = R.div`
  overflow: auto;
  position: relative;
  flex: 1 1 0;
`, mn = [{
  key: "state",
  icon: /* @__PURE__ */ u(we, {})
}, {
  key: "setting",
  icon: /* @__PURE__ */ u(Ae, {})
}, {
  key: "layout",
  icon: /* @__PURE__ */ u(lt, {})
}], gn = [{
  key: "state",
  icon: /* @__PURE__ */ u(we, {})
}, {
  key: "setting",
  icon: /* @__PURE__ */ u(Ae, {})
}];
function bn(t) {
  const {
    uiComp: e
  } = t, r = Ve(), n = pe(Be), {
    readOnly: o,
    hideHeader: l
  } = pe(Ye), a = X(Ge), s = X(qe), i = We().state, c = i == null ? void 0 : i.showNewUserGuide, d = X(ze), [p, h] = M(!1), m = N(() => h(!p), [p]), [v, P] = M(
    "state"
    /* State */
  ), [C, O] = M(), H = Xe(), [E, U] = M(() => c ? ae : Ke()), [te, xe] = M(ae), Re = r.viewMode === "view", ne = n.getAppSettingsComp(), {
    showHeaderInPublic: He
  } = ne.getView(), re = N((g) => {
    let S;
    g ? (S = Object.assign({}, E), S[g] = !E[g]) : Object.values(E).some((je) => je) ? (xe(E), S = {
      left: !1,
      bottom: !1,
      right: !1
    }) : S = te, U(S), se(S);
  }, [E, te]), [$, Ie] = M(() => Qe()), Me = N((g) => {
    Ie(g || "both"), Je(g || "both");
  }, [$]), _e = N((g) => {
    n.setDraggingCompType(g);
  }, [n]), ke = N((g) => {
    n.setShowPropertyPane(g === "property");
  }, [n]), q = ut(() => Object.keys(n.getHooksComp().children).map((g) => (
    // use appId as key, remount hook comp when app change. fix hookStateComp empty value
    /* @__PURE__ */ u("div", {
      children: n.getHooksComp().children[g].getView()
    }, g + "-" + (a == null ? void 0 : a.applicationId))
  )), [n]);
  dt(() => {
    function g() {
      O(window.innerHeight);
    }
    const S = "orientationchange" in window ? "orientationchange" : "resize";
    return window.addEventListener(S, g), g(), () => window.removeEventListener(S, g);
  }, []);
  const W = Ze() || Re && (!He || !s.showHeaderInPublicApps), oe = window.location.hostname === "app.lowcoder.cloud", ie = window.location.hostname === "localhost";
  if (o && l)
    return /* @__PURE__ */ b(ye, {
      children: [e.getView(), /* @__PURE__ */ u("div", {
        style: {
          zIndex: V.hooksCompContainer
        },
        children: q
      })]
    });
  if (o && !d)
    return /* @__PURE__ */ b(ye, {
      children: [/* @__PURE__ */ b(G, {
        children: [a && /* @__PURE__ */ u("title", {
          children: a.name
        }), oe || ie && [
          // Adding Support for iframely to be able to embedd apps as iframes
          a != null && a.name ? [/* @__PURE__ */ u("meta", {
            property: "iframely:title",
            content: a.name
          }, "iframely:title"), /* @__PURE__ */ u("meta", {
            property: "iframely:description",
            content: a.description
          }, "iframely:description")] : [/* @__PURE__ */ u("meta", {
            property: "iframely:title",
            content: "Lowcoder 3"
          }, "iframely:title"), /* @__PURE__ */ u("meta", {
            property: "iframely:description",
            content: "Lowcoder | rapid App & VideoMeeting builder for everyone."
          }, "iframely:description")],
          /* @__PURE__ */ u("link", {
            rel: "iframely",
            type: "text/html",
            href: window.location.href,
            media: "(aspect-ratio: 1280/720)"
          }),
          /* @__PURE__ */ u("link", {
            rel: "preconnect",
            href: "https://fonts.googleapis.com"
          }, "preconnect-googleapis"),
          /* @__PURE__ */ u("link", {
            rel: "preconnect",
            href: "https://fonts.gstatic.com",
            crossOrigin: "anonymous"
          }, "preconnect-gstatic"),
          /* @__PURE__ */ u("link", {
            href: "https://fonts.googleapis.com/css2?family=Ubuntu:ital,wght@0,300;0,400;0,700;1,400&display=swap",
            rel: "stylesheet"
          }, "font-ubuntu"),
          // adding Clearbit Support for Analytics
          /* @__PURE__ */ u("script", {
            src: "https://tag.clearbitscripts.com/v1/pk_dfbc0aeefb28dc63475b67134facf127/tags.js",
            referrerPolicy: "strict-origin-when-cross-origin",
            type: "text/javascript"
          }, "clearbit-script")
        ]]
      }), /* @__PURE__ */ b(pt, {
        fallback: /* @__PURE__ */ u(et, {}),
        children: [!W && /* @__PURE__ */ u(sn, {}), /* @__PURE__ */ b(ve, {
          children: [/* @__PURE__ */ u(Ee, {
            $hideBodyHeader: W,
            $height: C,
            children: e.getView()
          }), /* @__PURE__ */ u("div", {
            style: {
              zIndex: V.hooksCompContainer
            },
            children: q
          })]
        })]
      })]
    });
  const Ne = E.right || d;
  let z;
  d ? z = /* @__PURE__ */ u(Ee, {
    $hideBodyHeader: W,
    $height: C,
    children: /* @__PURE__ */ u(tn, {
      children: e.getView()
    })
  }) : z = e.getView();
  const De = (g) => {
    let S = !0;
    E.left && g.key === v && (S = !1), U({
      ...E,
      left: S
    }), se({
      ...E,
      left: S
    }), P(g.key);
  };
  return /* @__PURE__ */ b(ce, {
    children: [/* @__PURE__ */ b(G, {
      children: [a && /* @__PURE__ */ u("title", {
        children: a.name
      }), oe || ie && [
        // Adding Support for iframely to be able to embedd apps as iframes
        a != null && a.name ? [/* @__PURE__ */ u("meta", {
          property: "iframely:title",
          content: a.name
        }, "iframely:title"), /* @__PURE__ */ u("meta", {
          property: "iframely:description",
          content: a.description
        }, "iframely:description")] : [/* @__PURE__ */ u("meta", {
          property: "iframely:title",
          content: "Lowcoder 3"
        }, "iframely:title"), /* @__PURE__ */ u("meta", {
          property: "iframely:description",
          content: "Lowcoder | rapid App & VideoMeeting builder for everyone."
        }, "iframely:description")],
        /* @__PURE__ */ u("link", {
          rel: "iframely",
          type: "text/html",
          href: window.location.href,
          media: "(aspect-ratio: 1280/720)"
        }, "iframely"),
        /* @__PURE__ */ u("link", {
          rel: "preconnect",
          href: "https://fonts.googleapis.com"
        }, "preconnect-googleapis"),
        /* @__PURE__ */ u("link", {
          rel: "preconnect",
          href: "https://fonts.gstatic.com",
          crossOrigin: "anonymous"
        }, "preconnect-gstatic"),
        /* @__PURE__ */ u("link", {
          href: "https://fonts.googleapis.com/css2?family=Ubuntu:ital,wght@0,300;0,400;0,700;1,400&display=swap",
          rel: "stylesheet"
        }, "font-ubuntu"),
        // adding Clearbit Support for Analytics
        /* @__PURE__ */ u("script", {
          src: "https://tag.clearbitscripts.com/v1/pk_dfbc0aeefb28dc63475b67134facf127/tags.js",
          referrerPolicy: "strict-origin-when-cross-origin",
          type: "text/javascript"
        }, "clearbit-script")
      ]]
    }), /* @__PURE__ */ b(nn, {
      onDragEnd: (g) => {
        n.setDragging(!1), tt.clearData();
      },
      children: [/* @__PURE__ */ u(nt, {
        togglePanel: re,
        panelStatus: E,
        toggleEditorModeStatus: Me,
        editorModeStatus: $
      }), c && /* @__PURE__ */ u(Kt, {}), /* @__PURE__ */ u(Jt, {
        disabled: o,
        togglePanel: re,
        panelStatus: E,
        toggleShortcutList: m,
        children: /* @__PURE__ */ b(en, {
          children: [/* @__PURE__ */ u(ln, {
            children: /* @__PURE__ */ b(rt, {
              width: 40,
              children: [/* @__PURE__ */ u(ot, {
                theme: "dark",
                mode: "inline",
                defaultSelectedKeys: [
                  "state"
                  /* State */
                ],
                selectedKeys: E.left ? [v] : [""],
                items: a && !le(ue[a.applicationType]) ? mn : gn,
                disabled: d,
                onClick: (g) => De(g)
              }), !d && /* @__PURE__ */ u(un, {
                children: /* @__PURE__ */ u(an, {
                  showShortcutList: p,
                  setShowShortcutList: h,
                  isEdit: !0
                })
              })]
            })
          }), E.left && $ !== "layout" && /* @__PURE__ */ b(rn, {
            children: [v === "state" && /* @__PURE__ */ u(Wt, {
              uiComp: e
            }), v === "setting" && /* @__PURE__ */ b(dn, {
              children: [/* @__PURE__ */ b(it, {
                children: [a && !le(ue[a.applicationType]) && /* @__PURE__ */ b(ce, {
                  children: [ne.getPropertyView(), /* @__PURE__ */ u(at, {})]
                }), /* @__PURE__ */ u(pn, {
                  children: de("leftPanel.toolbarTitle")
                }), t.preloadComp.getPropertyView(), /* @__PURE__ */ b(fn, {
                  onClick: () => H(qt({
                    showScriptsAndStyleModal: !0
                  })),
                  children: [/* @__PURE__ */ u(st, {}), de("leftPanel.toolbarPreload")]
                })]
              }), t.preloadComp.getJSLibraryPropertyView()]
            }), v === "layout" && /* @__PURE__ */ u(zt, {
              uiComp: e
            })]
          }), /* @__PURE__ */ b(on, {
            children: [/* @__PURE__ */ u(hn, {
              className: ct,
              children: /* @__PURE__ */ u(Zt, {
                disabled: o,
                children: /* @__PURE__ */ b(ve, {
                  children: [z, /* @__PURE__ */ u(cn, {
                    children: q
                  })]
                })
              })
            }), E.bottom && $ !== "layout" && /* @__PURE__ */ u(Qt, {})]
          }), Ne && /* @__PURE__ */ u(Xt, {
            uiComp: e,
            onCompDrag: _e,
            showPropertyPane: n.showPropertyPane,
            onTabChange: ke
          })]
        })
      })]
    })]
  });
}
export {
  hn as EditorWrapper,
  bn as default
};
