// lowcoder-sdk/dist/6d3d41c3.js
import { _ as te, kV as oe, a8 as h, kW as ie, F as n, kX as ce, gf as N, kY as de, kZ as ue, k_ as pe, k$ as ge, l0 as he, l1 as me, l2 as ye, l3 as fe, l4 as Ce, l5 as xe, l6 as be, l7 as ve, fg as ke, l8 as re, X as ae, it as Se, ah as Ne, l9 as Be, iI as Ie, la as Te, iE as Q, iu as we, dp as De, iw as Fe, Y as L, ix as Ae, G as Ee, da as Ve, iB as $e, iC as Re, iD as Ke, lb as Pe, al as Ue, bV as le, gW as We, gy as ze, cv as Le, aj as Me } from "./c988e26e.js";
import { useState as p, useEffect as f, useContext as Oe, useRef as je, useCallback as G, useMemo as Ge } from "react";
import { D as He, C as Je, N as Xe } from "./e050e560.js";
import { c as Ye } from "./182e4b4f.js";
import { a as _e } from "./2d92f8b6.js";
import "react-dom";
import "./8f0aa994.js";
import "./14897c7e.js";
function ee(H) {
  const {
    themeSettingKey: r,
    name: M,
    desc: O,
    color: v,
    radius: B,
    configChange: C,
    showVarName: E = !0,
    margin: k,
    padding: I,
    gridColumns: S,
    borderStyle: T,
    borderWidth: w,
    borderColor: j,
    fontFamily: V
  } = H, J = te.debounce(C, 0), [x, b] = p(v), [X, $] = p(B), [D, F] = p(k), [Y, R] = p(I), [_, K] = p(S), [P, U] = p(T), [W, A] = p(w), [ne, q] = p(j), [Z, z] = p(V), l = `(${r})`, s = () => {
    !x || !oe(x) ? b(v) : (b(re(x)), C({
      themeSettingKey: r,
      color: re(x)
    }));
  }, o = (e) => {
    let t = "";
    !e || Number(e) === 0 ? t = "0" : /^[0-9]+$/.test(e) ? t = Number(e) + "px" : /^[0-9]+(px|%)$/.test(e) ? t = e : t = "0", $(t), C({
      themeSettingKey: r,
      radius: t
    });
  }, i = (e) => {
    let t = "";
    !e || Number(e) === 0 ? t = "0" : /^[0-9]+$/.test(e) ? t = Number(e) + "px" : /^[0-9]+(px|%)$/.test(e) ? t = e : t = "3px", F(t), C({
      themeSettingKey: r,
      margin: t
    });
  }, c = (e) => {
    let t = "";
    !e || Number(e) === 0 ? t = "0" : /^[0-9]+$/.test(e) ? t = Number(e) + "px" : /^[0-9]+(px|%)$/.test(e) ? t = e : t = "3px", R(t), C({
      themeSettingKey: r,
      padding: t
    });
  }, d = (e) => {
    let t = "";
    e ? t = e : t = "24", K(t), C({
      themeSettingKey: r,
      gridColumns: t
    });
  }, u = (e) => {
    let t = "";
    e ? t = e : t = "solid", U(t), C({
      themeSettingKey: r,
      borderStyle: t
    });
  }, y = (e) => {
    let t = "";
    !e || Number(e) === 0 ? t = "0" : /^[0-9]+$/.test(e) ? t = Number(e) + "px" : /^[0-9]+(px|%)$/.test(e) ? t = e : t = "1px", A(e), C({
      themeSettingKey: r,
      borderWidth: t
    });
  }, m = (e) => {
    let t = "";
    e ? t = e : t = "Roboto, sans-serif", z(t), C({
      themeSettingKey: r,
      fontFamily: t
    });
  };
  return f(() => {
    x && oe(x) && J({
      themeSettingKey: r,
      color: x
    });
  }, [x]), f(() => {
    b(v);
  }, [v]), f(() => {
    $(B);
  }, [B]), f(() => {
    F(k);
  }, [k]), f(() => {
    R(I);
  }, [I]), f(() => {
    K(S);
  }, [S]), f(() => {
    U(T);
  }, [T]), f(() => {
    A(w);
  }, [w]), f(() => {
    q(j);
  }, [j]), f(() => {
    z(V);
  }, [V]), /* @__PURE__ */ h(ie, {
    className: H.className,
    children: [/* @__PURE__ */ h("div", {
      className: "text-desc",
      children: [/* @__PURE__ */ h("div", {
        className: "name",
        children: [M, " ", E && /* @__PURE__ */ n("span", {
          children: l
        })]
      }), /* @__PURE__ */ n("div", {
        className: "desc",
        children: O
      })]
    }), r !== "radius" && r !== "margin" && r !== "padding" && r !== "gridColumns" && r !== "borderStyle" && r !== "borderWidth" && r !== "fontFamily" && /* @__PURE__ */ h("div", {
      className: "config-input",
      children: [/* @__PURE__ */ n(ce, {
        changeColor: te.debounce(b, 500, {
          leading: !0,
          trailing: !0
        }),
        color: x,
        trigger: "hover"
      }), /* @__PURE__ */ n(N, {
        value: x,
        onChange: (e) => b(e.target.value),
        onBlur: s,
        onKeyUp: (e) => e.nativeEvent.key === "Enter" && s()
      })]
    }), r === "radius" && /* @__PURE__ */ h("div", {
      className: "config-input",
      children: [/* @__PURE__ */ n(de, {
        $radius: B || "0",
        children: /* @__PURE__ */ n("div", {
          children: /* @__PURE__ */ n(ue, {
            style: {
              width: "16px",
              margin: "-2px 0 2px -2px",
              padding: "0px"
            }
          })
        })
      }), /* @__PURE__ */ n(N, {
        value: X,
        onChange: (e) => $(e.target.value),
        onBlur: (e) => o(e.target.value),
        onKeyUp: (e) => e.nativeEvent.key === "Enter" && o(e.currentTarget.value)
      })]
    }), r === "borderStyle" && /* @__PURE__ */ h("div", {
      className: "config-input",
      children: [/* @__PURE__ */ n(pe, {
        $borderStyle: T || "solid",
        children: /* @__PURE__ */ n("div", {
          children: /* @__PURE__ */ n(ge, {
            style: {
              width: "16px",
              margin: "2px 0 0 2px",
              padding: "0px"
            }
          })
        })
      }), /* @__PURE__ */ n(N, {
        value: P,
        onChange: (e) => U(e.target.value),
        onBlur: (e) => u(e.target.value),
        onKeyUp: (e) => e.nativeEvent.key === "Enter" && u(e.currentTarget.value)
      })]
    }), r === "borderWidth" && /* @__PURE__ */ h("div", {
      className: "config-input",
      children: [/* @__PURE__ */ n(he, {
        $borderWidth: w || "1px",
        children: /* @__PURE__ */ n("div", {
          children: /* @__PURE__ */ n(me, {
            style: {
              width: "16px",
              margin: "2px 0 0 2px",
              padding: "0px"
            }
          })
        })
      }), /* @__PURE__ */ n(N, {
        value: W,
        onChange: (e) => A(e.target.value),
        onBlur: (e) => y(e.target.value),
        onKeyUp: (e) => e.nativeEvent.key === "Enter" && y(e.currentTarget.value)
      })]
    }), r === "margin" && /* @__PURE__ */ h("div", {
      className: "config-input",
      children: [/* @__PURE__ */ n(ye, {
        $margin: k || "4px",
        children: /* @__PURE__ */ n("div", {
          children: /* @__PURE__ */ n(fe, {
            title: ""
          })
        })
      }), /* @__PURE__ */ n(N, {
        value: D,
        onChange: (e) => F(e.target.value),
        onBlur: (e) => i(e.target.value),
        onKeyUp: (e) => e.nativeEvent.key === "Enter" && i(e.currentTarget.value)
      })]
    }), r === "padding" && /* @__PURE__ */ h("div", {
      className: "config-input",
      children: [/* @__PURE__ */ n(Ce, {
        $padding: I || "4px",
        children: /* @__PURE__ */ n("div", {
          children: /* @__PURE__ */ n(xe, {
            title: ""
          })
        })
      }), /* @__PURE__ */ n(N, {
        value: Y,
        onChange: (e) => R(e.target.value),
        onBlur: (e) => c(e.target.value),
        onKeyUp: (e) => e.nativeEvent.key === "Enter" && c(e.currentTarget.value)
      })]
    }), r === "gridColumns" && /* @__PURE__ */ h("div", {
      className: "config-input",
      children: [/* @__PURE__ */ n(be, {
        $gridColumns: S || "24",
        children: /* @__PURE__ */ n("div", {
          children: /* @__PURE__ */ n(ve, {
            title: ""
          })
        })
      }), /* @__PURE__ */ n(ke, {
        style: {
          width: "90%",
          margin: "8px 5% 0 5%"
        },
        min: 8,
        max: 48,
        value: parseInt(_ || "24"),
        onChange: (e) => K(e.toString()),
        onChangeComplete: (e) => d(e.toString())
      })]
    }), r === "fontFamily" && /* @__PURE__ */ n("div", {
      className: "config-input",
      children: /* @__PURE__ */ n(N, {
        value: Z,
        onChange: (e) => z(e.target.value),
        onBlur: (e) => m(e.target.value),
        onKeyUp: (e) => e.nativeEvent.key === "Enter" && m(e.currentTarget.value)
      })
    })]
  });
}
const qe = [{
  label: "Hide Component",
  key: "hidden"
}, {
  label: "Disable Component",
  key: "disable"
}, {
  label: "Margin",
  key: "style.margin"
}, {
  label: "Padding",
  key: "style.padding"
}, {
  label: "Font Size",
  key: "style.textSize"
}, {
  label: "Font Weight",
  key: "style.textWeight"
}, {
  label: "Font Family",
  key: "style.fontFamily"
}, {
  label: "Font Style",
  key: "style.fontStyle"
}, {
  label: "Text Transform",
  key: "style.textTransform"
}, {
  label: "Text Decoration",
  key: "style.textDecoration"
}, {
  label: "Border Radius",
  key: "style.borderRadius"
}, {
  label: "Border Width",
  key: "style.borderWidth"
}, {
  label: "Border Style",
  key: "style.borderStyle"
}, {
  label: "Background Image",
  key: "style.backgroundImage"
}, {
  label: "Background Image Repeat",
  key: "style.backgroundImageRepeat"
}, {
  label: "Background Image Size",
  key: "style.backgroundImageSize"
}, {
  label: "Background Image Position",
  key: "style.backgroundImagePosition"
}, {
  label: "Background Image Origin",
  key: "style.backgroundImageOrigin"
}], Ze = ae.div`
  height: calc(100vh - ${Se});
`, Qe = ae(Me)`
  .ant-dropdown-menu-item-icon {
    width: 14px !important;
    height: 14px !important; 
    max-width: 14px !important;
  }
`, it = (H) => {
  const r = Oe(Ne), [M, O] = p([]);
  Be(), Ie();
  const [v, B] = p(r.getCollisionStatus()), [C, E] = p([]), [k, I] = p(""), [S, T] = p(null), [w, j] = p(""), V = je(null);
  f(() => {
    Te(v);
  }, [v]);
  const J = G((l) => {
    T(l);
  }, []), x = (l) => {
    r.rootComp.children.settings.children.disableCollision.dispatchChangeValueAction(l), B(l);
  }, b = (l, s, o) => {
    const {
      items: i,
      children: c
    } = l;
    if (Object.keys(i).length)
      for (const d in i) {
        const u = {
          title: i[d].children.name.getView(),
          type: i[d].children.compType.getView(),
          key: d,
          children: []
        };
        if (o) {
          const y = Q(s, o);
          u.disabled = !0, y == null || y.children.push(u);
        } else
          s.push(u);
      }
    if (Object.keys(c).length)
      for (const d in c)
        b(c[d], s, d);
    return s;
  }, X = G((l) => {
    r.setSelectedCompNames(/* @__PURE__ */ new Set([l]), "leftPanel");
  }, [r]), $ = (l, s) => (s.find((o) => o.name === l.title), /* @__PURE__ */ n(Xe, {
    children: /* @__PURE__ */ n("span", {
      children: /* @__PURE__ */ n("span", {
        children: l.title
      })
    })
  })), [D, F] = p([]);
  f(() => {
    const l = Y();
    F(l);
  }, [r]);
  const Y = () => {
    const l = r.getUIComp().getTree(), s = b(l, []), o = r.rootComp.toJsonValue();
    return o.ui.compType === "module" ? s.forEach((i) => {
      i.pos = o.ui.comp.container.layout[i.key].pos;
    }) : s.forEach((i) => {
      i.pos = o.ui.layout[i.key].pos;
    }), s.sort((i, c) => {
      const d = (i == null ? void 0 : i.pos) || 0, u = (c == null ? void 0 : c.pos) || 0;
      return d < u ? -1 : d > u ? 1 : 0;
    }), s;
  }, R = (l) => {
    const {
      expandedKeys: s
    } = l;
    s.includes(l.node.key) || O(s);
  }, _ = (l) => {
    const s = l.node.pos.split("-"), o = l.dragNode.pos.split("-");
    s.length === o.length && F((i) => {
      let c = Ye(i);
      const d = Number(s[s.length - 1]), u = Number(o[o.length - 1]), y = s.slice(0, -1).join("-"), m = y === "0" ? {
        children: c
      } : K(c, y);
      if (m && m.children) {
        const g = m.children.findIndex((a) => a.key === l.dragNode.key);
        if (g !== -1) {
          const [a] = m.children.splice(g, 1);
          m.children.splice(d > u ? d - 1 : d, 0, a);
        }
      }
      const e = r.rootComp.toJsonValue();
      let t = {};
      if (e.ui.compType === "module") {
        m.children.forEach((a, se) => {
          t[a.key] = {
            ...e.ui.comp.container.layout[a.key],
            pos: se
          };
        });
        const g = r.rootComp.children.ui.getModuleLayoutComp();
        g == null || g.children.container.dispatchChangeValueAction({
          ...e.ui.comp.container,
          layout: t
        });
      } else
        m.children.forEach((g, a) => {
          t[g.key] = {
            ...e.ui.layout[g.key],
            pos: a
          };
        }), r.rootComp.children.ui.dispatchChangeValueAction({
          ...e.ui,
          layout: t
        });
      return c;
    });
  }, K = (l, s) => {
    const o = s.split("-").map((c) => Number(c));
    let i = {
      children: l
    };
    for (let c = 0; c < o.length; c++)
      i = i.children[o[c]];
    return i;
  };
  G((l) => {
    switch (l) {
      case "hidden":
      case "disable":
        return "true | false";
      case "style.border":
        return "e.g., #ffffff";
      case "style.borderRadius":
      case "style.radius":
        return "e.g., 4px";
      case "style.borderWidth":
        return "e.g., 2px";
      case "style.borderStyle":
        return "solid | dashed | dotted";
      case "style.backgroundImage":
        return "URL as string";
      case "style.backgroundImageRepeat":
        return "repeat | repeat-x | repeat-y | no-repeat";
      case "style.backgroundImageSize":
        return "cover | contain | % | px";
      case "style.backgroundImagePosition":
        return "top | bottom | center | % | px";
      case "style.backgroundImageOrigin":
        return "padding-box | border-box | content-box";
      case "style.margin":
      case "style.padding":
        return "e.g., 4px 8px 16px 32px";
      case "style.textSize":
        return "e.g., 16px";
      case "style.textWeight":
        return "bold | 900 | normal | 400";
      case "style.fontFamily":
        return "Arial, sans-serif";
      case "style.fontStyle":
        return "normal | italic | oblique";
      case "style.textTransform":
        return "none | capitalize | uppercase | lowercase";
      case "style.textDecoration":
        return "none | underline | overline | line-through";
      default:
        return "Action Value";
    }
  }, []);
  const P = (l, s) => {
    var o, i, c, d, u;
    for (let y of W()) {
      const m = Q(D, y), e = r.getUICompByName(m.title);
      if (e) {
        const {
          children: t
        } = e.children.comp, g = e.children.compType.getView(), a = s.split(".");
        a.length === 1 ? (o = t[a[0]]) == null || o.dispatchChangeValueAction(l) : a.length === 2 && (t[a[0]] ? t[a[0]].children[a[1]] ? (d = t[a[0]].children[a[1]]) == null || d.dispatchChangeValueAction(l) : (u = t[a[0]][a[1]]) == null || u.dispatchChangeValueAction(l) : (i = t[g].children[a[0]]) != null && i.children[a[1]] && ((c = t[g].children[a[0]].children[a[1]]) == null || c.dispatchChangeValueAction(l)));
      }
    }
  };
  f(() => {
    E([]);
    const l = [], s = r.getUIComp().getTree(), o = b(s, []);
    for (let i of r.selectedCompNames)
      for (let c of o)
        c.title === i && l.push(c.key);
    E(l);
  }, [r]);
  const U = (l, s) => {
    E(l);
    const o = /* @__PURE__ */ new Set();
    for (let i of s.checkedNodes)
      o.add(i.title);
    r.setSelectedCompNames(o, "leftPanel");
  }, W = () => C, A = () => k, ne = G((l) => {
    var o, i, c, d, u;
    const s = A();
    for (let y of W()) {
      const m = Q(D, y), e = r.getUICompByName(m.title);
      if (e) {
        const {
          children: t
        } = e.children.comp, g = e.children.compType.getView(), a = l.split(".");
        a.length === 1 ? (o = t[a[0]]) == null || o.dispatchChangeValueAction(s) : a.length === 2 && (t[a[0]] ? t[a[0]].children[a[1]] ? (d = t[a[0]].children[a[1]]) == null || d.dispatchChangeValueAction(s) : (u = t[a[0]][a[1]]) == null || u.dispatchChangeValueAction(s) : (i = t[g].children[a[0]]) != null && i.children[a[1]] && ((c = t[g].children[a[0]].children[a[1]]) == null || c.dispatchChangeValueAction(s)));
      }
    }
  }, [A, W]), q = () => {
    const l = te.sortBy(r.uiCompInfoList(), [(o) => o.name]), s = r.getCollisionStatus();
    return /* @__PURE__ */ h(Ee, {
      children: [/* @__PURE__ */ h("div", {
        style: {
          margin: "0px 16px"
        },
        children: [/* @__PURE__ */ h("div", {
          style: {
            marginBottom: "10px"
          },
          children: [L("leftPanel.activatelayers"), /* @__PURE__ */ n(Ve, {
            style: {
              margin: "0px 10px"
            },
            size: "small",
            defaultChecked: v,
            disabled: !1,
            onChange: (o) => {
              x(o);
            }
          })]
        }), /* @__PURE__ */ n(He, {
          checkable: !0,
          onCheck: U,
          selectable: !1,
          checkedKeys: C,
          draggable: s,
          onDragEnter: R,
          onDrop: (o) => _(o),
          treeData: D,
          icon: (o) => o.type && /* @__PURE__ */ n("div", {
            style: {
              margin: "3px 4px 0 -4px"
            },
            children: Je[o.type] || /* @__PURE__ */ n($e, {})
          }),
          switcherIcon: (o) => o.expanded ? /* @__PURE__ */ n(Re, {}) : /* @__PURE__ */ n(Ke, {}),
          expandedKeys: M,
          onExpand: (o) => O(o),
          titleRender: (o) => $(o, l)
        }), /* @__PURE__ */ n("div", {
          style: {
            margin: "10px 0px"
          },
          children: /* @__PURE__ */ h(Pe, {
            gap: "small",
            vertical: !0,
            children: [/* @__PURE__ */ n(Qe, {
              dropdownRender: () => /* @__PURE__ */ n(Ue, {
                items: qe,
                onClick: ({
                  key: o
                }) => {
                  J(o);
                }
              }),
              children: /* @__PURE__ */ n(le, {
                size: "small",
                children: /* @__PURE__ */ h(We, {
                  children: ["Action ", /* @__PURE__ */ n(_e, {})]
                })
              })
            }), /* @__PURE__ */ n(ze, {
              ref: V,
              value: k,
              onChange: (o) => I(o.target.value),
              placeholder: w
            }), /* @__PURE__ */ n(le, {
              type: "primary",
              disabled: !S,
              onClick: () => S && ne(S),
              children: "Apply Action"
            }), /* @__PURE__ */ n("br", {}), /* @__PURE__ */ n(ee, {
              themeSettingKey: "background",
              name: L("componentDoc.styleBackgroundColor"),
              color: "#ffffff",
              configChange: (o) => P(o.color, "style.background")
            }), /* @__PURE__ */ n(ee, {
              themeSettingKey: "border",
              name: L("componentDoc.styleBorderColor"),
              color: "#ffffff",
              configChange: (o) => P(o.color, "style.border")
            }), /* @__PURE__ */ n(ee, {
              themeSettingKey: "text",
              name: L("style.textColor"),
              color: "#ffffff",
              configChange: (o) => P(o.color, "style.text")
            })]
          })
        })]
      }), /* @__PURE__ */ n(Le, {})]
    });
  }, Z = Ge(() => {
    if (!we(r.getAppType()))
      return q();
  }, [r, X, M, D, k]), z = /* @__PURE__ */ n(De, {
    children: /* @__PURE__ */ n("div", {
      style: {
        paddingBottom: 80
      },
      children: /* @__PURE__ */ n(Fe, {
        name: L("leftPanel.layers"),
        width: 288,
        noMargin: !0,
        children: /* @__PURE__ */ n("div", {
          className: Ae,
          children: Z
        })
      })
    })
  });
  return /* @__PURE__ */ n(Ze, {
    className: "cypress-left-content",
    children: z
  });
};
export {
  it as LeftLayersContent
};
