// lowcoder-sdk/dist/73ee57e8.js
import { X as d, ah as y, dm as P, a8 as h, F as n, dn as u, dq as m, dw as z, dB as W, av as H, aw as R, ax as S, dv as l } from "./c988e26e.js";
import { useContext as F } from "react";
import "react-dom";
const G = (o) => H`
    border-radius: ${o.radius ? o.radius : "4px"};
    border: ${o.borderWidth ? o.borderWidth : "0px"} ${o.borderStyle ? o.borderStyle : "solid"} ${o.border};
    color: ${o.text};
    font-size: ${o.textSize} !important;
    font-weight: ${o.textWeight} !important;
    font-family: ${o.fontFamily} !important;
    font-style:${o.fontStyle} !important;
    text-transform:${o.textTransform} !important;
    text-decoration:${o.textDecoration} !important;
    background-color: ${o.background};
    .markdown-body a {
      color: ${o.links};
    }
    .markdown-body {
      margin: ${o.margin} !important;	
      padding: ${o.padding};	
      width: ${R(o.margin)};	
      // height: ${S(o.margin)};
      h1 {
        line-height: 1.5;
      }
      h5 {
        line-height: 2.2;
      }
    }

    .markdown-body {
      &,
      p,
      div,
      h1,
      h2,
      h3,
      h4,
      h5,
      h6 {
        color: ${o.text};
      }
      img,
      pre {
        background-color: ${o.background};
        code {
          color: #000000;
        }
      }
    }
  `, O = d.div`
${(o) => o.$animationStyle && o.$animationStyle}
  display: flex;
  flex-flow: column;
  height: 100%;
  border: ${(o) => o.$style.borderWidth} ${(o) => o.$style.borderStyle ? o.$style.borderStyle : "solid"} ${(o) => o.$style.border};
  border-radius: ${(o) => o.$style.radius};
  background-color: ${(o) => o.$style.background};
  padding: ${(o) => o.$style.padding};
  margin: ${(o) => o.$style.margin};
  ${(o) => o.$style.backgroundImage && `background-image: url(${o.$style.backgroundImage});`}
  ${(o) => o.$style.backgroundImageRepeat && `background-repeat: ${o.$style.backgroundImageRepeat};`}
  ${(o) => o.$style.backgroundImageSize && `background-size: ${o.$style.backgroundImageSize};`}
  ${(o) => o.$style.backgroundImagePosition && `background-position: ${o.$style.backgroundImagePosition};`}
  ${(o) => o.$style.backgroundImageOrigin && `background-origin: ${o.$style.backgroundImageOrigin};`}
`, p = d.div`
  ${(o) => o.$style && G(o.$style)}
  text-align: ${(o) => o.$horizontalAlignment};
  padding: ${(o) => o.$style.padding};
  margin: ${(o) => o.$style.margin};
`, T = d(l)`
  overflow: visible;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
`, A = d(l)`
  border-top: ${(o) => `${o.$showBorder ? o.$borderWidth : 0} solid ${o.$borderColor}`};
  flex: 1;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
`, j = d(l)`
  border-top: ${(o) => `${o.$showBorder ? o.$borderWidth : 0} solid ${o.$borderColor}`};
  overflow: visible;
  ${(o) => o.$backgroundColor && `background-color: ${o.$backgroundColor};`}
  border-radius: 0;
  ${(o) => o.$footerBackgroundImage && `background-image: url(${o.$footerBackgroundImage});`}
  ${(o) => o.$footerBackgroundImageRepeat && `background-repeat: ${o.$footerBackgroundImageRepeat};`}
  ${(o) => o.$footerBackgroundImageSize && `background-size: ${o.$footerBackgroundImageSize};`}
  ${(o) => o.$footerBackgroundImagePosition && `background-position: ${o.$footerBackgroundImagePosition};`}
  ${(o) => o.$footerBackgroundImageOrigin && `background-origin: ${o.$footerBackgroundImageOrigin};`}
`;
function E(o) {
  const {
    container: a,
    text: b,
    animationStyle: k
  } = o, {
    showHeader: t,
    showFooter: g,
    horizontalGridCells: $
  } = a, s = a.showBody || !t && !g, {
    items: I,
    ...f
  } = a.header, {
    items: B,
    ...w
  } = a.body[0].children.view.getView(), {
    items: x,
    ...C
  } = a.footer, v = F(y).getAppSettings().maxWidth;
  P(v);
  const {
    style: e,
    headerStyle: i,
    bodyStyle: c,
    footerStyle: r
  } = a;
  return /* @__PURE__ */ h(O, {
    $style: e,
    $animationStyle: k,
    children: [t && /* @__PURE__ */ n(u.Provider, {
      value: a.style.background,
      children: /* @__PURE__ */ n(T, {
        ...f,
        horizontalGridCells: $,
        items: m(I),
        autoHeight: !0,
        emptyRows: 5,
        minHeight: "46px",
        containerPadding: [0, 0],
        showName: {
          bottom: g ? 20 : 0
        },
        $backgroundColor: (i == null ? void 0 : i.headerBackground) || "transparent",
        style: {
          padding: i.containerHeaderPadding
        }
      })
    }), s && /* @__PURE__ */ n(u.Provider, {
      value: a.style.background,
      children: /* @__PURE__ */ h("div", {
        style: {
          overflowY: "scroll",
          background: `${a.style.background}`
        },
        children: [/* @__PURE__ */ n(A, {
          $showBorder: !1,
          ...w,
          horizontalGridCells: $,
          items: m(B),
          autoHeight: a.autoHeight,
          emptyRows: 14,
          minHeight: t ? "143px" : "142px",
          containerPadding: [0, 0],
          hintPlaceholder: o.hintPlaceholder ?? z,
          $backgroundColor: (c == null ? void 0 : c.background) || "transparent",
          $borderColor: e == null ? void 0 : e.border,
          $borderWidth: e == null ? void 0 : e.borderWidth,
          style: {
            float: `${o.float}`,
            width: `${o.float === "none" ? "100%" : `${o.width}%`}`,
            height: "100%",
            ...a.bodyStyle
          }
        }), /* @__PURE__ */ n(p, {
          $style: o.style,
          $horizontalAlignment: o.horizontalAlignment,
          children: /* @__PURE__ */ n("p", {
            children: o.type === "markdown" ? /* @__PURE__ */ n(W, {
              children: b.value
            }) : b.value
          })
        })]
      })
    }), g && /* @__PURE__ */ n(u.Provider, {
      value: a.style.background,
      children: /* @__PURE__ */ n(j, {
        $showBorder: t,
        ...C,
        horizontalGridCells: $,
        items: m(x),
        autoHeight: !0,
        emptyRows: 5,
        minHeight: "48px",
        containerPadding: [0, 0],
        showName: {
          top: t ? 20 : 0
        },
        $backgroundColor: (r == null ? void 0 : r.footerBackground) || "transparent",
        $footerBackgroundImage: r == null ? void 0 : r.footerBackgroundImage,
        $footerBackgroundImageRepeat: r == null ? void 0 : r.footerBackgroundImageRepeat,
        $footerBackgroundImageSize: r == null ? void 0 : r.footerBackgroundImageSize,
        $footerBackgroundImagePosition: r == null ? void 0 : r.footerBackgroundImagePosition,
        $footerBackgroundImageOrigin: r == null ? void 0 : r.footerBackgroundImageOrigin,
        $borderColor: e == null ? void 0 : e.border,
        $borderWidth: e == null ? void 0 : e.borderWidth,
        style: {
          padding: r.containerFooterPadding
        }
      })
    })]
  });
}
export {
  E as TriContainer
};
