// lowcoder-sdk/dist/7970ae5a.js
import { mJ as B, i6 as M } from "./c988e26e.js";
import D from "react";
function A(c, a) {
  for (var f = 0; f < a.length; f++) {
    const i = a[f];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const u in i)
        if (u !== "default" && !(u in c)) {
          const o = Object.getOwnPropertyDescriptor(i, u);
          o && Object.defineProperty(c, u, o.get ? o : {
            enumerable: !0,
            get: () => i[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(c, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, v = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var j = { icon: function(a, f) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M308 412v268c0 36.78 9.68 71.96 27.8 102.9a205.39 205.39 0 0073.3 73.3A202.68 202.68 0 00512 884c36.78 0 71.96-9.68 102.9-27.8a205.39 205.39 0 0073.3-73.3A202.68 202.68 0 00716 680V412H308zm484 172v96c0 6.5-.22 12.95-.66 19.35C859.94 728.64 908 796.7 908 876a8 8 0 01-8 8h-56a8 8 0 01-8-8c0-44.24-23.94-82.89-59.57-103.7a278.63 278.63 0 01-22.66 49.02 281.39 281.39 0 01-100.45 100.45C611.84 946.07 563.55 960 512 960s-99.84-13.93-141.32-38.23a281.39 281.39 0 01-100.45-100.45 278.63 278.63 0 01-22.66-49.02A119.95 119.95 0 00188 876a8 8 0 01-8 8h-56a8 8 0 01-8-8c0-79.3 48.07-147.36 116.66-176.65A284.12 284.12 0 01232 680v-96H84a8 8 0 01-8-8v-56a8 8 0 018-8h148V412c-76.77 0-139-62.23-139-139a8 8 0 018-8h60a8 8 0 018 8 63 63 0 0063 63h560a63 63 0 0063-63 8 8 0 018-8h60a8 8 0 018 8c0 76.77-62.23 139-139 139v100h148a8 8 0 018 8v56a8 8 0 01-8 8H792zM368 272a8 8 0 01-8 8h-56a8 8 0 01-8-8c0-40.04 8.78-76.75 25.9-108.07a184.57 184.57 0 0174.03-74.03C427.25 72.78 463.96 64 504 64h16c40.04 0 76.75 8.78 108.07 25.9a184.57 184.57 0 0174.03 74.03C719.22 195.25 728 231.96 728 272a8 8 0 01-8 8h-56a8 8 0 01-8-8c0-28.33-5.94-53.15-17.08-73.53a112.56 112.56 0 00-45.39-45.4C573.15 141.95 548.33 136 520 136h-16c-28.33 0-53.15 5.94-73.53 17.08a112.56 112.56 0 00-45.4 45.39C373.95 218.85 368 243.67 368 272z", fill: a } }, { tag: "path", attrs: { d: "M308 412v268c0 36.78 9.68 71.96 27.8 102.9a205.39 205.39 0 0073.3 73.3A202.68 202.68 0 00512 884c36.78 0 71.96-9.68 102.9-27.8a205.39 205.39 0 0073.3-73.3A202.68 202.68 0 00716 680V412H308z", fill: f } }] };
}, name: "bug", theme: "twotone" };
_.default = j;
(function(c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), Object.defineProperty(c, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var a = /* @__PURE__ */ y(D), f = /* @__PURE__ */ o(_), i = /* @__PURE__ */ o(B);
  function u(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function o(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function y(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, l = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = l ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function h(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(l) {
        return Object.getOwnPropertyDescriptor(r, l).enumerable;
      }))), n.forEach(function(l) {
        u(e, l, r[l]);
      });
    }
    return e;
  }
  function P(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(l) {
        return Object.getOwnPropertyDescriptor(e, l).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function m(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : P(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var T = function(e, t) {
    return /* @__PURE__ */ a.createElement(i.default, m(h({}, e), {
      ref: t,
      icon: f.default
    }));
  }, g = /* @__PURE__ */ a.forwardRef(T);
  process.env.NODE_ENV !== "production" && (g.displayName = "BugTwoTone");
  var b = g;
})(v);
(function(c, a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), a.default = void 0;
  const f = i(v);
  function i(o) {
    return o && o.__esModule ? o : { default: o };
  }
  const u = f;
  a.default = u, c.exports = u;
})(d, d.exports);
var w = d.exports;
const C = /* @__PURE__ */ M(w), x = /* @__PURE__ */ A({
  __proto__: null,
  default: C
}, [w]);
export {
  x as B
};
