// lowcoder-sdk/dist/7a651ced.js
import { Y as e, a4 as t, w as o, cM as b, bI as l, a2 as r, S as d, a8 as n, a9 as g, aa as i } from "./c988e26e.js";
import "react";
import "react-dom";
const s = [{
  label: e("badge.SizeDefault"),
  value: "default"
}, {
  label: e("badge.SizeSmall"),
  value: "small"
}], p = [{
  label: e("badge.number"),
  value: "number"
}, {
  label: e("badge.dot"),
  value: "dot"
}], C = {
  badgeType: t(p, "number"),
  badgeCount: o(b("badgeCount"), "0"),
  badgeSize: t(s, "default"),
  showZero: l,
  overflowCount: o(r, 99),
  badgeTitle: o(d, "")
}, S = (a) => /* @__PURE__ */ n(g, {
  name: i.advanced,
  children: [a.badgeType.propertyView({
    label: e("badge.Type"),
    radioButton: !0
  }), a.badgeCount.propertyView({
    label: e("badge.Count")
  }), a.badgeType.getView() === "number" && a.overflowCount.propertyView({
    label: e("badge.overflowCount")
  }), a.badgeType.getView() === "number" && a.badgeSize.propertyView({
    label: e("badge.Size")
  }), a.badgeTitle.propertyView({
    label: e("badge.Title")
  })]
});
export {
  S as BadgeBasicSection,
  C as badgeChildren
};
