// lowcoder-sdk/dist/7d1c8402.js
import { a8 as i, Y as e, F as l } from "./c988e26e.js";
import "react";
import "react-dom";
const b = /* @__PURE__ */ i("li", {
  children: [e("timeLine.Introduction"), ":", /* @__PURE__ */ l("br", {}), "1. title - ", e("timeLine.helpTitle"), /* @__PURE__ */ l("br", {}), "2. subTitle - ", e("timeLine.helpsubTitle"), /* @__PURE__ */ l("br", {}), "3. label - ", e("timeLine.helpLabel"), /* @__PURE__ */ l("br", {}), "4. color - ", e("timeLine.helpColor"), /* @__PURE__ */ l("br", {}), "5. dot - ", e("timeLine.helpDot"), /* @__PURE__ */ l("br", {}), "6. titleColor - ", e("timeLine.helpTitleColor"), /* @__PURE__ */ l("br", {}), "7. subTitleColor - ", e("timeLine.helpSubTitleColor"), /* @__PURE__ */ l("br", {}), "8. labelColor - ", e("timeLine.helpLabelColor")]
}), n = [{
  title: "Majiang Releases",
  subTitle: "Majiang Published in China",
  label: "2022-6-10"
}, {
  title: "Openblocks public release",
  subTitle: "Openblocks open source in GitHub",
  label: "2022-11-28"
}, {
  title: "Last code submission",
  subTitle: "Openblocks project abandoned",
  dot: "ExclamationCircleOutlined",
  label: "2023-3-28",
  color: "red",
  titleColor: "red",
  subTitleColor: "red",
  labelColor: "red"
}, {
  title: "Lowcoder 2.0",
  subTitle: "Lowcoder, keep moving forward",
  dot: "LogoutOutlined",
  color: "green",
  label: "2023-6-20"
}];
export {
  b as TimelineDataTooltip,
  n as timelineDate
};
