// lowcoder-sdk/dist/863db66c.js
import { mJ as M, i6 as k } from "./c988e26e.js";
import j from "react";
function S(f, u) {
  for (var c = 0; c < u.length; c++) {
    const l = u[c];
    if (typeof l != "string" && !Array.isArray(l)) {
      for (const i in l)
        if (i !== "default" && !(i in f)) {
          const a = Object.getOwnPropertyDescriptor(l, i);
          a && Object.defineProperty(f, i, a.get ? a : {
            enumerable: !0,
            get: () => l[i]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, v = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M573.7 252.5C422.5 197.4 201.3 96.7 201.3 96.7c-15.7-4.1-17.9 11.1-17.9 11.1-5 61.1 33.6 160.5 53.6 182.8 19.9 22.3 319.1 113.7 319.1 113.7S326 357.9 270.5 341.9c-55.6-16-37.9 17.8-37.9 17.8 11.4 61.7 64.9 131.8 107.2 138.4 42.2 6.6 220.1 4 220.1 4s-35.5 4.1-93.2 11.9c-42.7 5.8-97 12.5-111.1 17.8-33.1 12.5 24 62.6 24 62.6 84.7 76.8 129.7 50.5 129.7 50.5 33.3-10.7 61.4-18.5 85.2-24.2L565 743.1h84.6L603 928l205.3-271.9H700.8l22.3-38.7c.3.5.4.8.4.8S799.8 496.1 829 433.8l.6-1h-.1c5-10.8 8.6-19.7 10-25.8 17-71.3-114.5-99.4-265.8-154.5z" } }] }, name: "dingtalk", theme: "outlined" };
s.default = E;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return h;
    }
  });
  var u = /* @__PURE__ */ P(j), c = /* @__PURE__ */ a(s), l = /* @__PURE__ */ a(M);
  function i(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function P(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, o = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = o ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function D(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(o) {
        return Object.getOwnPropertyDescriptor(r, o).enumerable;
      }))), n.forEach(function(o) {
        i(e, o, r[o]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(o) {
        return Object.getOwnPropertyDescriptor(e, o).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ u.createElement(l.default, b(D({}, e), {
      ref: t,
      icon: c.default
    }));
  }, g = /* @__PURE__ */ u.forwardRef(w);
  process.env.NODE_ENV !== "production" && (g.displayName = "DingtalkOutlined");
  var h = g;
})(v);
(function(f, u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), u.default = void 0;
  const c = l(v);
  function l(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const i = c;
  u.default = i, f.exports = i;
})(O, O.exports);
var y = O.exports;
const x = /* @__PURE__ */ k(y), $ = /* @__PURE__ */ S({
  __proto__: null,
  default: x
}, [y]);
export {
  $ as D
};
