// lowcoder-sdk/dist/87e9c629.js
import { mJ as D, i6 as T } from "./c988e26e.js";
import M from "react";
function z(i, u) {
  for (var c = 0; c < u.length; c++) {
    const o = u[c];
    if (typeof o != "string" && !Array.isArray(o)) {
      for (const l in o)
        if (l !== "default" && !(l in i)) {
          const a = Object.getOwnPropertyDescriptor(o, l);
          a && Object.defineProperty(i, l, a.get ? a : {
            enumerable: !0,
            get: () => o[l]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(i, Symbol.toStringTag, { value: "Module" }));
}
var p = { exports: {} }, b = {}, O = {};
Object.defineProperty(O, "__esModule", { value: !0 });
var j = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "defs", attrs: {}, children: [{ tag: "style", attrs: {} }] }, { tag: "path", attrs: { d: "M117 368h231v64H117zm559 0h241v64H676zm-264 0h200v64H412zm0 224h200v64H412zm264 0h241v64H676zm-559 0h231v64H117zm295-160V179h-64v666h64V592zm264-64V179h-64v666h64V432z" } }] }, name: "borderless-table", theme: "outlined" };
O.default = j;
(function(i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), Object.defineProperty(i, "default", {
    enumerable: !0,
    get: function() {
      return B;
    }
  });
  var u = /* @__PURE__ */ g(M), c = /* @__PURE__ */ a(O), o = /* @__PURE__ */ a(D);
  function l(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function g(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var s in e)
      if (s !== "default" && Object.prototype.hasOwnProperty.call(e, s)) {
        var d = f ? Object.getOwnPropertyDescriptor(e, s) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, s, d) : n[s] = e[s];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(f) {
        return Object.getOwnPropertyDescriptor(r, f).enumerable;
      }))), n.forEach(function(f) {
        l(e, f, r[f]);
      });
    }
    return e;
  }
  function h(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function P(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : h(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ u.createElement(o.default, P(m({}, e), {
      ref: t,
      icon: c.default
    }));
  }, v = /* @__PURE__ */ u.forwardRef(w);
  process.env.NODE_ENV !== "production" && (v.displayName = "BorderlessTableOutlined");
  var B = v;
})(b);
(function(i, u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), u.default = void 0;
  const c = o(b);
  function o(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const l = c;
  u.default = l, i.exports = l;
})(p, p.exports);
var y = p.exports;
const E = /* @__PURE__ */ T(y), x = /* @__PURE__ */ z({
  __proto__: null,
  default: E
}, [y]);
export {
  x as B
};
