// lowcoder-sdk/dist/8cb32968.js
import { a8 as l, Y as e, F as a } from "./c988e26e.js";
import "react";
import "react-dom";
const i = /* @__PURE__ */ l("li", {
  children: ["1. label - ", e("tree.helpLabel"), /* @__PURE__ */ a("br", {}), "2. value - ", e("tree.helpValue"), /* @__PURE__ */ a("br", {}), "3. children - ", e("tree.helpChildren"), /* @__PURE__ */ a("br", {}), "4. disabled - ", e("tree.helpDisabled"), /* @__PURE__ */ a("br", {}), "5. selectable - ", e("tree.helpSelectable"), /* @__PURE__ */ a("br", {}), "6. checkable - ", e("tree.helpCheckable"), /* @__PURE__ */ a("br", {}), "7. disableCheckbox - ", e("tree.helpDisableCheckbox")]
}), h = [{
  label: e("tree.treeDataAsia"),
  value: "asia",
  children: [{
    label: e("tree.treeDataChina"),
    value: "china",
    children: [{
      label: e("tree.treeDataBeijing"),
      value: "beijing"
    }, {
      label: e("tree.treeDataShanghai"),
      value: "shanghai"
    }]
  }, {
    label: e("tree.treeDataJapan"),
    value: "japan"
  }]
}, {
  label: e("tree.treeDataEurope"),
  value: "europe",
  disabled: !0,
  children: [{
    label: e("tree.treeDataEngland"),
    value: "england"
  }, {
    label: e("tree.treeDataFrance"),
    value: "france",
    checkable: !1
  }, {
    label: e("tree.treeDataGermany"),
    value: "germany",
    disableCheckbox: !0
  }]
}, {
  label: e("tree.treeDataNorthAmerica"),
  value: "northAmerica"
}];
export {
  h as defaultTreeData,
  i as treeDataTooltip
};
