// lowcoder-sdk/dist/91f867d0.js
import { av as y, Y as t, X as O, a5 as M, O as k, w as q, aA as _, S as C, cZ as j, Z as Q, lj as J, U, F as d, a8 as b, G as R, a9 as g, aa as x, en as Z, ah as S, ab as G, dE as I, ac as K, ae as T, ad as X, _ as Y, eA as L } from "./c988e26e.js";
import ee, { useRef as n, useCallback as w, useEffect as v, useContext as V, useState as H, Suspense as te } from "react";
import { formDataChildren as oe, FormDataPropertyView as re } from "./85e61f03.js";
import "react-dom";
function ae(e, o) {
  o === void 0 && (o = 0);
  var r = n(!1), a = n(), l = n(e), h = w(function() {
    return r.current;
  }, []), c = w(function() {
    r.current = !1, a.current && clearTimeout(a.current), a.current = setTimeout(function() {
      r.current = !0, l.current();
    }, o);
  }, [o]), u = w(function() {
    r.current = null, a.current && clearTimeout(a.current);
  }, []);
  return v(function() {
    l.current = e;
  }, [e]), v(function() {
    return c(), u;
  }, [o]), [h, u, c];
}
function ie(e, o, r) {
  o === void 0 && (o = 0), r === void 0 && (r = []);
  var a = ae(e, o), l = a[0], h = a[1], c = a[2];
  return v(c, r), [l, h];
}
const D = y`
  & .ql-snow {
    .ql-picker.ql-header {
      .ql-picker-label::before,
      .ql-picker-item::before {
        content: "${t("richTextEditor.content")}";
      }

      .ql-picker-label[data-value="1"]::before,
      .ql-picker-item[data-value="1"]::before {
        content: "${t("richTextEditor.title")} 1";
      }

      .ql-picker-label[data-value="2"]::before,
      .ql-picker-item[data-value="2"]::before {
        content: "${t("richTextEditor.title")} 2";
      }

      .ql-picker-label[data-value="3"]::before,
      .ql-picker-item[data-value="3"]::before {
        content: "${t("richTextEditor.title")} 3";
      }

      .ql-picker-item[data-value="1"]::before {
        font-size: 26px;
      }

      .ql-picker-item[data-value="3"]::before {
        font-size: 19px;
      }

      .ql-picker-item[data-value="3"]::before {
        font-size: 15px;
      }
    }
    & .ql-tooltip.ql-editing a.ql-action::after {
      content: "${t("richTextEditor.save")}";
    }
    & .ql-tooltip::before {
      content: "${t("richTextEditor.link")}";
    }
    & .ql-tooltip {
      a.ql-action::after {
        content: "${t("richTextEditor.edit")}";
      }
      a.ql-remove::before {
        content: "${t("richTextEditor.remove")}";
      }
    }
  }
`, P = (e) => y`
  height: 100%;

  & .ql-editor {
    min-height: 85px;
  }
  & .ql-snow .ql-tooltip.ql-editing {
    input[type="text"] {
      width: 130px;
    }
  }
  & .ql-snow {
    &.ql-container,
    &.ql-toolbar {
      border-color: ${e.border};
      background-color: ${e.background};
      
    }
  }
  & .ql-toolbar {
    border-radius: ${e.radius} ${e.radius} 0 0;
    border-width: ${e.borderWidth ? e.borderWidth : "1px"};
  }
  & .ql-container {
    border-radius: 0 0 ${e.radius} ${e.radius};
    background-color: ${e.background};
    border-width: ${e.borderWidth ? e.borderWidth : "1px"};
  }
`, F = (e) => y`
  .ql-snow.ql-toolbar {
    height: 0;
    overflow: hidden;
    padding: 0;
    border-bottom: 0;
    border: none;
  }
  .quill .ql-snow.ql-container {
    border-radius: ${e.radius};
    border: 1px solid ${e.border};
  }
`, ne = O.div`
  ${D}
  ${(e) => P(e.$style)}
  & .ql-container .ql-editor {
    min-height: 125px;
  }
  ${(e) => e.$hideToolbar ? F(e.$style) : ""};
`, le = O.div`
  ${D}
  ${(e) => P(e.$style)}
  & .quill {
    display: flex;
    flex-direction: column;
    height: 100%;
  }
  & .ql-snow {
    &.ql-container {
      flex: 1;
      overflow: auto;
    }
  }
  ${(e) => e.$hideToolbar ? F(e.$style) : ""};
`, ce = [[{
  header: [1, 2, 3, !1]
}], ["bold", "italic", "underline", "strike", "blockquote"], [{
  list: "ordered"
}, {
  list: "bullet"
}], [{
  indent: "-1"
}, {
  indent: "+1"
}], [{
  color: []
}, {
  background: []
}, {
  align: []
}], ["link", "image"], ["clean"]], de = {
  value: M("value"),
  hideToolbar: k,
  readOnly: k,
  autoHeight: q(_, "fixed"),
  placeholder: q(C, t("richTextEditor.placeholder")),
  toolbar: q(C, JSON.stringify(ce)),
  onEvent: j,
  style: Q(J, "style"),
  ...oe
}, ue = ee.lazy(() => import("./08fa5cf1.js").then((e) => e.i));
function se(e) {
  const [o, r] = H(0), [a, l] = H(""), h = n(null), c = n(null), u = n(0), f = n(e.onChange);
  f.current = e.onChange;
  const N = n(Y.debounce((i) => {
    var s;
    window.clearTimeout(u.current), u.current = window.setTimeout(() => u.current = 0, 100), (s = f.current) == null || s.call(f, i);
  }));
  ie(() => {
    r(Math.random());
  }, 500, [e.placeholder]);
  const W = (i, s) => {
    try {
      s.parentNode;
    } catch {
      return !1;
    }
    return i.contains(s);
  }, z = (i) => {
    l(i), N.current(i);
  };
  v(() => {
    let i = e.value;
    /^<\w+>.+<\/\w+>$/.test(e.value) || (i = `<p class="">${e.value}</p>`), l(i);
  }, [e.value]);
  const A = (i) => {
    var m, p;
    (p = ((m = c.current) == null ? void 0 : m.editor).theme.pickers) == null || p.forEach((E) => {
      W(E.container, i.nativeEvent.target) || E.close();
    });
  }, $ = "rtf-editor", B = e.autoHeight ? ne : le;
  return /* @__PURE__ */ d(B, {
    id: $,
    onClick: A,
    ref: h,
    $hideToolbar: e.hideToolbar,
    $style: e.$style,
    children: /* @__PURE__ */ d(te, {
      fallback: /* @__PURE__ */ d(L, {}),
      children: /* @__PURE__ */ d(ue, {
        ref: c,
        bounds: `#${$}`,
        modules: {
          toolbar: JSON.parse(e.toolbar)
        },
        theme: "snow",
        value: a,
        placeholder: e.placeholder,
        readOnly: e.readOnly,
        onChange: z
      }, o)
    })
  });
}
const he = new U(de, (e) => {
  const o = (r) => {
    e.value.onChange(r), e.onEvent("change");
  };
  return /* @__PURE__ */ d(se, {
    autoHeight: e.autoHeight,
    hideToolbar: e.hideToolbar,
    toolbar: e.toolbar,
    readOnly: e.readOnly,
    value: e.value.value,
    placeholder: e.placeholder,
    onChange: o,
    $style: e.style
  });
}).setPropertyViewFn((e) => /* @__PURE__ */ b(R, {
  children: [/* @__PURE__ */ b(g, {
    name: x.basic,
    children: [e.value.propertyView({
      label: t("richTextEditor.defaultValue")
    }), Z(e)]
  }), /* @__PURE__ */ d(re, {
    ...e
  }), ["logic", "both"].includes(V(S).editorModeStatus) && /* @__PURE__ */ b(g, {
    name: x.interaction,
    children: [e.onEvent.getPropertyView(), G(e), I(e)]
  }), ["layout", "both"].includes(V(S).editorModeStatus) && /* @__PURE__ */ b(R, {
    children: [/* @__PURE__ */ b(g, {
      name: x.layout,
      children: [e.autoHeight.getPropertyView(), e.toolbar.propertyView({
        label: t("richTextEditor.toolbar"),
        tooltip: t("richTextEditor.toolbarDescription")
      }), e.hideToolbar.propertyView({
        label: t("richTextEditor.hideToolbar")
      })]
    }), /* @__PURE__ */ d(g, {
      name: x.style,
      children: e.style.getPropertyView()
    })]
  })]
})).build();
class be extends he {
  autoHeight() {
    return this.children.autoHeight.getView();
  }
}
const me = K(be, [new T("value", t("export.richTextEditorValueDesc")), new T("readOnly", t("export.richTextEditorReadOnlyDesc")), new T("hideToolbar", t("export.richTextEditorHideToolBarDesc")), X]);
export {
  me as RichTextEditorComp
};
