// lowcoder-sdk/dist/9248d5f9.js
import { mJ as q, i6 as D } from "./c988e26e.js";
import F from "react";
function M(c, l) {
  for (var o = 0; o < l.length; o++) {
    const i = l[o];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const u in i)
        if (u !== "default" && !(u in c)) {
          const a = Object.getOwnPropertyDescriptor(i, u);
          a && Object.defineProperty(c, u, a.get ? a : {
            enumerable: !0,
            get: () => i[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(c, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, y = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var U = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM690 624h-46.9c-10.2 0-19.9-4.9-25.9-13.2L512 465.4 406.8 610.8c-6 8.3-15.6 13.2-25.9 13.2H334c-6.5 0-10.3-7.4-6.5-12.7l178-246c3.2-4.4 9.7-4.4 12.9 0l178 246c3.9 5.3.1 12.7-6.4 12.7z" } }] }, name: "up-square", theme: "filled" };
_.default = U;
(function(c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), Object.defineProperty(c, "default", {
    enumerable: !0,
    get: function() {
      return S;
    }
  });
  var l = /* @__PURE__ */ P(F), o = /* @__PURE__ */ a(_), i = /* @__PURE__ */ a(q);
  function u(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? t : r;
    })(e);
  }
  function P(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = O(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = f ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function m(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(f) {
        return Object.getOwnPropertyDescriptor(t, f).enumerable;
      }))), n.forEach(function(f) {
        u(e, f, t[f]);
      });
    }
    return e;
  }
  function b(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function w(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : b(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var h = function(e, r) {
    return /* @__PURE__ */ l.createElement(i.default, w(m({}, e), {
      ref: r,
      icon: o.default
    }));
  }, v = /* @__PURE__ */ l.forwardRef(h);
  process.env.NODE_ENV !== "production" && (v.displayName = "UpSquareFilled");
  var S = v;
})(y);
(function(c, l) {
  Object.defineProperty(l, "__esModule", {
    value: !0
  }), l.default = void 0;
  const o = i(y);
  function i(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const u = o;
  l.default = u, c.exports = u;
})(s, s.exports);
var g = s.exports;
const j = /* @__PURE__ */ D(g), N = /* @__PURE__ */ M({
  __proto__: null,
  default: j
}, [g]);
export {
  N as U
};
