// lowcoder-sdk/dist/92a49c47.js
import { i7 as V, R as x, du as E, i8 as I, e5 as w, a8 as A, i9 as C, F as r, ia as R, ib as g, ic as v, id as H, ie as L, ig as P } from "./c988e26e.js";
import { useRef as T, useMemo as b, useEffect as u } from "react";
import "react-dom";
const D = V();
function j(p) {
  const {
    dsl: h,
    moduleDsl: m,
    appId: i,
    moduleInputs: o,
    onCompChange: n,
    onModuleEventTriggered: c
  } = p, a = T(c);
  a.current = c;
  const f = b(
    () => ({
      Comp: x,
      initialValue: h,
      reduceContext: {
        readOnly: !0,
        applicationId: i,
        parentApplicationPath: [],
        moduleDSL: m || {}
      },
      initHandler: (t) => t.preload(`app-${i}`),
      actionPreInterceptor: (t) => {
        var s;
        return t.type === E.TRIGGER_MODULE_EVENT ? ((s = a.current) == null || s.call(a, t.name), !1) : !0;
      }
    }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [i]
  ), [e] = I(f);
  return u(() => {
    n == null || n(e);
  }, [e]), u(() => {
    const t = e == null ? void 0 : e.children.ui.getModuleLayoutComp();
    if (!t || !o)
      return;
    t.children.io.children.inputs.getView().forEach((d) => {
      const {
        name: y
      } = d.getView(), l = o[y];
      l !== void 0 && d.children.defaultValue.children.comp.dispatch(w(d.children.defaultValue.children.comp.changeValueAction(l)));
    });
  }, [o]), /* @__PURE__ */ A(C, {
    children: [/* @__PURE__ */ r(R, {}), /* @__PURE__ */ r(g, {
      store: v,
      children: /* @__PURE__ */ r(H.Provider, {
        value: {
          applicationId: i,
          appType: 1,
          readOnly: !0,
          hideHeader: !0
        },
        children: /* @__PURE__ */ r(L, {
          history: D,
          children: /* @__PURE__ */ r(P, {
            path: "/",
            render: () => e == null ? void 0 : e.getView()
          })
        })
      })
    })]
  });
}
export {
  j as AppView
};
