// lowcoder-sdk/dist/98214bb6.js
import { mJ as S, i6 as q } from "./c988e26e.js";
import M from "react";
function D(c, a) {
  for (var f = 0; f < a.length; f++) {
    const i = a[f];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const u in i)
        if (u !== "default" && !(u in c)) {
          const o = Object.getOwnPropertyDescriptor(i, u);
          o && Object.defineProperty(c, u, o.get ? o : {
            enumerable: !0,
            get: () => i[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(c, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, g = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var U = { icon: function(a, f) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z", fill: a } }, { tag: "path", attrs: { d: "M184 840h656V184H184v656zm143.5-228.7l178-246c3.2-4.4 9.7-4.4 12.9 0l178 246c3.9 5.3.1 12.7-6.4 12.7h-46.9c-10.2 0-19.9-4.9-25.9-13.2L512 465.4 406.8 610.8c-6 8.3-15.6 13.2-25.9 13.2H334c-6.5 0-10.3-7.4-6.5-12.7z", fill: f } }, { tag: "path", attrs: { d: "M334 624h46.9c10.3 0 19.9-4.9 25.9-13.2L512 465.4l105.2 145.4c6 8.3 15.7 13.2 25.9 13.2H690c6.5 0 10.3-7.4 6.4-12.7l-178-246a7.95 7.95 0 00-12.9 0l-178 246c-3.8 5.3 0 12.7 6.5 12.7z", fill: a } }] };
}, name: "up-square", theme: "twotone" };
_.default = U;
(function(c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), Object.defineProperty(c, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var a = /* @__PURE__ */ y(M), f = /* @__PURE__ */ o(_), i = /* @__PURE__ */ o(S);
  function u(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function o(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function y(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, p = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var l in e)
      if (l !== "default" && Object.prototype.hasOwnProperty.call(e, l)) {
        var s = p ? Object.getOwnPropertyDescriptor(e, l) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, l, s) : n[l] = e[l];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(p) {
        return Object.getOwnPropertyDescriptor(r, p).enumerable;
      }))), n.forEach(function(p) {
        u(e, p, r[p]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(p) {
        return Object.getOwnPropertyDescriptor(e, p).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function T(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var h = function(e, t) {
    return /* @__PURE__ */ a.createElement(i.default, T(P({}, e), {
      ref: t,
      icon: f.default
    }));
  }, v = /* @__PURE__ */ a.forwardRef(h);
  process.env.NODE_ENV !== "production" && (v.displayName = "UpSquareTwoTone");
  var b = v;
})(g);
(function(c, a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), a.default = void 0;
  const f = i(g);
  function i(o) {
    return o && o.__esModule ? o : { default: o };
  }
  const u = f;
  a.default = u, c.exports = u;
})(d, d.exports);
var w = d.exports;
const j = /* @__PURE__ */ q(w), x = /* @__PURE__ */ D({
  __proto__: null,
  default: j
}, [w]);
export {
  x as U
};
