// lowcoder-sdk/dist/a6ed71f1.js
import { generateInsertSql as s } from "./bf2ffa97.js";
import { timeCompSelection as c, timestampCompSelection as o, dateCompSelection as i, numberCompSelection as r, booleanCompSelection as n, allCompSelection as m } from "./90e6d3f7.js";
function l(e) {
  if (e) {
    switch (e.toLowerCase()) {
      case "bool":
      case "boolean":
        return n();
      case "smallint":
      case "int2":
      case "integer":
      case "int":
      case "int4":
      case "bigint":
      case "int8":
      case "smallserial":
      case "serial2":
      case "serial":
      case "serial4":
      case "bigserial":
      case "serial8":
      case "real":
      case "float4":
      case "double precision":
      case "float8":
      case "numeric":
      case "decimal":
        return r(!0);
      case "date":
        return i();
      case "timestamp":
      case "timestamptz":
        return o();
      case "time":
      case "timetz":
        return c();
    }
    return m();
  }
}
function p(e, t, a) {
  return {
    compType: "postgres",
    comp: {
      sql: s(t, a),
      commandType: "INSERT",
      mode: "GUI",
      command: {
        table: t,
        changeSet: {
          compType: "OBJECT",
          comp: `{{ ${e}.data }}`
        }
      }
    }
  };
}
const S = {
  type: "postgres",
  getCompSelection: l,
  getQueryInitData: p
};
export {
  S as postgreSqlConfig
};
