// lowcoder-sdk/dist/a759a2f6.js
import { X as i, f3 as b, cW as I, iS as r, iT as k, iU as m, F as e, eA as C, iV as u, iW as L, Y as c, iX as j, iY as T, iZ as v, i_ as P, iI as S, i$ as E, a8 as d, G as U, j0 as _, j1 as H, j2 as g, j3 as B, j4 as D, j5 as R, j6 as V, j7 as F, j8 as $, gg as l, j9 as M } from "./c988e26e.js";
import { useState as W } from "react";
import "react-dom";
const z = i.div`
  font-weight: 500;
  font-size: 14px;
  color: ${(s) => b(s.$bgColor) ? "#ffffff" : "#000000"};
  font-style: normal;
  line-height: 24px;
  margin-right: 8px;
  margin-left: 20px;
  max-width: 264px;
  ${I};
`, N = i.a`
  display: flex;
  align-items: center;
`, O = i($)`
  height: 28px;
  width: 28px;
`, G = i(l)`
  min-width: 60px;
  padding: 0;
  height: 28px;
  margin-right: 4px;
`, X = i(l)`
  min-width: 60px;
  height: 28px;
`, Y = i(l)`
  &&& {
    color: #ffffff;
    background: #8b8fa34c;
    border: none;
    height: 28px;
    margin-right: 8px;
    min-width: 60px;
    padding: 0;
    cursor: pointer;
  
    &:hover {
      background: #666666;
      color: #ffffff;
      border: none;
    }
  
    &:focus {
      background: #666666;
      color: #ffffff;
      border: none;
    }
  }
`, Z = i(M)`
  margin-right: 4px;

  g g {
    fill: #ffffff;
  }
`, q = i(l)`
  min-width: 60px;
  height: 28px;
  margin-right: 24px;
  padding: 0;
`, J = i.div`
  display: inherit;
  @media screen and (max-width: 500px) {
    > div:nth-of-type(1),
    > button {
      display: none;
    }
  }
`;
function K(s) {
  const {
    user: n
  } = s, t = r(k), a = r(m);
  return t ? /* @__PURE__ */ e(C.Avatar, {
    shape: "circle",
    size: 28
  }) : /* @__PURE__ */ e("div", {
    children: n.isAnonymous ? a ? null : /* @__PURE__ */ e(G, {
      buttonType: "primary",
      onClick: () => u.push(L),
      children: c("userAuth.login")
    }) : /* @__PURE__ */ e(j, {
      user: n,
      profileSide: 28,
      fontSize: 12
    })
  });
}
const ne = () => {
  const s = T(), n = r(v), t = r(P), a = S(), h = r(m), o = r(E), [x, f] = W(!1), p = s.viewMode === "view_marketplace", w = /* @__PURE__ */ d(U, {
    children: [/* @__PURE__ */ e(N, {
      onClick: () => u.push(_),
      children: /* @__PURE__ */ e(O, {
        branding: !0
      })
    }), p && /* @__PURE__ */ e(H, {
      setEdit: () => {
      },
      isViewMarketplaceMode: p
    }), !p && /* @__PURE__ */ e(z, {
      $bgColor: (o == null ? void 0 : o.headerColor) ?? "#2c2c2c",
      children: t && t.name
    })]
  }), y = /* @__PURE__ */ d(J, {
    children: [g(n, t) && /* @__PURE__ */ e(B, {
      applicationId: a,
      visible: x,
      onVisibleChange: (A) => !A && f(!1)
    }), g(n, t) && /* @__PURE__ */ e(Y, {
      onClick: () => f(!0),
      children: D
    }), R(n, t) && /* @__PURE__ */ d(q, {
      buttonType: "primary",
      onClick: () => (
        // redirection to app by JS will cause the problem that queries don't execute on initialization
        // so just open a new window.
        window.open(V(a, "edit"))
      ),
      children: [/* @__PURE__ */ e(Z, {}), c("edit")]
    }), !!h && /* @__PURE__ */ e(X, {
      style: {
        marginRight: n.isAnonymous ? "" : "24px"
      },
      buttonType: "primary",
      onClick: () => {
        window.open(c("template.cloneUrl") + h);
      },
      children: c("header.clone")
    }), /* @__PURE__ */ e(K, {
      user: n
    })]
  });
  return /* @__PURE__ */ e(F, {
    headerStart: w,
    headerEnd: y,
    style: {
      backgroundColor: o == null ? void 0 : o.headerColor
    }
  });
};
export {
  K as HeaderProfile,
  ne as PreviewHeader
};
