// lowcoder-sdk/dist/a79eca63.js
import { X as m, go as h, ah as z, F as r, lc as R, a8 as g, ld as B, le as H, G as C, lf as P, iZ as E, lg as I, lh as $ } from "./c988e26e.js";
import { useContext as L, useRef as M, useState as c, useMemo as F } from "react";
import { u as O, H as U } from "./1443d5c6.js";
import "react-dom";
const W = m.div`
  right: calc(313px + 4px); // FIXME: don't rely on the width of the right panel
  bottom: ${(t) => t.$bottom ? t.$bottom + 4 + "px" : 285 + 4 + "px"};
  position: fixed;
  z-index: ${h.queryResultPanel};

  display: flex;
  flex-direction: column;
  width: 592px;
  height: fit-content;
  max-height: 250px;
  background: #ffffff;
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.1);
  border-radius: 8px;
  pointer-events: auto;
  padding-bottom: 16px;
`, j = (t) => {
  const {
    bottom: e
  } = t, a = L(z), o = a.showResultComp(), i = o == null ? void 0 : o.result(), l = M(null), [u, n] = c(!0), [f, x] = c({
    left: 0,
    top: 0,
    bottom: 0,
    right: 0
  }), {
    header: w,
    body: y
  } = O({
    ...i ?? {
      data: "",
      dataType: "default",
      success: !0
    },
    onClose: () => a.setShowResultCompName(void 0)
  });
  return i ? /* @__PURE__ */ r(R, {
    disabled: u,
    bounds: f,
    enableUserSelectHack: !1,
    onStart: (G, d) => {
      var p;
      const {
        clientWidth: v,
        clientHeight: S
      } = window.document.documentElement, s = (p = l.current) == null ? void 0 : p.getBoundingClientRect();
      s && x({
        left: -s.left + d.x,
        right: v - (s.right - d.x),
        top: -s.top + d.y,
        bottom: S - (s.bottom - d.y)
      });
    },
    children: /* @__PURE__ */ g(W, {
      $bottom: e,
      ref: l,
      children: [/* @__PURE__ */ r(U, {
        onMouseOver: () => n(!1),
        onMouseOut: () => n(!0),
        children: w
      }), y]
    })
  }) : null;
}, k = m($)`
  position: relative;
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.1);
  border-top: 1px solid #e1e3eb;
  z-index: ${h.bottomPanel};

  .react-resizable-handle {
    position: absolute;
    border-top: transparent solid 3px;
    width: 100%;
    padding: 0 3px 3px 0;
    top: 0;
    cursor: row-resize;
  }
`, b = (t) => {
  t.preventDefault();
}, D = () => {
  window.addEventListener("mousedown", b);
}, T = () => {
  window.removeEventListener("mousedown", b);
};
function X(t) {
  const e = F(() => H(), []), a = document.documentElement.clientHeight, o = (u, n) => {
    I({
      ...e,
      bottom: {
        h: n.size.height
      }
    }), l(n.size.height), T();
  }, [i, l] = c(e.bottom.h);
  return /* @__PURE__ */ g(C, {
    children: [/* @__PURE__ */ r(j, {
      bottom: i
    }), /* @__PURE__ */ r(k, {
      width: 1 / 0,
      height: e.bottom.h,
      resizeHandles: ["n"],
      minConstraints: [680, 285],
      maxConstraints: [1 / 0, a - 48 - 40],
      onResizeStart: D,
      onResizeStop: o,
      children: /* @__PURE__ */ r(P, {})
    })]
  });
}
const q = (t) => ({
  orgId: E(t).currentOrgId,
  datasourceInfos: t.entities.datasource.data
}), K = B(q, null)(X);
export {
  K as default
};
