// lowcoder-sdk/dist/afb3f8fb.js
import { mJ as D, i6 as M } from "./c988e26e.js";
import j from "react";
function E(o, a) {
  for (var l = 0; l < a.length; l++) {
    const f = a[l];
    if (typeof f != "string" && !Array.isArray(f)) {
      for (const u in f)
        if (u !== "default" && !(u in o)) {
          const i = Object.getOwnPropertyDescriptor(f, u);
          i && Object.defineProperty(o, u, i.get ? i : {
            enumerable: !0,
            get: () => f[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(o, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, v = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var S = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z" } }] }, name: "right", theme: "outlined" };
s.default = S;
(function(o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), Object.defineProperty(o, "default", {
    enumerable: !0,
    get: function() {
      return R;
    }
  });
  var a = /* @__PURE__ */ h(j), l = /* @__PURE__ */ i(s), f = /* @__PURE__ */ i(D);
  function u(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function i(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function h(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, c = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = c ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(c) {
        return Object.getOwnPropertyDescriptor(r, c).enumerable;
      }))), n.forEach(function(c) {
        u(e, c, r[c]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(c) {
        return Object.getOwnPropertyDescriptor(e, c).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ a.createElement(f.default, b(P({}, e), {
      ref: t,
      icon: l.default
    }));
  }, g = /* @__PURE__ */ a.forwardRef(w);
  process.env.NODE_ENV !== "production" && (g.displayName = "RightOutlined");
  var R = g;
})(v);
(function(o, a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), a.default = void 0;
  const l = f(v);
  function f(i) {
    return i && i.__esModule ? i : { default: i };
  }
  const u = l;
  a.default = u, o.exports = u;
})(O, O.exports);
var y = O.exports;
const x = /* @__PURE__ */ M(y), $ = /* @__PURE__ */ E({
  __proto__: null,
  default: x
}, [y]);
export {
  $ as R
};
