// lowcoder-sdk/dist/b416b998.js
import { dd as m, w as n, cs as e } from "./c988e26e.js";
import { TableComp as s } from "./e830a3ab.js";
import { n as c, c as u } from "./f915fcd2.js";
import "react";
import "react-dom";
import "./911053df.js";
import "./6a2c46a1.js";
import "./b0ddbdcf.js";
import "./22fa10de.js";
import "./8c51b61b.js";
import "./5230605f.js";
import "./be799735.js";
import "./0417d842.js";
import "./975a080d.js";
import "./82a4bca8.js";
import "./71bdd49a.js";
import "./2088ebd5.js";
import "./d3b0afba.js";
import "./85e61f03.js";
import "./36c2248f.js";
import "./70f9cad3.js";
import "./80c67fbe.js";
import "./094988fd.js";
import "./734fbd0f.js";
import "./80229ccd.js";
import "./16b77c22.js";
import "./b25be8d9.js";
import "./0c3664e6.js";
import "./c504e788.js";
import "./ad2208e2.js";
import "./e5bcb9b0.js";
import "./51dea655.js";
import "./fcc9e192.js";
import "./325a473d.js";
import "./33706f61.js";
import "./765e17fb.js";
import "./98885dce.js";
import "./9ee13d0e.js";
import "./9f467dd0.js";
import "./ada1f35c.js";
import "./464239f7.js";
import "./68ba86a3.js";
import "./608865ad.js";
import "./765a8ca5.js";
import "./3e9c3444.js";
const i = {
  toolbar: {
    showDownload: !0,
    showFilter: !0,
    showRefresh: !0
  }
}, p = {
  ...i,
  data: JSON.stringify(m.table.defaultData, null, " "),
  columns: m.table.columns.map((o) => c(o.key, u(o.key, m.table.defaultData), o.title, o.isTag))
}, pt = n(s, p);
function at(o, b, r) {
  const t = r == null ? void 0 : r.selectedQueryComp(), a = t == null ? void 0 : t.children.data.getView(), l = t == null ? void 0 : t.children.name.getView();
  return e.isArrayLikeObject(a) ? {
    ...i,
    data: `{{ ${l}.data }}`
  } : p;
}
export {
  pt as MockTableComp,
  at as defaultTableData
};
