// lowcoder-sdk/dist/b65f18d9.js
import { mJ as D, i6 as M } from "./c988e26e.js";
import j from "react";
function E(i, u) {
  for (var l = 0; l < u.length; l++) {
    const o = u[l];
    if (typeof o != "string" && !Array.isArray(o)) {
      for (const a in o)
        if (a !== "default" && !(a in i)) {
          const c = Object.getOwnPropertyDescriptor(o, a);
          c && Object.defineProperty(i, a, c.get ? c : {
            enumerable: !0,
            get: () => o[a]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(i, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, y = {}, O = {};
Object.defineProperty(O, "__esModule", { value: !0 });
var H = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M594.3 601.5a111.8 111.8 0 0029.1-75.5c0-61.9-49.9-112-111.4-112s-111.4 50.1-111.4 112c0 29.1 11 55.5 29.1 75.5a158.09 158.09 0 00-74.6 126.1 8 8 0 008 8.4H407c4.2 0 7.6-3.3 7.9-7.5 3.8-50.6 46-90.5 97.2-90.5s93.4 40 97.2 90.5c.3 4.2 3.7 7.5 7.9 7.5H661a8 8 0 008-8.4c-2.8-53.3-32-99.7-74.7-126.1zM512 578c-28.5 0-51.7-23.3-51.7-52s23.2-52 51.7-52 51.7 23.3 51.7 52-23.2 52-51.7 52zm416-354H768v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H548v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H328v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H96c-17.7 0-32 14.3-32 32v576c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V256c0-17.7-14.3-32-32-32zm-40 568H136V296h120v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h148v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h148v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h120v496z" } }] }, name: "contacts", theme: "outlined" };
O.default = H;
(function(i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), Object.defineProperty(i, "default", {
    enumerable: !0,
    get: function() {
      return C;
    }
  });
  var u = /* @__PURE__ */ h(j), l = /* @__PURE__ */ c(O), o = /* @__PURE__ */ c(D);
  function a(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function c(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function v(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (v = function(n) {
      return n ? r : t;
    })(e);
  }
  function h(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = v(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = f ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(f) {
        return Object.getOwnPropertyDescriptor(r, f).enumerable;
      }))), n.forEach(function(f) {
        a(e, f, r[f]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ u.createElement(o.default, b(P({}, e), {
      ref: t,
      icon: l.default
    }));
  }, _ = /* @__PURE__ */ u.forwardRef(w);
  process.env.NODE_ENV !== "production" && (_.displayName = "ContactsOutlined");
  var C = _;
})(y);
(function(i, u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), u.default = void 0;
  const l = o(y);
  function o(c) {
    return c && c.__esModule ? c : { default: c };
  }
  const a = l;
  u.default = a, i.exports = a;
})(d, d.exports);
var g = d.exports;
const S = /* @__PURE__ */ M(g), N = /* @__PURE__ */ E({
  __proto__: null,
  default: S
}, [g]);
export {
  N as C
};
