// lowcoder-sdk/dist/b6dbdda8.js
import { mJ as S, i6 as D } from "./c988e26e.js";
import M from "react";
function j(c, o) {
  for (var p = 0; p < o.length; p++) {
    const f = o[p];
    if (typeof f != "string" && !Array.isArray(f)) {
      for (const i in f)
        if (i !== "default" && !(i in c)) {
          const a = Object.getOwnPropertyDescriptor(f, i);
          a && Object.defineProperty(c, i, a.get ? a : {
            enumerable: !0,
            get: () => f[i]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(c, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, g = {}, v = {};
Object.defineProperty(v, "__esModule", { value: !0 });
var E = { icon: function(o, p) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M696 472c0 4.4-3.6 8-8 8h-56c-4.4 0-8-3.6-8-8v-88H400v88c0 4.4-3.6 8-8 8h-56c-4.4 0-8-3.6-8-8v-88h-96v456h560V384h-96v88z", fill: p } }, { tag: "path", attrs: { d: "M832 312H696v-16c0-101.6-82.4-184-184-184s-184 82.4-184 184v16H192c-17.7 0-32 14.3-32 32v536c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V344c0-17.7-14.3-32-32-32zm-432-16c0-61.9 50.1-112 112-112s112 50.1 112 112v16H400v-16zm392 544H232V384h96v88c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-88h224v88c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-88h96v456z", fill: o } }] };
}, name: "shopping", theme: "twotone" };
v.default = E;
(function(c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), Object.defineProperty(c, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var o = /* @__PURE__ */ w(M), p = /* @__PURE__ */ a(v), f = /* @__PURE__ */ a(S);
  function i(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function w(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, u = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var l in e)
      if (l !== "default" && Object.prototype.hasOwnProperty.call(e, l)) {
        var s = u ? Object.getOwnPropertyDescriptor(e, l) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, l, s) : n[l] = e[l];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function y(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(u) {
        return Object.getOwnPropertyDescriptor(r, u).enumerable;
      }))), n.forEach(function(u) {
        i(e, u, r[u]);
      });
    }
    return e;
  }
  function P(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(u) {
        return Object.getOwnPropertyDescriptor(e, u).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function m(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : P(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var T = function(e, t) {
    return /* @__PURE__ */ o.createElement(f.default, m(y({}, e), {
      ref: t,
      icon: p.default
    }));
  }, O = /* @__PURE__ */ o.forwardRef(T);
  process.env.NODE_ENV !== "production" && (O.displayName = "ShoppingTwoTone");
  var b = O;
})(g);
(function(c, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const p = f(g);
  function f(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const i = p;
  o.default = i, c.exports = i;
})(d, d.exports);
var h = d.exports;
const x = /* @__PURE__ */ D(h), N = /* @__PURE__ */ j({
  __proto__: null,
  default: x
}, [h]);
export {
  N as S
};
