// lowcoder-sdk/dist/bdfb6952.js
import { m as yt, u as $t, b as vt, w as wt, a as kt } from "./1fb07b21.js";
import { useCallback as nt, forwardRef as Mt, useMemo as st, useState as St, useEffect as Dt } from "react";
import { u as xt, b as Wt, P as Yt, r as It } from "./2242e3df.js";
import { lx as tt, i6 as et, F as rt } from "./c988e26e.js";
import "react-dom";
import "./951b5247.js";
import "./14897c7e.js";
import "./9caec5b5.js";
var ut = { exports: {} };
(function(n, f) {
  (function(t, l) {
    n.exports = l();
  })(tt, function() {
    var t = 1e3, l = 6e4, p = 36e5, $ = "millisecond", d = "second", h = "minute", I = "hour", v = "day", W = "week", D = "month", w = "quarter", S = "year", Y = "date", P = "Invalid Date", Z = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, B = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, q = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(i) {
      var r = ["th", "st", "nd", "rd"], e = i % 100;
      return "[" + i + (r[(e - 20) % 10] || r[e] || r[0]) + "]";
    } }, E = function(i, r, e) {
      var s = String(i);
      return !s || s.length >= r ? i : "" + Array(r + 1 - s.length).join(e) + i;
    }, F = { s: E, z: function(i) {
      var r = -i.utcOffset(), e = Math.abs(r), s = Math.floor(e / 60), o = e % 60;
      return (r <= 0 ? "+" : "-") + E(s, 2, "0") + ":" + E(o, 2, "0");
    }, m: function i(r, e) {
      if (r.date() < e.date())
        return -i(e, r);
      var s = 12 * (e.year() - r.year()) + (e.month() - r.month()), o = r.clone().add(s, D), u = e - o < 0, c = r.clone().add(s + (u ? -1 : 1), D);
      return +(-(s + (e - o) / (u ? o - c : c - o)) || 0);
    }, a: function(i) {
      return i < 0 ? Math.ceil(i) || 0 : Math.floor(i);
    }, p: function(i) {
      return { M: D, y: S, w: W, d: v, D: Y, h: I, m: h, s: d, ms: $, Q: w }[i] || String(i || "").toLowerCase().replace(/s$/, "");
    }, u: function(i) {
      return i === void 0;
    } }, N = "en", T = {};
    T[N] = q;
    var j = "$isDayjsObject", H = function(i) {
      return i instanceof L || !(!i || !i[j]);
    }, _ = function i(r, e, s) {
      var o;
      if (!r)
        return N;
      if (typeof r == "string") {
        var u = r.toLowerCase();
        T[u] && (o = u), e && (T[u] = e, o = u);
        var c = r.split("-");
        if (!o && c.length > 1)
          return i(c[0]);
      } else {
        var y = r.name;
        T[y] = r, o = y;
      }
      return !s && o && (N = o), o || !s && N;
    }, M = function(i, r) {
      if (H(i))
        return i.clone();
      var e = typeof r == "object" ? r : {};
      return e.date = i, e.args = arguments, new L(e);
    }, a = F;
    a.l = _, a.i = H, a.w = function(i, r) {
      return M(i, { locale: r.$L, utc: r.$u, x: r.$x, $offset: r.$offset });
    };
    var L = function() {
      function i(e) {
        this.$L = _(e.locale, null, !0), this.parse(e), this.$x = this.$x || e.x || {}, this[j] = !0;
      }
      var r = i.prototype;
      return r.parse = function(e) {
        this.$d = function(s) {
          var o = s.date, u = s.utc;
          if (o === null)
            return /* @__PURE__ */ new Date(NaN);
          if (a.u(o))
            return /* @__PURE__ */ new Date();
          if (o instanceof Date)
            return new Date(o);
          if (typeof o == "string" && !/Z$/i.test(o)) {
            var c = o.match(Z);
            if (c) {
              var y = c[2] - 1 || 0, g = (c[7] || "0").substring(0, 3);
              return u ? new Date(Date.UTC(c[1], y, c[3] || 1, c[4] || 0, c[5] || 0, c[6] || 0, g)) : new Date(c[1], y, c[3] || 1, c[4] || 0, c[5] || 0, c[6] || 0, g);
            }
          }
          return new Date(o);
        }(e), this.init();
      }, r.init = function() {
        var e = this.$d;
        this.$y = e.getFullYear(), this.$M = e.getMonth(), this.$D = e.getDate(), this.$W = e.getDay(), this.$H = e.getHours(), this.$m = e.getMinutes(), this.$s = e.getSeconds(), this.$ms = e.getMilliseconds();
      }, r.$utils = function() {
        return a;
      }, r.isValid = function() {
        return this.$d.toString() !== P;
      }, r.isSame = function(e, s) {
        var o = M(e);
        return this.startOf(s) <= o && o <= this.endOf(s);
      }, r.isAfter = function(e, s) {
        return M(e) < this.startOf(s);
      }, r.isBefore = function(e, s) {
        return this.endOf(s) < M(e);
      }, r.$g = function(e, s, o) {
        return a.u(e) ? this[s] : this.set(o, e);
      }, r.unix = function() {
        return Math.floor(this.valueOf() / 1e3);
      }, r.valueOf = function() {
        return this.$d.getTime();
      }, r.startOf = function(e, s) {
        var o = this, u = !!a.u(s) || s, c = a.p(e), y = function(V, C) {
          var z = a.w(o.$u ? Date.UTC(o.$y, C, V) : new Date(o.$y, C, V), o);
          return u ? z : z.endOf(v);
        }, g = function(V, C) {
          return a.w(o.toDate()[V].apply(o.toDate("s"), (u ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(C)), o);
        }, x = this.$W, k = this.$M, m = this.$D, U = "set" + (this.$u ? "UTC" : "");
        switch (c) {
          case S:
            return u ? y(1, 0) : y(31, 11);
          case D:
            return u ? y(1, k) : y(0, k + 1);
          case W:
            var A = this.$locale().weekStart || 0, O = (x < A ? x + 7 : x) - A;
            return y(u ? m - O : m + (6 - O), k);
          case v:
          case Y:
            return g(U + "Hours", 0);
          case I:
            return g(U + "Minutes", 1);
          case h:
            return g(U + "Seconds", 2);
          case d:
            return g(U + "Milliseconds", 3);
          default:
            return this.clone();
        }
      }, r.endOf = function(e) {
        return this.startOf(e, !1);
      }, r.$set = function(e, s) {
        var o, u = a.p(e), c = "set" + (this.$u ? "UTC" : ""), y = (o = {}, o[v] = c + "Date", o[Y] = c + "Date", o[D] = c + "Month", o[S] = c + "FullYear", o[I] = c + "Hours", o[h] = c + "Minutes", o[d] = c + "Seconds", o[$] = c + "Milliseconds", o)[u], g = u === v ? this.$D + (s - this.$W) : s;
        if (u === D || u === S) {
          var x = this.clone().set(Y, 1);
          x.$d[y](g), x.init(), this.$d = x.set(Y, Math.min(this.$D, x.daysInMonth())).$d;
        } else
          y && this.$d[y](g);
        return this.init(), this;
      }, r.set = function(e, s) {
        return this.clone().$set(e, s);
      }, r.get = function(e) {
        return this[a.p(e)]();
      }, r.add = function(e, s) {
        var o, u = this;
        e = Number(e);
        var c = a.p(s), y = function(k) {
          var m = M(u);
          return a.w(m.date(m.date() + Math.round(k * e)), u);
        };
        if (c === D)
          return this.set(D, this.$M + e);
        if (c === S)
          return this.set(S, this.$y + e);
        if (c === v)
          return y(1);
        if (c === W)
          return y(7);
        var g = (o = {}, o[h] = l, o[I] = p, o[d] = t, o)[c] || 1, x = this.$d.getTime() + e * g;
        return a.w(x, this);
      }, r.subtract = function(e, s) {
        return this.add(-1 * e, s);
      }, r.format = function(e) {
        var s = this, o = this.$locale();
        if (!this.isValid())
          return o.invalidDate || P;
        var u = e || "YYYY-MM-DDTHH:mm:ssZ", c = a.z(this), y = this.$H, g = this.$m, x = this.$M, k = o.weekdays, m = o.months, U = o.meridiem, A = function(C, z, G, X) {
          return C && (C[z] || C(s, u)) || G[z].slice(0, X);
        }, O = function(C) {
          return a.s(y % 12 || 12, C, "0");
        }, V = U || function(C, z, G) {
          var X = C < 12 ? "AM" : "PM";
          return G ? X.toLowerCase() : X;
        };
        return u.replace(B, function(C, z) {
          return z || function(G) {
            switch (G) {
              case "YY":
                return String(s.$y).slice(-2);
              case "YYYY":
                return a.s(s.$y, 4, "0");
              case "M":
                return x + 1;
              case "MM":
                return a.s(x + 1, 2, "0");
              case "MMM":
                return A(o.monthsShort, x, m, 3);
              case "MMMM":
                return A(m, x);
              case "D":
                return s.$D;
              case "DD":
                return a.s(s.$D, 2, "0");
              case "d":
                return String(s.$W);
              case "dd":
                return A(o.weekdaysMin, s.$W, k, 2);
              case "ddd":
                return A(o.weekdaysShort, s.$W, k, 3);
              case "dddd":
                return k[s.$W];
              case "H":
                return String(y);
              case "HH":
                return a.s(y, 2, "0");
              case "h":
                return O(1);
              case "hh":
                return O(2);
              case "a":
                return V(y, g, !0);
              case "A":
                return V(y, g, !1);
              case "m":
                return String(g);
              case "mm":
                return a.s(g, 2, "0");
              case "s":
                return String(s.$s);
              case "ss":
                return a.s(s.$s, 2, "0");
              case "SSS":
                return a.s(s.$ms, 3, "0");
              case "Z":
                return c;
            }
            return null;
          }(C) || c.replace(":", "");
        });
      }, r.utcOffset = function() {
        return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
      }, r.diff = function(e, s, o) {
        var u, c = this, y = a.p(s), g = M(e), x = (g.utcOffset() - this.utcOffset()) * l, k = this - g, m = function() {
          return a.m(c, g);
        };
        switch (y) {
          case S:
            u = m() / 12;
            break;
          case D:
            u = m();
            break;
          case w:
            u = m() / 3;
            break;
          case W:
            u = (k - x) / 6048e5;
            break;
          case v:
            u = (k - x) / 864e5;
            break;
          case I:
            u = k / p;
            break;
          case h:
            u = k / l;
            break;
          case d:
            u = k / t;
            break;
          default:
            u = k;
        }
        return o ? u : a.a(u);
      }, r.daysInMonth = function() {
        return this.endOf(D).$D;
      }, r.$locale = function() {
        return T[this.$L];
      }, r.locale = function(e, s) {
        if (!e)
          return this.$L;
        var o = this.clone(), u = _(e, s, !0);
        return u && (o.$L = u), o;
      }, r.clone = function() {
        return a.w(this.$d, this);
      }, r.toDate = function() {
        return new Date(this.valueOf());
      }, r.toJSON = function() {
        return this.isValid() ? this.toISOString() : null;
      }, r.toISOString = function() {
        return this.$d.toISOString();
      }, r.toString = function() {
        return this.$d.toUTCString();
      }, i;
    }(), R = L.prototype;
    return M.prototype = R, [["$ms", $], ["$s", d], ["$m", h], ["$H", I], ["$W", v], ["$M", D], ["$y", S], ["$D", Y]].forEach(function(i) {
      R[i[1]] = function(r) {
        return this.$g(r, i[0], i[1]);
      };
    }), M.extend = function(i, r) {
      return i.$i || (i(r, L, M), i.$i = !0), M;
    }, M.locale = _, M.isDayjs = H, M.unix = function(i) {
      return M(1e3 * i);
    }, M.en = T[N], M.Ls = T, M.p = {}, M;
  });
})(ut);
var Ct = ut.exports;
const b = /* @__PURE__ */ et(Ct);
var ct = { exports: {} };
(function(n, f) {
  (function(t, l) {
    n.exports = l();
  })(tt, function() {
    var t = "day";
    return function(l, p, $) {
      var d = function(v) {
        return v.add(4 - v.isoWeekday(), t);
      }, h = p.prototype;
      h.isoWeekYear = function() {
        return d(this).year();
      }, h.isoWeek = function(v) {
        if (!this.$utils().u(v))
          return this.add(7 * (v - this.isoWeek()), t);
        var W, D, w, S, Y = d(this), P = (W = this.isoWeekYear(), D = this.$u, w = (D ? $.utc : $)().year(W).startOf("year"), S = 4 - w.isoWeekday(), w.isoWeekday() > 4 && (S += 7), w.add(S, t));
        return Y.diff(P, "week") + 1;
      }, h.isoWeekday = function(v) {
        return this.$utils().u(v) ? this.day() || 7 : this.day(this.day() % 7 ? v : v - 7);
      };
      var I = h.startOf;
      h.startOf = function(v, W) {
        var D = this.$utils(), w = !!D.u(W) || W;
        return D.p(v) === "isoweek" ? w ? this.date(this.date() - (this.isoWeekday() - 1)).startOf("day") : this.date(this.date() - 1 - (this.isoWeekday() - 1) + 7).endOf("day") : I.bind(this)(v, W);
      };
    };
  });
})(ct);
var _t = ct.exports;
const lt = /* @__PURE__ */ et(_t);
var ft = { exports: {} };
(function(n, f) {
  (function(t, l) {
    n.exports = l();
  })(tt, function() {
    return function(t, l) {
      l.prototype.isoWeeksInYear = function() {
        var p = this.isLeapYear(), $ = this.endOf("y").day();
        return $ === 4 || p && $ === 5 ? 53 : 52;
      };
    };
  });
})(ft);
var Ot = ft.exports;
const dt = /* @__PURE__ */ et(Ot);
var ht = { exports: {} };
(function(n, f) {
  (function(t, l) {
    n.exports = l();
  })(tt, function() {
    return function(t, l) {
      l.prototype.isLeapYear = function() {
        return this.$y % 4 == 0 && this.$y % 100 != 0 || this.$y % 400 == 0;
      };
    };
  });
})(ht);
var bt = ht.exports;
const pt = /* @__PURE__ */ et(bt), K = "TILL_NOW";
b.extend(lt);
b.extend(dt);
b.extend(pt);
const J = {
  year: 0,
  month: 1,
  day: 2,
  hour: 3,
  minute: 4,
  second: 5
};
function Tt(n, f, t, l, p, $, d) {
  const h = [], I = f.getFullYear(), v = f.getMonth() + 1, W = f.getDate(), D = f.getHours(), w = f.getMinutes(), S = f.getSeconds(), Y = t.getFullYear(), P = t.getMonth() + 1, Z = t.getDate(), B = t.getHours(), q = t.getMinutes(), E = t.getSeconds(), F = J[l], N = parseInt(n[0]), T = b(ot([n[0], n[1], "1"])), j = parseInt(n[1]), H = parseInt(n[2]), _ = parseInt(n[3]), M = parseInt(n[4]), a = N === I, L = N === Y, R = a && j === v, i = L && j === P, r = R && H === W, e = i && H === Z, s = r && _ === D, o = e && _ === B, u = s && M === w, c = o && M === q, y = (g, x, k) => {
    let m = [];
    for (let O = g; O <= x; O++)
      m.push(O);
    const U = n.slice(0, J[k]), A = $ == null ? void 0 : $[k];
    return A && typeof A == "function" && (m = m.filter((O) => A(O, {
      get date() {
        const V = [...U, O.toString()];
        return ot(V);
      }
    }))), m;
  };
  if (F >= J.year) {
    const k = y(I, Y, "year");
    h.push(k.map((m) => ({
      label: p("year", m),
      value: m.toString()
    })));
  }
  if (F >= J.month) {
    const k = y(a ? v : 1, L ? P : 12, "month");
    h.push(k.map((m) => ({
      label: p("month", m),
      value: m.toString()
    })));
  }
  if (F >= J.day) {
    const g = R ? W : 1, x = i ? Z : T.daysInMonth(), k = y(g, x, "day");
    h.push(k.map((m) => ({
      label: p("day", m),
      value: m.toString()
    })));
  }
  if (F >= J.hour) {
    const k = y(r ? D : 0, e ? B : 23, "hour");
    h.push(k.map((m) => ({
      label: p("hour", m),
      value: m.toString()
    })));
  }
  if (F >= J.minute) {
    const k = y(s ? w : 0, o ? q : 59, "minute");
    h.push(k.map((m) => ({
      label: p("minute", m),
      value: m.toString()
    })));
  }
  if (F >= J.second) {
    const k = y(u ? S : 0, c ? E : 59, "second");
    h.push(k.map((m) => ({
      label: p("second", m),
      value: m.toString()
    })));
  }
  if (d && (h[0].push({
    label: p("now", null),
    value: K
  }), K === (n == null ? void 0 : n[0])))
    for (let g = 1; g < h.length; g += 1)
      h[g] = [];
  return h;
}
function Ht(n) {
  return n ? [n.getFullYear().toString(), (n.getMonth() + 1).toString(), n.getDate().toString(), n.getHours().toString(), n.getMinutes().toString(), n.getSeconds().toString()] : [];
}
function ot(n) {
  var f, t, l, p, $, d;
  const h = (f = n[0]) !== null && f !== void 0 ? f : "1900", I = (t = n[1]) !== null && t !== void 0 ? t : "1", v = (l = n[2]) !== null && l !== void 0 ? l : "1", W = (p = n[3]) !== null && p !== void 0 ? p : "0", D = ($ = n[4]) !== null && $ !== void 0 ? $ : "0", w = (d = n[5]) !== null && d !== void 0 ? d : "0";
  return new Date(parseInt(h), parseInt(I) - 1, parseInt(v), parseInt(W), parseInt(D), parseInt(w));
}
b.extend(lt);
b.extend(dt);
b.extend(pt);
const Q = {
  year: 0,
  week: 1,
  "week-day": 2
};
function At(n, f, t, l, p, $) {
  const d = [], h = f.getFullYear(), I = t.getFullYear(), v = Q[l], W = parseInt(n[0]), D = W === h, w = W === I, S = b(f), Y = b(t), P = S.isoWeek(), Z = Y.isoWeek(), B = S.isoWeekday(), q = Y.isoWeekday(), E = parseInt(n[1]), F = D && E === P, N = w && E === Z, T = b(`${W}-01-01`).isoWeeksInYear(), j = (H, _, M) => {
    let a = [];
    for (let i = H; i <= _; i++)
      a.push(i);
    const L = n.slice(0, Q[M]), R = $ == null ? void 0 : $[M];
    return R && typeof R == "function" && (a = a.filter((i) => R(i, {
      get date() {
        const r = [...L, i.toString()];
        return mt(r);
      }
    }))), a;
  };
  if (v >= Q.year) {
    const M = j(h, I, "year");
    d.push(M.map((a) => ({
      label: p("year", a),
      value: a.toString()
    })));
  }
  if (v >= Q.week) {
    const M = j(D ? P : 1, w ? Z : T, "week");
    d.push(M.map((a) => ({
      label: p("week", a),
      value: a.toString()
    })));
  }
  if (v >= Q["week-day"]) {
    const M = j(F ? B : 1, N ? q : 7, "week-day");
    d.push(M.map((a) => ({
      label: p("week-day", a),
      value: a.toString()
    })));
  }
  return d;
}
function Pt(n) {
  if (!n)
    return [];
  const f = b(n);
  return [f.isoWeekYear().toString(), f.isoWeek().toString(), f.isoWeekday().toString()];
}
function mt(n) {
  var f, t, l;
  const p = (f = n[0]) !== null && f !== void 0 ? f : "1900", $ = (t = n[1]) !== null && t !== void 0 ? t : "1", d = (l = n[2]) !== null && l !== void 0 ? l : "1";
  return b().year(parseInt(p)).isoWeek(parseInt($)).isoWeekday(parseInt(d)).hour(0).minute(0).second(0).toDate();
}
const Ft = {
  year: 1,
  month: 2,
  day: 3,
  hour: 4,
  minute: 5,
  second: 6
}, Nt = (n, f) => {
  if (f.includes("week"))
    return Pt(n);
  {
    const t = f;
    return Ht(n).slice(0, Ft[t]);
  }
}, it = (n, f) => {
  if ((n == null ? void 0 : n[0]) === K) {
    const t = /* @__PURE__ */ new Date();
    return t.tillNow = !0, t;
  }
  return f.includes("week") ? mt(n) : ot(n);
}, jt = (n, f, t, l, p, $, d) => l.startsWith("week") ? At(n, f, t, l, p, $) : Tt(n, f, t, l, p, $, d);
function Lt(n) {
  const {
    locale: f
  } = xt();
  return nt((t, l) => {
    if (n)
      return n(t, l);
    switch (t) {
      case "minute":
      case "second":
      case "hour":
        return ("0" + l.toString()).slice(-2);
      case "now":
        return f.DatePicker.tillNow;
      default:
        return l.toString();
    }
  }, [n]);
}
const at = (/* @__PURE__ */ new Date()).getFullYear(), Rt = {
  min: new Date((/* @__PURE__ */ new Date()).setFullYear(at - 10)),
  max: new Date((/* @__PURE__ */ new Date()).setFullYear(at + 10)),
  precision: "day",
  defaultValue: null
}, gt = Mt((n, f) => {
  const t = yt(Rt, n), {
    renderLabel: l
  } = t, [p, $] = $t({
    value: t.value,
    defaultValue: t.defaultValue,
    onChange: (w) => {
      var S;
      w !== null && ((S = t.onConfirm) === null || S === void 0 || S.call(t, w));
    }
  }), d = st(() => /* @__PURE__ */ new Date(), []), h = Lt(l), I = st(() => {
    let w = p ?? d;
    return w.tillNow ? [K] : (w = new Date(Wt(w.getTime(), t.min.getTime(), t.max.getTime())), Nt(w, t.precision));
  }, [p, t.precision, t.min, t.max]), v = nt((w) => {
    const S = it(w, t.precision);
    $(S, !0);
  }, [$, t.precision]), W = vt((w) => {
    var S;
    const Y = it(w, t.precision);
    (S = t.onSelect) === null || S === void 0 || S.call(t, Y);
  }), D = nt((w) => jt(w, t.min, t.max, t.precision, h, t.filter, t.tillNow), [t.min, t.max, t.precision, h, t.tillNow]);
  return wt(t, /* @__PURE__ */ rt(Yt, {
    ref: f,
    columns: D,
    value: I,
    onCancel: t.onCancel,
    onClose: t.onClose,
    closeOnMaskClick: t.closeOnMaskClick,
    visible: t.visible,
    confirmText: t.confirmText,
    cancelText: t.cancelText,
    onConfirm: v,
    onSelect: W,
    getContainer: t.getContainer,
    loading: t.loading,
    loadingContent: t.loadingContent,
    afterShow: t.afterShow,
    afterClose: t.afterClose,
    onClick: t.onClick,
    title: t.title,
    stopPropagation: t.stopPropagation,
    mouseWheel: t.mouseWheel,
    destroyOnClose: t.destroyOnClose,
    forceRender: t.forceRender,
    children: (w, S) => {
      var Y;
      return (Y = t.children) === null || Y === void 0 ? void 0 : Y.call(t, p, S);
    }
  }));
});
function Vt(n) {
  return new Promise((f) => {
    const l = It(/* @__PURE__ */ rt(() => {
      const [p, $] = St(!1);
      return Dt(() => {
        $(!0);
      }, []), /* @__PURE__ */ rt(gt, {
        ...Object.assign({}, n, {
          visible: p,
          onConfirm: (d) => {
            var h;
            (h = n.onConfirm) === null || h === void 0 || h.call(n, d), f(d);
          },
          onClose: () => {
            var d;
            (d = n.onClose) === null || d === void 0 || d.call(n), $(!1), f(null);
          },
          afterClose: () => {
            var d;
            (d = n.afterClose) === null || d === void 0 || d.call(n), l();
          }
        })
      });
    }, {}));
  });
}
const Qt = kt(gt, {
  prompt: Vt,
  DATE_NOW: K
});
export {
  Qt as default
};
