// lowcoder-sdk/dist/be0c5653.js
import { mJ as q, i6 as L } from "./c988e26e.js";
import D from "react";
function F(i, u) {
  for (var o = 0; o < u.length; o++) {
    const l = u[o];
    if (typeof l != "string" && !Array.isArray(l)) {
      for (const f in l)
        if (f !== "default" && !(f in i)) {
          const a = Object.getOwnPropertyDescriptor(l, f);
          a && Object.defineProperty(i, f, a.get ? a : {
            enumerable: !0,
            get: () => l[f]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(i, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, y = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var M = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM624 380.9c0 10.2-4.9 19.9-13.2 25.9L465.4 512l145.4 105.2c8.3 6 13.2 15.6 13.2 25.9V690c0 6.5-7.4 10.3-12.7 6.5l-246-178a7.95 7.95 0 010-12.9l246-178c5.3-3.8 12.7 0 12.7 6.5v46.8z" } }] }, name: "left-square", theme: "filled" };
_.default = M;
(function(i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), Object.defineProperty(i, "default", {
    enumerable: !0,
    get: function() {
      return h;
    }
  });
  var u = /* @__PURE__ */ P(D), o = /* @__PURE__ */ a(_), l = /* @__PURE__ */ a(q);
  function f(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function P(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, c = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = c ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(c) {
        return Object.getOwnPropertyDescriptor(r, c).enumerable;
      }))), n.forEach(function(c) {
        f(e, c, r[c]);
      });
    }
    return e;
  }
  function b(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(c) {
        return Object.getOwnPropertyDescriptor(e, c).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function w(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : b(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var S = function(e, t) {
    return /* @__PURE__ */ u.createElement(l.default, w(m({}, e), {
      ref: t,
      icon: o.default
    }));
  }, v = /* @__PURE__ */ u.forwardRef(S);
  process.env.NODE_ENV !== "production" && (v.displayName = "LeftSquareFilled");
  var h = v;
})(y);
(function(i, u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), u.default = void 0;
  const o = l(y);
  function l(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const f = o;
  u.default = f, i.exports = f;
})(s, s.exports);
var g = s.exports;
const j = /* @__PURE__ */ L(g), N = /* @__PURE__ */ F({
  __proto__: null,
  default: j
}, [g]);
export {
  N as L
};
