// lowcoder-sdk/dist/c2f0a2ba.js
import { a as x } from "./1daf7233.js";
import { X as F, aw as V, ax as D, eC as w, w as u, Y as r, S as E, cZ as P, P as k, Z as c, eD as N, c7 as U, O as C, a as z, eE as B, U as I, F as l, aB as R, a8 as i, eA as M, eF as H, G as b, a9 as n, aa as s, ah as g, ab as j, ac as A, ae as L, ad as O, cB as W, cC as Z } from "./c988e26e.js";
import _, { useState as S, Suspense as G, useContext as h } from "react";
import { formDataChildren as T, FormDataPropertyView as X } from "./85e61f03.js";
import "react-dom";
const Y = F.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  position: relative;
  border: ${(e) => e.$style.borderWidth ? e.$style.borderWidth : "1px"} solid ${(e) => e.$style.border};
  border-radius: ${(e) => e.$style.radius};
  overflow: hidden;
  width: 100%;
  height: 100%;
  width: ${(e) => V(e.$style.margin)};	
  height: ${(e) => D(e.$style.margin)};	
  margin: ${(e) => e.$style.margin};	
  padding: ${(e) => e.$style.padding};
  .signature {
    background-color: ${(e) => e.$style.background};
    opacity: ${(e) => e.$isEmpty ? 0 : 1};
    width: 100%;
    height: 100%;
  }

  .sigCanvas {
    flex-grow: 1;
  }
  .footer {
    position: absolute;
    bottom: 0;
    right: 0;
    padding-right: 8px;
    > span {
      margin: 0 8px 16px 8px;
      cursor: pointer;
      svg {
        color: ${(e) => e.$style.footerIcon};
        width: 14px;
        height: 14px;
        path {
          fill: ${(e) => e.$style.footerIcon};
        }
      }
      &:hover svg {
        color: ${(e) => w(e.$style.footerIcon)};
        path {
          fill: ${(e) => w(e.$style.footerIcon)};
        }
      }
    }
  }
  .empty {
    position: absolute;
    height: 100%;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: ${(e) => e.$style.background};
    z-index: -1;
    color: ${(e) => e.$style.tips};
  }
`, q = {
  tips: u(E, r("signature.signHere")),
  onEvent: P,
  label: u(k, {
    position: "column",
    text: ""
  }),
  style: c(N, "style"),
  labelStyle: c(U, "labelStyle"),
  showUndo: u(C, !0),
  showClear: u(C, !0),
  value: z(""),
  inputFieldStyle: c(B, "inputFieldStyle"),
  ...T
}, J = _.lazy(() => import("./3072ddca.js").then((e) => e.i));
let y = function() {
  return new I(q, (e, $) => {
    let t = null;
    const [m, p] = S(!1), [f, v] = S([0, 0]), d = (a = !1) => {
      const o = a || (t == null ? void 0 : t.toData().length) === 0;
      t && (o && (t == null || t.clear()), $(W({
        value: Z(o ? "" : t.toDataURL(), !1)
      })));
    };
    return e.label({
      style: e.style,
      labelStyle: e.labelStyle,
      inputFieldStyle: e.inputFieldStyle,
      children: /* @__PURE__ */ l(R, {
        onResize: (a, o) => {
          a && o && v([a, o]), d(!0);
        },
        children: /* @__PURE__ */ i(Y, {
          onMouseDown: (a) => {
            a.preventDefault();
          },
          $style: e.inputFieldStyle,
          $isEmpty: !e.value && !m,
          children: [/* @__PURE__ */ l("div", {
            className: "signature",
            children: /* @__PURE__ */ l(G, {
              fallback: /* @__PURE__ */ l(M, {}),
              children: /* @__PURE__ */ l(J, {
                ref: (a) => {
                  t = a;
                },
                penColor: e.inputFieldStyle.pen,
                clearOnResize: !1,
                canvasProps: {
                  className: "sigCanvas",
                  width: f[0],
                  height: f[1]
                },
                onEnd: () => {
                  d(), p(!1), e.onEvent("change");
                },
                onBegin: () => p(!0)
              })
            })
          }), (e.showClear || e.showUndo) && /* @__PURE__ */ i("div", {
            className: "footer",
            children: [e.showUndo && /* @__PURE__ */ l("span", {
              className: "anticon",
              children: /* @__PURE__ */ l(H, {
                onClick: () => {
                  const a = t == null ? void 0 : t.toData();
                  a && (a == null || a.pop(), t == null || t.fromData(a), d(), e.onEvent("change"));
                }
              })
            }), e.showClear && /* @__PURE__ */ l(x, {
              onClick: () => {
                d(!0), e.onEvent("change");
              }
            })]
          }), !(m || e.value) && /* @__PURE__ */ l("div", {
            className: "empty",
            children: e.tips
          })]
        })
      })
    });
  }).setPropertyViewFn((e) => /* @__PURE__ */ i(b, {
    children: [/* @__PURE__ */ l(n, {
      name: s.basic,
      children: e.tips.propertyView({
        label: r("signature.tips")
      })
    }), /* @__PURE__ */ l(X, {
      ...e
    }), ["logic", "both"].includes(h(g).editorModeStatus) && /* @__PURE__ */ i(n, {
      name: s.interaction,
      children: [e.onEvent.getPropertyView(), j(e), e.showUndo.propertyView({
        label: r("signature.showUndo")
      }), e.showClear.propertyView({
        label: r("signature.showClear")
      })]
    }), ["layout", "both"].includes(h(g).editorModeStatus) && e.label.getPropertyView(), ["layout", "both"].includes(h(g).editorModeStatus) && /* @__PURE__ */ i(b, {
      children: [/* @__PURE__ */ l(n, {
        name: s.style,
        children: e.style.getPropertyView()
      }), /* @__PURE__ */ l(n, {
        name: s.labelStyle,
        children: e.labelStyle.getPropertyView()
      }), /* @__PURE__ */ l(n, {
        name: s.inputFieldStyle,
        children: e.inputFieldStyle.getPropertyView()
      })]
    })]
  })).build();
}();
y = class extends y {
  autoHeight() {
    return !1;
  }
};
const le = A(y, [new L("value", r("value")), O]);
export {
  le as SignatureComp
};
