// lowcoder-sdk/dist/c37ca80e.js
import { mJ as S, i6 as q } from "./c988e26e.js";
import L from "react";
function M(u, a) {
  for (var c = 0; c < a.length; c++) {
    const i = a[c];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const f in i)
        if (f !== "default" && !(f in u)) {
          const o = Object.getOwnPropertyDescriptor(i, f);
          o && Object.defineProperty(u, f, o.get ? o : {
            enumerable: !0,
            get: () => i[f]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(u, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, g = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var D = { icon: function(a, c) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z", fill: a } }, { tag: "path", attrs: { d: "M184 840h656V184H184v656zm181.3-334.5l246-178c5.3-3.8 12.7 0 12.7 6.5v46.9c0 10.3-4.9 19.9-13.2 25.9L465.4 512l145.4 105.2c8.3 6 13.2 15.7 13.2 25.9V690c0 6.5-7.4 10.3-12.7 6.4l-246-178a7.95 7.95 0 010-12.9z", fill: c } }, { tag: "path", attrs: { d: "M365.3 518.4l246 178c5.3 3.9 12.7.1 12.7-6.4v-46.9c0-10.2-4.9-19.9-13.2-25.9L465.4 512l145.4-105.2c8.3-6 13.2-15.6 13.2-25.9V334c0-6.5-7.4-10.3-12.7-6.5l-246 178a7.95 7.95 0 000 12.9z", fill: a } }] };
}, name: "left-square", theme: "twotone" };
_.default = D;
(function(u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), Object.defineProperty(u, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var a = /* @__PURE__ */ y(L), c = /* @__PURE__ */ o(_), i = /* @__PURE__ */ o(S);
  function f(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function o(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function y(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, l = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = l ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function P(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(l) {
        return Object.getOwnPropertyDescriptor(r, l).enumerable;
      }))), n.forEach(function(l) {
        f(e, l, r[l]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(l) {
        return Object.getOwnPropertyDescriptor(e, l).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function T(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var h = function(e, t) {
    return /* @__PURE__ */ a.createElement(i.default, T(P({}, e), {
      ref: t,
      icon: c.default
    }));
  }, v = /* @__PURE__ */ a.forwardRef(h);
  process.env.NODE_ENV !== "production" && (v.displayName = "LeftSquareTwoTone");
  var b = v;
})(g);
(function(u, a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), a.default = void 0;
  const c = i(g);
  function i(o) {
    return o && o.__esModule ? o : { default: o };
  }
  const f = c;
  a.default = f, u.exports = f;
})(d, d.exports);
var w = d.exports;
const j = /* @__PURE__ */ q(w), V = /* @__PURE__ */ M({
  __proto__: null,
  default: j
}, [w]);
export {
  V as L
};
