// lowcoder-sdk/dist/c515902a.js
import { mJ as D, i6 as F } from "./c988e26e.js";
import M from "react";
function j(a, c) {
  for (var u = 0; u < c.length; u++) {
    const o = c[u];
    if (typeof o != "string" && !Array.isArray(o)) {
      for (const l in o)
        if (l !== "default" && !(l in a)) {
          const i = Object.getOwnPropertyDescriptor(o, l);
          i && Object.defineProperty(a, l, i.get ? i : {
            enumerable: !0,
            get: () => o[l]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(a, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, g = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm5.4 670c-110 0-173.4-73.2-173.4-194.9v-52.3C344 364.2 407.4 290 517.3 290c94.3 0 162.7 60.7 162.7 147.4 0 2.6-2.1 4.7-4.7 4.7h-56.7c-4.2 0-7.6-3.2-8-7.4-4-49.5-40-83.4-93-83.4-65.3 0-102.1 48.5-102.1 135.5v52.6c0 85.7 36.9 133.6 102.1 133.6 52.8 0 88.7-31.7 93-77.8.4-4.1 3.8-7.3 8-7.3h56.8c2.6 0 4.7 2.1 4.7 4.7 0 82.6-68.7 141.4-162.7 141.4z" } }] }, name: "copyright-circle", theme: "filled" };
_.default = E;
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return w;
    }
  });
  var c = /* @__PURE__ */ h(M), u = /* @__PURE__ */ i(_), o = /* @__PURE__ */ i(D);
  function l(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function i(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function y(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (y = function(n) {
      return n ? t : r;
    })(e);
  }
  function h(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = y(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = f ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function C(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(f) {
        return Object.getOwnPropertyDescriptor(t, f).enumerable;
      }))), n.forEach(function(f) {
        l(e, f, t[f]);
      });
    }
    return e;
  }
  function P(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function m(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : P(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var b = function(e, r) {
    return /* @__PURE__ */ c.createElement(o.default, m(C({}, e), {
      ref: r,
      icon: u.default
    }));
  }, O = /* @__PURE__ */ c.forwardRef(b);
  process.env.NODE_ENV !== "production" && (O.displayName = "CopyrightCircleFilled");
  var w = O;
})(g);
(function(a, c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), c.default = void 0;
  const u = o(g);
  function o(i) {
    return i && i.__esModule ? i : { default: i };
  }
  const l = u;
  c.default = l, a.exports = l;
})(s, s.exports);
var v = s.exports;
const S = /* @__PURE__ */ F(v), R = /* @__PURE__ */ j({
  __proto__: null,
  default: S
}, [v]);
export {
  R as C
};
