// lowcoder-sdk/dist/c57644a9.js
import { mJ as E, i6 as x } from "./c988e26e.js";
import M from "react";
function F(l, c) {
  for (var i = 0; i < c.length; i++) {
    const f = c[i];
    if (typeof f != "string" && !Array.isArray(f)) {
      for (const a in f)
        if (a !== "default" && !(a in l)) {
          const o = Object.getOwnPropertyDescriptor(f, a);
          o && Object.defineProperty(l, a, o.get ? o : {
            enumerable: !0,
            get: () => f[a]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(l, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, g = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var D = { icon: function(c, i) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M534 352V136H232v752h560V394H576a42 42 0 01-42-42zm51.6 120h35.7a12.04 12.04 0 0110.1 18.5L546.1 623l84 130.4c3.6 5.6 2 13-3.6 16.6-2 1.2-4.2 1.9-6.5 1.9h-37.5c-4.1 0-8-2.1-10.2-5.7L510 664.8l-62.7 101.5c-2.2 3.5-6 5.7-10.2 5.7h-34.5a12.04 12.04 0 01-10.2-18.4l83.4-132.8-82.3-130.4c-3.6-5.7-1.9-13.1 3.7-16.6 1.9-1.3 4.1-1.9 6.4-1.9H442c4.2 0 8.1 2.2 10.3 5.8l61.8 102.4 61.2-102.3c2.2-3.6 6.1-5.8 10.3-5.8z", fill: i } }, { tag: "path", attrs: { d: "M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM602 137.8L790.2 326H602V137.8zM792 888H232V136h302v216a42 42 0 0042 42h216v494z", fill: c } }, { tag: "path", attrs: { d: "M514.1 580.1l-61.8-102.4c-2.2-3.6-6.1-5.8-10.3-5.8h-38.4c-2.3 0-4.5.6-6.4 1.9-5.6 3.5-7.3 10.9-3.7 16.6l82.3 130.4-83.4 132.8a12.04 12.04 0 0010.2 18.4h34.5c4.2 0 8-2.2 10.2-5.7L510 664.8l62.3 101.4c2.2 3.6 6.1 5.7 10.2 5.7H620c2.3 0 4.5-.7 6.5-1.9 5.6-3.6 7.2-11 3.6-16.6l-84-130.4 85.3-132.5a12.04 12.04 0 00-10.1-18.5h-35.7c-4.2 0-8.1 2.2-10.3 5.8l-61.2 102.3z", fill: c } }] };
}, name: "file-excel", theme: "twotone" };
_.default = D;
(function(l) {
  Object.defineProperty(l, "__esModule", {
    value: !0
  }), Object.defineProperty(l, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var c = /* @__PURE__ */ y(M), i = /* @__PURE__ */ o(_), f = /* @__PURE__ */ o(E);
  function a(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function o(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function y(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, u = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = u ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function h(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(u) {
        return Object.getOwnPropertyDescriptor(r, u).enumerable;
      }))), n.forEach(function(u) {
        a(e, u, r[u]);
      });
    }
    return e;
  }
  function P(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(u) {
        return Object.getOwnPropertyDescriptor(e, u).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function T(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : P(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var m = function(e, t) {
    return /* @__PURE__ */ c.createElement(f.default, T(h({}, e), {
      ref: t,
      icon: i.default
    }));
  }, v = /* @__PURE__ */ c.forwardRef(m);
  process.env.NODE_ENV !== "production" && (v.displayName = "FileExcelTwoTone");
  var b = v;
})(g);
(function(l, c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), c.default = void 0;
  const i = f(g);
  function f(o) {
    return o && o.__esModule ? o : { default: o };
  }
  const a = i;
  c.default = a, l.exports = a;
})(s, s.exports);
var w = s.exports;
const j = /* @__PURE__ */ x(w), S = /* @__PURE__ */ F({
  __proto__: null,
  default: j
}, [w]);
export {
  S as F
};
