// lowcoder-sdk/dist/c5f25aa6.js
import { mJ as M, i6 as D } from "./c988e26e.js";
import z from "react";
function j(f, o) {
  for (var u = 0; u < o.length; u++) {
    const i = o[u];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const c in i)
        if (c !== "default" && !(c in f)) {
          const a = Object.getOwnPropertyDescriptor(i, c);
          a && Object.defineProperty(f, c, a.get ? a : {
            enumerable: !0,
            get: () => i[c]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, v = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var V = { icon: function(o, u) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M320 480c0 49.1 19.1 95.3 53.9 130.1 34.7 34.8 81 53.9 130.1 53.9h16c49.1 0 95.3-19.1 130.1-53.9 34.8-34.7 53.9-81 53.9-130.1V184H320v296zM184 352c0 41 26.9 75.8 64 87.6-37.1-11.9-64-46.7-64-87.6zm364 382.5C665 721.8 758.4 630.2 773.8 514 758.3 630.2 665 721.7 548 734.5zM250.2 514C265.6 630.2 359 721.8 476 734.5 359 721.7 265.7 630.2 250.2 514z", fill: u } }, { tag: "path", attrs: { d: "M868 160h-92v-40c0-4.4-3.6-8-8-8H256c-4.4 0-8 3.6-8 8v40h-92a44 44 0 00-44 44v148c0 81.7 60 149.6 138.2 162C265.7 630.2 359 721.7 476 734.5v105.2H280c-17.7 0-32 14.3-32 32V904c0 4.4 3.6 8 8 8h512c4.4 0 8-3.6 8-8v-32.3c0-17.7-14.3-32-32-32H548V734.5C665 721.7 758.3 630.2 773.8 514 852 501.6 912 433.7 912 352V204a44 44 0 00-44-44zM248 439.6a91.99 91.99 0 01-64-87.6V232h64v207.6zM704 480c0 49.1-19.1 95.4-53.9 130.1-34.8 34.8-81 53.9-130.1 53.9h-16c-49.1 0-95.4-19.1-130.1-53.9-34.8-34.8-53.9-81-53.9-130.1V184h384v296zm136-128c0 41-26.9 75.8-64 87.6V232h64v120z", fill: o } }] };
}, name: "trophy", theme: "twotone" };
_.default = V;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var o = /* @__PURE__ */ T(z), u = /* @__PURE__ */ a(_), i = /* @__PURE__ */ a(M);
  function c(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function y(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (y = function(n) {
      return n ? t : r;
    })(e);
  }
  function T(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = y(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, p = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var l in e)
      if (l !== "default" && Object.prototype.hasOwnProperty.call(e, l)) {
        var d = p ? Object.getOwnPropertyDescriptor(e, l) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, l, d) : n[l] = e[l];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function w(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(p) {
        return Object.getOwnPropertyDescriptor(t, p).enumerable;
      }))), n.forEach(function(p) {
        c(e, p, t[p]);
      });
    }
    return e;
  }
  function g(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(p) {
        return Object.getOwnPropertyDescriptor(e, p).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function P(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : g(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var m = function(e, r) {
    return /* @__PURE__ */ o.createElement(i.default, P(w({}, e), {
      ref: r,
      icon: u.default
    }));
  }, O = /* @__PURE__ */ o.forwardRef(m);
  process.env.NODE_ENV !== "production" && (O.displayName = "TrophyTwoTone");
  var b = O;
})(v);
(function(f, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const u = i(v);
  function i(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const c = u;
  o.default = c, f.exports = c;
})(s, s.exports);
var h = s.exports;
const E = /* @__PURE__ */ D(h), x = /* @__PURE__ */ j({
  __proto__: null,
  default: E
}, [h]);
export {
  x as T
};
