// lowcoder-sdk/dist/d1e456ed.js
import { mJ as M, i6 as D } from "./c988e26e.js";
import j from "react";
function z(f, u) {
  for (var l = 0; l < u.length; l++) {
    const a = u[l];
    if (typeof a != "string" && !Array.isArray(a)) {
      for (const c in a)
        if (c !== "default" && !(c in f)) {
          const i = Object.getOwnPropertyDescriptor(a, c);
          i && Object.defineProperty(f, c, i.get ? i : {
            enumerable: !0,
            get: () => a[c]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var O = { exports: {} }, g = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M952 792H72c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h880c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0-632H72c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h880c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-24 500c8.8 0 16-7.2 16-16V380c0-8.8-7.2-16-16-16H416c-8.8 0-16 7.2-16 16v264c0 8.8 7.2 16 16 16h512zM472 436h400v152H472V436zM80 646c0 4.4 3.6 8 8 8h224c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H88c-4.4 0-8 3.6-8 8v56zm8-204h224c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H88c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8z" } }] }, name: "pic-right", theme: "outlined" };
s.default = E;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return R;
    }
  });
  var u = /* @__PURE__ */ h(j), l = /* @__PURE__ */ i(s), a = /* @__PURE__ */ i(M);
  function c(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function i(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function h(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, o = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = o ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function y(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(o) {
        return Object.getOwnPropertyDescriptor(r, o).enumerable;
      }))), n.forEach(function(o) {
        c(e, o, r[o]);
      });
    }
    return e;
  }
  function m(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(o) {
        return Object.getOwnPropertyDescriptor(e, o).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function b(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : m(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ u.createElement(a.default, b(y({}, e), {
      ref: t,
      icon: l.default
    }));
  }, v = /* @__PURE__ */ u.forwardRef(w);
  process.env.NODE_ENV !== "production" && (v.displayName = "PicRightOutlined");
  var R = v;
})(g);
(function(f, u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), u.default = void 0;
  const l = a(g);
  function a(i) {
    return i && i.__esModule ? i : { default: i };
  }
  const c = l;
  u.default = c, f.exports = c;
})(O, O.exports);
var P = O.exports;
const H = /* @__PURE__ */ D(P), N = /* @__PURE__ */ z({
  __proto__: null,
  default: H
}, [P]);
export {
  N as P
};
