// lowcoder-sdk/dist/dbb92130.js
import { mJ as h, i6 as q } from "./c988e26e.js";
import M from "react";
function F(c, i) {
  for (var o = 0; o < i.length; o++) {
    const u = i[o];
    if (typeof u != "string" && !Array.isArray(u)) {
      for (const l in u)
        if (l !== "default" && !(l in c)) {
          const a = Object.getOwnPropertyDescriptor(u, l);
          a && Object.defineProperty(c, l, a.get ? a : {
            enumerable: !0,
            get: () => u[l]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(c, Symbol.toStringTag, { value: "Module" }));
}
var b = { exports: {} }, v = {}, s = {};
Object.defineProperty(s, "__esModule", { value: !0 });
var z = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M498.6 432c-40.8-72.5-84.7-133.4-91.2-142.3-68.8 32.5-120.3 95.9-136.2 172.2 11 .2 112.4.7 227.4-29.9zm66.5 21.8c5.7 11.7 11.2 23.6 16.3 35.6 1.8 4.2 3.6 8.4 5.3 12.7 81.8-10.3 163.2 6.2 171.3 7.9-.5-58.1-21.3-111.4-55.5-153.3-5.3 7.1-46.5 60-137.4 97.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM512 800c-158.8 0-288-129.2-288-288s129.2-288 288-288 288 129.2 288 288-129.2 288-288 288zm89.7-259.1c32.2 88.4 45.3 160.4 47.8 175.4 55.2-37.3 94.5-96.4 105.4-164.9-8.4-2.6-76.1-22.8-153.2-10.5zm-72.5-26.4c3.2-1 6.4-2 9.7-2.9-6.2-14-12.9-28-19.9-41.7-122.8 36.8-242.1 35.2-252.8 35-.1 2.5-.1 5-.1 7.5 0 63.2 23.9 120.9 63.2 164.5 5.5-9.6 73-121.4 199.9-162.4zm145.9-186.2a245.2 245.2 0 00-220.8-55.1c6.8 9.1 51.5 69.9 91.8 144 87.5-32.8 124.5-82.6 129-88.9zM554 552.8c-138.7 48.3-188.6 144.6-193 153.6 41.7 32.5 94.1 51.9 151 51.9 34.1 0 66.6-6.9 96.1-19.5-3.7-21.6-17.9-96.8-52.5-186.6l-1.6.6z" } }] }, name: "dribbble-square", theme: "filled" };
s.default = z;
(function(c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), Object.defineProperty(c, "default", {
    enumerable: !0,
    get: function() {
      return S;
    }
  });
  var i = /* @__PURE__ */ g(M), o = /* @__PURE__ */ a(s), u = /* @__PURE__ */ a(h);
  function l(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? t : r;
    })(e);
  }
  function g(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = _(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = f ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function m(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(f) {
        return Object.getOwnPropertyDescriptor(t, f).enumerable;
      }))), n.forEach(function(f) {
        l(e, f, t[f]);
      });
    }
    return e;
  }
  function P(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function D(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : P(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var w = function(e, r) {
    return /* @__PURE__ */ i.createElement(u.default, D(m({}, e), {
      ref: r,
      icon: o.default
    }));
  }, O = /* @__PURE__ */ i.forwardRef(w);
  process.env.NODE_ENV !== "production" && (O.displayName = "DribbbleSquareFilled");
  var S = O;
})(v);
(function(c, i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), i.default = void 0;
  const o = u(v);
  function u(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const l = o;
  i.default = l, c.exports = l;
})(b, b.exports);
var y = b.exports;
const j = /* @__PURE__ */ q(y), N = /* @__PURE__ */ F({
  __proto__: null,
  default: j
}, [y]);
export {
  N as D
};
