// lowcoder-sdk/dist/de2e4e92.js
import { generateInsertSql as m } from "./bf2ffa97.js";
import { timeCompSelection as c, dateTimeCompSelection as o, dateCompSelection as s, numberCompSelection as i, allCompSelection as n } from "./90e6d3f7.js";
function r(e) {
  if (e) {
    switch (e.toLowerCase()) {
      case "bit":
      case "tinyint":
      case "smallint":
      case "int":
      case "bigint":
      case "dec":
      case "decimal":
      case "numeric":
      case "smallmoney":
      case "money":
      case "float":
      case "real":
        return i(!0);
      case "date":
        return s();
      case "datetime":
      case "datetime2":
      case "datetimeoffset":
      case "smalldatetime":
        return o();
      case "time":
        return c();
    }
    return n();
  }
}
function l(e, t, a) {
  return {
    compType: "mssql",
    comp: {
      sql: m(t, a),
      commandType: "INSERT",
      mode: "SQL"
    }
  };
}
const u = {
  type: "mssql",
  getCompSelection: r,
  getQueryInitData: l
};
export {
  u as msSqlConfig
};
