// lowcoder-sdk/dist/df82265d.js
import { d8 as l, O as p, h5 as i, a4 as r, h4 as n, a8 as e, G as s, F as m, a9 as a, Y as o } from "./c988e26e.js";
import "react";
import { TourStepControl as u } from "./432d6870.js";
import { PlacementOptions as b, TypeOptions as w } from "./3f72c732.js";
import { TourMaskTooltip as c, TourPlacementTooltip as d, TourArrowTooltip as y } from "./ebbc0fd0.js";
import "react-dom";
const I = {
  open: l("open"),
  options: u,
  // indicatorsRender: AlkjdfControl, // todo get this working later
  disabledInteraction: p,
  mask: i,
  placement: r(b, "bottom"),
  arrow: n,
  type: r(w, "default")
}, M = (t) => /* @__PURE__ */ e(s, {
  children: [/* @__PURE__ */ m(a, {
    name: o("tour.section1Title"),
    children: t.options.propertyView({})
  }), /* @__PURE__ */ e(a, {
    name: "customization",
    children: [t.disabledInteraction.propertyView({
      label: o("tour.disabledInteraction.label"),
      tooltip: o("tour.disabledInteraction.tooltip")
    }), t.mask.propertyView({
      label: o("tour.mask.label"),
      tooltip: c
    }), t.placement.propertyView({
      label: o("tour.placement.label"),
      tooltip: d,
      radioButton: !1
    }), t.arrow.propertyView({
      label: o("tour.arrow.label"),
      tooltip: y
    }), t.type.propertyView({
      label: o("tour.type.label"),
      tooltip: o("tour.type.tooltip")
    })]
  })]
});
export {
  I as TourChildrenMap,
  M as TourPropertyView
};
