// lowcoder-sdk/dist/e050e560.js
import { F as e, jn as s, jo as n, jp as i, gu as f, jq as w, jr as p, js as r, jt as F, ju as R, jv as l, jw as $, jx as c, jy as h, jz as k, jA as g, jB as x, jC as m, jD as j, jE as u, jF as b, jG as v, jH as C, jI as y, jJ as E, jK as S, jL as B, jM as L, jN as T, jO as D, jP as G, jQ as q, jR as z, jS as I, jT as M, jU as N, jV as P, jW as A, jX as V, jY as o, jZ as t, j_ as H, j$ as J, k0 as K, iB as O, k1 as Q, k2 as U, k3 as X, k4 as W, k5 as Y, k6 as Z, k7 as _, k8 as ee, k9 as ae, ka as re, kb as oe, kc as te, kd as de, ke as se, kf as ne, kg as ie, kh as fe, ki as we, kj as pe, kk as Fe, kl as Re, km as le, kn as $e, ko as d, kp as ce, kq as he, kr as ke, ks as ge, kt as xe, ku as me, kv as je, kw as ue, kx as be, ky as ve, kz as Ce, kA as ye, kB as Ee, kC as Se, kD as Be, kE as Le, kF as Te, kG as De, kH as Ge, kI as qe, kJ as ze, kK as Ie, kL as Me, kM as Ne, kN as Pe, kO as Ae, kP as Ve, kQ as He, kR as Je, kS as Ke, kT as Oe, X as a, kU as Qe } from "./c988e26e.js";
import "react";
const We = {
  audio: /* @__PURE__ */ e(s, {}),
  autocomplete: /* @__PURE__ */ e(n, {}),
  button: /* @__PURE__ */ e(i, {}),
  calendar: /* @__PURE__ */ e(f, {}),
  carousel: /* @__PURE__ */ e(w, {}),
  cascader: /* @__PURE__ */ e(p, {}),
  chart: /* @__PURE__ */ e(r, {}),
  checkbox: /* @__PURE__ */ e(F, {}),
  collapsibleContainer: /* @__PURE__ */ e(R, {}),
  columnLayout: /* @__PURE__ */ e(l, {}),
  comment: /* @__PURE__ */ e($, {}),
  container: /* @__PURE__ */ e(c, {}),
  controlButton: /* @__PURE__ */ e(h, {}),
  custom: /* @__PURE__ */ e(k, {}),
  date: /* @__PURE__ */ e(g, {}),
  dateRange: /* @__PURE__ */ e(x, {}),
  divider: /* @__PURE__ */ e(m, {}),
  drawer: /* @__PURE__ */ e(j, {}),
  dropdown: /* @__PURE__ */ e(u, {}),
  file: /* @__PURE__ */ e(b, {}),
  fileViewer: /* @__PURE__ */ e(v, {}),
  floatTextContainer: /* @__PURE__ */ e(C, {}),
  form: /* @__PURE__ */ e(y, {}),
  grid: /* @__PURE__ */ e(E, {}),
  icon: /* @__PURE__ */ e(S, {}),
  iframe: /* @__PURE__ */ e(B, {}),
  image: /* @__PURE__ */ e(L, {}),
  imageEditor: /* @__PURE__ */ e(T, {}),
  input: /* @__PURE__ */ e(D, {}),
  jsonEditor: /* @__PURE__ */ e(G, {}),
  jsonExplorer: /* @__PURE__ */ e(q, {}),
  jsonLottie: /* @__PURE__ */ e(z, {}),
  jsonSchemaForm: /* @__PURE__ */ e(I, {}),
  link: /* @__PURE__ */ e(M, {}),
  listView: /* @__PURE__ */ e(N, {}),
  meeting: /* @__PURE__ */ e(P, {}),
  mention: /* @__PURE__ */ e(A, {}),
  mermaid: /* @__PURE__ */ e(V, {}),
  modal: /* @__PURE__ */ e(o, {}),
  module: /* @__PURE__ */ e(t, {}),
  moduleContainer: /* @__PURE__ */ e(t, {}),
  // Placeholder or specify if there's a dedicated icon
  multiSelect: /* @__PURE__ */ e(H, {}),
  navigation: /* @__PURE__ */ e(J, {}),
  numberInput: /* @__PURE__ */ e(K, {}),
  npmPlugin: /* @__PURE__ */ e(O, {}),
  // Placeholder or specify if there's a dedicated icon
  pageLayout: /* @__PURE__ */ e(Q, {}),
  password: /* @__PURE__ */ e(U, {}),
  progress: /* @__PURE__ */ e(X, {}),
  progressCircle: /* @__PURE__ */ e(W, {}),
  qrCode: /* @__PURE__ */ e(Y, {}),
  radio: /* @__PURE__ */ e(Z, {}),
  rangeSlider: /* @__PURE__ */ e(_, {}),
  rating: /* @__PURE__ */ e(ee, {}),
  responsiveLayout: /* @__PURE__ */ e(ae, {}),
  richTextEditor: /* @__PURE__ */ e(re, {}),
  scanner: /* @__PURE__ */ e(oe, {}),
  segmentedControl: /* @__PURE__ */ e(te, {}),
  select: /* @__PURE__ */ e(de, {}),
  sharingcomponent: /* @__PURE__ */ e(se, {}),
  signature: /* @__PURE__ */ e(ne, {}),
  slider: /* @__PURE__ */ e(ie, {}),
  step: /* @__PURE__ */ e(fe, {}),
  switch: /* @__PURE__ */ e(we, {}),
  tabbedContainer: /* @__PURE__ */ e(pe, {}),
  table: /* @__PURE__ */ e(Fe, {}),
  text: /* @__PURE__ */ e(Re, {}),
  textArea: /* @__PURE__ */ e(le, {}),
  time: /* @__PURE__ */ e($e, {}),
  timeRange: /* @__PURE__ */ e(d, {}),
  timeline: /* @__PURE__ */ e(ce, {}),
  toggleButton: /* @__PURE__ */ e(he, {}),
  tour: /* @__PURE__ */ e(o, {}),
  tree: /* @__PURE__ */ e(ke, {}),
  treeSelect: /* @__PURE__ */ e(ge, {}),
  video: /* @__PURE__ */ e(xe, {}),
  videocomponent: /* @__PURE__ */ e(me, {}),
  hillchart: /* @__PURE__ */ e(je, {}),
  ganttChart: /* @__PURE__ */ e(ue, {}),
  openLayersGeoMap: /* @__PURE__ */ e(be, {}),
  chartsGeoMap: /* @__PURE__ */ e(ve, {}),
  bpmnEditor: /* @__PURE__ */ e(Ce, {}),
  turnstileCaptcha: /* @__PURE__ */ e(ye, {}),
  pivotTable: /* @__PURE__ */ e(Ee, {}),
  funnelChart: /* @__PURE__ */ e(Se, {}),
  gaugeChart: /* @__PURE__ */ e(Be, {}),
  sankeyChart: /* @__PURE__ */ e(Le, {}),
  candleStickChart: /* @__PURE__ */ e(Te, {}),
  radarChart: /* @__PURE__ */ e(De, {}),
  heatmapChart: /* @__PURE__ */ e(Ge, {}),
  graphChart: /* @__PURE__ */ e(qe, {}),
  treeChart: /* @__PURE__ */ e(ze, {}),
  treemapChart: /* @__PURE__ */ e(Ie, {}),
  sunburstChart: /* @__PURE__ */ e(Me, {}),
  themeriverChart: /* @__PURE__ */ e(Ne, {}),
  basicChart: /* @__PURE__ */ e(r, {}),
  avatar: /* @__PURE__ */ e(Pe, {}),
  avatarGroup: /* @__PURE__ */ e(Ae, {}),
  colorPicker: /* @__PURE__ */ e(Ve, {}),
  floatingButton: /* @__PURE__ */ e(He, {}),
  transfer: /* @__PURE__ */ e(Je, {}),
  card: /* @__PURE__ */ e(Ke, {}),
  timer: /* @__PURE__ */ e(d, {}),
  shape: /* @__PURE__ */ e(Oe, {})
}, Ye = a(Qe)`
  font-size: 13px;
  color: #333;
  .ant-tree-treenode {
    position: relative;
  }
  .ant-tree-switcher {
    width: 12px;
    margin: 0 2px;
    height: 26px;
    display: flex;
    align-items: center;
  }
  .ant-tree-title {
    padding-right: 6px;
    width: 100%;
  }
  .ant-tree-node-content-wrapper {
    padding: 0;
    display: flex;
    min-height: 26px;
    position: unset;
    .ant-tree-iconEle {
      width: 16px;
      height: 16px;
      margin: 0px 0px 0px 4px;
      display: flex;
      align-items: center;
      svg {
        width: 16px;
        height: 16px;
        stroke: #000;
      }
    }

  }
  .ant-tree-checkbox+span {
    padding-left: 0;
  }
  .ant-tree-treenode {
    padding: 0;
    max-width: 288px;
  }
  .ant-tree-indent-unit {
    width: 16px;
  }
  &.ant-tree.ant-tree-directory {
    .ant-tree-treenode:hover::before,
    .ant-tree-treenode-selected::before {
      bottom: 0;
    }
    .ant-tree-treenode:hover {
      &::before {
        background-color: rgba(242, 247, 252, 0.5);
      }
      .ant-tree-title svg {
        visibility: visible;
      }
    }
    .ant-tree-treenode.ant-tree-treenode-selected::before {
      background-color: #f2f7fc;
      height: 26px;
    }
    .ant-tree-node-content-wrapper.ant-tree-node-selected {
      color: #333;
    }
    .ant-tree-treenode-disabled {
      .ant-tree-node-content-wrapper {
        color: inherit;
      }
    }
  }
`, Ze = a.span`
  font-weight: 600;
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-wrap: wrap;
  > span {
    span {
      height: 26px;
      text-overflow: ellipsis;
      overflow: hidden;
      white-space: nowrap;
      flex: 1;
      width: 0;
    }
    width: 100%;
    display: flex;
    justify-content: space-between;
  }
  .no-data svg,
  .show-data svg {
    width: 40px;
    padding: 0 14px;
    height: 26px;
  }
  .show-data {
    height: 26px;
  }
  .no-data {
    visibility: hidden;
    height: 26px;
    &:hover {
      g {
        stroke: #222;
      }
    }
  }
  .show-data:hover svg{
    path {
      fill: #222;
    }
  }
`;
a.div`
  width: 100%; 
  border: 1px solid #E1E3EB;
  border-radius: 4px;
  overflow: hidden;
  background: #fff;
  padding: 0px;
  position: relative;
  margin: 4px 0px 4px 0};
  .simplebar-content > div {
    > .ant-collapse > .ant-collapse-item {
      > .ant-collapse-header {
        display: none;
      }
      > .ant-collapse-content > .ant-collapse-content-box {
        padding-left: 1px;
      }
    }
    &:hover,
    .ant-collapse-content-box > div > div:hover {
      background-color: #fff;
    }
  }
  .simplebar-track.simplebar-horizontal {
    display: none;
  }
  .simplebar-content > div {
    // padding: 0;
  }
`;
export {
  We as C,
  Ye as D,
  Ze as N
};
