// lowcoder-sdk/dist/e11dcd98.js
import { mJ as M, i6 as j } from "./c988e26e.js";
import E from "react";
function S(o, i) {
  for (var l = 0; l < i.length; l++) {
    const f = i[l];
    if (typeof f != "string" && !Array.isArray(f)) {
      for (const a in f)
        if (a !== "default" && !(a in o)) {
          const u = Object.getOwnPropertyDescriptor(f, a);
          u && Object.defineProperty(o, a, u.get ? u : {
            enumerable: !0,
            get: () => f[a]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(o, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, v = {}, O = {};
Object.defineProperty(O, "__esModule", { value: !0 });
var x = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M304 176h80v672h-80zm408 0h-64c-4.4 0-8 3.6-8 8v656c0 4.4 3.6 8 8 8h64c4.4 0 8-3.6 8-8V184c0-4.4-3.6-8-8-8z" } }] }, name: "pause", theme: "outlined" };
O.default = x;
(function(o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), Object.defineProperty(o, "default", {
    enumerable: !0,
    get: function() {
      return D;
    }
  });
  var i = /* @__PURE__ */ g(E), l = /* @__PURE__ */ u(O), f = /* @__PURE__ */ u(M);
  function a(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function u(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function _(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (_ = function(n) {
      return n ? r : t;
    })(e);
  }
  function g(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = _(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, c = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = c ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(c) {
        return Object.getOwnPropertyDescriptor(r, c).enumerable;
      }))), n.forEach(function(c) {
        a(e, c, r[c]);
      });
    }
    return e;
  }
  function b(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(c) {
        return Object.getOwnPropertyDescriptor(e, c).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function h(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : b(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var w = function(e, t) {
    return /* @__PURE__ */ i.createElement(f.default, h(m({}, e), {
      ref: t,
      icon: l.default
    }));
  }, P = /* @__PURE__ */ i.forwardRef(w);
  process.env.NODE_ENV !== "production" && (P.displayName = "PauseOutlined");
  var D = P;
})(v);
(function(o, i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), i.default = void 0;
  const l = f(v);
  function f(u) {
    return u && u.__esModule ? u : { default: u };
  }
  const a = l;
  i.default = a, o.exports = a;
})(d, d.exports);
var y = d.exports;
const N = /* @__PURE__ */ j(y), q = /* @__PURE__ */ S({
  __proto__: null,
  default: N
}, [y]);
export {
  q as P
};
