// lowcoder-sdk/dist/e846960a.js
import { mJ as C, i6 as M } from "./c988e26e.js";
import D from "react";
function j(a, c) {
  for (var l = 0; l < c.length; l++) {
    const i = c[l];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const u in i)
        if (u !== "default" && !(u in a)) {
          const o = Object.getOwnPropertyDescriptor(i, u);
          o && Object.defineProperty(a, u, o.get ? o : {
            enumerable: !0,
            get: () => i[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(a, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, P = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var E = { icon: function(c, l) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z", fill: c } }, { tag: "path", attrs: { d: "M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm192 396c0 4.4-3.6 8-8 8H544v152c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V544H328c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h152V328c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v152h152c4.4 0 8 3.6 8 8v48z", fill: l } }, { tag: "path", attrs: { d: "M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z", fill: c } }] };
}, name: "plus-circle", theme: "twotone" };
_.default = E;
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var c = /* @__PURE__ */ w(D), l = /* @__PURE__ */ o(_), i = /* @__PURE__ */ o(C);
  function u(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function o(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function w(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = f ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function y(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(f) {
        return Object.getOwnPropertyDescriptor(r, f).enumerable;
      }))), n.forEach(function(f) {
        u(e, f, r[f]);
      });
    }
    return e;
  }
  function h(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function m(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : h(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var T = function(e, t) {
    return /* @__PURE__ */ c.createElement(i.default, m(y({}, e), {
      ref: t,
      icon: l.default
    }));
  }, v = /* @__PURE__ */ c.forwardRef(T);
  process.env.NODE_ENV !== "production" && (v.displayName = "PlusCircleTwoTone");
  var b = v;
})(P);
(function(a, c) {
  Object.defineProperty(c, "__esModule", {
    value: !0
  }), c.default = void 0;
  const l = i(P);
  function i(o) {
    return o && o.__esModule ? o : { default: o };
  }
  const u = l;
  c.default = u, a.exports = u;
})(d, d.exports);
var g = d.exports;
const S = /* @__PURE__ */ M(g), N = /* @__PURE__ */ j({
  __proto__: null,
  default: S
}, [g]);
export {
  N as P
};
