// lowcoder-sdk/dist/e8faa8a6.js
import { mJ as C, i6 as D } from "./c988e26e.js";
import F from "react";
function j(a, u) {
  for (var o = 0; o < u.length; o++) {
    const c = u[o];
    if (typeof c != "string" && !Array.isArray(c)) {
      for (const l in c)
        if (l !== "default" && !(l in a)) {
          const i = Object.getOwnPropertyDescriptor(c, l);
          i && Object.defineProperty(a, l, i.get ? i : {
            enumerable: !0,
            get: () => c[l]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(a, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, y = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var E = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm192 472c0 4.4-3.6 8-8 8H328c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h368c4.4 0 8 3.6 8 8v48z" } }] }, name: "minus-circle", theme: "filled" };
_.default = E;
(function(a) {
  Object.defineProperty(a, "__esModule", {
    value: !0
  }), Object.defineProperty(a, "default", {
    enumerable: !0,
    get: function() {
      return h;
    }
  });
  var u = /* @__PURE__ */ P(F), o = /* @__PURE__ */ i(_), c = /* @__PURE__ */ i(C);
  function l(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function i(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? t : r;
    })(e);
  }
  function P(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = O(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = f ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function m(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(f) {
        return Object.getOwnPropertyDescriptor(t, f).enumerable;
      }))), n.forEach(function(f) {
        l(e, f, t[f]);
      });
    }
    return e;
  }
  function M(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function b(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : M(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var w = function(e, r) {
    return /* @__PURE__ */ u.createElement(c.default, b(m({}, e), {
      ref: r,
      icon: o.default
    }));
  }, v = /* @__PURE__ */ u.forwardRef(w);
  process.env.NODE_ENV !== "production" && (v.displayName = "MinusCircleFilled");
  var h = v;
})(y);
(function(a, u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), u.default = void 0;
  const o = c(y);
  function c(i) {
    return i && i.__esModule ? i : { default: i };
  }
  const l = o;
  u.default = l, a.exports = l;
})(d, d.exports);
var g = d.exports;
const S = /* @__PURE__ */ D(g), R = /* @__PURE__ */ j({
  __proto__: null,
  default: S
}, [g]);
export {
  R as M
};
