// lowcoder-sdk/dist/ea869b84.js
import { mJ as q, i6 as C } from "./c988e26e.js";
import D from "react";
function F(i, u) {
  for (var o = 0; o < u.length; o++) {
    const l = u[o];
    if (typeof l != "string" && !Array.isArray(l)) {
      for (const c in l)
        if (c !== "default" && !(c in i)) {
          const a = Object.getOwnPropertyDescriptor(l, c);
          a && Object.defineProperty(i, c, a.get ? a : {
            enumerable: !0,
            get: () => l[c]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(i, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, y = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var M = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM695.5 365.7l-210.6 292a31.8 31.8 0 01-51.7 0L308.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H689c6.5 0 10.3 7.4 6.5 12.7z" } }] }, name: "check-square", theme: "filled" };
_.default = M;
(function(i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), Object.defineProperty(i, "default", {
    enumerable: !0,
    get: function() {
      return S;
    }
  });
  var u = /* @__PURE__ */ h(D), o = /* @__PURE__ */ a(_), l = /* @__PURE__ */ a(q);
  function c(e, r, t) {
    return r in e ? Object.defineProperty(e, r, {
      value: t,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[r] = t, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var r = /* @__PURE__ */ new WeakMap(), t = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? t : r;
    })(e);
  }
  function h(e, r) {
    if (!r && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var t = O(r);
    if (t && t.has(e))
      return t.get(e);
    var n = {
      __proto__: null
    }, f = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = f ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, t && t.set(e, n), n;
  }
  function P(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r] != null ? arguments[r] : {}, n = Object.keys(t);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(f) {
        return Object.getOwnPropertyDescriptor(t, f).enumerable;
      }))), n.forEach(function(f) {
        c(e, f, t[f]);
      });
    }
    return e;
  }
  function m(e, r) {
    var t = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      r && (n = n.filter(function(f) {
        return Object.getOwnPropertyDescriptor(e, f).enumerable;
      })), t.push.apply(t, n);
    }
    return t;
  }
  function b(e, r) {
    return r = r ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : m(Object(r)).forEach(function(t) {
      Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
    }), e;
  }
  var w = function(e, r) {
    return /* @__PURE__ */ u.createElement(l.default, b(P({}, e), {
      ref: r,
      icon: o.default
    }));
  }, v = /* @__PURE__ */ u.forwardRef(w);
  process.env.NODE_ENV !== "production" && (v.displayName = "CheckSquareFilled");
  var S = v;
})(y);
(function(i, u) {
  Object.defineProperty(u, "__esModule", {
    value: !0
  }), u.default = void 0;
  const o = l(y);
  function l(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const c = o;
  u.default = c, i.exports = c;
})(s, s.exports);
var g = s.exports;
const k = /* @__PURE__ */ C(g), x = /* @__PURE__ */ F({
  __proto__: null,
  default: k
}, [g]);
export {
  x as C
};
