// lowcoder-sdk/dist/ebbc0fd0.js
import { a8 as i, Y as t, F as o } from "./c988e26e.js";
import "react";
import "react-dom";
const l = `(current, total) => (
  <span>
    {current + 1} / {total}
  </span>
)`, h = /* @__PURE__ */ i("div", {
  children: [t("tour.indicatorsRender.tooltip"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), t("tour.indicatorsRender.tooltipValidTypes"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("h4", {
    children: t("tour.tooltipSignatureHeader")
  }), /* @__PURE__ */ o("code", {
    children: t("tour.indicatorsRender.tooltipFunctionSignature")
  }), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("h4", {
    children: t("tour.tooltipExampleHeader")
  }), /* @__PURE__ */ o("code", {
    children: l
  })]
});
let r = {
  style: {
    boxShadow: "inset 0 0 15px #fff"
  },
  color: "rgba(40, 0, 255, .4)"
};
const p = /* @__PURE__ */ i("div", {
  children: [t("tour.options.mask.tooltip"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), t("tour.options.mask.tooltipValidTypes"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("h3", {
    children: "Example:"
  }), /* @__PURE__ */ o("code", {
    children: JSON.stringify(r, null, 1)
  })]
}), a = /* @__PURE__ */ i("div", {
  children: [t("tour.mask.tooltip"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), t("tour.mask.tooltipValidTypes"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("h4", {
    children: "Example:"
  }), /* @__PURE__ */ o("code", {
    children: JSON.stringify(r, null, 1)
  })]
}), s = /* @__PURE__ */ i("div", {
  children: [t("tour.placement.tooltip"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("h4", {
    children: t("tour.placement.tooltipValidOptions")
  }), /* @__PURE__ */ o("h5", {
    children: t("tour.placement.tooltipValidOptionsAbove")
  }), /* @__PURE__ */ i("ul", {
    children: [/* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "topLeft"
      })
    }), /* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "top"
      })
    }), /* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "topRight"
      })
    })]
  }), /* @__PURE__ */ o("h5", {
    children: t("tour.placement.tooltipValidOptionsLeft")
  }), /* @__PURE__ */ i("ul", {
    children: [/* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "leftTop"
      })
    }), /* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "left"
      })
    }), /* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "leftBottom"
      })
    })]
  }), /* @__PURE__ */ o("h5", {
    children: t("tour.placement.tooltipValidOptionsRight")
  }), /* @__PURE__ */ i("ul", {
    children: [/* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "rightTop"
      })
    }), /* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "right"
      })
    }), /* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "rightBottom"
      })
    })]
  }), /* @__PURE__ */ o("h5", {
    children: t("tour.placement.tooltipValidOptionsBelow")
  }), /* @__PURE__ */ i("ul", {
    children: [/* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "bottomLeft"
      })
    }), /* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "bottom"
      })
    }), /* @__PURE__ */ o("li", {
      children: /* @__PURE__ */ o("code", {
        children: "bottomRight"
      })
    })]
  }), /* @__PURE__ */ o("h5", {
    children: t("tour.placement.tooltipValidOptionsOnTop")
  }), /* @__PURE__ */ o("ul", {
    children: /* @__PURE__ */ o("li", {
      children: "center"
    })
  })]
}), e = "boolean | { pointAtCenter: boolean }", u = /* @__PURE__ */ i("div", {
  children: [t("tour.options.arrow.tooltip"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("h4", {
    children: t("tour.tooltipSignatureHeader")
  }), /* @__PURE__ */ o("code", {
    children: e
  })]
}), b = /* @__PURE__ */ i("div", {
  children: [t("tour.arrow.tooltip"), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("br", {}), /* @__PURE__ */ o("h4", {
    children: t("tour.tooltipSignatureHeader")
  }), /* @__PURE__ */ o("code", {
    children: e
  })]
});
export {
  h as IndicatorsRenderTooltip,
  b as TourArrowTooltip,
  a as TourMaskTooltip,
  s as TourPlacementTooltip,
  u as TourStepArrowTooltip,
  p as TourStepMaskTooltip
};
