// lowcoder-sdk/dist/ed56efe5.js
import { bI as u, Z as y, az as h, F as a, bW as w, a8 as c, G as m, ah as i, a9 as o, aa as r, bJ as f, ab as S, ac as C, ad as g, hB as b, ae as P, Y as p, e1 as v, cg as V, bX as x, B as D } from "./c988e26e.js";
import { useState as N, useContext as s } from "react";
import { ContainerCompBuilder as H } from "./3d641873.js";
import { PageLayout as B } from "./225b8bb4.js";
import "react-dom";
import "./e6ae73db.js";
import "./a8b829c9.js";
const F = function() {
  const t = {
    disabled: u,
    animationStyle: y(h, "animationStyle")
  };
  return new H(t, (e, n) => {
    const [l, d] = N(!1);
    return /* @__PURE__ */ a(w.Provider, {
      value: e.disabled,
      children: /* @__PURE__ */ a(B, {
        ...e,
        siderCollapsed: l,
        setSiderCollapsed: d
      })
    });
  }).setPropertyViewFn((e) => /* @__PURE__ */ c(m, {
    children: [(s(i).editorModeStatus === "logic" || s(i).editorModeStatus === "both") && /* @__PURE__ */ c(o, {
      name: r.interaction,
      children: [f(e), S(e), e.container.appSelectorPropertyView()]
    }), (s(i).editorModeStatus === "layout" || s(i).editorModeStatus === "both") && /* @__PURE__ */ c(m, {
      children: [/* @__PURE__ */ a(o, {
        name: r.layout,
        children: e.container.getPropertyView()
      }), /* @__PURE__ */ a(o, {
        name: r.style,
        children: e.container.stylePropertyView()
      }), /* @__PURE__ */ a(o, {
        name: r.animationStyle,
        hasTooltip: !0,
        children: e.animationStyle.getPropertyView()
      }), e.container.children.showHeader.getView() && /* @__PURE__ */ a(o, {
        name: "Header Style",
        children: e.container.headerStylePropertyView()
      }), e.container.children.showSider.getView() && /* @__PURE__ */ a(o, {
        name: "Sider Style",
        children: e.container.siderStylePropertyView()
      }), /* @__PURE__ */ a(o, {
        name: "Body Style",
        children: e.container.bodyStylePropertyView()
      }), e.container.children.showFooter.getView() && /* @__PURE__ */ a(o, {
        name: "Footer Style",
        children: e.container.footerStylePropertyView()
      })]
    })]
  })).build();
}();
function L(t) {
  let e = D(t);
  if (e.value) {
    const n = e.value.container;
    if (n && (n.hasOwnProperty("layout") || n.hasOwnProperty("items"))) {
      const l = e.value.autoHeight, d = e.value.scrollbars;
      return {
        ...e,
        value: {
          container: {
            showHeader: !0,
            body: {
              0: {
                view: n
              }
            },
            showFooter: !1,
            showSider: !0,
            autoHeight: l,
            contentScrollbars: d
          }
        }
      };
    }
  }
  return e;
}
class M extends F {
  constructor(e) {
    super(L(e));
  }
}
const T = C(M, [g, b, new P("container", p("export.ratingValueDesc")), new v("siderCollapsed", (t) => ({
  data: t.children.container.children.siderCollapsed.nodeWithoutCache()
}), (t) => t.data.value, p("listView.itemsDesc"))]), G = V(T, [{
  method: {
    name: "setSiderCollapsed",
    description: "Set the Sider of the PageLayout to be collapsed or not",
    params: [{
      name: "collapsed",
      type: "boolean"
    }]
  },
  execute: (t, e) => {
    e[0];
  }
}]);
function J(t, e) {
  return {
    container: {
      header: x([{
        item: {
          compType: "navigation",
          name: e.genItemName("pageNavigation"),
          comp: {
            logoUrl: "",
            hidden: !1,
            items: [{
              label: "Home",
              hidden: !1,
              active: !1
            }, {
              label: "Services",
              hidden: !1,
              active: !1,
              items: [{
                label: "Lowcode Platform",
                hidden: !1,
                active: !1
              }, {
                label: "App Development",
                hidden: !1,
                active: !1
              }]
            }, {
              label: "About",
              hidden: !1,
              active: !1
            }]
          }
        },
        layoutItem: {
          i: "",
          h: 6,
          w: 24,
          x: 0,
          y: 0
        }
      }])
    }
  };
}
export {
  F as ContainerBaseComp,
  G as PageLayoutComp,
  J as defaultPageLayoutData
};
