// lowcoder-sdk/dist/f0b9af3b.js
import { mJ as M, i6 as D } from "./c988e26e.js";
import L from "react";
function z(f, o) {
  for (var c = 0; c < o.length; c++) {
    const i = o[c];
    if (typeof i != "string" && !Array.isArray(i)) {
      for (const u in i)
        if (u !== "default" && !(u in f)) {
          const a = Object.getOwnPropertyDescriptor(i, u);
          a && Object.defineProperty(f, u, a.get ? a : {
            enumerable: !0,
            get: () => i[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(f, Symbol.toStringTag, { value: "Module" }));
}
var s = { exports: {} }, v = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var j = { icon: function(o, c) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M384 185h456v136H384zm-200 0h136v656H184zm696-73H144c-17.7 0-32 14.3-32 32v1c0-17.7 14.3-32 32-32h736c17.7 0 32 14.3 32 32v-1c0-17.7-14.3-32-32-32zM384 385h456v456H384z", fill: c } }, { tag: "path", attrs: { d: "M880 113H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V145c0-17.7-14.3-32-32-32zM320 841H184V185h136v656zm520 0H384V385h456v456zm0-520H384V185h456v136z", fill: o } }] };
}, name: "layout", theme: "twotone" };
_.default = j;
(function(f) {
  Object.defineProperty(f, "__esModule", {
    value: !0
  }), Object.defineProperty(f, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var o = /* @__PURE__ */ g(L), c = /* @__PURE__ */ a(_), i = /* @__PURE__ */ a(M);
  function u(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function y(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (y = function(n) {
      return n ? r : t;
    })(e);
  }
  function g(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = y(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, l = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var d = l ? Object.getOwnPropertyDescriptor(e, p) : null;
        d && (d.get || d.set) ? Object.defineProperty(n, p, d) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function m(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(l) {
        return Object.getOwnPropertyDescriptor(r, l).enumerable;
      }))), n.forEach(function(l) {
        u(e, l, r[l]);
      });
    }
    return e;
  }
  function P(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(l) {
        return Object.getOwnPropertyDescriptor(e, l).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function h(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : P(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var T = function(e, t) {
    return /* @__PURE__ */ o.createElement(i.default, h(m({}, e), {
      ref: t,
      icon: c.default
    }));
  }, O = /* @__PURE__ */ o.forwardRef(T);
  process.env.NODE_ENV !== "production" && (O.displayName = "LayoutTwoTone");
  var b = O;
})(v);
(function(f, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const c = i(v);
  function i(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const u = c;
  o.default = u, f.exports = u;
})(s, s.exports);
var w = s.exports;
const H = /* @__PURE__ */ D(w), x = /* @__PURE__ */ z({
  __proto__: null,
  default: H
}, [w]);
export {
  x as L
};
