// lowcoder-sdk/dist/f0d61f39.js
import { X as l, ah as n, F as i, dm as r, fl as c, f5 as m } from "./c988e26e.js";
import { getStyle as d } from "./7032c5eb.js";
import { useUIView as h } from "./89fd4803.js";
import s, { useContext as f } from "react";
import { timeZoneOptions as u } from "./8c5a44af.js";
import "react-dom";
const x = l(c)`
  width: 100%;
  ${(e) => e.$style && d(e.$style)}
`, y = s.lazy(() => import("./4fe456d5.js").then((e) => ({
  default: e.TimeMobileUIView
}))), T = l(m)`
  width: 100%;
  margin: 10px 0; 
  .ant-select-selector {
    font-size: 14px; 
    padding: 8px; 
  }
`, V = (e) => {
  const t = f(n), o = Array.isArray(e.placeholder) ? e.placeholder[0] : e.placeholder;
  return h(/* @__PURE__ */ i(y, {
    ...e
  }), /* @__PURE__ */ i(x, {
    ...e,
    ref: e.viewRef,
    hideDisabledOptions: !0,
    inputReadOnly: r(t == null ? void 0 : t.getAppSettings().maxWidth),
    placeholder: o,
    renderExtraFooter: () => e.timeZone === "UserChoice" && /* @__PURE__ */ i(T, {
      placeholder: "Select Time Zone",
      options: u.filter((a) => a.value !== "UserChoice"),
      onChange: e == null ? void 0 : e.handleTimeZoneChange,
      defaultValue: "Etc/UTC"
    })
  }));
};
export {
  V as TimeUIView
};
