// lowcoder-sdk/dist/f753b62a.js
import { mJ as M, i6 as D } from "./c988e26e.js";
import j from "react";
function E(i, o) {
  for (var f = 0; f < o.length; f++) {
    const c = o[f];
    if (typeof c != "string" && !Array.isArray(c)) {
      for (const u in c)
        if (u !== "default" && !(u in i)) {
          const a = Object.getOwnPropertyDescriptor(c, u);
          a && Object.defineProperty(i, u, a.get ? a : {
            enumerable: !0,
            get: () => c[u]
          });
        }
    }
  }
  return Object.freeze(Object.defineProperty(i, Symbol.toStringTag, { value: "Module" }));
}
var d = { exports: {} }, v = {}, _ = {};
Object.defineProperty(_, "__esModule", { value: !0 });
var S = { icon: function(o, f) {
  return { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M474.8 357.7l-24.5 24.5-34.4-3.8c-9.6-1.1-19.3-1.6-28.9-1.6-29 0-57.5 4.7-84.7 14.1-14 4.8-27.4 10.8-40.3 17.9l353.1 353.3a259.92 259.92 0 0030.4-153.9l-3.8-34.4 24.5-24.5L800 415.5 608.5 224 474.8 357.7z", fill: f } }, { tag: "path", attrs: { d: "M878.3 392.1L631.9 145.7c-6.5-6.5-15-9.7-23.5-9.7s-17 3.2-23.5 9.7L423.8 306.9c-12.2-1.4-24.5-2-36.8-2-73.2 0-146.4 24.1-206.5 72.3a33.23 33.23 0 00-2.7 49.4l181.7 181.7-215.4 215.2a15.8 15.8 0 00-4.6 9.8l-3.4 37.2c-.9 9.4 6.6 17.4 15.9 17.4.5 0 1 0 1.5-.1l37.2-3.4c3.7-.3 7.2-2 9.8-4.6l215.4-215.4 181.7 181.7c6.5 6.5 15 9.7 23.5 9.7 9.7 0 19.3-4.2 25.9-12.4 56.3-70.3 79.7-158.3 70.2-243.4l161.1-161.1c12.9-12.8 12.9-33.8 0-46.8zM666.2 549.3l-24.5 24.5 3.8 34.4a259.92 259.92 0 01-30.4 153.9L262 408.8c12.9-7.1 26.3-13.1 40.3-17.9 27.2-9.4 55.7-14.1 84.7-14.1 9.6 0 19.3.5 28.9 1.6l34.4 3.8 24.5-24.5L608.5 224 800 415.5 666.2 549.3z", fill: o } }] };
}, name: "pushpin", theme: "twotone" };
_.default = S;
(function(i) {
  Object.defineProperty(i, "__esModule", {
    value: !0
  }), Object.defineProperty(i, "default", {
    enumerable: !0,
    get: function() {
      return b;
    }
  });
  var o = /* @__PURE__ */ y(j), f = /* @__PURE__ */ a(_), c = /* @__PURE__ */ a(M);
  function u(e, t, r) {
    return t in e ? Object.defineProperty(e, t, {
      value: r,
      enumerable: !0,
      configurable: !0,
      writable: !0
    }) : e[t] = r, e;
  }
  function a(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  function O(e) {
    if (typeof WeakMap != "function")
      return null;
    var t = /* @__PURE__ */ new WeakMap(), r = /* @__PURE__ */ new WeakMap();
    return (O = function(n) {
      return n ? r : t;
    })(e);
  }
  function y(e, t) {
    if (!t && e && e.__esModule)
      return e;
    if (e === null || typeof e != "object" && typeof e != "function")
      return {
        default: e
      };
    var r = O(t);
    if (r && r.has(e))
      return r.get(e);
    var n = {
      __proto__: null
    }, l = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for (var p in e)
      if (p !== "default" && Object.prototype.hasOwnProperty.call(e, p)) {
        var s = l ? Object.getOwnPropertyDescriptor(e, p) : null;
        s && (s.get || s.set) ? Object.defineProperty(n, p, s) : n[p] = e[p];
      }
    return n.default = e, r && r.set(e, n), n;
  }
  function g(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t] != null ? arguments[t] : {}, n = Object.keys(r);
      typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(r).filter(function(l) {
        return Object.getOwnPropertyDescriptor(r, l).enumerable;
      }))), n.forEach(function(l) {
        u(e, l, r[l]);
      });
    }
    return e;
  }
  function h(e, t) {
    var r = Object.keys(e);
    if (Object.getOwnPropertySymbols) {
      var n = Object.getOwnPropertySymbols(e);
      t && (n = n.filter(function(l) {
        return Object.getOwnPropertyDescriptor(e, l).enumerable;
      })), r.push.apply(r, n);
    }
    return r;
  }
  function T(e, t) {
    return t = t ?? {}, Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : h(Object(t)).forEach(function(r) {
      Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
    }), e;
  }
  var m = function(e, t) {
    return /* @__PURE__ */ o.createElement(c.default, T(g({}, e), {
      ref: t,
      icon: f.default
    }));
  }, P = /* @__PURE__ */ o.forwardRef(m);
  process.env.NODE_ENV !== "production" && (P.displayName = "PushpinTwoTone");
  var b = P;
})(v);
(function(i, o) {
  Object.defineProperty(o, "__esModule", {
    value: !0
  }), o.default = void 0;
  const f = c(v);
  function c(a) {
    return a && a.__esModule ? a : { default: a };
  }
  const u = f;
  o.default = u, i.exports = u;
})(d, d.exports);
var w = d.exports;
const x = /* @__PURE__ */ D(w), R = /* @__PURE__ */ E({
  __proto__: null,
  default: x
}, [w]);
export {
  R as P
};
